 rust
struct Foo;
struct Bar(Foo);
impl !Send for Foo;
fn is_send<T:Send>();
fn main() {
    is_send::<Bar>(); //~ ERROR ...
    is_send::<(int, Foo)>(); //~ ERROR ...
    is_send::<Box<Foo>>(); //~ ERROR ...
    is_send::<Box<Bar>>(); //~ ERROR ...

}
