
-    //~^^^ ERROR mismatched types in range
-    //~| NOTE expected char, found integral variable
+    //~^^^ ERROR mismatched types
+    //~| NOTE expected type `_`
+    //~| NOTE found type `char`
