\n"},"level":"error","spans":[{"file_name":"/Users/dpc/Code/rust/src/test/ui/lint/outer-forbid.rs","byte_start":897,"byte_end":903,"line_start":22,"line_end":22,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"#[allow(unused)] //~ ERROR overruled","highlight_start":9,"highlight_end":15}],"label":"overruled by previous forbid","suggested_replacement":null,"expansion":null},{"file_name":"/Users/dpc/Code/rust/src/test/ui/lint/outer-forbid.rs","byte_start":803,"byte_end":809,"line_start":17,"line_end":17,"column_start":11,"column_end":17,"is_primary":false,"text":[{"text":"#![forbid(unused, non_snake_case)]","highlight_start":11,"highlight_end":17}],"label":"`forbid` level set here","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0453]: allow(unused) overruled by outer forbid(unused)\n  --> /Users/dpc/Code/rust/src/test/ui/lint/outer-forbid.rs:22:9\n   |\n17 | #![forbid(unused, non_snake_case)]\n   |           ------ `forbid` level set here\n...\n22 | #[allow(unused)] //~ ERROR overruled\n   |         ^^^^^^ overruled by previous forbid\n\n"}
{"message":"allow(bad_style) overruled by outer forbid(non_snake_case)","code":{"code":"E0453","explanation":"\nA lint check attribute was overruled by a `forbid` directive set as an\nattribute on an enclosing scope, or on the command line with the `-F` option.\n\nExample of erroneous code:\n\n