\n"},"level":"error","spans":[{"file_name":"/Users/dpc/Code/rust/src/test/ui/lint/outer-forbid.rs","byte_start":947,"byte_end":956,"line_start":25,"line_end":25,"column_start":9,"column_end":18,"is_primary":true,"text":[{"text":"#[allow(bad_style)] //~ ERROR overruled","highlight_start":9,"highlight_end":18}],"label":"overruled by previous forbid","suggested_replacement":null,"expansion":null},{"file_name":"/Users/dpc/Code/rust/src/test/ui/lint/outer-forbid.rs","byte_start":811,"byte_end":825,"line_start":17,"line_end":17,"column_start":19,"column_end":33,"is_primary":false,"text":[{"text":"#![forbid(unused, non_snake_case)]","highlight_start":19,"highlight_end":33}],"label":"`forbid` level set here","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0453]: allow(bad_style) overruled by outer forbid(non_snake_case)\n  --> /Users/dpc/Code/rust/src/test/ui/lint/outer-forbid.rs:25:9\n   |\n17 | #![forbid(unused, non_snake_case)]\n   |                   -------------- `forbid` level set here\n...\n25 | #[allow(bad_style)] //~ ERROR overruled\n   |         ^^^^^^^^^ overruled by previous forbid\n\n"}
{"message":"aborting due to 6 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 6 previous errors\n\n"}
