\n\nThe number of supplied parameters must exactly match the number of defined type\nparameters.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0087.rs","byte_start":511,"byte_end":521,"line_start":15,"line_end":15,"column_start":5,"column_end":15,"is_primary":false,"text":[{"text":"    foo::<f64>(); //~ ERROR expected at most 0 type parameters, found 1 type parameter [E0087]","highlight_start":5,"highlight_end":15}],"label":"expected 0 type parameters","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/error-codes/E0087.rs","byte_start":517,"byte_end":520,"line_start":15,"line_end":15,"column_start":11,"column_end":14,"is_primary":true,"text":[{"text":"    foo::<f64>(); //~ ERROR expected at most 0 type parameters, found 1 type parameter [E0087]","highlight_start":11,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0087]: too many type parameters provided: expected at most 0 type parameters, found 1 type parameter\n  --> /checkout/src/test/ui/error-codes/E0087.rs:15:11\n   |\nLL |     foo::<f64>(); //~ ERROR expected at most 0 type parameters, found 1 type parameter [E0087]\n   |     ------^^^- expected 0 type parameters\n\n"}
[00:47:21] {"message":"too many type parameters provided: expected at most 1 type parameter, found 2 type parameters","code":{"code":"E0087","explanation":"\nToo many type parameters were supplied for a function. For example:\n\n