plain
Receiving objects: 100% (250/250), 23.39 KiB | 23.39 MiB/s, done.
---
Resolving deltas: 100% (221/221), completed with 68 local objects.
---
[00:00:57] configure: rust.quiet-tests     := True
---
[00:39:03] .........................................................................i..........................
[00:39:09] ................i...................................................................................
---
[00:39:43] .............................................................................................i......
[00:39:50] ................................................................i...................................
---
[00:40:42] .............................................i......................................................
---
[00:44:24] .............................i......................................................................
[00:44:38] ..............................................................i.....................................
[00:44:53] ...............................................i....................................................
[00:45:12] ....................................................................................................
[00:45:32] ....................................................................................................
[00:45:53] ....................................................................................................
[00:46:16] .i................................................................................................i.
[00:46:45] ...........................................................................................test [run-pass] run-pass/mir_heavy_promoted.rs has been running for over 60 seconds
[00:46:50] .........
[00:47:18] ....................................................................................................
[00:47:52] ...............................................................ii...................................
[00:48:39] ..........................i....................................................i.ii.................
[00:49:16] .......................................................................................iiiiiii......
---
[00:51:09] ....................................i...........................................................F...
[00:51:17] ..F.................................................................................................
[00:51:24] ..................i............................................................ii.iii...............
[00:51:31] ................................F...................................................................
[00:51:38] ........i.......F......................i...........F.................................F..............
[00:51:46] .....................F..............................................................................
[00:51:52] .....................i..............................................................................
[00:52:00] ....................................................................................................
[00:52:09] ....................................................................................................
[00:52:19] ....................................................................................................
[00:52:29] ...........F........................................................................................
[00:52:42] ....................................................................................................
[00:52:50] ..............i...............FFF......FF...........................................................
[00:52:59] .............F...i..ii.................................F............................................
[00:53:09] ..........................................F.......F..................................F..............
[00:53:19] ....................................................................................................
[00:53:28] .........................................................F....................F.....i...............
[00:53:38] ..............................i.....................................................................
[00:53:49] ....................................................................................................
[00:53:59] .............................................................................F......................
[00:54:09] ......................F.............................................................................
t.rs:2901:9
[00:54:11]
[00:54:11] ---- [compile-fail] compile-fail/elided-test.rs stdout ----
[00:54:11]
[00:54:11] error: compiler encountered internal error
[00:54:11] status: exit code: 101
[00:54:11] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/compile-fail/elided-test.rs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/compile-fail" "--target=x86_64-unknown-linux-gnu" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/compile-fail/elided-test.stage2-x86_64-unknown-linux-gnu" "-Crpath" "-O" "-Zmiri" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/compile-fail/elided-test.stage2-x86_64-unknown-linux-gnu.aux" "-A" "unused"
---
[00:54:11] error[E0601]: `main` function not found in crate `elided_test`
[00:54:11]    |
[00:54:11]    = note: consider adding a `main` function to `/checkout/src/test/compile-fail/elided-test.rs`
[00:54:11]
[00:54:11] thread 'rustc' panicked at 'value was not set', libcore/option.rs:916:5
[00:54:11] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:54:11] error: aborting due to previous error
[00:54:11]
[00:54:11] For more information about this error, try `rustc --explain E0601`.
[00:54:11]
[00:54:11] error: internal compiler error: unexpected panic
[00:54_start":19,"line_end":19,"column_start":1,"column_end":12,"is_primary":true,"text":[{"text":"#![feature] //~ ERROR: malformed feature","highlight_start":1,"highlight_end":12}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0555]: malformed feature attribute, expected #![feature(...)]\n  --> /checkout/src/test/compile-fail/gated-bad-feature.rs:19:1\n   |\nLL | #![feature] //~ ERROR: malformed feature\n   | ^^^^^^^^^^^\n\n"}
[00:54:11] {"message":"malformed feature attribute, expected #![feature(...)]","code":{"code":"E0555","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/compile-fail/gated-bad-feature.rs","byte_start":635,"byte_end":654,"line_start":20,"line_end":20,"column_start":1,"column_end":20,"is_primary":true,"text":[{"text":"#![feature = \"foo\"] //~ ERROR: malformed feature","highlight_start":1,"highlight_end":20}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0555]: malformed feature attribute, expected #![feature(...)]\n  --> /checkout/src/test/compile-fail/gated-bad-feature.rs:20:1\n   |\nLL | #![feature = \"foo\"] //~ ERROR: malformed feature\n   | ^^^^^^^^^^^^^^^^^^^\n\n"}
[00:54:11] {"message":"feature has been removed","code":{"code":"E0557","explanation":"\nA feature attribute named a feature that has been removed.\n\nErroneous code example:\n\n