\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs","byte_start":2783,"byte_end":2790,"line_start":85,"line_end":85,"column_start":74,"column_end":81,"is_primary":true,"text":[{"text":"    std::rc::Rc::new(&mut Box::new(&no_method_suggested_traits::Bar::X)).method3();","highlight_start":74,"highlight_end":81}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&no_method_suggested_traits::Bar>>` in the current scope\n  --> /checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs:85:74\n   |\nLL |     std::rc::Rc::new(&mut Box::new(&no_method_suggested_traits::Bar::X)).method3();\n   |                                                                          ^^^^^^^\n\n"}
[00:40:47] {"message":"aborting due to 24 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 24 previous errors\n\n"}
[00:40:47] {"message":"For more information about this error, try `rustc --explain E0599`ent":null,"expansion":null},{"file_name":"/checkout/src/test/ui/method-call-err-msg.rs","byte_start":746,"byte_end":750,"line_start":22,"line_end":22,"column_start":7,"column_end":11,"is_primary":true,"text":[{"text":"    x.zero(0)   //~ ERROR this function takes 0 parameters but 1 parameter was supplied","highlight_start":7,"highlight_end":11}],"label":"expected 0 parameters","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0061]: this function takes 0 parameters but 1 parameter was supplied\n  --> /checkout/src/test/ui/method-call-err-msg.rs:22:7\n   |\nLL |     fn zero(self) -> Foo { self }\n   |     -------------------- defined here\n...\nLL |     x.zero(0)   //~ ERROR this function takes 0 parameters but 1 parameter was supplied\n   |       ^^^^ expected 0 parameters\n\n"}
[00:40:47] {"message":"this function takes 1 parameter but 0 parameters were supplied","code":{"code":"E0061","explanation":"\nThe number of arguments passed to a function must match the number of arguments\nspecified in the function signature.\n\nFor example, a function like:\n\n