\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs","byte_start":2327,"byte_end":2334,"line_start":79,"line_end":79,"column_start":13,"column_end":20,"is_primary":true,"text":[{"text":"    1_usize.method3(); //~ ERROR no method named","highlight_start":13,"highlight_end":20}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `method3` found for type `usize` in the current scope\n  --> /checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs:79:13\n   |\nLL |     1_usize.method3(); //~ ERROR no method named\n   |             ^^^^^^^\n\n"}
[00:40:47] {"message":"no method named `method3` found for type `std::rc::Rc<&mutl":"error","spans":[{"file_name":"/checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs","byte_start":2483,"byte_end":2490,"line_start":81,"line_end":81,"column_start":37,"column_end":44,"is_primary":true,"text":[{"text":"    no_method_suggested_traits::Foo.method3();  //~ ERROR no method named","highlight_start":37,"highlight_end":44}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `method3` found for type `no_method_suggested_traits::Foo` in the current scope\n  --> /checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs:81:37\n   |\nLL |     no_method_suggested_traits::Foo.method3();  //~ ERROR no method named\n   |                                     ^^^^^^^\n\n"}
[00:40:47] {"message":"no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&no_method_suggested_traits::Foo>>` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n