\n\nDelete the offending feature attribute.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/compile-fail/gated-bad-feature.rs","byte_start":696,"byte_end":716,"line_start":22,"line_end":22,"column_start":12,"column_end":32,"is_primary":true,"text":[{"text":"#![feature(test_removed_feature)] //~ ERROR: feature has been removed","highlight_start":12,"highlight_end":32}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0557]: feature has been removed\n  --> /checkout/src/test/compile-fail/gated-bad-feature.rs:22:12\n   |\nLL | #![feature(test_removed_feature)] //~ ERROR: feature has been removed\n   |            ^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:54:11] {"message":"`main` function not found in crate `gated_bad_feature`","code":{"code":"E0601","explanation":"\nNo `main` function was found in a binary crate. To fix this error, add a\n`main` function. For example:\n\n