\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lifetime-errors/liveness-assign-imm-local-notes.rs","byte_start":1174,"byte_end":1179,"line_start":45,"line_end":45,"column_start":13,"column_end":18,"is_primary":true,"text":[{"text":"            x = 1;      //~ ERROR (Ast) [E0384]","highlight_start":13,"highlight_end":18}],"label":"cannot assign twice to immutable variable","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0384]: cannot assign twice to immutable variable `x` (Mir)\n  --> /checkout/src/test/ui/lifetime-errors/liveness-assign-imm-local-notes.rs:45:13\n   |\nLL |             x = 1;      //~ ERROR (Ast) [E0384]\n   |             ^^^^^ cannot assign twice to immutable variable\n\n"}
[00:42:43] {"message":"cannot assign twice to immutable variable `x` (Mir)","code":{"code":"E0384","explanation":"\nThis error occurs when an attempt is made to reassign an immutable variable.\nFor example:\n\n