\n\nWith this approach, x and y share ownership of the data via the `Rc` (reference\ncount type). `RefCell` essentially performs runtime borrow checking: ensuring\nthat at most one writer or multiple readers can access the data at any one time.\n\nIf you wish to learn more about ownership in Rust, start with the chapter in the\nBook:\n\nhttps://doc.rust-lang.org/book/first-edition/ownership.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/issue-41962.rs","byte_start":593,"byte_end":598,"line_start":17,"line_end":17,"column_start":21,"column_end":26,"is_primary":false,"text":[{"text":"        if let Some(thing) = maybe {","highlight_start":21,"highlight_end":26}],"label":"value moved here","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/issue-41962.rs","byte_start":581,"byte_end":966,"line_start":17,"line_end":23,"column_start":9,"column_end":10,"is_primary":true,"text":[{"text":"        if let Some(thing) = maybe {","highlight_start":9,"highlight_end":37},{"text":"        //~^ ERROR use of partially moved value: `maybe` (Ast) [E0382]","highlight_start":1,"highlight_end":71},{"text":"        //~| ERROR use of moved value: `(maybe as std::prelude::v1::Some).0` (Ast) [E0382]","highlight_start":1,"highlight_end":91},{"text":"        //~| ERROR use of moved value: `maybe` (Mir) [E0382]","highlight_start":1,"highlight_end":61},{"text":"        //~| ERROR use of moved value: `maybe` (Mir) [E0382]","highlight_start":1,"highlight_end":61},{"text":"        //~| ERROR use of moved value: `maybe.0` (Mir) [E0382]","highlight_start":1,"highlight_end":63},{"text":"        }","highlight_start":1,"highlight_end":10}],"label":"value used here after move","suggested_replacement":null,"expansion":null}],"children":[{"message":"move occurs because `maybe` has type `std::option::Option<std::vec::Vec<bool>>`, which does not implement the `Copy` trait","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0382]: use of moved value: `maybe` (Mir)\n  --> /checkout/src/test/ui/borrowck/issue-41962.rs:17:9\n   |\nLL |           if let Some(thing) = maybe {\n   |           ^           ----- value moved here\n   |  _________|\n   | |\nLL | |         //~^ ERROR use of partially moved value: `maybe` (Ast) [E0382]\nLL | |         //~| ERROR use of moved value: `(maybe as std::prelude::v1::Some).0` (Ast) [E0382]\nLL | |         //~| ERROR use of moved value: `maybe` (Mir) [E0382]\nLL | |         //~| ERROR use of moved value: `maybe` (Mir) [E0382]\nLL | |         //~| ERROR use of moved value: `maybe.0` (Mir) [E0382]\nLL | |         }\n   | |_________^ value used here after move\n   |\n   = note: move occurs because `maybe` has type `std::option::Option<std::vec::Vec<bool>>`, which does not implement the `Copy` trait\n\n"}
[00:42:43] {"message":"use of moved value: `maybe` (Mir)","code":{"code":"E0382","explanation":"\nThis error occurs when an attempt is made to use a variable after its contents\nhave been moved elsewhere. For example:\n\n