\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lifetime-errors/liveness-assign-imm-local-notes.rs","byte_start":774,"byte_end":779,"line_start":23,"line_end":23,"column_start":9,"column_end":14,"is_primary":true,"text":[{"text":"        x = 3;      //~ ERROR (Ast) [E0384]","highlight_start":9,"highlight_end":14}],"label":"cannot assign twice to immutable variable","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/lifetime-errors/liveness-assign-imm-local-notes.rs","byte_start":759,"byte_end":764,"line_start":22,"line_end":22,"column_start":9,"column_end":14,"is_primary":false,"text":[{"text":"        x = 2;","highlight_start":9,"highlight_end":14}],"label":"first assignment to `x`","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0384]: cannot assign twice to immutable variable `x` (Ast)\n  --> /checkout/src/test/ui/lifetime-errors/liveness-assign-imm-local-notes.rs:23:9\n   |\nLL |         x = 2;\n   |         ----- first assignment to `x`\nLL |         x = 3;      //~ ERROR (Ast) [E0384]\n   |         ^^^^^ cannot assign twice to immutable variable\n\n"}
[00:42:43] {"message":"cannot assign twice to immutable variable `x` (Mir)","code":{"code":"E0384","explanation":"\nThis error occurs when an attempt is made to reassign an immutable variable.\nFor example:\n\n