\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/generator/yield-while-iterating.rs","byte_start":948,"byte_end":955,"line_start":23,"line_end":23,"column_start":13,"column_end":20,"is_primary":false,"text":[{"text":"            yield();","highlight_start":13,"highlight_end":20}],"label":"possible yield occurs here","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/generator/yield-while-iterating.rs","byte_start":921,"byte_end":923,"line_start":22,"line_end":22,"column_start":18,"column_end":20,"is_primary":true,"text":[{"text":"        for p in &x { //~ ERROR","highlight_start":18,"highlight_end":20}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0626]: borrow may still be in use when generator yields\n  --> /checkout/src/test/ui/generator/yield-while-iterating.rs:22:18\n   |\nLL |         for p in &x { //~ ERROR\n   |                  ^^\nLL |             yield();\n   |             ------- possible yield occurs here\n\n"}
[00:44:39] {"message":"cannot borrow `x` as immutable because it is also borrowed as mutable","code":{"code":"E0502","explanation":"\nThis error indicates that you are trying to borrow a variable as mutable when it\nhas already been borrowed as immutable.\n\nExample of erroneous code:\n\n