\n\nIf the trait `Foo` was deriving from something like `Super<String>` or\n`Super<T>` (where `Foo` itself is `Foo<T>`), this is okay, because given a type\n`get_a()` will definitely return an object of that type.\n\nHowever, if it derives from `Super<Self>`, even though `Super` is object safe,\nthe method `get_a()` would return an object of unknown type when called on the\nfunction. `Self` type parameters let us make object safe traits no longer safe,\nso they are forbidden when specifying supertraits.\n\nThere's no easy fix for this, generally code will need to be refactored so that\nyou no longer need to derive from `Super<Self>`.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/arbitrary-self-types-not-object-safe.rs","byte_start":931,"byte_end":947,"line_start":40,"line_end":40,"column_start":13,"column_end":29,"is_primary":true,"text":[{"text":"    let x = Box::new(5usize) as Box<Foo>;","highlight_start":13,"highlight_end":29}],"label":"the trait `Foo` cannot be made into an object","suggested_replacement":null,"expansion":null}],"children":[{"message":"method `foo` has a non-standard `self` type","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"required because of the requirements on the impl of `std::ops::CoerceUnsized<std::boxed::Box<Foo>>` for `std::boxed::Box<usize>`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0038]: the trait `Foo` cannot be made into an object\n  --> /checkout/src/test/ui/arbitrary-self-types-not-object-safe.rs:40:13: in fn make_foo\n   |\nLL |     let x = Box::new(5usize) as Box<Foo>;\n   |             ^^^^^^^^^^^^^^^^ the trait `Foo` cannot be made into an object\n   |\n   = note: method `foo` has a non-standard `self` type\n   = note: required because of the requirements on the impl of `std::ops::CoerceUnsized<std::boxed::Box<Foo>>` for `std::boxed::Box<usize>`\n\n"}
[00:48:30] {"message":"aborting due to 2 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 2 previous errors\n\n"}
[00:48:30] {"message":"For more information about this error, try `rustc --explain E0038`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0038`.\n"}
[00:48:30] ------------------------------------------
[00:48:30] 
[00:48:30] thread '[ui] ui/arbitrary-self-types-not-object-safe.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3033:9
[00:48:30] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:48:30] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:48:30] 
[00:48:30] ---- [ui] ui/anonymous-higher-ranked-lifetime.rs stdout ----
[00:48:30]  diff of stderr:
[00:48:30] 
[00:48:30] 1 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:12:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:12:5: in fn main
[00:48:30] 3    |
[00:48:30] 4 LL |     f1(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 5    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 13    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 14 
[00:48:30] 15 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:13:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:13:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:13:5: in fn main
[00:48:30] 17    |
[00:48:30] 18 LL |     f2(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 19    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 27    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 28 
[00:48:30] 29 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:14:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:14:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:14:5: in fn main
[00:48:30] 31    |
[00:48:30] 32 LL |     f3(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 33    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 41    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 42 
[00:48:30] 43 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:15:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:15:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:15:5: in fn main
[00:48:30] 45    |
[00:48:30] 46 LL |     f4(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 47    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 55    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 56 
[00:48:30] 57 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:16:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:16:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:16:5: in fn main
[00:48:30] 59    |
[00:48:30] 60 LL |     f5(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 61    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 69    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 70 
[00:48:30] 71 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:17:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:17:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:17:5: in fn main
[00:48:30] 73    |
[00:48:30] 74 LL |     g1(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 75    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 83    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 84 
[00:48:30] 85 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:18:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:18:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:18:5: in fn main
[00:48:30] 87    |
[00:48:30] 88 LL |     g2(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 89    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 97    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 98 
[00:48:30] 99 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:19:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:19:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:19:5: in fn main
[00:48:30] 101    |
[00:48:30] 102 LL |     g3(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 103    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 111    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 112 
[00:48:30] 113 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:20:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:20:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:20:5: in fn main
[00:48:30] 115    |
[00:48:30] 116 LL |     g4(|_: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 117    |     ^^ -------------- found signature of `fn((), ()) -> _`
[00:48:30] 125    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 126 
[00:48:30] 127 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:21:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:21:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:21:5: in fn main
[00:48:30] 129    |
[00:48:30] 130 LL |     h1(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 131    |     ^^ ---------------------------- found signature of `fn((), (), (), ()) -> _`
[00:48:30] 139    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:48:30] 140 
[00:48:30] 141 error[E0631]: type mismatch in closure arguments
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:22:5
[00:48:30] -   --> $DIR/anonymous-higher-ranked-lifetime.rs:22:5
[00:48:30] +   --> $DIR/anonymous-higher-ranked-lifetime.rs:22:5: in fn main
[00:48:30] 143    |
[00:48:30] 144 LL |     h2(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch
[00:48:30] 145    |     ^^ ---------------------------- found signature of `fn((), (), (), ()) -> _`
[00:48:30] 
[00:48:30] The actual stderr differed from the expected stderr.
[00:48:30] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/anonymous-higher-ranked-lifetime.stderr
[00:48:30] To update references, run this command from build directory:
[00:48:30] To update references, run this command from build directory:
[00:48:30] /checkout/src/test/ui/update-references.sh '/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui' 'anonymous-higher-ranked-lifetime.rs'
[00:48:30] error: 1 errors occurred comparing output.
[00:48:30] status: exit code: 101
[00:48:30] status: exit code: 101
[00:48:30] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/anonymous-higher-ranked-lifetime.stage2-x86_64-unknown-linux-gnu" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/anonymous-higher-ranked-lifetime.stage2-x86_64-unknown-linux-gnu.aux" "-A" "unused"
[00:48:30] ------------------------------------------
[00:48:30] 
[00:48:30] ------------------------------------------
[00:48:30] stderr:
[00:48:30] stderr:
[00:48:30] ------------------------------------------
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":483,"byte_end":485,"line_start":12,"line_end":12,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    f1(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'r, 's> fn(&'r (), &'s ()) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":486,"byte_end":500,"line_start":12,"line_end":12,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    f1(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `f1`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1080,"byte_end":1116,"line_start":26,"line_end":26,"column_start":1,"column_end":37,"is_primary":true,"text":[{"text":"fn f1<F>(_: F) where F: Fn(&(), &()) {}","highlight_start":1,"highlight_end":37}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:12:5: in fn main\n   |\nLL |     f1(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'r, 's> fn(&'r (), &'s ()) -> _`\n   |\nnote: required by `f1`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:26:1\n   |\nLL | fn f1<F>(_: F) where F: Fn(&(), &()) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":534,"byte_end":536,"line_start":13,"line_end":13,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    f2(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'a, 'r> fn(&'a (), &'r ()) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":537,"byte_end":551,"line_start":13,"line_end":13,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    f2(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `f2`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1120,"byte_end":1167,"line_start":27,"line_end":27,"column_start":1,"column_end":48,"is_primary":true,"text":[{"text":"fn f2<F>(_: F) where F: for<'a> Fn(&'a (), &()) {}","highlight_start":1,"highlight_end":48}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:13:5: in fn main\n   |\nLL |     f2(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'a, 'r> fn(&'a (), &'r ()) -> _`\n   |\nnote: required by `f2`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:27:1\n   |\nLL | fn f2<F>(_: F) where F: for<'a> Fn(&'a (), &()) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":585,"byte_end":587,"line_start":14,"line_end":14,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    f3(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'r> fn(&(), &'r ()) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":588,"byte_end":602,"line_start":14,"line_end":14,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    f3(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `f3`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1171,"byte_end":1214,"line_start":28,"line_end":28,"column_start":1,"column_end":44,"is_primary":true,"text":[{"text":"fn f3<'a, F>(_: F) where F: Fn(&'a (), &()) {}","highlight_start":1,"highlight_end":44}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:14:5: in fn main\n   |\nLL |     f3(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'r> fn(&(), &'r ()) -> _`\n   |\nnote: required by `f3`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:28:1\n   |\nLL | fn f3<'a, F>(_: F) where F: Fn(&'a (), &()) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":636,"byte_end":638,"line_start":15,"line_end":15,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    f4(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'s, 'r> fn(&'s (), &'r ()) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":639,"byte_end":653,"line_start":15,"line_end":15,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    f4(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `f4`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1218,"byte_end":1265,"line_start":29,"line_end":29,"column_start":1,"column_end":48,"is_primary":true,"text":[{"text":"fn f4<F>(_: F) where F: for<'r> Fn(&(), &'r ()) {}","highlight_start":1,"highlight_end":48}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:15:5: in fn main\n   |\nLL |     f4(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'s, 'r> fn(&'s (), &'r ()) -> _`\n   |\nnote: required by `f4`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:29:1\n   |\nLL | fn f4<F>(_: F) where F: for<'r> Fn(&(), &'r ()) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":687,"byte_end":689,"line_start":16,"line_end":16,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    f5(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'r> fn(&'r (), &'r ()) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":690,"byte_end":704,"line_start":16,"line_end":16,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    f5(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `f5`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1269,"byte_end":1319,"line_start":30,"line_end":30,"column_start":1,"column_end":51,"is_primary":true,"text":[{"text":"fn f5<F>(_: F) where F: for<'r> Fn(&'r (), &'r ()) {}","highlight_start":1,"highlight_end":51}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:16:5: in fn main\n   |\nLL |     f5(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'r> fn(&'r (), &'r ()) -> _`\n   |\nnote: required by `f5`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:30:1\n   |\nLL | fn f5<F>(_: F) where F: for<'r> Fn(&'r (), &'r ()) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":738,"byte_end":740,"line_start":17,"line_end":17,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    g1(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'r> fn(&'r (), std::boxed::Box<for<'s> std::ops::Fn(&'s ()) + 'static>) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":741,"byte_end":755,"line_start":17,"line_end":17,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    g1(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `g1`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1334,"byte_end":1379,"line_start":33,"line_end":33,"column_start":1,"column_end":46,"is_primary":true,"text":[{"text":"fn g1<F>(_: F) where F: Fn(&(), Box<Fn(&())>) {}","highlight_start":1,"highlight_end":46}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:17:5: in fn main\n   |\nLL |     g1(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'r> fn(&'r (), std::boxed::Box<for<'s> std::ops::Fn(&'s ()) + 'static>) -> _`\n   |\nnote: required by `g1`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:33:1\n   |\nLL | fn g1<F>(_: F) where F: Fn(&(), Box<Fn(&())>) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":789,"byte_end":791,"line_start":18,"line_end":18,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    g2(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'r> fn(&'r (), for<'s> fn(&'s ())) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":792,"byte_end":806,"line_start":18,"line_end":18,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    g2(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `g2`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1383,"byte_end":1423,"line_start":34,"line_end":34,"column_start":1,"column_end":41,"is_primary":true,"text":[{"text":"fn g2<F>(_: F) where F: Fn(&(), fn(&())) {}","highlight_start":1,"highlight_end":41}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:18:5: in fn main\n   |\nLL |     g2(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'r> fn(&'r (), for<'s> fn(&'s ())) -> _`\n   |\nnote: required by `g2`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:34:1\n   |\nLL | fn g2<F>(_: F) where F: Fn(&(), fn(&())) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":840,"byte_end":842,"line_start":19,"line_end":19,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    g3(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'s> fn(&'s (), std::boxed::Box<for<'r> std::ops::Fn(&'r ()) + 'static>) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":843,"byte_end":857,"line_start":19,"line_end":19,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    g3(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `g3`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1427,"byte_end":1483,"line_start":35,"line_end":35,"column_start":1,"column_end":57,"is_primary":true,"text":[{"text":"fn g3<F>(_: F) where F: for<'s> Fn(&'s (), Box<Fn(&())>) {}","highlight_start":1,"highlight_end":57}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:19:5: in fn main\n   |\nLL |     g3(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'s> fn(&'s (), std::boxed::Box<for<'r> std::ops::Fn(&'r ()) + 'static>) -> _`\n   |\nnote: required by `g3`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:35:1\n   |\nLL | fn g3<F>(_: F) where F: for<'s> Fn(&'s (), Box<Fn(&())>) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":891,"byte_end":893,"line_start":20,"line_end":20,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    g4(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'s> fn(&'s (), for<'r> fn(&'r ())) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":894,"byte_end":908,"line_start":20,"line_end":20,"column_start":8,"column_end":22,"is_primary":false,"text":[{"text":"    g4(|_: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":22}],"label":"found signature of `fn((), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `g4`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1487,"byte_end":1538,"line_start":36,"line_end":36,"column_start":1,"column_end":52,"is_primary":true,"text":[{"text":"fn g4<F>(_: F) where F: Fn(&(), for<'r> fn(&'r ())) {}","highlight_start":1,"highlight_end":52}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:20:5: in fn main\n   |\nLL |     g4(|_: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ -------------- found signature of `fn((), ()) -> _`\n   |     |\n   |     expected signature of `for<'s> fn(&'s (), for<'r> fn(&'r ())) -> _`\n   |\nnote: required by `g4`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:36:1\n   |\nLL | fn g4<F>(_: F) where F: Fn(&(), for<'r> fn(&'r ())) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":942,"byte_end":944,"line_start":21,"line_end":21,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    h1(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'r, 's> fn(&'r (), std::boxed::Box<for<'t0> std::ops::Fn(&'t0 ()) + 'static>, &'s (), for<'t0, 't1> fn(&'t0 (), &'t1 ())) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":945,"byte_end":973,"line_start":21,"line_end":21,"column_start":8,"column_end":36,"is_primary":false,"text":[{"text":"    h1(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":36}],"label":"found signature of `fn((), (), (), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `h1`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1552,"byte_end":1616,"line_start":39,"line_end":39,"column_start":1,"column_end":65,"is_primary":true,"text":[{"text":"fn h1<F>(_: F) where F: Fn(&(), Box<Fn(&())>, &(), fn(&(), &())) {}","highlight_start":1,"highlight_end":65}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:21:5: in fn main\n   |\nLL |     h1(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ ---------------------------- found signature of `fn((), (), (), ()) -> _`\n   |     |\n   |     expected signature of `for<'r, 's> fn(&'r (), std::boxed::Box<for<'t0> std::ops::Fn(&'t0 ()) + 'static>, &'s (), for<'t0, 't1> fn(&'t0 (), &'t1 ())) -> _`\n   |\nnote: required by `h1`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:39:1\n   |\nLL | fn h1<F>(_: F) where F: Fn(&(), Box<Fn(&())>, &(), fn(&(), &())) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"type mismatch in closure arguments","code":{"code":"E0631","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1007,"byte_end":1009,"line_start":22,"line_end":22,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    h2(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":5,"highlight_end":7}],"label":"expected signature of `for<'r, 't0> fn(&'r (), std::boxed::Box<for<'s> std::ops::Fn(&'s ()) + 'static>, &'t0 (), for<'s, 't1> fn(&'s (), &'t1 ())) -> _`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1010,"byte_end":1038,"line_start":22,"line_end":22,"column_start":8,"column_end":36,"is_primary":false,"text":[{"text":"    h2(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch","highlight_start":8,"highlight_end":36}],"label":"found signature of `fn((), (), (), ()) -> _`","suggested_replacement":null,"expansion":null}],"children":[{"message":"required by `h2`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs","byte_start":1620,"byte_end":1697,"line_start":40,"line_end":40,"column_start":1,"column_end":78,"is_primary":true,"text":[{"text":"fn h2<F>(_: F) where F: for<'t0> Fn(&(), Box<Fn(&())>, &'t0 (), fn(&(), &())) {}","highlight_start":1,"highlight_end":78}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0631]: type mismatch in closure arguments\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:22:5: in fn main\n   |\nLL |     h2(|_: (), _: (), _: (), _: ()| {}); //~ ERROR type mismatch\n   |     ^^ ---------------------------- found signature of `fn((), (), (), ()) -> _`\n   |     |\n   |     expected signature of `for<'r, 't0> fn(&'r (), std::boxed::Box<for<'s> std::ops::Fn(&'s ()) + 'static>, &'t0 (), for<'s, 't1> fn(&'s (), &'t1 ())) -> _`\n   |\nnote: required by `h2`\n  --> /checkout/src/test/ui/anonymous-higher-ranked-lifetime.rs:40:1\n   |\nLL | fn h2<F>(_: F) where F: for<'t0> Fn(&(), Box<Fn(&())>, &'t0 (), fn(&(), &())) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:48:30] {"message":"aborting due to 11 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 11 previous errors\n\n"}
[00:48:30] {"message":"For more information about this error, try `rustc --explain E0631`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0631`.\n"}
[00:48:30] ------------------------------------------
[00:48:30] 
[00:48:30] thread '[ui] ui/anonymous-higher-ranked-lifetime.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3033:9
[00:48:30] 
[00:48:30] 
[00:48:30] ---- [ui] ui/associated-types-ICE-when-projecting-out-of-err.rs stdout ----
[00:48:30]  diff of stderr:
[00:48:30] 
[00:48:30] 1 error[E0277]: the trait bound `(): Add<A>` is not satisfied
[00:48:30] -   --> $DIR/associated-types-ICE-when-projecting-out-of-err.rs:33:11
[00:48:30] +   --> $DIR/associated-types-ICE-when-projecting-out-of-err.rs:33:11: in fn ice
[00:48:30] 3    |
[00:48:30] 4 LL |     r = r + a;
[00:48:30] 5    |           ^ the trait `Add<A>` is not implemented for `()`
[00:48:30] 
[00:48:30] The actual stderr differed from the expected stderr.
[00:48:30] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/associated-types-ICE-when-projecting-out-of-err.stderr
[00:48:30] To update references, run this command from build directory:
[00:48:30] To update references, run this command from build directory:
[00:48:30] /checkout/src/test/ui/update-references.sh '/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui' 'associated-types-ICE-when-projecting-out-of-err.rs'
[00:48:30] 
[00:48:30] error: 1 errors occurred comparing output.
[00:48:30] status: exit code: 101
[00:48:30] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/associated-types-ICE-when-projecting-out-of-err.rs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/associated-types-ICE-when-projecting-out-of-err.stage2-x86_64-unknown-linux-gnu" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/associated-types-ICE-when-projecting-out-of-err.stage2-x86_64-unknown-linux-gnu.aux" "-A" "unused"
[00:48:30] ------------------------------------------
[00:48:30] 
[00:48:30] ------------------------------------------
[00:48:30] stderr:
[00:48:30] stderr:
[00:48:30] ------------------------------------------
[00:48:30] {"message":"the trait bound `(): Add<A>` is not satisfied","code":{"code":"E0277","explanation":"\nYou tried to use a type which doesn't implement some trait in a place which\nexpected that trait. Erroneous code example:\n\n