\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/span/borrowck-object-mutability.rs","byte_start":593,"byte_end":594,"line_start":19,"line_end":19,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    x.borrowed_mut(); //~ ERROR cannot borrow","highlight_start":5,"highlight_end":6}],"label":"cannot borrow as mutable","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0596]: cannot borrow immutable item `*x` as mutable\n  --> /checkout/src/test/ui/span/borrowck-object-mutability.rs:19:5\n   |\nLL |     x.borrowed_mut(); //~ ERROR cannot borrow\n   |     ^ cannot borrow as mutable\n\n"}
[00:49:37] {"message":"cannot borrow immutable item `*x` as mutable","code":{"code":"E0596","explanation":"\nThis error occurs because you tried to mutably borrow a non-mutable variable.\n\nExample of erroneous code:\n\n