\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-39544.rs","byte_start":954,"byte_end":966,"line_start":36,"line_end":36,"column_start":17,"column_end":29,"is_primary":true,"text":[{"text":"        let _ = &mut other.x; //~ ERROR cannot borrow","highlight_start":17,"highlight_end":29}],"label":"cannot borrow as mutable","suggested_replacement":null,"expansion":null}],"children":[{"message":"the value which is causing this path not to be mutable is...: `*other`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0596]: cannot borrow immutable item `other.x` as mutable\n  --> /checkout/src/test/ui/did_you_mean/issue-39544.rs:36:17\n   |\nLL |         let _ = &mut other.x; //~ ERROR cannot borrow\n   |                 ^^^^^^^^^^^^ cannot borrow as mutable\n   |\n   = note: the value which is causing this path not to be mutable is...: `*other`\n\n"}
[00:49:37] {"message":"cannot borrow immutable item `self.x` as mutable","code":{"code":"E0596","explanation":"\nThis error occurs because you tried to mutably borrow a non-mutable variable.\n\nExample of erroneous code:\n\n