\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/span/borrowck-call-is-borrow-issue-12224.rs","byte_start":688,"byte_end":689,"line_start":22,"line_end":22,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    f(Box::new(|| {","highlight_start":5,"highlight_end":6}],"label":"first mutable borrow occurs here","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/borrowck-call-is-borrow-issue-12224.rs","byte_start":699,"byte_end":701,"line_start":22,"line_end":22,"column_start":16,"column_end":18,"is_primary":true,"text":[{"text":"    f(Box::new(|| {","highlight_start":16,"highlight_end":18}],"label":"second mutable borrow occurs here","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/borrowck-call-is-borrow-issue-12224.rs","byte_start":772,"byte_end":773,"line_start":24,"line_end":24,"column_start":9,"column_end":10,"is_primary":false,"text":[{"text":"        f((Box::new(|| {})))","highlight_start":9,"highlight_end":10}],"label":"borrow occurs due to use of `f` in closure","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/borrowck-call-is-borrow-issue-12224.rs","byte_start":688,"byte_end":800,"line_start":22,"line_end":25,"column_start":5,"column_end":8,"is_primary":false,"text":[{"text":"    f(Box::new(|| {","highlight_start":5,"highlight_end":20},{"text":"    //~^ ERROR: cannot borrow `f` as mutable more than once","highlight_start":1,"highlight_end":60},{"text":"        f((Box::new(|| {})))","highlight_start":1,"highlight_end":29},{"text":"    }));","highlight_start":1,"highlight_end":8}],"label":"borrow later used here","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0499]: cannot borrow `f` as mutable more than once at a time\n  --> /checkout/src/test/ui/span/borrowck-call-is-borrow-issue-12224.rs:22:16\n   |\nLL |       f(Box::new(|| {\n   |       -          ^^ second mutable borrow occurs here\n   |       |\n   |  _____first mutable borrow occurs here\n   | |\nLL | |     //~^ ERROR: cannot borrow `f` as mutable more than once\nLL | |         f((Box::new(|| {})))\n   | |         - borrow occurs due to use of `f` in closure\nLL | |     }));\n   | |_______- borrow later used here\n\n"}
[00:49:37] {"message":"cannot borrow immutable item `*f` as mutable","code":{"code":"E0596","explanation":"\nThis error occurs because you tried to mutably borrow a non-mutable variable.\n\nExample of erroneous code:\n\n