\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-35937.rs","byte_start":557,"byte_end":560,"line_start":17,"line_end":17,"column_start":5,"column_end":8,"is_primary":true,"text":[{"text":"    f.v.push(\"cat\".to_string()); //~ ERROR cannot borrow","highlight_start":5,"highlight_end":8}],"label":"cannot borrow as mutable","suggested_replacement":null,"expansion":null}],"children":[{"message":"the value which is causing this path not to be mutable is...: `f`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0596]: cannot borrow immutable item `f.v` as mutable\n  --> /checkout/src/test/ui/did_you_mean/issue-35937.rs:17:5\n   |\nLL |     f.v.push(\"cat\".to_string()); //~ ERROR cannot borrow\n   |     ^^^ cannot borrow as mutable\n   |\n   = note: the value which is causing this path not to be mutable is...: `f`\n\n"}
[00:49:37] {"message":"cannot assign twice to immutable variable `s.x`","code":{"code":"E0384","explanation":"\nThis error occurs when an attempt is made to reassign an immutable variable.\nFor example:\n\n