\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-35937.rs","byte_start":662,"byte_end":673,"line_start":25,"line_end":25,"column_start":13,"column_end":24,"is_primary":false,"text":[{"text":"    let s = S { x: 42 };","highlight_start":13,"highlight_end":24}],"label":"first assignment to `s.x`","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/did_you_mean/issue-35937.rs","byte_start":679,"byte_end":687,"line_start":26,"line_end":26,"column_start":5,"column_end":13,"is_primary":true,"text":[{"text":"    s.x += 1; //~ ERROR cannot assign","highlight_start":5,"highlight_end":13}],"label":"cannot assign twice to immutable variable","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0384]: cannot assign twice to immutable variable `s.x`\n  --> /checkout/src/test/ui/did_you_mean/issue-35937.rs:26:5\n   |\nLL |     let s = S { x: 42 };\n   |             ----------- first assignment to `s.x`\nLL |     s.x += 1; //~ ERROR cannot assign\n   |     ^^^^^^^^ cannot assign twice to immutable variable\n\n"}
[00:49:37] {"message":"cannot assign twice to immutable variable `s.x`","code":{"code":"E0384","explanation":"\nThis error occurs when an attempt is made to reassign an immutable variable.\nFor example:\n\n