\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issue-36163.rs","byte_start":0,"byte_end":0,"line_start":1,"line_end":1,"column_start":1,"column_end":1,"is_primary":true,"text":[{"text":"// Copyright 2012 The Rust Project Developers. See the COPYRIGHT","highlight_start":1,"highlight_end":1}],"label":"cyclic reference","suggested_replacement":null,"expansion":null}],"children":[{"message":"the cycle begins when const-evaluating `Foo::B::{{initializer}}`...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/issue-36163.rs","byte_start":521,"byte_end":522,"line_start":14,"line_end":14,"column_start":9,"column_end":10,"is_primary":true,"text":[{"text":"    B = A, //~ ERROR E0391","highlight_start":9,"highlight_end":10}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null},{"message":"...which then requires processing `Foo::B::{{initializer}}`...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/issue-36163.rs","byte_start":521,"byte_end":522,"line_start":14,"line_end":14,"column_start":9,"column_end":10,"is_primary":true,"text":[{"text":"    B = A, //~ ERROR E0391","highlight_start":9,"highlight_end":10}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null},{"message":"...which then requires const-evaluating `A`...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/issue-36163.rs","byte_start":467,"byte_end":500,"line_start":11,"line_end":11,"column_start":1,"column_end":34,"is_primary":true,"text":[{"text":"const A: isize = Foo::B as isize;","highlight_start":1,"highlight_end":34}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null},{"message":"...which then requires computing layout of `Foo`...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/issue-36163.rs","byte_start":484,"byte_end":490,"line_start":11,"line_end":11,"column_start":18,"column_end":24,"is_primary":true,"text":[{"text":"const A: isize = Foo::B as isize;","highlight_start":18,"highlight_end":24}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null},{"message":"...which then again requires const-evaluating `Foo::B::{{initializer}}`, completing the cycle.","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0391]: cyclic dependency detected\n   |\nnote: the cycle begins when const-evaluating `Foo::B::{{initializer}}`...\n  --> /checkout/src/test/ui/issue-36163.rs:14:9\n   |\nLL |     B = A, //~ ERROR E0391\n   |         ^\nnote: ...which then requires processing `Foo::B::{{initializer}}`...\n  --> /checkout/src/test/ui/issue-36163.rs:14:9\n   |\nLL |     B = A, //~ ERROR E0391\n   |         ^\nnote: ...which then requires const-evaluating `A`...\n  --> /checkout/src/test/ui/issue-36163.rs:11:1\n   |\nLL | const A: isize = Foo::B as isize;\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nnote: ...which then requires computing layout of `Foo`...\n  --> /checkout/src/test/ui/issue-36163.rs:11:18\n   |\nLL | const A: isize = Foo::B as isize;\n   |                  ^^^^^^\n   = note: ...which then again requires const-evaluating `Foo::B::{{initializer}}`, completing the cycle.\n\n"}
[00:40:53] {"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to previous error\n\n"}
[00:40:53] {"message":"For more information about this error, try `rustc --explain E0391`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0391`.\n"}
---
[00:40:53] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-3.9/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zmiri -Zunstable-options " "--target-rustcflags" "-Crpath -O -Zmiri -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "3.9.1\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[00:40:53] expected success, got: exit code: 101
[00:40:53]
[00:40:53]
[00:40:53] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[00:40:53] Build completed unsuccessfully in 0:01:42
[00:40:53] make: *** [check] Error 1
[00:40:53] Makefile:58: recipe for target 'check' failed
---
$ ls -lat $HOME/Library/Logs/DiagnosticReports/
ls: cannot access /home/travis/Library/Logs/DiagnosticReports/: No such file or directory
travis_time:end:021b3be9:start=1523643112757680693,finish=1523643112764024147,duration=6343454
travis_fold:end:after_failure.2
travis_fold:start:after_failure.3
travis_time:start:0ebffe27
$ find $HOME/Library/Logs/DiagnosticReports -type f -name '*.crash' -not -name '*.stage2-*.crash' -not -name 'com.apple.CoreSimulator.CoreSimulatorService-*.crash' -exec printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" {} \; -exec head -750 {} \; -exec echo travis_fold":"end:crashlog \; || true
find: `/home/travis/Library/Logs/DiagnosticReports': No such file or directory
travis_time:end:0ebffe27:start=1523643112769610858,finish=1523643112775245414,duration=5634556
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:173cd2b8
$ dmesg | grep -i kill
[   10.180488] init: failsafe main process (1093) killed by TERM signal
