plain
travis_time:start:test_ui
Check compiletest suite=ui mode=ui (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
[00:40:44] 
[00:40:44] running 1552 tests
[00:40:49] .........................................................................F...F....................i.
[00:41:00] ....................................................................................................
[00:41:04] ....................................................................................................
[00:41:08] ....................................................................................................
[00:41:11] ....................................................................................................
---
[00:41:44] ......................i.............................................................................
[00:41:49] ....................................................................................................
[00:41:55] ....................................................................................................
[00:42:03] ..................................................................i.................................
e":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/chalkify/lower_env1.rs","byte_start":529,"byte_end":558,"line_start":16,"line_end":16,"column_start":1,"column_end":30,"is_primary":true,"text":[{"text":"#[rustc_dump_program_clauses] //~ ERROR program clause dump","highlight_start":1,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"FromEnv(Self: Foo) :- FromEnv(Self: Bar).","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"FromEnv(Self: Foo) :- FromEnv(Self: Bar).","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"Implemented(Self: Bar) :- FromEnv(Self: Bar).","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"WellFormed(Self: Bar) :- Implemented(Self: Bar), WellFormed(Self: Foo), WellFormed(Self: Foo).","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: program clause dump\n  --> /checkout/src/test/ui/chalkify/lower_env1.rs:16:1\n   |\nLL | #[rustc_dump_program_clauses] //~ ERROR program clause dump\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = note: FromEnv(Self: Foo) :- FromEnv(Self: Bar).\n   = note: FromEnv(Self: Foo) :- FromEnv(Self: Bar).\n   = note: Implemented(Self: Bar) :- FromEnv(Self: Bar).\n   = note: WellFormed(Self: Bar) :- Implemented(Self: Bar), WellFormed(Self: Foo), WellFormed(Self: Foo).\n\n"}
[00:42:05] {"message":"program clause dump","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/chalkify/lower_env1.rs","byte_start":620,"byte_end":653,"line_Sat, 07 Jul 2018 03:47:33 GMT

The command "date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
" exited with 0.
travis_fold:start:after_failure.1
