\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/stable/book/references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/get_default.rs","byte_start":1610,"byte_end":1632,"line_start":51,"line_end":51,"column_start":17,"column_end":39,"is_primary":true,"text":[{"text":"                map.set(String::new()); // Ideally, just AST would error here","highlight_start":17,"highlight_end":39}],"label":"mutable borrow occurs here","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/get_default.rs","byte_start":1314,"byte_end":1317,"line_start":43,"line_end":43,"column_start":15,"column_end":18,"is_primary":false,"text":[{"text":"        match map.get() {","highlight_start":15,"highlight_end":18}],"label":"immutable borrow occurs here","suggested_replacement":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the anonymous lifetime #1 defined on the function body at 41:1...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/nll/get_default.rs","byte_start":1254,"byte_end":1813,"line_start":41,"line_end":57,"column_start":1,"column_end":2,"is_primary":true,"text":[{"text":"fn err(map: &mut Map) -> &String {","highlight_start":1,"highlight_end":35},{"text":"    loop {","highlight_start":1,"highlight_end":11},{"text":"        match map.get() {","highlight_start":1,"highlight_end":26},{"text":"            Some(v) => {","highlight_start":1,"highlight_end":25},{"text":"                map.set(String::new()); // Both AST and MIR error here","highlight_start":1,"highlight_end":71},{"text":"                //~^ ERROR borrowed as immutable (Mir)","highlight_start":1,"highlight_end":55},{"text":"                //~| ERROR borrowed as immutable (Ast)","highlight_start":1,"highlight_end":55},{"text":"                return v;","highlight_start":1,"highlight_end":26},{"text":"            }","highlight_start":1,"highlight_end":14},{"text":"            None => {","highlight_start":1,"highlight_end":22},{"text":"                map.set(String::new()); // Ideally, just AST would error here","highlight_start":1,"highlight_end":78},{"text":"                //~^ ERROR borrowed as immutable (Ast)","highlight_start":1,"highlight_end":55},{"text":"                //~| ERROR borrowed as immutable (Mir)","highlight_start":1,"highlight_end":55},{"text":"            }","highlight_start":1,"highlight_end":14},{"text":"        }","highlight_start":1,"highlight_end":10},{"text":"    }","highlight_start":1,"highlight_end":6},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0502]: cannot borrow `*map` as mutable because it is also borrowed as immutable (Mir)\n  --> /checkout/src/test/ui/nll/get_default.rs:51:17\n   |\nLL |         match map.get() {\n   |               --- immutable borrow occurs here\n...\nLL |                 map.set(String::new()); // Ideally, just AST would error here\n   |                 ^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here\n   |\nnote: borrowed value must be valid for the anonymous lifetime #1 defined on the function body at 41:1...\n  --> /checkout/src/test/ui/nll/get_default.rs:41:1\n   |\nLL | / fn err(map: &mut Map) -> &String {\nLL | |     loop {\nLL | |         match map.get() {\nLL | |             Some(v) => {\n...  |\nLL | |     }\nLL | | }\n   | |_^\n\n"}
[00:47:19] {"message":"aborting due to 6 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 6 previous errors\n\n"}
[00:47:19] {"message":"For more information about this error, try `rustc --explain E0502`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0502`.\n"}
[00:47:19] ------------------------------------------
[00:47:19] 
[00:47:19] thread '[ui (nll)] ui/nll/get_default.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3033:9
[00:47:19] note: Run with `RUST_BACKTRACE=1` for a backtrace.
---
[00:47:19] test result: FAILED. 1378 passed; 1 failed; 7 ignored; 0 measured; 0 filtered out
[00:47:19] 
[00:47:19] 
[00:47:19] 
[00:47:19] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-3.9/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options " "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "3.9.1\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always" "--compare-mode" "nll"
[00:47:19] 
[00:47:19] 
[00:47:19] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[00:47:19] Build completed unsuccessfully in 0:03:47
[00:47:19] Build completed unsuccessfully in 0:03:47
[00:47:19] make: *** [check] Error 1
[00:47:19] Makefile:58: recipe for target 'check' failed

The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:0f0095a0
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
