plain
[00:43:59] ....................................................................................................
[00:44:03] ....................................................................................................
[00:44:08] ....................................................................................................
[00:44:14] ....................................................................................................
[00:44:19] ...................................................................F................................
[00:44:30] ...............i....................................................................................
[00:44:35] ..................................ii................................................................
[00:44:35] ..................................ii................................................................
[00:44:41] ..................F.................................................................................
n/rustc" "/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.stage2-x86_64-unknown-linux-gnu" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.stage2-x86_64-unknown-linux-gnu.aux" "-A" "unused"
[00:44:48] ------------------------------------------
[00:44:48] 
[00:44:48] ------------------------------------------
[00:44:48] stderr:
[00:44:48] stderr:
[00:44:48] ------------------------------------------
[00:44:48] {"message":"unused variable: `i_think_continually`","code":{"code":"unused_variables","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":823,"byte_end":842,"line_start":32,"line_end":32,"column_start":9,"column_end":28,"is_primary":true,"text":[{"text":"    let i_think_continually = 2;","highlight_start":9,"highlight_end":28}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":557,"byte_end":563,"line_start":16,"line_end":16,"column_start":9,"column_end":"expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unused variable: `unused_var`\n  --> /checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs:41:19\n   |\nLL |     let (mut var, unused_var) = (1, 2);\n   |                   ^^^^^^^^^^ help: consider using `_unused_var` instead: `_unused_var`\n\n"}
[00:44:48] {"message":"unused variable: `corridors_of_light`","code":{"code":"unused_variables","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1103,"byte_end":1121,"line_start":43,"line_end":43,"column_start":26,"column_end":44,"is_primary":true,"text":[{"text":"    if let SoulHistory { corridors_of_light,","highlight_start":26,"highlight_end":44}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[{"message":"try ignoring the field","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1103,"byte_end":1121,"line_start":43,"line_end":43,"column_start":26,"column_end":44,"is_primary":true,"text":[{"text":"    if let SoulHistory { corridors_of_light,","highlight_start":26,"highlight_end":44}],"label":null,"suggested_replacement":"corridors_of_light: _","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unused variable: `corridors_of_light`\n  --> /checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs:43:26\n   |\nLL |     if let SoulHistory { corridors_of_light,\n   |                          ^^^^^^^^^^^^^^^^^^ help: try ignoring the field: `corridors_of_light: _`\n\n"}
[00:44:48] {"message":"variable `hours_are_suns` is assigned to, but never used","code":{"code":"unused_variables","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1152,"byte_end":1166,"line_start":44,"line_end":44,"column_start":30,"column_end":44,"is_primary":true,"text":[{"text":"                         mut hours_are_suns,","highlight_start":30,"highlight_end":44}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[{"message":"consider using `_hours_are_suns` instead","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: variable `hours_are_suns` is assigned to, but never used\n  --> /checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs:44:30\n   |\nLL |                          mut hours_are_suns,\n   |                              ^^^^^^^^^^^^^^\n   |\n   = note: consider using `_hours_are_suns` instead\n\n"}
[00:44:48] {"message":"value assigned to `hours_are_suns` is never read","code":{"code":"unused_assignments","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1262,"byte_end":1276,"line_start":46,"line_end":46,"column_start":9,"column_end":23,"is_primary":true,"text":[{"text":"        hours_are_suns = false;","highlight_start":9,"highlight_end":23}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":ile_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1754,"byte_end":1758,"line_start":75,"line_end":75,"column_start":24,"column_end":28,"is_primary":true,"text":[{"text":"        [Large::Suit { case }] => {}","highlight_start":24,"highlight_end":28}],"label":null,"suggested_replacement":"case: _","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unused variable: `case`\n  --> /checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs:75:24\n   |\nLL |         [Large::Suit { case }] => {}\n   |                        ^^^^ help: try ignoring the field: `case: _`\n\n"}
[00:44:48] {"message":"unused variable: `case`","code":{"code":"unused_variables","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1863,"byte_end":1867,"line_start":80,"line_end":80,"column_start":29,"column_end":33,"is_primary":true,"text":[{"text":"        Tuple(Large::Suit { case }, ()) => {}","highlight_start":29,"highlight_end":33}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[{"message":"try ignoring the field","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1863,"byte_end":1867,"line_start":80,"line_end":80,"column_start":29,"column_end":33,"is_primary":true,"text":[{"text":"        Tuple(Large::Suit { case }, ()) => {}","highlight_start":29,"highlight_end":33}],"label":null,"suggested_replacement":"case: _","expansion":null}],"children":[],"rendered"text":[{"text":"fn main() { //~ ERROR compilation successful","highlight_start":1,"highlight_end":45},{"text":"    let theTwo = 2; //~ WARN should have a snake case name","highlight_start":1,"highlight_end":59},{"text":"    let theOtherTwo = 2; //~ WARN should have a snake case name","highlight_start":1,"highlight_end":64},{"text":"    //~^ WARN unused variable","highlight_start":1,"highlight_end":30},{"text":"    println!(\"{}\", theTwo);","highlight_start":1,"highlight_end":28},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error: compilation successful\n  --> /checkout/src/test/ui/span/issue-24690.rs:21:1\n   |\nLL | / fn main() { //~ ERROR compilation successful\nLL | |     let theTwo = 2; //~ WARN should have a snake case name\nLL | |     let theOtherTwo = 2; //~ WARN should have a snake case name\nLL | |     //~^ WARN unused variable\nLL | |     println!(\"{}\", theTwo);\nLL | | }\n   | |_^\n\n"}
[00:44:48] {"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to previous error\n\n"}
[00:44:48] ------------------------------------------
[00:44:48] 
[00:44:48] thread '[ui] ui/span/issue-24690.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3033:9
[00:44:48] 
---
[00:44:48] 
[00:44:48] thread 'main' panicked at 'Some tests failed', tools/compiletest/src/main.rs:492:22
[00:44:48] 
[00:44:48] 
[00:44:48] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-3.9/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options " "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "3.9.1\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[00:44:48] 
[00:44:48] 
[00:44:48] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[00:44:48] Build completed unsuccessfully in 0:02:26
[00:44:48] Build completed unsuccessfully in 0:02:26
[00:44:48] make: *** [check] Error 1
[00:44:48] Makefile:58: recipe for target 'check' failed
75492 ./obj/build/x86_64-unknown-linux-gnu/stage0-tools/release
71536 ./.git/modules/src/tools
70976 ./obj/build/x86_64-unknown-linux-gnu/native
70716 ./obj/build/x86_64-unknown-linux-gnu/stage0-rustc
70716 ./obj/build/x86_64-unknown-linux-gnu/stage0-rustc
70300 ./obj/build/x86_64-unknown-linux-gnu/native/jemalloc
68788 ./src/llvm/lib
65416 ./src/llvm-emscripten/test/CodeGen
60840 ./src/llvm-emscripten/lib
56088 ./obj/build/x86_64-unknown-linux-gnu/stage0/lib/rustlib/x86_64-unknown-linux-gnu/bin
55448 ./obj/build/x86_64-unknown-linux-gnu/stage0-rustc/release
53648 ./obj/build/x86_64-unknown-linux-gnu/stage1-rustc/x86_64-unknown-linux-gnu/release/incremental/syntax-1q0trsiw2bs8m
53644 ./obj/build/x86_64-unknown-linux-gnu/stage1-rustc/x86_64-unknown-linux-gnu/release/incremental/syntax-1q0trsiw2bs8m/s-f0zi6d8c75-6us3yb-28ws1xvt2npoa
48604 ./obj/build/x86_64-unknown-linux-gnu/stage0/bin
47984 ./obj/build/x86_64-unknown-linux-gnu/stage0-std
47236 ./src/test
46812 ./obj/build/x86_64-unknown-linux-gnu/stage1-std/release
