\n\nRust only looks at the signature of the called function, as such it must\nalready specify all requirements that will be used for every type parameter.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-trivial_bounds.rs","byte_start":1309,"byte_end":1351,"line_start":62,"line_end":62,"column_start":1,"column_end":43,"is_primary":true,"text":[{"text":"struct TwoStrs(str, str) where str: Sized; //~ ERROR","highlight_start":1,"highlight_end":43}],"label":"`str` does not have a constant size known at compile-time","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the trait `std::marker::Sized` is not implemented for `str`","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"see issue #48214","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"add #![feature(trivial_bounds)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied\n  --> /checkout/src/test/ui/feature-gate-trivial_bounds.rs:62:1\n   |\nLL | struct TwoStrs(str, str) where str: Sized; //~ ERROR\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `str` does not have a constant size known at compile-time\n   |\n   = help: the trait `std::marker::Sized` is not implemented for `str`\n   = help: see issue #48214\n   = help: add #![feature(trivial_bounds)] to the crate attributes to enable\n\n"}
[01:00:06] {"message":"the trait bound `A + 'static: std::marker::Sized` is not satisfied in `Dst<A + 'static>`","code":{"code":"E0277","explanation":"\nYou tried to use a type which doesn't implement some trait in a place which\nexpected that trait. Erroneous code example:\n\n