\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":2973,"byte_end":3003,"line_start":55,"line_end":55,"column_start":5,"column_end":35,"is_primary":true,"text":[{"text":"    extern \"vectorcall\" fn m3() {} //~ ERROR vectorcall is experimental and subject to change","highlight_start":5,"highlight_end":35}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_vectorcall)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: vectorcall is experimental and subject to change\n  --> /checkout/src/test/ui/feature-gate-abi.rs:55:5\n   |\nLL |     extern \"vectorcall\" fn m3() {} //~ ERROR vectorcall is experimental and subject to change\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_vectorcall)] to the crate attributes to enable\n\n"}
[00:43:58] {"message":"rust-call ABI is subject to change (see issue #29625)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n