\n"},"level":"error","spans":[{"file_name":"/checkout/src/teste-abi.rs","byte_start":2883,"byte_end":2921,"line_start":54,"line_end":54,"column_start":5,"column_end":43,"is_primary":true,"text":[{"text":"    extern \"platform-intrinsic\" fn m2() {} //~ ERROR platform intrinsics are experimental","highlight_start":5,"highlight_end":43}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(platform_intrinsics)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: platform intrinsics are experimental and possibly buggy (see issue #27731)\n  --> /checkout/src/test/ui/feature-gate-abi.rs:54:5\n   |\nLL |     extern \"platform-intrinsic\" fn m2() {} //~ ERROR platform intrinsics are experimental\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(platform_intrinsics)] to the crate attributes to enable\n\n"}
[00:43:58] {"message":"vectorcall is experimental and subject to change","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n