\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":898,"byte_end":927,"line_start":22,"line_end":22,"column_start":1,"column_end":30,"is_primary":true,"text":[{"text":"extern \"rust-call\" fn f4() {} //~ ERROR rust-call ABI is sub ABI is experimental","highlight_start":1,"highlight_end":37}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_msp430_interrupt)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: msp430-interrupt ABI is experimental and subject to change (see issue #38487)\n  --> /checkout/src/test/ui/feature-gate-abi.rs:23:1\n   |\nLL | extern \"msp430-interrupt\" fn f5() {} //~ ERROR msp430-interrupt ABI is experimental\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_msp430_interrupt)] to the crate attributes to enable\n\n"}
[00:43:58] {"message":"PTX ABIs are experimental and subject to change (see issue #38788)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n