plain
[00:50:00] ....................................................................................................
[00:50:05] ....................................................................................................
[00:50:10] ....................................................................................i...............
[00:50:15] .............................................................i......................................
[00:50:20] ................................................................F...................................
[00:50:25] .....................F..............................................................................
[00:50:32] ..........................................................................................i......F..
   } else { //~ ERROR: incorrect close delimiter: `}`","highlight_start":9,"highlight_end":10}],"label":"unexpected token","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: expected one of `.`, `;`, `?`, `}`, or an operator, found `)`\n  --> /checkout/src/test/ui/resolve/token-error-correct-3.rs:30:9\n   |\nLL |             fs::create_dir_all(path.as_ref()).map(|()| true) //~ ERROR: mismatched types\n   |                                                             - expected one of `.`, `;`, `?`, `}`, or an operator here\n...\nLL |         } else { //~ ERROR: incorrect close delimiter: `}`\n   |         ^ unexpected token\n\n"}
[00:50:34] {"message":"cannot find function `is_directory` in this scope","code":{"code":"E0425","explanation":"\nAn unresolved name was used.\n\nErroneous code examples:\n\n