\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":1871,"byte_end":1902,"line_start":36,"line_end":36,"column_start":5,"column_end":36,"is_primary":true,"text":[{"text":"    extern \"x86-interrupt\" fn m7(); //~ ERROR x86-interrupt ABI is experimental","highlight_start":5,"highlight_end":36}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_x86_interrupt)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: x86-interrupt ABI is experimental and subject to change (see issue #40180)\n  --> /checkout/src/test/ui/feature-gate-abi.rs:36:5\n   |\nLL |     extern \"x86-interrupt\" fn m7(); //~ ERROR x86-interrupt ABI is experimental\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_x86_interrupt)] to the crate attributes to enable\n\n"}
[00:43:58] {"message":"thiscall is experimental and subject to change","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n