\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":1780,"byte_end":1808,"line_start":35,"line_end":35,"column_start":5,"column_end":33,"is_primary":true,"text":[{"text":"    extern \"ptx-kernel\" fn m6(); //~ ERROR PTX ABIs are experimental and subject to change","highlight_start":5,"highlight_end":33}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_ptx)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: PTX ABIs are experimental and subject to change (see issue #38788)\n  --> /checkout/src/test/ui/feature-gate-abi.rs:35:5\n   |\nLL |     extern \"ptx-kernel\" fn m6(); //~ ERROR PTX ABIs are experimental and subject to change\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_ptx)] to the crate attributes to enable\n\n"}
[00:43:58] {"message":"x86-interrupt ABI is experimental and subject to change (see issue #40180)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n