\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1683,"byte_end":1694,"line_start":54,"line_end":54,"column_start":13,"column_end":24,"is_primary":true,"text":[{"text":"    let _ = E::A as f32; //~ ERROR is invalid","highlight_start":13pansion":null}],"children":[],"rendered":"error[E0606]: casting `&dyn Foo` as `*const str` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:64:13\n   |\nLL |     let _ = foo as *const str; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^^^^^^\n\n"}
[00:48:08] {"message":"casting `&dyn Foo` as `*mut str` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code example:\n\n