\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":533,"byte_end":546,"line_start":13,"line_end":13,"column_start":5,"column_end":18,"is_primary":true,"text":[{"text":"    u as *const V //~ ERROR is invalid","highlight_start":5,"highlight_end":18}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"vtable kinds may not match","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0606]: casting `*const U` as `*const V` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:13:5\n   |\nLL |     u as *const V //~ ERROR is invalid\n   |     ^^^^^^^^^^^^^\n   |\n   = note: vtable kinds may not match\n\n"}
[00:48:08] {"message":"casting `*const U` as `*const str` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code example:\n\n