\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1581,"byte_end":1596,"line_start":51,"line_end":51,"column_start":13,"column_end":28,"is_primary":true,"text":[{"text":"    let _ = 0x61u32 as char; //~ ERROR can be cast as","highlight_start":13,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0604]: only `u8` can be cast as `char`, not `u32`\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:51:13\n   |\nLL |     let _ = 0x61u32 as char; //~ ERROR can be cast as\n   |             ^^^^^^^^^^^^^^^\n\n"}
[00:48:08] {"message":"casting `bool` as `f32` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code example:\n\n