\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1636,"byte_end":1648,"line_start":53,"line_end":53,"column_start":13,"column_end":25,"is_primary":true,"text":[{"text":"    let _ = false as f32; //~ ERROR is invalid","highlight_start":13,"highlight_end":25}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"cast through an integer first","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0606]: casting `bool` as `f32` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:53:13\n   |\nLL |     let _ = false as f32; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^\n   |\n   = help: cast through an integer first\n\n"}
[00:48:08] {"message":"casting `E` as `f32` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code example:\n\n