\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":2371,"byte_end":2386,"line_start":69,"line_end":69,"column_start":13,"column_end":28,"is_primary":true,"text":[{"text":"    let _ = fat_sv as usize; //~ ERROR is invalid","highlight_start":13,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"cast through a thin pointer first","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0606]: casting `*const [i8]` as `usize` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:69:13\n   |\nLL |     let _ = fat_sv as usize; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^^^^\n   |\n   = help: cast through a thin pointer first\n\n"}
[00:48:08] {"message":"casting `*const dyn Foo` as `*const [u16]` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code example:\n\n