\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":2643,"byte_end":2661,"line_start":78,"line_end":78,"column_start":13,"column_end":31,"is_primary":true,"text":[{"text":"    let _ = cf as *const [u16]; //~ ERROR is invalid","highlight_start":13,"highlight_end":31}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"vtable kinds may not match","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0606]: casting `*const dyn Foo` as `*const [u16]` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:78:13\n   |\nLL |     let _ = cf as *const [u16]; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^^^^^^^\n   |\n   = note: vtable kinds may not match\n\n"}
[00:48:08] {"message":"casting `*const dyn Foo` as `*const dyn Bar` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code example:\n\n