\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1533,"byte_end":1545,"line_start":50,"line_end":50,"column_start":13,"column_end":25,"is_primary":true,"text":[{"text":"    let _ = E::A as bool; //~ ERROR cannot cast","highlight_start":13,"highlight_end":25}],"label":"unsupported cast","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"compare with zero instead","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0054]: cannot cast as `bool`\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:50:13\n   |\nLL |     let _ = E::A as bool; //~ ERROR cannot cast\n   |             ^^^^^^^^^^^^ unsupported cast\n   |\n   = help: compare with zero instead\n\n"}
[00:48:08] {"message":"only `u8` can be cast as `char`, not `u32`","code":{"code":"E0604","explanation":"\nA cast to `char` was attempted on a type other than `u8`.\n\nErroneous code example:\n\n