\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1300,"byte_end":1308,"line_start":45,"line_end":45,"column_start":13,"column_end":21,"is_primary":true,"text":[{"text":"    let _ = v as f32; //~ ERROR is invalid","highlight_start":13,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0606]: casting `*const u8` as `f32` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:45:13\n   |\nLL |     let _ = v as f32; //~ ERROR is invalid\n   |             ^^^^^^^^\n\n"}
[00:48:08] {"message":"casting `fn() {main}` as `f64` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code ecast-rfc0401.rs","byte_start":1435,"byte_end":1449,"line_start":48,"line_end":48,"column_start":13,"column_end":27,"is_primary":true,"text":[{"text":"    let _ = f as *const u8; //~ ERROR is invalid","highlight_start":13,"highlight_end":27}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0606]: casting `f32` as `*const u8` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:48:13\n   |\nLL |     let _ = f as *const u8; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^^^\n\n"}
[00:48:08] {"message":"cannot cast as `bool`","code":{"code":"E0054","explanation":"\nIt is not allowed to cast to a bool. If you are trying to cast a numeric type\nto a bool, you can compare it with zero instead:\n\n