\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1484,"byte_end":1497,"line_start":49,"line_end":49,"column_start":13,"column_end":26,"is_primary":true,"text":[{"text":"    let _ = 3_i32 as bool; //~ ERROR cannot cast","highlight_start":13,"highlight_end":26}],"label":"unsupported cast","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"compare with zero instead","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0054]: cannot cast as `bool`\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:49:13\n   |\nLL |     let _ = 3_i32 as bool; //~ ERROR cannot cast\n   |             ^^^^^^^^^^^^^ unsupported cast\n   |\n   = help: compare with zero instead\n\n"}
[00:48:08] {"message":"cannot cast as `bool`","code":{"code":"E0054","explanation":"\nIt is not allowed to cast to a bool. If you are trying to cast a numeric type\nto a bool, you can compare it with zero instead:\n\n