\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":2170,"byte_end":2185,"line_start":65,"line_end":65,"column_start":13,"column_end":28,"is_primary":true,"text":[{"text":"    let _ = foo as *mut str; //~ ERROR is invalid","highlight_start":13,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0606]: casting `&dyn Foo` as `*mut str` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:65:13\n   |\nLL |     let _ = foo as *mut str; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^^^^\n\n"}
[00:48:08] {"message":"casting `fn() {main}` as `*mut str` is invalend":2336,"line_start":68,"line_end":68,"column_start":13,"column_end":29,"is_primary":true,"text":[{"text":"    let _ = &f as *const f64; //~ ERROR is invalid","highlight_start":13,"highlight_end":29}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0606]: casting `&f32` as `*const f64` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:68:13\n   |\nLL |     let _ = &f as *const f64; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^^^^^\n\n"}
[00:48:08] {"message":"casting `*const [i8]` as `usize` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code example:\n\n