\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-31424.rs","byte_start":791,"byte_end":795,"line_start":23,"line_end":23,"column_start":15,"column_end":19,"is_primary":true,"text":[{"text":"        (&mut self).bar(); //~ ERROR cannot borrow","highlight_start":15,"highlight_end":19}],"label":"cannot borrow mutably","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider removing the `&mut`, as it is an immutable binding to a mutable reference","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-31424.rs","byte_start":785,"byte_end":796,"line_start":23,"line_end":23,"column_start":9,"column_end":20,"is_primary":true,"text":[{"text":"        (&mut self).bar(); //~ ERROR cannot borrow","highlight_start":9,"highlight_end":20}],"label":null,"suggested_replacement":"self: &mut Self","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0596]: cannot borrow immutable argument `self` as mutable\n  --> /checkout/src/test/ui/did_you_mean/issue-31424.rs:23:15\n   |\nLL |         (&mut self).bar(); //~ ERROR cannot borrow\n   |               ^^^^ cannot borrow mutably\nhelp: consider removing the `&mut`, as it is an immutable binding to a mutable reference\n   |\nLL |         self: &mut Self.bar(); //~ ERROR cannot borrow\n   |         ^^^^^^^^^^^^^^^\n\n"}
[00:43:19] {"message":"aborting due to 2 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 2 previous errors\n\n"}
[00:43:19] {"message":"For more information about this error, try `rustc --explain E0596`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0596`.\n"}
[00:43:19] ------------------------------------------
[00:43:19] 
[00:43:19] thread '[ui] ui/did_you_mean/issue-31424.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3096:9
[00:43:19] 
---
[00:43:19] test result: FAILED. 1465 passed; 2 failed; 14 ignored; 0 measured; 0 filtered out
[00:43:19] 
[00:43:19] 
[00:43:19] 
[00:43:19] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-3.9/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options " "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "3.9.1\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" ./.git/modules/src/libcompiler_builtins
32156 ./.git/modules/src/libcompiler_builtins/modules/compiler-rt
31728 ./.git/modules/src/libcompiler_builtins/modules/compiler-rt/objects
31720 ./.git/modules/src/libcompiler_builtins/modules/compiler-rt/objects/pack
31532 ./src/libcompiler_builtins/compiler-rt/test
