\n\nIf the item you are importing is not defined in some super-module of the\ncurrent module, then it must also be declared as public (e.g., `pub fn`).\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/hygiene/globs.rs","byte_start":734,"byte_end":735,"line_start":32,"line_end":32,"column_start":9,"column_end":10,"is_primary":true,"text":[{"text":"        f(); //~ ERROR cannot find function `f` in this scope","highlight_start":9,"highlight_end":10}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0425]: cannot find function `f` in this scope\n  --> /checkout/src/test/ui/hygiene/globs.rs:32:9\n   |\nLL |         f(); //~ ERROR cannot find function `f` in this scope\n   |         ^ not found in this scope\n\n"}
[00:49:33] {"message":"cannot find function `g` in this scope","code":{"code":"E0425","explanation":"\nAn unresolved name was used.\n\nErroneous code examples:\n\n