\n\nIf the item you are importing is not defined in some super-module of the\ncurrent module, then it must also be declared as public (e.g., `pub fn`).\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/hygiene/globs.rs","byte_start":615,"byte_end":616,"line_start":25,"line_end":25,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    g(); //~ ERROR cannot find function `g` in this scope","highlight_start":5,"highlight_end":6}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/hygiene/globs.rs","byte_start":688,"byte_end":793,"line_start":29,"line_end":33,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    m! {","highlight_start":5,"highlight_end":9},{"text":"        use bar::*;","highlight_start":1,"highlight_end":20},{"text":"        g();","highlight_start":1,"highlight_end":13},{"text":"        f(); //~ ERROR cannot find function `f` in this scope","highlight_start":1,"highlight_end":62},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"m!","def_site_span":{"file_name":"/checkout/src/test/ui/hygiene/globs.rs","byte_start":554,"byte_end":670,"line_start":21,"line_end":26,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro m($($t:tt)*) {","highlight_start":1,"highlight_end":21},{"text":"    $($t)*","highlight_start":1,"highlight_end":11},{"text":"    use foo::*;","highlight_start":1,"highlight_end":16},{"text":"    f();","highlight_start":1,"highlight_end":9},{"text":"    g(); //~ ERROR cannot find function `g` in this scope","highlight_start":1,"highlight_end":58},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[],"rendered":"error[E0425]: cannot find function `g` in this scope\n  --> /checkout/src/test/ui/hygiene/globs.rs:25:5\n   |\nLL |       g(); //~ ERROR cannot find function `g` in this scope\n   |       ^ not found in this scope\n...\nLL | /     m! {\nLL | |         use bar::*;\nLL | |         g();\nLL | |         f(); //~ ERROR cannot find function `f` in this scope\nLL | |     }\n   | |_____- in this macro invocation\n\n"}
[00:49:33] {"message":"cannot find function `f` in this scope","code":{"code":"E0425","explanation":"\nAn unresolved name was used.\n\nErroneous code examples:\n\n