\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issue-38293.rs","byte_start":589,"byte_end":593,"line_start":16,"line_end":16,"column_start":14,"column_end":18,"is_primary":true,"text":[{"text":"use foo::f::{self}; //~ ERROR unresolved import `foo::f`","highlight_start":14,"highlight_end":18}],"label":"no `f` in `foo`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0432]: unresolved import `foo::f`\n  --> /checkout/src/test/ui/issue-38293.rs:16:14\n   |\nLL | use foo::f::{self}; //~ ERROR unresolved import `foo::f`\n   |              ^^^^ no `f` in `foo`\n\n"}
[00:49:33] {"message":"expected function, found module `baz`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n