\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1775,"byte_end":1793,"line_start":57,"line_end":57,"column_start":13,"column_end":31,"is_primary":true,"text":[{"text":"    let _ = false as *const u8; //~ ERROR is invalid","highlight_start":13,"highlight_end":31}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0606]: casting `bool` as `*const u8` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:57:13\n   |\nLL |     let _ = false as *const u8; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^^^^^^^\n\n"}
[00:46:20] {"message":"casting `E` as `*const u8` is invalid","code":{"code":"E0606","explanation":"\nAn incompatible cast was attempted.\n\nErroneous code example:\n\n