\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1932,"byte_end":1954,"line_start":61,"line_end":61,"column_start":13,"column_end":35,"is_primary":true,"text":[{"text":"    let _ = 42usize as *const [u8]; //~ ERROR is invalid","highlight_start":13,"highlight_end":35}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0606]: casting `usize` as `*const [u8]` is invalid\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:61:13\n   |\nLL |     let _ = 42usize as *const [u8]; //~ ERROR is invalid\n   |             ^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:46:20] {"message":"cannot cast thin pointer `*const u8` to fat pointer `*invalid\n