plain
[00:45:31] ....................................................................................................
[00:45:36] ....................................................................................................
[00:45:41] ....................................................................................................
[00:45:46] ....................................................................................................
[00:45:52] ....................................................................................F...............
[00:46:02] ....................................................................................................
[00:46:08] ....................................................................................................
[00:46:14] ....................................................................................................
[00:46:14] ....................................................................................................
st-rfc0401.rs","byte_start":1080,"byte_end":1086,"line_start":40,"line_end":40,"column_start":13,"column_end":19,"is_primary":true,"text":[{"text":"    let _ = v as E; //~ ERROR non-primitive cast","highlight_start":13,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"an `as` expression can only be used to convert between primitive types. Consider using the `From` trait","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0605]: non-primitive cast: `*const u8` as `E`\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:40:13\n   |\nLL |     let _ = v as E; //~ ERROR non-primitive cast\n   |             ^^^^^^\n   |\n   = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait\n\n"}
[00:46:20] {"message":"non-primitive cast: `*const u8` as `fn()`","code":{"code":"E0605","explanation":"\nAn invalid cast was attempted.\n\nErroneous code examples:\n\n