\n\nFor more information about casts, take a look at The Book:\nhttps://doc.rust-lang.org/book/first-edition/casting-between-types.html\n"},"level":"error","spans":[{"file_name":"/checkout/el":"error","spans":[{"file_name":"/checkout/src/test/ui/mismatched_types/cast-rfc0401.rs","byte_start":1181,"byte_end":1192,"line_start":42,"line_end":42,"column_start":13,"column_end":24,"is_primary":true,"text":[{"text":"    let _ = v as (u32,); //~ ERROR non-primitive cast","highlight_start":13,"highlight_end":24}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"an `as` expression can only be used to convert between primitive types. Consider using the `From` trait","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0605]: non-primitive cast: `*const u8` as `(u32,)`\n  --> /checkout/src/test/ui/mismatched_types/cast-rfc0401.rs:42:13\n   |\nLL |     let _ = v as (u32,); //~ ERROR non-primitive cast\n   |             ^^^^^^^^^^^\n   |\n   = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait\n\n"}
[00:46:20] {"message":"non-primitive cast: `std::option::Option<&*const u8>` as `*const u8`","code":{"code":"E0605","explanation":"\nAn invalid cast was attempted.\n\nErroneous code examples:\n\n