\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/const-eval/infinite_loop.rs","byte_start":878,"byte_end":1016,"line_start":20,"line_end":22,"column_start":9,"column_end":10,"is_primary":true,"text":[{"text":"        while n != 0 { //~ ERROR constant contains unimplemented expression type","highlight_start":9,"highlight_end":81},{"text":"            n = if n % 2 == 0 { n/2 } else { 3*n + 1 };","highlight_start":1,"highlight_end":56},{"text":"        }","highlight_start":1,"highlight_end":10}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0019]: constant contains unimplemented expression type\n  --> /checkout/src/test/ui/const-eval/infinite_loop.rs:20:9\n   |\nLL | /         while n != 0 { //~ ERROR constant contains unimplemented expression type\nLL | |             n = if n % 2 == 0 { n/2 } else { 3*n + 1 };\nLL | |         }\n   | |_________^\n\n"}
[00:48:21] {"message":"Constant evaluating a complex constant, this might take some time","code":null,"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/const-eval/infinite_loop.rs","byte_start":666,"byte_end":1032,"line_start":16,"line_end":24,"column_start":18,"column_end":6,"is_primary":true,"text":[{"text":"    let _ = [(); {","highlight_start":18,"highlight_end":19},{"text":"        //~^ WARNING Constant evaluating a complex constant, this might take some time","highlight_start":1,"highlight_end":87},{"text":"        //~| ERROR could not evaluate repeat length","highlight_start":1,"highlight_end":52},{"text":"        let mut n = 113383; // #20 in https://oeis.org/A006884","highlight_start":1,"highlight_end":63},{"text":"        while n != 0 { //~ ERROR constant contains unimplemented expression type","highlight_start":1,"highlight_end":81},{"text":"            n = if n % 2 == 0 { n/2 } else { 3*n + 1 };","highlight_start":1,"highlight_end":56},{"text":"        }","highlight_start":1,"highlight_end":10},{"text":"        n","highlight_start":1,"highlight_end":10},{"text":"    }];","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"warning: Constant evaluating a complex constant, this might take some time\n  --> /checkout/src/test/ui/const-eval/infinite_loop.rs:16:18\n   |\nLL |       let _ = [(); {\n   |  __________________^\nLL | |         //~^ WARNING Constant evaluating a complex constant, this might take some time\nLL | |         //~| ERROR could not evaluate repeat length\nLL | |         let mut n = 113383; // #20 in https://oeis.org/A006884\n...  |\nLL | |         n\nLL | |     }];\n   | |_____^\n\n"}
[00:48:21] {"message":"could not evaluate repeat length","code":{"code":"E0080","explanation":"\nThis error indicates that the compiler was unable to sensibly evaluate an\nconstant expression that had to be evaluated. Attempting to divide by 0\nor causing integer overflow are two ways to induce this error. For example:\n\n