plain
[00:47:00] 
[00:47:00] running 1535 tests
[00:47:06] ................................................................................................i...
[00:47:13] .............................................................i......................................
[00:47:18] ...................................F................................................................
[00:47:25] ....................................................................................................
[00:47:28] ....................................................................................................
[00:47:34] ....................................................................................................
[00:47:38] ....................................................................................................
---
[00:48:21] ---- [ui] ui/const-eval/infinite_loop.rs stdout ----
[00:48:21] diff of stderr:
[00:48:21] 
[00:48:21] 13    |  __________________^
[00:48:21] 14 LL | |         //~^ WARNING Constant evaluating a complex constant, this might take some time
[00:48:21] 15 LL | |         //~| ERROR could not evaluate repeat length
[00:48:21] - LL | |         let mut n = 113383; // #20 in A006884
[00:48:21] + LL | |         let mut n = 113383; // #20 in https://oeis.org/A006884
[00:48:21] 17 ...  |
[00:48:21] 18 LL | |         n
[00:48:21] 19 LL | |     }];
[00:48:21] 26    |  __________________^
[00:48:21] 26    |  __________________^
[00:48:21] 27 LL | |         //~^ WARNING Constant evaluating a complex constant, this might take some time
[00:48:21] 28 LL | |         //~| ERROR could not evaluate repeat length
[00:48:21] - LL | |         let mut n = 113383; // #20 in A006884
[00:48:21] + LL | |         let mut n = 113383; // #20 in https://oeis.org/A006884
[00:48:21] 30 LL | |         while n != 0 { //~ ERROR constant contains unimplemented expression type
[00:48:21] 31 LL | |             n = if n % 2 == 0 { n/2 } else { 3*n + 1 };
[00:48:21] 32    | |                    ---------- n's value must be known at compile-time. Erroneous code\nexample:\n\n