\n\nIn this case, `foo` is defined, but is not a struct, so Rust can't use it as\none.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-async-await.rs","byte_start":583,"byte_end":588,"line_start":17,"line_end":17,"column_start":13,"column_end":18,"is_primary":true,"text":[{"text":"    let _ = async {}; //~ ERROR async blocks are unstable","highlight_start":13,"highlight_end":18}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0422]: cannot find struct, variant or union type `async` in this scope\n  --> /checkout/src/test/ui/feature-gate-async-await.rs:17:13\n   |\nLL |     let _ = async {}; //~ ERROR async blocks are unstable\n   |             ^^^^^ not found in this scope\n\n"}
[00:47:46] {"message":"cannot find value `async` in this scope","code":{"code":"E0425","explanation":"\nAn unresolved name was used.\n\nErroneous code examples:\n\n