\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/deref-suggestion.rs","byte_start":508,"byte_end":511,"line_start":12,"line_end":12,"column_start":20,"column_end":23,"is_primary":true,"text":[{"text":"    ($x:expr) => { &$x } //~ ERROR mismatched types","highlight_start":20,"highlight_end":23}],"label":"expected u32, found &{integer}","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/deref-suggestion.rs","byte_start":906,"byte_end":916,"line_start":32,"line_end":32,"column_start":10,"column_end":20,"is_primary":false,"text":[{"text":"    foo3(borrow!(0));","highlight_start":10,"highlight_end":20}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"borrow!","def_site_span":{"file_name":"/checkout/src/test/ui/deref-suggestion.rs","byte_start":467,"byte_end":542,"line_start":11,"line_end":13,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! borrow {","highlight_start":1,"highlight_end":22},{"text":"    ($x:expr) => { &$x } //~ ERROR mismatched types","highlight_start":1,"highlight_end":52},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[{"message":"expected type `u32`\n   found type `&{integer}`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"consider removing the borrow","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/deref-suggestion.rs","byte_start":508,"byte_end":511,"line_start":12,"line_end":12,"column_start":20,"column_end":23,"is_primary":true,"text":[{"text":"    ($x:expr) => { &$x } //~ ERROR mismatched types","highlight_start":20,"highlight_end":23}],"label":null,"suggested_replacement":"0","suggestion_applicability":"Unspecified","expansion":{"span":{"file_name":"/checkout/src/test/ui/deref-suggestion.rs","byte_start":906,"byte_end":916,"line_start":32,"line_end":32,"column_start":10,"column_end":20,"is_primary":false,"text":[{"text":"    foo3(borrow!(0));","highlight_start":10,"highlight_end":20}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"borrow!","def_site_span":{"file_name":"/checkout/src/test/ui/deref-suggestion.rs","byte_start":467,"byte_end":542,"line_start":11,"line_end":13,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! borrow {","highlight_start":1,"highlight_end":22},{"text":"    ($x:expr) => { &$x } //~ ERROR mismatched types","highlight_start":1,"highlight_end":52},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/deref-suggestion.rs:12:20\n   |\nLL |     ($x:expr) => { &$x } //~ ERROR mismatched types\n   |                    ^^^\n   |                    |\n   |                    expected u32, found &{integer}\n   |                    help: consider removing the borrow: `0`\n...\nLL |     foo3(borrow!(0));\n   |          ---------- in this macro invocation\n   |\n   = note: expected type `u32`\n              found type `&{integer}`\n\n"}
[00:48:07] {"message":"aborting due to 5 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 5 previous errors\n\n"}
[00:48:07] {"message":"For more information about this error, try `rustc --explain E0308`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0308`.\n"}
[00:48:07] ------------------------------------------
[00:48:07] 
[00:48:07] thread '[ui] ui/deref-suggestion.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3139:9
[00:48:07] note: Run with `RUST_BACKTRACE=1` for a backtrace.
---
travis_time:end:071f36d0:start=1530149885150194183,finish=1530149885158686354,duration=8492171
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:0304de1a
$ head -30 ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
head: cannot open ‘./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers’ for reading: No such file or directory
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:12325830
$ dmesg | grep -i kill
