\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/deref-suggestion.rs","byte_start":596,"byte_end":597,"line_start":18,"line_end":18,"column_start":9,"column_end":10,"is_primary":true,"text":[{"text":"    foo(s); //~ ERROR mismatched types","highlight_start":9,"highlight_end":10}],"label":"expected struct `std::string::String`, found reference","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `std::string::String`\n   found type `&std::string::String`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"try using a conversion method","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/deref-suggestion.rs","byte_start":596,"byte_end":597,"line_start":18,"line_end":18,"column_start":9,"column_end":10,"is_primary":true,"text":[{"text":"    foo(s); //~ ERROR mismatched types","highlight_start":9,"highlight_end":10}],"label":null,"suggested_replacement":"s.to_string()","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/deref-suggestion.rs:18:9\n   |\nLL |     foo(s); //~ ERROR mismatched types\n   |         ^\n   |         |\n   |         expected struct `std::string::String`, found reference\n   |         help: try using a conversion method: `s.to_string()`\n   |\n   = note: expected type `std::string::String`\n              found type `&std::string::String`\n\n"}
[00:48:07] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing express:"    foo(&\"aaa\".to_owned()); //~ ERROR mismatched types","highlight_start":9,"highlight_end":26}],"label":null,"suggested_replacement":"\"aaa\".to_owned()","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/deref-suggestion.rs:30:9\n   |\nLL |     foo(&\"aaa\".to_owned()); //~ ERROR mismatched types\n   |         ^^^^^^^^^^^^^^^^^\n   |         |\n   |         expected struct `std::string::String`, found reference\n   |         help: consider removing the borrow: `\"aaa\".to_owned()`\n   |\n   = note: expected type `std::string::String`\n              found type `&std::string::String`\n\n"}
[00:48:07] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n