\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issue-20831-debruijn.rs","byte_start":1159,"byte_end":1555,"line_start":38,"line_end":46,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    fn subscribe(&mut self, t : Box<Subscriber<Input=<Self as Publisher>::Output> + 'a>) {","highlight_start":5,"highlight_end":91},{"text":"        // Not obvious, but there is an implicit lifetime here -------^","highlight_start":1,"highlight_end":72},{"text":"        //~^^ ERROR cannot infer","highlight_start":1,"highlight_end":33},{"text":"        //","highlight_start":1,"highlight_end":11},{"text":"        // The fact that `Publisher` is using an implicit lifetime is","highlight_start":1,"highlight_end":70},{"text":"        // what was causing the debruijn accounting to be off, so","highlight_start":1,"highlight_end":66},{"text":"        // leave it that way!","highlight_start":1,"highlight_end":30},{"text":"        self.sub = t;","highlight_start":1,"highlight_end":22},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":"lifetime mismatch","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `'a`\n   found type ``","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"the anonymous lifetime #2 defined on the method body at 38:5...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/issue-20831-debruijn.rs","byte_start":1159,"byte_end":1555,"line_start":38,"line_end":46,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    fn subscribe(&mut self, t : Box<Subscriber<Input=<Self as Publisher>::Output> + 'a>) {","highlight_start":5,"highlight_end":91},{"text":"        // Not obvious, but there is an implicit lifetime here -------^","highlight_start":1,"highlight_end":72},{"text":"        //~^^ ERROR cannot infer","highlight_start":1,"highlight_end":33},{"text":"        //","highlight_start":1,"highlight_end":11},{"text":"        // The fact that `Publisher` is using an implicit lifetime is","highlight_start":1,"highlight_end":70},{"text":"        // what was causing the debruijn accounting to be off, so","highlight_start":1,"highlight_end":66},{"text":"        // leave it that way!","highlight_start":1,"highlight_end":30},{"text":"        self.sub = t;","highlight_start":1,"highlight_end":22},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"...does not necessarily outlive the lifetime 'a as defined on the impl at 36:6","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/issue-20831-debruijn.rs","byte_start":1095,"byte_end":1097,"line_start":36,"line_end":36,"column_start":6,"column_end":8,"is_primary":true,"text":[{"text":"impl<'a> Publisher<'a> for MyStruct<'a> {","highlight_start":6,"highlight_end":8}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/issue-20831-debruijn.rs:38:5\n   |\nLL | /     fn subscribe(&mut self, t : Box<Subscriber<Input=<Self as Publisher>::Output> + 'a>) {\nLL | |         // Not obvious, but there is an implicit lifetime here -------^\nLL | |         //~^^ ERROR cannot infer\nLL | |         //\n...  |\nLL | |         self.sub = t;\nLL | |     }\n   | |_____^ lifetime mismatch\n   |\n   = note: expected type `'a`\n              found type ``\nnote: the anonymous lifetime #2 defined on the method body at 38:5...\n  --> /checkout/src/test/ui/issue-20831-debruijn.rs:38:5\n   |\nLL | /     fn subscribe(&mut self, t : Box<Subscriber<Input=<Self as Publisher>::Output> + 'a>) {\nLL | |         // Not obvious, but there is an implicit lifetime here -------^\nLL | |         //~^^ ERROR cannot infer\nLL | |         //\n...  |\nLL | |         self.sub = t;\nLL | |     }\n   | |_____^\nnote: ...does not necessarily outlive the lifetime 'a as defined on the impl at 36:6\n  --> /checkout/src/test/ui/issue-20831-debruijn.rs:36:6\n   |\nLL | impl<'a> Publisher<'a> for MyStruct<'a> {\n   |      ^^\n\n"}
[00:41:43] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n