\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-festival.rs","byte_start":713,"byte_end":721,"line_start":32,"line_end":32,"column_start":5,"column_end":13,"is_primary":true,"text":[{"text":"    foo::FOO;","highlight_start":5,"highlight_end":13}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"erro_decl_name":"mac_trait_impl!","def_site_span":{"file_name":"/checkout/src/test/ui/hygiene/assoc_item_ctxt.rs","byte_start":952,"byte_end":1161,"line_start":43,"line_end":47,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    macro mac_trait_impl() {","highlight_start":5,"highlight_end":29},{"text":"        impl Tr for u8 { //~ ERROR not all trait items implemented, missing: `method`","highlight_start":1,"highlight_end":86},{"text":"            fn method() {} //~ ERROR method `method` is not a member of trait `Tr`","highlight_start":1,"highlight_end":83},{"text":"        }","highlight_start":1,"highlight_end":10},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[],"rendered":"error[E0407]: method `method` is not a member of trait `Tr`\n  --> /checkout/src/test/ui/hygiene/assoc_item_ctxt.rs:45:13\n   |\nLL |             fn method() {} //~ ERROR method `method` is not a member of trait `Tr`\n   |             ^^^^^^^^^^^^^^ not a member of trait `Tr`\n...\nLL |     mac_trait_impl!();\n   |     ------------------ in this macro invocation\n\n"}
[00:45:01] {"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to previous error\n\n"}
[00:45:01] {"message":"For more information about this error, try `rustc --explain E0407`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0407`.\n"}
[00:45:01] ------------------------------------------
[00:45:01] ------------------------------------------
[00:45kout/src/test/ui/hygiene/no_implicit_prelude.rs","byte_start":639,"byte_end":642,"line_start":21,"line_end":21,"column_start":9,"column_end":12,"is_primary":true,"text":[{"text":"        Vec::new(); //~ ERROR failed to resolve","highlight_start":9,"highlight_end":12}],"label":"Use of undeclared type or module `Vec`","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/hygiene/no_implicit_prelude.rs","byte_start":560,"byte_end":572,"line_start":15,"line_end":15,"column_start":14,"column_end":26,"is_primary":false,"text":[{"text":"    fn f() { ::bar::m!(); }","highlight_start":14,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"::bar::m!","def_site_span":{"file_name":"/checkout/src/test/ui/hygiene/no_implicit_prelude.rs","byte_start":615,"byte_end":743,"line_start":20,"line_end":23,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    pub macro m() {","highlight_start":5,"highlight_end":20},{"text":"        Vec::new(); //~ ERROR failed to resolve","highlight_start":1,"highlight_end":48},{"text":"        ().clone() //~ ERROR no method named `clone` found","highlight_start":1,"highlight_end":59},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[],"rendered":"error[E0433]: failed to resolve. Use of undeclared type or module `Vec`\n  --> /checkout/src/test/ui/hygiene/no_implicit_prelude.rs:21:9\n   |\nLL |     fn f() { ::bar::m!(); }\n   |              ------------ in this macro invocation\n...\nLL |         Vec::new(); //~ ERROR failed to resolve\n   |         ^^^ Use of undeclared type or module `Vec`\n\n"}
[00:45:01] {"message":"`main` function not found in crate `no_implicit_prelude`","code":{"code":"E0601","explanation":"\nNo `main` function was found in a binary crate. To fix this error, add a\n`main` function. For example:\n\n