\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/promote-ref-mut-in-let-issue-46557.rs","byte_start":607,"byte_end":616,"line_start":15,"line_end":15,"column_start":9,"column_end":18,"is_primary":true,"text":[{"text":"    let ref mut x = 1234543; //~ ERROR","highlight_start":9,"highlight_end":18}],"label":"temporary value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/promote-ref-mut-in-let-issue-46557.rs","byte_start":644,"byte_end":645,"line_start":17,"line_end":17,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"temporary value only lives until here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the static lifetime...","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0597]: borrowed value does not live long enough\n  --> /checkout/src/test/ui/borrowck/promote-ref-mut-in-let-issue-46557.rs:15:9\n   |\nLL |     let ref mut x = 1234543; //~ ERROR\n   |         ^^^^^^^^^ temporary value does not live long enough\nLL |     x\nLL | }\n   | - temporary value only lives until here\n   |\n   = note: borrowed value must be valid for the static lifetime...\n\n"}
[00:40:29] {"message":"borrowed value does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n