\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/promote-ref-mut-in-let-issue-46557.rs","byte_start":711,"byte_end":720,"line_start":20,"line_end":20,"column_start":10,"column_end":19,"is_primary":true,"text":[{"text":"    let (ref mut x, ) = (1234543, ); //~ ERROR","highlight_start":10,"highlight_end":19}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/promote-ref-mut-in-let-issue-46557.rs","byte_start":755,"byte_end":756,"line_start":22,"line_end":22,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"borrowed value only lives until here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the static lifetime...","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0597]: borrowed value does not live long enough\n  --> /checkout/src/test/ui/borrowck/promote-ref-mut-in-let-issue-46557.rs:20:10\n   |\nLL |     let (ref mut x, ) = (1234543, ); //~ ERROR\n   |          ^^^^^^^^^ borrowed value does not live long enough\nLL |     x\nLL | }\n   | - borrowed value only lives until here\n   |\n   = note: borrowed value must be valid for the static lifetime...\n\n"}
[00:46:35] {"message":"borrowed value does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n