\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/existential_type.rs","byte_start":1336,"byte_end":1338,"line_start":54,"line_end":54,"column_start":9,"column_end":11,"is_primary":true,"text":[{"text":"        \"\" //~ ERROR mismatched types","highlight_start":9,"highlight_end":11}],"label":"expected anonymized type, found reference","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/existential_type.rs","byte_start":1317,"byte_end":1325,"line_start":53,"line_end":53,"column_start":18,"column_end":26,"is_primary":false,"text":[{"text":"    fn bomp() -> boo::Boo {","highlight_start":18,"highlight_end":26}],"label":"expected `Boo` because of return type","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `Boo`\n   found type `&'static str`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/existential_type.rs:54:9\n   |\nLL |     fn bomp() -> boo::Boo {\n   |                  -------- expected `Boo` because of return type\nLL |         \"\" //~ ERROR mismatched types\n   |         ^^ expected anonymized type, found reference\n   |\n   = note: expected type `Boo`\n              found type `&'static str`\n\n"}
[00:46:14] {"message":"the trait bound `T: Trait` is not satisfied","code":{"code":"E0277","explanation":"\nYou tried to use a type which doesn't implement some trait in a place which\nexpected that trait. Erroneous code example:\n\n