\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/existential_type.rs","byte_start":1007,"byte_end":1009,"line_start":36,"line_end":36,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    \"\" //~ ERROR mismatched types","highlight_start":5,"highlight_end":7}],"label":"expected anonymized type, found reference","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/existential_type.rs","byte_start":992,"byte_end":1000,"line_start":35,"line_end":35,"column_start":14,"column_end":22,"is_primary":false,"text":[{"text":"fn bomp() -> boo::Boo {","highlight_start":14,"highlight_end":22}],"label":"expected `Boo` because of return type","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `Boo`\n   found type `&'static str`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/existential_type.rs:36:5\n   |\nLL | fn bomp() -> boo::Boo {\n   |              -------- expected `Boo` because of return type\nLL |     \"\" //~ ERROR mismatched types\n   |     ^^ expected anonymized type, found reference\n   |\n   = note: expected type `Boo`\n              found type `&'static str`\n\n"}
[00:46:14] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n