\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":5078,"byte_end":5101,"line_start":85,"line_end":85,"column_start":11,"column_end":34,"is_primary":true,"text":[{"text":"type A4 = extern \"rust-call\" fn(); //~ ERROR rust-call ABI is subject to change","highlight_start":11,"highlight_end":34}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(unboxed_closures)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: rust-call ABI is subject to change (see issue #29625)\n  --> /checkout/src/test/ui/feature-gate-abi.rs:85:11\n   |\nLL | type A4 = extern \"rust-call\" fn(); //~ ERROR rust-call ABI is subject to change\n   |           ^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(unboxed_closures)] to the crate attributes to enable\n\n"}
[00:41:08] {"message":"msp430-interrupt ABI is experimental and subject to change (see issue #38487)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n