\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":2441,"byte_end":2472,"line_start":44,"line_end":44,"column_start":5,"column_end":36,"is_primary":true,"text":[{"text":"    extern \"vectorcall\" fn dm3() {} //~ ERROR vectorcall is experimental and subject to change","highlight_start":5,"highlight_end":36}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_vectorcall)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: vectorcall is experimental and subject to change\n  --> /checkout/src/test/ui/feature-gate-abi.rs:44:5\n   |\nLL |     extern \"vectorcall\" fn dm3() {} //~ ERROR vectorcall is experimental and subject to change\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_vectorcall)] to the crate attributes to enable\n\n"}
[00:41:08] {"message":"rust-call ABI is subject to change (see issue #29625)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n