\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":4892,"byte_end":4924,"line_start":83,"line_end":83,"column_start":11,"column_end":43,"is_primary":true,"text":[{"text":"type A2 = extern \"platform-intrinsic\" fn(); //~ ERROR platform intrinsics are experimental","highlight_start":11,"highlight_end":43}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(platform_intrinsics)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: platform intrinsics are experimental and possibly buggy (see issue #27731)\n  --> /checkout/src/test/ui/feature-gate-abi.rs:83:11\n   |\nLL | type A2 = extern \"platform-intrinsic\" fn(); //~ ERROR platform intrinsics are experimental\n   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(platform_intrinsics)] to the crate attributes to enable\n\n"}
[00:41:08] {"message":"vectorcall is experimental and subject to change","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n