\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":4405,"byte_end":4436,"line_start":75,"line_end":75,"column_start":5,"coture-gate-abi.rs","byte_start":4809,"byte_end":4837,"line_start":82,"line_end":82,"column_start":11,"column_end":39,"is_primary":true,"text":[{"text":"type A1 = extern \"rust-intrinsic\" fn(); //~ ERROR intrinsics are subject to change","highlight_start":11,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(intrinsics)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: intrinsics are subject to change\n  --> /checkout/src/test/ui/feature-gate-abi.rs:82:11\n   |\nLL | type A1 = extern \"rust-intrinsic\" fn(); //~ ERROR intrinsics are subject to change\n   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(intrinsics)] to the crate attributes to enable\n\n"}
[00:41:08] {"message":"platform intrinsics are experimental and possibly buggy (see issue #27731)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n