\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gate-abi.rs","byte_start":3395,"byte_end":3424,"line_start":60,"line_end":60,"column_start":5,"column_end":34,"is_primary":true,"text":[{"text":"    extern \"rust-call\" fn m4() {} //~ ERROR rust-call ABI is subject to change","highlight_start":5,"highlight_end":34}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(unboxed_closures)] to the crate attributes to enable","code":n"expansion":null}],"children":[{"message":"add #![feature(abi_thiscall)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: thiscall is experimental and subject to change\n  --> /checkout/src/test/ui/feature-gate-abi.rs:64:5\n   |\nLL |     extern \"thiscall\" fn m8() {} //~ ERROR thiscall is experimental and subject to change\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_thiscall)] to the crate attributes to enable\n\n"}
[00:41:08] {"message":"amdgpu-kernel ABI is experimental and subject to change (see issue #51575)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n