\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/stable/book/references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/get_default.rs","byte_start":1610,"byte_end":1613,"line_start":51,"line_end":51,"column_start":17,"column_end":20,"is_primary":true,"text":[{"text":"                map.set(String::new()); // Ideally, just AST would error here","highlight_start":17,"highlight_end":20}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/get_default.rs","byte_start":1314,"byte_end":1317,"line_start":43,"line_end":43,"column_start":15,"column_end":18,"is_primary":false,"text":[{"text":"        match map.get() {","highlight_start":15,"highlight_end":18}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the anonymous lifetime #1 defined on the function body at 41:1...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/nll/get_default.rs","byte_start":1254,"byte_end":1813,"line_start":41,"line_end":57,"column_start":1,"column_end":2,"is_primary":true,"text":[{"text":"fn err(map: &mut Map) -> &String {","highlight_start":1,"highlight_end":35},{"text":"    loop {","highlight_start":1,"highlight_end":11},{"text":"        match map.get() {","highlight_start":1,"highlight_end":26},{"text":"            Some(v) => {","highlight_start":1,"highlight_end":25},{"text":"                map.set(String::new()); // Both AST and MIR error here","highlight_start":1,"highlight_end":71},{"text":"                //~^ ERROR borrowed as immutable (Mir)","highlight_start":1,"highlight_end":55},{"text":"                //~| ERROR borrowed as immutable (Ast)","highlight_start":1,"highlight_end":55},{"text":"                return v;","highlight_start":1,"highlight_end":26},{"text":"            }","highlight_start":1,"highlight_end":14},{"text":"            None => {","highlight_start":1,"highlight_end":22},{"text":"                map.set(String::new()); // Ideally, just AST would error here","highlight_start":1,"highlight_end":78},{"text":"                //~^ ERROR borrowed as immutable (Ast)","highlight_start":1,"highlight_end":55},{"text":"                //~| ERROR borrowed as immutable (Mir)","highlight_start":1,"highlight_end":55},{"text":"            }","highlight_start":1,"highlight_end":14},{"text":"        }","highlight_start":1,"highligsue-47184.rs
[00:41:17] 
[00:41:17] test result: FAILED. 1535 passed; 2 failed; 5 ignored; 0 measured; 0 filtered out
[00:41:17] 
[00:41:17] 
[00:41:17] 
[00:41:17] 
[00:41:17] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-3.9/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options " "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "3.9.1\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[00:41:17] 
[00:41:17] 
[00:41:17] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[00:41:17] Build completed unsuccessfully in 0:02:00
