\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":708,"byte_end":727,"line_start":21,"line_end":21,"column_start":1,"column_end":20,"is_primary":true,"text":[{"text":"generate_mod::check!(); //~ ERROR cannot find type `FromOutside` in this scope","highlight_start":1,"highlight_end":20}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(use_extern_macros)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: non-ident macro paths are experimental (see issue #35896)\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:21:1\n   |\nLL | generate_mod::check!(); //~ ERROR cannot find type `FromOutside` in this scope\n   | ^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(use_extern_macros)] to the crate attributes to enable\n\n"}
[00:52:35] {"message":"non-ident macro paths are experimental (see issue #35896)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n