\n\nThis does not pose a problem by itself because they can't be\naccessed directly.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/const-eval/match-test-ptr-null.rs","byte_start":766,"byte_end":791,"line_start":16,"line_end":16,"column_start":15,"column_end":40,"is_primary":true,"text":[{"text":"        match &1 as *const i32 as usize { //~ ERROR raw pointers cannot be cast to integers","highlight_start":15,"highlight_end":40}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":rror[E0019]: constant contains unimplemented expression type\n  --> /checkout/src/test/ui/const-eval/match-test-ptr-null.rs:17:13\n   |\nLL |             0 => 42, //~ ERROR constant contains unimplemented expression type\n   |             ^\n\n"}
[00:46:39] {"message":"could not evaluate repeat length","code":{"code":"E0080","explanation":"\nThis error indicates that the compiler was unable to sensibly evaluate an\nconstant expression that had to be evaluated. Attempting to divide by 0\nor causing integer overflow are two ways to induce this error. For example:\n\n