\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issue-24446.rs","byte_start":509,"byte_end":619,"line_start":12,"line_end":16,"column_start":31,"column_end":6,"is_primary":true,"text":[{"text":"    static foo: Fn() -> u32 = || -> u32 {","highlight_start":31,"highlight_end":42},{"text":"        //~^ ERROR mismatched types","highlight_start":1,"highlight_end":36},{"text":"        //~| ERROR the size for values of type","highlight_start":1,"highlight_end":47},{"text":"        0","highlight_start":1,"highlight_end":10},{"text":"    };","highlight_start":1,"highlight_end":6}],"label":"expected trait std::ops::Fn, found closure","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `(dyn std::ops::Fn() -> u32 + 'static)`\n   found type `[closure@/checkout/src/test/ui/issue-24446.rs:12:31: 16:6]`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/issue-24446.rs:12:31\n   |\nLL |       static foo: Fn() -> u32 = || -> u32 {\n   |  _______________________________^\nLL | |         //~^ ERROR mismatched types\nLL | |         //~| ERROR the size for values of type\nLL | |         0\nLL | |     };\n   | |_____^ expected trait std::ops::Fn, found closure\n   |\n   = note: expected type `(dyn std::ops::Fn() -> u32 + 'static)`\n              found type `[closure@/checkout/src/test/ui/issue-24446.rs:12:31: 16:6]`\n\n"}
[00:43:30] {"message":"the size for values of type `(dyn std::ops::Fn() -> u32 + 'static)` cannot be known at compilation time","code":{"code":"E0277","explanation":"\nYou tried to use a type which doesn't implement some trait in a place which\nexpected that trait. Erroneous code example:\n\n