\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/augmented-assignments.rs","byte_start":828,"byte_end":829,"line_start":30,"line_end":30,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    y   //~ error: cannot borrow immutable local variable `y` as mutable","highlight_start":5,"highlight_end":6}],"label":"cannot borrow mutably","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/augmented-assignments.rs","byte_start":769,"byte_end":770,"line_start":28,"line_end":28,"column_start":9,"column_end":10,"is_primary":false,"text":[{"text":"    let y = Int(2);","highlight_start":9,"highlight_end":10}],"label":"consider changing this to `mut y`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0596]: cannot borrow immutable local variable `y` as mutable\n  --> /checkout/src/test/ui/augmented-assignments.rs:30:5\n   |\nLL |     let y = Int(2);\n   |         - consider changing this to `mut y`\nLL |     //~^ consider changing this to `mut y`\nLL |     y   //~ error: cannot borrow immutable local variable `y` as mutable\n   |     ^ cannot borrow mutably\n\n"}
[00:50:15] {"message":"use of moved value: `x`","code":{"code":"E0382","explanation":"\nThis error occurs when an attempt is made to use a variable after its contents\nhave been moved elsewhere. For example:\n\n