\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/array-break-length.rs","byte_start":506,"byte_end":511,"line_start":13,"line_end":13,"column_start":17,"column_end":22,"is_primary":true,"text":[{"text":"        |_: [_; break]| {} //~ ERROR: `break` outside of loop","highlight_start":17,"highlight_end":22}],"label":"cannot break outside of a loop","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0268]: `break` outside of loop\n  --> /checkout/src/test/ui/array-break-length.rs:13:17\n   |\nLL |         |_: [_; break]| {} //~ ERROR: `break` outside of loop\n   |                 ^^^^^ cannot break outside of a loop\n\n"}
[00:50:15] {"message":"`continue` outside of loop","code":{"code":"E0268","explanation":"\nThis error indicates the use of a loop keyword (`break` or `continue`) outside\nof a loop. Without a loop to break out of or continue in, no sensible action can\nbe taken. Erroneous code example:\n\n