\n\n* The `#[repr(C)]` attribute can only be placed on structs and enums.\n* The `#[repr(packed)]` and `#[repr(simd)]` attributes only work on structs.\n* The `#[repr(u8)]`, `#[repr(i16)]`, etc attributes only work on enums.\n\nThese attributes do not work on typedefs, since typedefs are just aliases.\n\nRepresentations like `#[repr(u8)]`, `#[repr(i64)]` are for selecting the\ndiscriminant size for enums with no data fields on any of the variants, e.g.\n`enum Color {Red, Blue, Green}`, effectively setting the size of the enum to\nthe size of the provided type. Such an enum can be cast to a value of the same\ntype as well. In short, `#[repr(u8)]` makes the enum behave like an integer\nwith a constrained set of allowed values.\n\nOnly field-less enums can be cast to numerical primitives, so this attribute\nwill not apply to structs.\n\n`#[repr(packed)]` reduces padding to make the struct size smaller. The\nrepresentation of enums isn't strictly defined in Rust, and this attribute\nwon't work on enums.\n\n`#[repr(simd)]` will give a struct consisting of a homogeneous series of machine\ntypes (i.e. `u8`, `i32`, etc) a representation that permits vectorization via\nSIMD. This doesn't make much sense for enums since they don't consist of a\nsingle list of data.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/attr-usage-repr.rs","byte_start":995,"byte_end":1016,"line_start":36,"line_end":36,"column_start":1,"column_end":22,"is_primary":false,"text":[{"text":"enum EPacked { A, B }","highlight_start":1,"highlight_end":22}],"label":"not a struct or union","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/attr-usage-repr.rs","byte_start":937,"byte_end":943,"line_start":35,"line_end":35,"column_start":8,"column_end":14,"is_primary":true,"text":[{"text":"#[repr(packed)] //~ ERROR: attribute should be applied to struct","highlight_start":8,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0517]: attribute should be applied to struct or union\n  --> /checkout/src/test/ui/attr-usage-repr.rs:35:8\n   |\nLL | #[repr(packed)] //~ ERROR: attribute should be applied to struct\n   |        ^^^^^^\nLL | enum EPacked { A, B }\n   | --------------------- not a struct or union\n\n"}
[00:50:15] {"message":"attribute should be applied to struct","code":{"code":"E0517","explanation":"\nThis error indicates that a `#[repr(..)]` attribute was placed on an\nunsupported item.\n\nExamples of erroneous code:\n\n