\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0423.rs","byte_start":521,"byte_end":524,"line_start":14,"line_end":14,"column_start":13,"column_end":16,"is_primary":true,"text":[{"text":"    let f = Foo(); //~ ERROR E0423","highlight_start":13,"highlight_end":16}],"label":"did you mean `foo`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/error-codes/E0423.rs","byte_start":521,"byte_end":524,"line_start":14,"line_end":14,"column_start":13,"column_end":16,"is_primary":true,"text":[{"text":"    let f = Foo(); //~ ERROR E0423","highlight_start":13,"highlight_end":16}],"label":"did you mean `Foo { /* fields */ }`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0423]: expected function, found struct `Foo`\n  --> /checkout/src/test/ui/error-codes/E0423.rs:14:13\n   |\nLL |     let f = Foo(); //~ ERROR E0423\n   |             ^^^\n   |             |\n   |             did you mean `foo`?\n   |             did you mean `Foo { /* fields */ }`?\n\n"}
[00:44:06] {"message":"expected value, found struct `S`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n