\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/index-mut-help.rs","byte_start":727,"byte_end":739,"line_start":21,"line_end":21,"column_start":5,"column_end":17,"is_primary":true,"text":[{"text":"    map[\"peter\"].clear();           //~ ERROR","highlight_start":5,"highlight_end":17}],"label":"cannot borrow as mutable","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"trait `IndexMut` is required to modify indexed content, but it is not implemented for `std::collections::HashMap<&str, std::string::String>`","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0596]: cannot borrow immutable indexed content as mutable\n  --> /checkout/src/test/ui/borrowck/index-mut-help.rs:21:5\n   |\nLL |     map[\"peter\"].clear();           //~ ERROR\n   |     ^^^^^^^^^^^^ cannot borrow as mutable\n   |\n   = help: trait `IndexMut` is required to modify indexed content, but it is not implemented for `std::collections::HashMap<&str, std::string::String>`\n\n"}
[00:44:40] {"message":"cannot assign to immutable indexed content","code":{"code":"E0594","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/index-mut-help.rs","byte_start":773,"byte_end":803,"line_start":22,"line_end":22,"column_start":5,"column_end":35,"is_primary":true,"text":[{"text":"    map[\"peter\"] = \"0\".to_string(); //~ ERROR","highlight_start":5,"highlight_end":35}],"label":"cannot borrow as mutable","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"trait `IndexMut` is required to modify indexed content, but it is not implemented for `std::collections::HashMap<&str, std::string::String>`","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0594]: cannot assign to immutable indexed content\n  --> /checkout/src/test/ui/borrowck/index-mut-help.rs:22:5\n   |\nLL |     map[\"peter\"] = \"0\".to_string(); //~ ERROR\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable\n   |\n   = help: trait `IndexMut` is required to modify indexed content, but it is not implemented for `std::collections::HashMap<&str, std::string::String>`\n\n"}
[00:44:40] {"message":"cannot borrow immutable indexed content as mutable","code":{"code":"E0596","explanation":"\nThis error occurs because you tried to mutably borrow a non-mutable variable.\n\nExample of erroneous code:\n\n