\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1761,"byte_end":1763,"line_start":83,"line_end":83,"column_start":13,"column_end":15,"is_primary":true,"text":[{"text":"    let y = &x; //~ ERROR","highlight_start":13,"highlight_end":15}],"label":"borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1738,"byte_end":1740,"line_start":82,"line_end":82,"column_start":13,"column_end":15,"is_primary":false,"text":[{"text":"    let f = || *x = 0;","highlight_start":13,"highlight_end":15}],"label":"closure construction occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1742,"byte_end":1743,"line_start":82,"line_end":82,"column_start":17,"column_end":18,"is_primary":false,"text":[{"text":"    let f = || *x = 0;","highlight_start":17,"highlight_end":18}],"label":"first borrow occurs due to use of `x` in closure","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1779,"byte_end":1780,"line_start":84,"line_end":84,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    f.use_ref();","highlight_start":5,"highlight_end":6}],"label":"borrow used here in later iteration of loop","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0501]: cannot borrow `x` as immutable because previous closure requires unique access\n  --> /checkout/src/test/ui/nll/closure-borrow-spans.rs:83:13\n   |\nLL |     let f = || *x = 0;\n   |             --  - first borrow occurs due to use of `x` in closure\n   |             |\n   |             closure construction occurs here\nLL |     let y = &x; //~ ERROR\n   |             ^^ borrow occurs here\nLL |     f.use_ref();\n   |     - borrow used here in later iteration of loop\n\n"}
[00:43:19] {"message":"cannot borrow `x` as mutable because previous closure requires unique access","code":{"code":"E0501","explanation":"\nThis error indicates that a mutable variable is being used while it is still\ncaptured by a closure. Because the closure has borrowed the variable, it is not\navailable for use until the closure goes out of scope.\n\nNote that a capture will either move or borrow a variable, but in this\nsituation, the closure is borrowing the variable. Take a look at\nhttp://rustbyexample.com/fn/closures/capture.html for more information about\ncapturing.\n\nExample of erroneous code:\n\n