\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/stable/book/references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":740,"byte_end":746,"line_start":23,"line_end":23,"column_start":13,"column_end":19,"is_primary":true,"text":[{"text":"    let y = &mut x; //~ ERROR","highlight_start":13,"highlight_end":19}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":722,"byte_end":724,"line_start":22,"line_end":22,"column_start":13,"column_end":15,"is_primary":false,"text":[{"text":"    let f = || x;","highlight_start":13,"highlight_end":15}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":725,"byte_end":726,"line_start":22,"line_end":22,"column_start":16,"column_end":17,"is_primary":false,"text":[{"text":"    let f = || x;","highlight_start":16,"highlight_end":17}],"label":"first borrow occurs due to use of `x` in closure","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":762,"byte_end":763,"line_start":24,"line_end":24,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    f.use_ref();","highlight_start":5,"highlight_end":6}],"label":"borrow used here in later iteration of loop","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `x` as mutable because it is also borrowed as immutable\n  --> /checkout/src/test/ui/nll/closure-borrow-spans.rs:23:13\n   |\nLL |     let f = || x;\n   |             -- - first borrow occurs due to use of `x` in closure\n   |             |\n   |             immutable borrow occurs here\nLL |     let y = &mut x; //~ ERROR\n   |             ^^^^^^ mutable borrow occurs here\nLL |     f.use_ref();\n   |     - borrow used here in later iteration \nYou can find more information about borrowing in the rust-book:\nhttp://doc.rust-lang.org/stable/book/references-and-borrowing.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1046,"byte_end":1048,"line_start":43,"line_end":43,"column_start":13,"column_end":15,"is_primary":false,"text":[{"text":"    let f = || x = 0;","highlight_start":13,"highlight_end":15}],"label":"borrow of `x` occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1068,"byte_end":1069,"line_start":44,"line_end":44,"column_start":13,"column_end":14,"is_primary":true,"text":[{"text":"    let y = x; //~ ERROR","highlight_start":13,"highlight_end":14}],"label":"use of borrowed `x`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1049,"byte_end":1050,"line_start":43,"line_end":43,"column_start":16,"column_end":17,"is_primary":false,"text":[{"text":"    let f = || x = 0;","highlight_start":16,"highlight_end":17}],"label":"borrow occurs due to use of `x` in closure","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1085,"byte_end":1086,"line_start":45,"line_end":45,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    f.use_ref();","highlight_start":5,"highlight_end":6}],"label":"borrow used here in later iteration of loop","suggested_replacement":null,"s y = &x; //~ ERROR","highlight_start":13,"highlight_end":15}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1156,"byte_end":1158,"line_start":49,"line_end":49,"column_start":13,"column_end":15,"is_primary":false,"text":[{"text":"    let f = || x = 0;","highlight_start":13,"highlight_end":15}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1159,"byte_end":1160,"line_start":49,"line_end":49,"column_start":16,"column_end":17,"is_primary":false,"text":[{"text":"    let f = || x = 0;","highlight_start":16,"highlight_end":17}],"label":"first borrow occurs due to use of `x` in closure","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-borrow-spans.rs","byte_start":1196,"byte_end":1197,"line_start":51,"line_end":51,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    f.use_ref();","highlight_start":5,"highlight_end":6}],"label":"borrow used here in later iteration of loop","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `x` as immutable because it is also borrowed as mutable\n  --> /checkout/src/test/ui/nll/closure-borrow-spans.rs:50:13\n   |\nLL |     let f = || x = 0;\n   |             -- - first borrow occurs due to use of `x` in closure\n   |             |\n  597\nstruct Foo<'a> {\n    x: Option<&'a u32>,\n}\n\nlet mut x = Foo { x: None };\nlet y = 0;\nx.x = Some(&y); // error: `y` does not live long enough\n