plain
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/2a/a2/dba13566ff2819757280c7c92ca225de3bef4ef4892697885156dbe36ac9/awscli-1.15.71-py2.py3-none-any.whl (1.3MB)
    0% |▎                               | 10kB 17.7MB/s eta 0:00:01
    1% |▌                               | 20kB 1.8MB/s eta 0:00:01
    2% |▊                               | 30kB 2.2MB/s eta 0:00:01
    3% |█                               | 40kB 1.9MB/s eta 0:00:01
---
travis_time:start:test_ui
Check compiletest suite=ui mode=ui (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
[00:41:51] 
[00:41:51] running 2228 tests
[00:41:54] ...................F................................................................................
[00:42:01] ....................................................................................................
[00:42:03] ....................................................................................................
[00:42:03] ....................................................................................................
[00:42:05] .................................................................................F.F................
[00:42:10] ....................................................................................................
[00:42:13] ....................................................................................................
[00:42:15] ....................................................................................................
[00:42:18] ...........i........................................................................................
[00:42:18] ...........i........................................................................................
[00:42:20] ....................................................................................................
[00:42:22] ....................................................................................................
[00:42:25] ....................................................................................................
[00:42:28] .......................................................................................F............
[00:42:33] ....................................................................................................
[00:42:37] ....................................................................................................
[00:42:40] .....i..............................................................................................
[00:42:40] .....i..............................................................................................
[00:42:43] .........i......................F...................................................................
[00:42:50] ....................................................................................................
[00:42:53] ...............................i....................................................................
[00:42:54] .......................i....
[00:42:54] failures:
[00:42:54] failures:
[00:42:54] 
[00:42:54] ---- [ui] ui/attr-usage-repr.rs stdout ----
[00:42:54] diff of stderr:
[00:42:54] 
[00:42:54] 1 error[E0517]: attribute should be applied to struct, enum or union
[00:42:54] +   --> $DIR/attr-usage-repr.rs:13:8
[00:42:54] 3    |
[00:42:54] 3    |
[00:42:54] 4 LL | #[repr(C)] //~ ERROR: attribute should be applied to struct, enum or union
[00:42:54] 
[00:42:54] 
[00:42:54] 7    | --------- not a struct, enum or union
[00:42:54] 9 error[E0517]: attribute should be applied to enum
[00:42:54] -   --> $DIR/attr-usage-repr.rs:26:8
[00:42:54] +   --> $DIR/attr-usage-repr.rs:25:8
[00:42:54] 11    |
[00:42:54] 11    |
[00:42:54] 12 LL | #[repr(i8)] //~ ERROR: attribute should be applied to enum
[00:42:54] 
[00:42:54] 
[00:42:54] 15    | ---------------------- not an enum
[00:42:54] 17 error[E0517]: attribute should be applied to struct or union
[00:42:54] -   --> $DIR/attr-usage-repr.rs:32:8
[00:42:54] +   --> $DIR/attr-usage-repr.rs:31:8
[00:42:54] 19    |
[00:42:54] 19    |
[00:42:54] 20 LL | #[repr(align(8))] //~ ERROR: attribute should be applied to struct
[00:42:54] 
[00:42:54] 
[00:42:54] 23    | -------------------- not a struct or union
mary":false,"text":[{"text":"struct SInt(f64, f64);","highlight_start":1,"highlight_end":23}],"label":"not an enum","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/attr-usage-repr.rs","byte_start":704,"byte_end":706,"line_start":25,"line_end":25,"column_start":8,"column_end":10,"is_primary":true,"text":[{"text":"#[repr(i8)] //~ ERROR: attribute should be applied to enum","highlight_start":8,"highlight_end":10}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0517]: attribute should be applied to enum\n  --> /checkout/src/test/ui/attr-usage-repr.rs:25:8\n   |\nLL | #[repr(i8)] //~ ERROR: attribute should be applied to enum\n   |        ^^\nLL | struct SInt(f64, f64);\n   | ---------------------- not an enum\n\n"}
[00:42:54] {"message":"attribute should be applied to struct or union","code":{"code":"E0517","explanation":"\nThis error indicates that a `#[repr(..)]` attribute was placed on an\nunsupported item.\n\nExamples of erroneous code:\n\n