\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0509.rs","byte_start":700,"byte_end":717,"line_start":26,"line_end":26,"column_start":23,"column_end":40,"is_primary":true,"text":[{"text":"    let fancy_field = drop_struct.fancy; //~ ERROR E0509","highlight_start":23,"highlight_end":40}],"label":"cannot move out of here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider borrowing here","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0509.rs","byte_start":700,"byte_end":717,"line_start":26,"line_end":26,"column_start":23,"column_end":40,"is_primary":true,"text":[{"text":"    let fancy_field = drop_struct.fancy; //~ ERROR E0509","highlight_start":23,"highlight_end":40}],"label":null,"suggested_replacement":"&drop_struct.fancy","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0509]: cannot move out of type `DropStruct`, which implements the `Drop` trait\n  --> /checkout/src/test/ui/error-codes/E0509.rs:26:23\n   |\nLL |     let fancy_field = drop_struct.fancy; //~ ERROR E0509\n   |                       ^^^^^^^^^^^^^^^^^\n   |                       |\n   |                       cannot move out of here\n   |                       help: consider borrowing here: `&drop_struct.fancy`\n\n"}
[00:46:35] {"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to previous error\n\n"}
[00:46:35] {"message":"For more information about this error, try `rustc --explain E05efCell;\n\nstruct TheDarkKnight;\n\nimpl TheDarkKnight {\n    fn nothing_is_true(self) {}\n}\n\nfn main() {\n    let x = RefCell::new(TheDarkKnight);\n    let x = x.into_inner(); // we get back ownership\n\n    x.nothing_is_true(); // ok!\n}\n