\n\nYou can find more information about borrowing in the rust-book:\nhttp://doc.rust-lang.org/book/first-edition/references-and-borrowing.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/std-uncopyable-atomics.rs","byte_start":585,"byte_end":588,"line_start":19,"line_end":19,"column_start":13,"column_end":16,"is_primary":true,"text":[{"text":"    let x = *&x; //~ ERROR: cannot move out of borrowed content","highlight_start":13,"highlight_end":16}],"label":"cannot move out of borrowed content","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider removing the `*`","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/std-uncopyable-atomics.rs","byte_start":585,"byte_end":588,"line_start":19,"line_end":19,"column_start":13,"column_end":16,"is_primary":true,"text":[{"text":"    let x = *&x; //~ ERROR: cannot move out of borrowed content","highlight_start":13,"highlight_end":16}],"label":null,"suggested_replacement":"&x","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0507]: cannot move out of borrowed content\n  --> /checkout/src/test/ui/std-uncopyable-atomics.rs:19:13\n   |\nLL |     let x = *&x; //~ ERROR: cannot move out of borrowed content\n   |             ^^^\n   |             |\n   |             cannot move out of borrowed content\n   |             help: consider removing the `*`: `&x`\n\n"}
[00:52:46] {"message":"cannot move out of borrowed content","code":{"code":"E0507","explanation":"\nYou tried to move out of a value which was borrowed. Erroneous code example:\n\n