\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/stable/book/references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs","byte_start":1868,"byte_end":1887,"line_start":58,"line_end":58,"column_start":23,"column_end":42,"is_primary":true,"text":[{"text":"        if let [_, _, ref mut from_begin2, ..] = *s { //~ERROR","highlight_start":23,"highlight_end":42}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs","byte_start":1546,"byte_end":1559,"line_start":51,"line_end":51,"column_start":35,"column_end":48,"is_primary":false,"text":[{"text":"    if let [.., _, ref from_end4, ref from_end3, _, ref from_end1] = *s {","highlight_start":35,"highlight_end":48}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs","byte_start":1951,"byte_end":1960,"line_start":59,"line_end":59,"column_start":43,"column_end":52,"is_primary":false,"text":[{"text":"            nry":true,"text":[{"text":"        if let [.., ref mut from_end2, _] = *s { //~ERROR","highlight_start":21,"highlight_end":38}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs","byte_start":2182,"byte_end":2197,"line_start":65,"line_end":65,"column_start":50,"column_end":65,"is_primary":false,"text":[{"text":"    if let [ref from_begin0, ref from_begin1, _, ref from_begin3, _, ..] = *s {","highlight_start":50,"highlight_end":65}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs","byte_start":2441,"byte_end":2452,"line_start":70,"line_end":70,"column_start":45,"column_end":56,"is_primary":false,"text":[{"text":"            nop(&[from_begin0, from_begin1, from_begin3, from_end2]);","highlight_start":45,"highlight_end":56}],"label":"borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `s[..]` as mutable because it is also borrowed as immutable\n  --> /checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs:69:21\n   |\nLL |     if let [ref from_begin0, ref from_begin1, _, ref from_begin3, _, ..] = *s {\n   |                                                  --------------- immutable borrow occurs here\n...\nLL |         if let [.., ref mut from_end2, _] = *s { //~ERROR\n   |                     ^^^^^^^^^^^^^^^^^ mutable borrow occurs here\nLL |             nop(&[from_begin0, from_begin1, from_begin3, from_end2]);\n   |                                             ----------- borrow later used here\n\n"}
[00:44:56] {"message":"cannot borrow `s[..]` as mutable because it is also borrowed as immutable","code":{"code":"E0502","explanation":"\nThis error indicates that you are trying to borrow a variable as mutable when it\nhas already been borrowed as immutable.\n\nExample of erroneous code:\n\n