\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/stable/book/references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs","byte_start":2497,"byte_end":2514,"line_start":72,"line_end":72,"column_start":21,"column_end":38,"is_primary":true,"text":[{"text":"        if let [.., ref mut from_end3, _,  _] = *s { //~ERROR","highlight_start":21,"highlight_end":38}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs","byte,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `s[..]` as mutable because it is also borrowed as immutable\n  --> /checkout/src/test/ui/borrowck/borrowck-slice-pattern-element-loan.rs:92:20\n   |\nLL |     if let [ref first, ref second, ..] = *s {\n   |                        ---------- immutable borrow occurs here\nLL |         if let [_, ref mut tail..] = *s { //~ERROR\n   |                    ^^^^^^^^^^^^ mutable borrow occurs here\nLL |             nop(&[first, second]);\n   |                          ------ borrow later used here\n\n"}
[00:44:56] {"message":"cannot borrow `s[..]` as mutable because it is also borrowed as immutable","code":{"code":"E0502","explanation":"\nThis error indicates that you are trying to borrow a variable as mutable when it\nhas already been borrowed as immutable.\n\nExample of erroneous code:\n\n