\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":506,"byte_end":515,"line_start":13,"line_end":13,"column_start":1,"column_end":10,"is_primary":false,"text":[{"text":"struct S;","highlight_start":1,"highlight_end":10}],"label":"associated item `A` not found for this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":2522,"byte_end":2526,"line_start":107,"line_end":107,"column_start":5,"column_end":9,"is_primary":true,"text":[{"text":"    S::A; //~ ERROR no associated item named `A` found for type `S` in the current scope","highlight_start":5,"highlight_end":9}],"label":"associated item not found in `S`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"items from traits can only be used if the trait is implemented and in scope","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"the following trait defines an item `A`, perhaps you need to implement it:\ncandidate #1: `assoc_const::A`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0599]: no associated item named `A` found for type `S` in the current scope\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:107:5\n   |\nLL | struct S;\n   | --------- associated item `A` not found for this\n...\nLL |     S::A; //~ ERROR no associated item named `A` found for type `S` in the current scope\n   |     ^^^^ associated item not found i   pub struct Foo;\n\n    impl Foo {\n        fn method(&self) {}\n    }\n\n    pub fn call_method(foo: &Foo) { // We create a public function.\n        foo.method(); // Which calls the item.\n    }\n}\n\nlet foo = inner::Foo;\ninner::call_method(&foo); // And since the function is public, we can call the\n                          // method through it.\n