\n\nIf the trait `Foo` was deriving from something like `Super<String>` or\n`Super<T>` (where `Foo` itself is `Foo<T>`), this is okay, because given a type\n`get_a()` will definits":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":3591,"byte_end":3595,"line_start":129,"line_end":129,"column_start":12,"column_end":16,"is_primary":true,"text":[{"text":"    let _: S::C; //~ ERROR ambiguous associated type","highlight_start":12,"highlight_end":16}],"label":"ambiguous associated type","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"specify the type using the syntax `<S as Trait>::C`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0223]: ambiguous associated type\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:129:12\n   |\nLL |     let _: S::C; //~ ERROR ambiguous associated type\n   |            ^^^^ ambiguous associated type\n   |\n   = note: specify the type using the syntax `<S as Trait>::C`\n\n"}
[00:47:07] {"message":"associated type `A` is private","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":3730,"byte_end":3734,"line_start":131,"line_end":131,"column_start":12,"column_end":16,"is_primary":true,"text":[{"text":"    let _: T::A; //~ ERROR associated type `A` is private","highlight_start":12,"highlight_end":16}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: associated type `A` is private\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:131:12\n   |\nLL |     let _: T::A; //~ ERROR associated type `A` is private\n   |            ^^^^\n\n"}
[00:47:07] {"message":"associated type `A` is private","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":3933,"byte_end":3939,"line_start":140,"line_end":140,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"        A = u8, //~ ERROR associated type `A` is private","highlight_start":9,"highlight_end":15}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: associated type `A` is private\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:140:9\n   |\nLL |         A = u8, //~ ERROR associated type `A` is private\n   |         ^^^^^^\n\n"}
[00:47:07] {"message":"the size for values of type `dyn assoc_ty::B<B=u8>` cannot be known at compilation time","code":{"code":"E0277","explanation":"\nYou tried to use a type which doesn't implement some trait in a place which\nexpected that trait. Erroneous code example:\n\n