\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":506,"byte_end":515,"line_start":13,"line_end":13,"column_start":1,"column_end":10,"is_primary":false,"text":[{"text":"struct S;","highlight_start":1,"highlight_end":10}],"label":"method `b` not found for this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":1530,"byte_end":1531,"line_start":78,"line_end":78,"column_start":7,"column_end":8,"is_primary":true,"text":[{"text":"    S.b(); //~ ERROR no method named `b` found for type `S` in the current scope","highlight_start":7,"highlight_end":8}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"items from traits can only be used if the trait is in scope","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"the following trait is implemented but not in scope, perhaps add a `use` for it:","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":506,"byte_end":506,"line_start":13,"line_end":13,"column_start":1,"column_end":1,"is_primary":true,"text":[{"text":"struct S;","highlight_start":1,"highlight_end":1}],"label":null,"suggested_replacement":"use method::B;\n\n","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0599]: no method named `b` found for type `S` in the current scope\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:78:7\n   |\nLL | struct S;\n   | --------- method `b` not found for this\n...\nLL |     S.b(); //~ ERROR no method named `b` found for type `S` in the current scope\n   |       ^\n   |\n   = help: items from traits can only be used if the trait is in scope\nhelp: the following trait is implemented but not in scope, perhaps add a `use` for it:\n   |\nLL | use method::B;\n   |\n\n"}
[00:46:20] {"message":"method `a` is private","code":{"code":"E0624",` found for type `S` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n