\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":2194,"byte_end":2198,"line_start":94,"line_end":94,"column_start":5,"column_end":9,"is_primary":true,"text":[{"text":"    C::a(&S); //~ ERROR method `a` is private","highlight_start":5,"highlight_end":9}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0624]: method `a` is private\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:94:5\n   |\nLL |     C::a(&S); //~ ERROR method `a` is private\n   |     ^^^^\n\n"}
[00:46:20] {"message":"no associated item named `A` found for type `S` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n