\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":506,"byte_end":515,"line_start":13,"line_end":13,"column_start":1,"column_end":10,"is_primary":false,"text":[{"text":"struct S;","highlight_start":1,"highlight_end":10}],"label":"associated item `A` not found for this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":2522,"byte_end":2526,"line_start":107,"line_end":107,"column_start":5,"column_end":9,"is_primary":true,"text":[{"text":"    S::A; //~ ERROR no associated item named `A` found for type `S` in the current scope","highlight_start":5,"highlight_end":9}],"label":"associated item not found in `S`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"items from traits can only be used if the trait is implemented and in scope","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"the following trait defines an item `A`, perhaps you need to implement it:\ncandidate #1: `assoc_const::A`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0599]: no associated item named `A` found for type `S` in the current scope\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:107:5\n   |\nLL | struct S;\n   | --------- associated item `A` not found for this\n...\nLL |     S::A; //~ ERROR no associated item named `A` found for type `S` in the current scope\n   |     ^^^^ associated item not found in `S`\n   |\n   = help: items from traits can only be used if the trait is implemented and in scope\n   = note: the following trait defines an item `A`, perhaps you need to implement it:\n           candidate #1: `assoc_const::A`\n\n"}
[00:46:20] {"message":"no associated item named `B` found for type `S` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n