\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/structs/structure-constructor-type-mismatch.rs","byte_start":1472,"byte_end":1473,"line_start":59,"line_end":59,"column_start":12,"column_end":13,"is_primary":true,"text":[{"text":"        x: 9,  //~ ERROR mismatched types","highlight_start":12,"highlight_end":13}],"label":"expected f32, found integral variable","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `f32`\n   found type `{integer}`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"use a float literal","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/structs/structure-constructor-type-mismatch.rs","byte_start":1472,"byte_end":1473,"line_start":59,"line_end":59,"column_start":12,"column_end":13,"is_primary":true,"text":found 1\n  --> /checkout/src/test/ui/structs/structure-constructor-type-mismatch.rs:64:18\n   |\nLL |         PointF::<u32> { .. } => {} //~ ERROR wrong number of type arguments\n   |                  ^^^ unexpected type argument\n\n"}
[00:46:53] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n