\n\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0404.rs","byte_start":497,"byte_end":500,"line_start":14,"line_end":14,"column_start":6,"column_end":9,"is_primary":true,"text":[{"text":"impl Foo for Bar {} //~ ERROR E0404","highlight_start":6,"highlight_end":9}],"label":"not a trait","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0404]: expected trait, found struct `Foo`\n  --> /checkout/src/test/ui/error-codes/E0404.rs:14:6\n   |\nLL | impl Foo for Bar {} //~ ERROR E0404\n   |      ^^^ not a trait\n\n"}
[00:47:57] {"message":"expected trait, found struct `Foo`","code":{"code":"E0404","explanation":"\nYou tried to use something which is not a trait in a trait position, such as\na bound or `impl`.\n\nErroneous code example:\n\n