\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":1021,"byte_end":1029,"line_start":23,"line_end":23,"column_start":14,"column_end":22,"is_primary":true,"text":[{"text":"        impl Iterator for Test {} //~ ERROR cannot find trait `Iterator` in this scope","highlight_start":14,"highlight_end":22}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"possible candidates are found in other modules, you can import them into scope","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":837,"byte_end":837,"line_start":20,"line_end":20,"column_start":9,"column_end":9,"is_primary":true,"text":[{"text":"        struct Test;","highlight_start":9,"highlight_end":9}],"label":null,"suggested_replacement":"use std::iter::Iterator;\n\n","suggestion_applicability":"Unspecified","expansion":null},{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":837,"byte_end":837,"line_start":20,"line_end":20,"column_start":9,"column_end":9,"is_primary":true,"text":[{"text":"        struct Test;","highlight_start":9,"highlight_end":9}],"label":null,"suggested_replacement":"use std::prelude::v1::Iterator;\n\n","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0405]: cannot find trait `Iterator` in this scope\n  --> /checkout/src/test/ui/no-implicit-prelude-nested.rs:23:14\n   |\nLL |         impl Iterator for Test {} //~ ERROR cannot  pub static foo : i32 = 0i32;\n}\n\nsomething_that_does_exist::foo; // ok!\n