\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":2216,"byte_end":2224,"line_start":51,"line_end":51,"column_start":14,"column_end":22,"is_primary":true,"text":[{"text":"        impl ToString for Test {} //~ ERROR cannot find trait `ToString` in this scope","highlight_start":14,"highlight_end":22}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"possible candidates are found in other modules, you can import them into scope","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":1945,"byte_end":1945,"line_start":47,"line_end":47,"column_start":9,"column_end":9,"is_primary":true,"text":[{"text":"        struct Test;","highlight_start":9,"highlight_end":9}],"label":null,"suggested_replacement":"use std::prelude::v1::ToString;\n\n","suggestion_applicability":"Unspecified","expansion":null},{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":1945,"byte_end":1945,"line_start":47,"line_end":47,"column_start":9,"column_end":9,"is_primary":true,"text":[{"text":"        struct Test;","highlight_start":9,"highlight_end":9}],"label":null,"suggested_replacement":"use std::string::ToString;\n\n","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0405]: cannot find trait `ToString` in this scope\n  --> /checkout/src/test/ui/no-implicit-prelude-nested.rs:51:14\n   |\nLL |         impl ToString for Test {} //~ ERROR cannot find trait `ToString` in this scope\n   |              ^^^^^^^^ not found in this scope\nhelp: possible candidates are found in other modules, you can import them into scope\n   |\nLL |         use std::prelude::v1::ToString;\n   |\nLL |         use std::string::ToString;\n   |\n\n"}
[00:47:57] {"message":"cannot find trait `Writer` in this scope","code":{"code":"E0405","explanation":"\nThe code refers to a trait that is not in scope.\n\nErroneous code example:\n\n