\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":940,"byte_end":945,"line_start":22,"line_end":22,"column_start":14,"column_end":19,"is_primary":true,"text":[{"text":"        impl Clone for Test {} //~ ERROR cannot find trait `Clone` in this scope","highlight_start":14,"highlight_end":19}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"possible candidates are found in other modules, you can import them into scope","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":837,"byte_end":837,"line_start":20,"line_end":20,"column_start":9,"column_end":9,"is_primary":true,"text":[{"text":"        struct Test;","highlight_start":9,"highlight_end":9}],"label":null,"suggested_replacement":"use std::clone::Clone;\n\n","suggestion_applicability":"Unspecified","expansion":null},{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":837,"byte_end":837,"line_start":20,"line_end":20,"column_start":9,"column_end":9,"is_primary":true,"text":[{"text":"        struct Test;","highlight_start":9,"highlight_end":9}],"label":null,"suggested_replacement":"use std::prelude::v1::Clone;\n\n","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0405]: cannot find trait `Clone` in this scope\n  --> /checkout/src/test/ui/no-implicit-prelude-nested.rs:22:14\n   |\nLL |         impl Clone for Test {} //~ ERROR cannot find trait `Clone` in this scope\n   |              ^^^^^ not found in this scope\nhelp: possible candidates are found in other modules, you can import them into scope\n   |\nLL |         use std::clone::Clone;\n   |\nLL |         use std::prelude::v1::Clone;\n   |\n\n"}
[00:47:57] {"message":"cannot find trait `Iterator` in this scope","code":{"code":"E0405","explanation":"\nThe code refers to a trait that is not in scope.\n\nErroneous code example:\n\n