\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/resolve/resolve-self-in-impl-2.rs","byte_start":514,"byte_end":518,"line_start":14,"line_end":14,"column_start":6,"column_end":10,"is_primary":true,"text":[{"text":"impl Self for S {} //~ ERROR expected trait, found self type `Self`","highlight_start":6,"highlight_end":10}],"label":"`Self` is only available in traits and impls","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0411]: expected trait, found self type `Self`\n  --> /checkout/src/test/ui/resolve/resolve-self-in-impl-2.rs:14:6\n   |\nLL | impl Self for S {} //~ ERROR expected trait, found self type `Self`\n   |      ^^^^ `Self` is only available in traits and impls\n\n"}
[00:47:57] {"message":"cannot find trait `N` in `Self`","code":{"code":"E0405","explanation":"\nThe code refers to a trait that is not in scope.\n\nErroneous code example:\n\n