\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":2129,"byte_end":2137,"line_start":50,"line_end":50,"column_start":14,"column_end":22,"is_primary":true,"text":[{"text":"        impl Iterator for Test {} //~ ERROR cannot find trait `Iterator` in this scope","highlight_start":14,"highlight_end":22}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"possible candidates are found in other modules, you can import them into scope","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":1945,"byte_end":1945,"line_start":47,"line_end":47,"column_start":9,"column_end":9,"is_primary":true,"text":[{"text":"        struct Test;","highlight_start":9,"highlight_end":9}],"label":null,"suggested_replacement":"use std::iter::Iterator;\n\n","suggestion_applicability":"Unspecified","expansion":null},{"file_name":"/checkout/src/test/ui/no-implicit-prelude-nested.rs","byte_start":1945,"byte_end":1945,"line_start":47,"line_end":47,"column_start":9,"column_end":9,"is_primary":true,"text":[{"text":"        struct Test;","highlight_start":9,"highlight_end":9}],"label":null,"suggested_replacement":"use std::prelude::v1::Iterator;\n\n","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0405]: cannot find trait `Iterator` in this scope\n  --> /checkout/src/test/ui/no-implicit-prelude-nested.rs:50:14\n   |\nLL |         impl Iterator for Test {} //~ ERROR cannot find trait `Iterator` in this scope\n   |              ^^^^^^^^ not found in this scope\nhelp: possible candidates are found in other modules, you can import them into scope\n   |\nLL |         use std::iter::Iterator;\n   |\nLL |         use std::prelude::v1::Iterator;\n   |\n\n"}
[00:47:57] {"message":"cannot find trait `ToString` in this scope","code":{"code":"E0405","explanation":"\nThe code refers to a trait that is not in scope.\n\nErroneous code example:\n\n