\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/self/self_type_keyword-2.rs","byte_start":471,"byte_end":488,"line_start":11,"line_end":11,"column_start":5,"column_end":22,"is_primary":true,"text":[{"text":"use self::Self as Foo; //~ ERROR unresolved import `self::Self`","highlight_start":5,"highlight_end":22}],"label":"no `Self` in the root","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0432]: unresolved import `self::Self`\n  --> /checkout/src/test/ui/self/self_type_keyword-2.rs:11:5\n   |\nLL | use self::Self as Foo; //~ ERROR unresolved import `self::Self`\n   |     ^^^^^^^^^^^^^^^^^ no `Self` in the root\n\n"}
[00:47:41] {"message":"cannot find unit struct/variant or constant `Self` in this scope","code":{"code":"E0531","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/self/self_type_keyword-2.rs","byte_start":556,"byte_end":560,"line_start":14,"line_end":14,"column_start":9,"column_end":13,"is_primary":true,"text":[{"text":"    let Self = 5;","highlight_start":9,"highlight_end":13}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0531]: cannot find unit struct/variant or constant `Self` in this scope\n  --> /checkout/src/test/ui/self/self_type_keyword-2.rs:14:9\n   |\nLL |     let Self = 5;\n   |         ^^^^ not found in this scope\n\n"}
[00:47:41] {"message":"cannot find unit struct/variant or constant `Self` in this scope","code":{"code":"E0531","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/self/self_type_keyword-2.rs","byte_start":749,"byte_end":753,"line_start":19,"line_end":19,"column_start":9,"column_end":13,"is_primary":true,"text":[{"text":"        Self => (),","highlight_start":9,"highlight_end":13}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0531]: cannot find unit struct/variant or constant `Self` in this scope\n  --> /checkout/src/test/ui/self/self_type_keyword-2.rs:19:9\n   |\nLL |         Self => (),\n   |         ^^^^ not found in this scope\n\n"}
[00:47:41] {"message":"cannot find unit struct/variant or constant `Self` in this scope","code":{"code":"E0531","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/self/self_type_keyword-2.rs","byte_start":945,"byte_end":949,"line_start":22,"struct_self_ctor)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: tuple struct Self constructors are unstable (see issue #51994)\n  --> /checkout/src/test/ui/self/self_type_keyword-2.rs:14:9\n   |\nLL |     let Self = 5;\n   |         ^^^^\n   |\n   = help: add #![feature(tuple_struct_self_ctor)] to the crate attributes to enable\n\n"}
[00:47:41] {"message":"tuple struct Self constructors are unstable (see issue #51994)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n