\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs","byte_start":911,"byte_end":933,"line_start":33,"line_end":33,"column_start":25,"column_end":47,"is_primary":true,"text":[{"text":"    Ref2Variant2(isize, RequireOutlives<'a, T>), //~ ERROR the parameter type `T` may not live long enough","highlight_start":25,"highlight_end":47}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"...so that the type `T` will meet its required lifetime bounds","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs","byte_start":911,"byte_end":933,"line_start":33,"line_end":33,"column_start":25,"column_end":47,"is_primary":true,"text":[{"text":"    Ref2Variant2(isize, RequireOutlives<'a, T>), //~ ERROR the parameter type `T` may not live long enough","highlight_start":25,"highlight_end":47}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"consider adding an explicit lifetime bound `T: 'a`...","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs","byte_start":864,"byte_end":865,"line_start":31,"line_end":31,"column_start":15,"column_end":16,"is_primary":true,"text":[{"text":"enum Ref2<'a, T> {","highlight_start":15,"highlight_end":16}],"label":null,"suggested_replacement":"T: 'a","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0309]: the parameter type `T` may not live long enough\n  --> /checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs:33:25\n   |\nLL | enum Ref2<'a, T> {\n   |               - help: consider adding an explicit lifetime bound `T: 'a`...\nLL |     Ref2Variant1,\nLL |     Ref2Variant2(isize, RequireOutlives<'a, T>), //~ ERROR the parameter type `T` may not live long enough\n   |                         ^^^^^^^^^^^^^^^^^^^^^^\n   |\nnote: ...so that the type `T` will meet its required lifetime bounds\n  --> /checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs:33:25\n   |\nLL |     Ref2Variant2(isize, RequireOutlives<'a, T>), //~ ERROR the parameter type `T` may not live long enough\n   |                         ^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:47:21] {"message":"the parameter type `T` may not live long enough","code":{"code":"E0309","explanation":"\nTypes in type definitions have lifetimes associated with them that represent\nhow long the data stored within them is guaranteed to be live. This lifetime\nmust be as long as the data needs to be alive, and missing the constraint that\ndenotes this will cause this error.\n\n