\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs","byte_start":1286,"byte_end":1312,"line_start":46,"line_end":46,"column_start":23,"column_end":49,"is_primary":true,"text":[{"text":"    RefDoubleVariant1(&'a RequireOutlives<'b, T>)","highlight_start":23,"highlight_end":49}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"...so that the type `T` will meet its required lifetime bounds","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs","byte_start":1286,"byte_end":1312,"line_start":46,"line_end":46,"column_start":23,"column_end":49,"is_primary":true,"text":[{"text":"    RefDoubleVariant1(&'a RequireOutlives<'b, T>)","highlight_start":23,"highlight_end":49}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"consider adding an explicit lifetime bound `T: 'b`...","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs","byte_start":1181,"byte_end":1182,"line_start":45,"line_end":45,"column_start":24,"column_end":25,"is_primary":true,"text":[{"text":"enum RefDouble<'a, 'b, T> { //~ ERROR 43:1: 46:2: the parameter type `T` may not live long enough [E0309]","highlight_start":24,"highlight_end":25}],"label":null,"suggested_replacement":"T: 'b","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0309]: the parameter type `T` may not live long enough\n  --> /checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs:46:23\n   |\nLL | enum RefDouble<'a, 'b, T> { //~ ERROR 43:1: 46:2: the parameter type `T` may not live long enough [E0309]\n   |                        - help: consider adding an explicit lifetime bound `T: 'b`...\nLL |     RefDoubleVariant1(&'a RequireOutlives<'b, T>)\n   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\nnote: ...so that the type `T` will meet its required lifetime bounds\n  --> /checkout/src/test/ui/rfc-2093-infer-outlives/regions-enum-not-wf.rs:46:23\n   |\nLL |     RefDoubleVariant1(&'a RequireOutlives<'b, T>)\n   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:47:21] {"message":"aborting due to 4 previous errors","code":null,"level":"error","spans":[],"
