\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/const-eval/issue-52475.rs","byte_start":716,"byte_end":875,"line_start":19,"line_end":22,"column_start":9,"column_end":10,"is_primary":true,"text":[{"text":"        while n < 5 { //~ ERROR constant contains unimplemented expression type","highlight_start":9,"highlight_end":80},{"text":"            n = (n + 1) % 5;","highlight_start":1,"highlight_end":29},{"text":"            x = &0; // Materialize a new AllocId","highlight_start":1,"highlight_end":49},{"text":"        }","highlight_start":1,"highlight_end":10}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0019]: constant contains unimplemented expression type\n  --> /checkout/src/test/ui/consts/const-eval/issue-52475.rs:19:9\n   |\nLL | /         while n < 5 { //~ ERROR constant contains unimplemented expression type\nLL | |             n = (n + 1) % 5;\nLL | |             x = &0; // Materialize a new AllocId\nLL | |         }\n   | |_________^\n\n"}
[00:46:41] {"message":"Constant evaluating a complex constant, this might take some time","code":null,"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/consts/const-eval/issue-52475.rs","byte_start":520,"byte_end":891,"line_start":14,"line_end":24,"column_station":"\nThis error indicates that the compiler was unable to sensibly evaluate an\nconstant expression that had to be evaluated. Attempting to divide by 0\nor causing integer overflow are two ways to induce this error. For example:\n\n