\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs","byte_start":1045,"byte_end":1050,"line_start":32,"line_end":32,"column_start":28,"column_end":33,"is_primary":true,"text":[{"text":"    let _: &'static u32 = &meh(); //~ ERROR does not live long enough","highlight_start":28,"highlight_end":33}],"label":"temporary value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs","byte_start":1202,"byte_end":1203,"line_start":35,"line_end":35,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"temporary value only lives until here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the static lifetime...","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0597]: borrowed value does not live long enough\n  --> /checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs:32:28\n   |\nLL |     let _: &'static u32 = &meh(); //~ ERROR does not live long enough\n   |                            ^^^^^ temporary value does not live long enough\n...\nLL | }\n   | - temporary value only lives until here\n   |\n   = note: borrowed value must be valid for the static lifetime...\n\n"}
[00:49:13] {"message":"borrowed value does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n