plain
[00:47:21] .................................................................i...i................ii............
[00:47:24] ...................................................................................iii..............
[00:47:27] ......................................................................i..........i..................
[00:47:30] ........i...........................................................................................
[00:47:35] ............................F...F...................................................................
[00:47:41] ...........................ii.iii...................................................................
[00:47:44] ....................................................................................................
[00:47:46] ....................................................................................................
[00:47:48] ....................................................................................................
---
[00:48:07] ....................................................................................................
[00:48:10] ....................................................................................................
[00:48:13] ...........................i........................................................................
[00:48:16] ....................................................................................................
[00:48:18] ............................................................................iiiiiiiii...............
[00:48:24] ....................................................................................................
[00:48:27] ....................................................................................................
[00:48:30] .........................................................i..........................................
[00:48:33] ....................................................................................................
---
[00:49:05] ....................................................................................................
[00:49:09] ....................................................................................................
[00:49:11] ................................................................................................i.ii
..
[00:49:13] 25 LL | }
[00:49:13] 26    | - temporary value only lives until here
[00:49:13] 
[00:49:13] 
[00:49:13] The actual stderr differed from the expected stderr.
[00:49:13] The actual stderr differed from the expected stderr.
[00:49:13] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-eval/dont_promote_unstable_const_fn.nll/dont_promote_unstable_const_fn.nll.stderr
[00:49:13] To update references, rerun the tests and pass the `--bless` flag
[00:49:13] To only update this specific test, also pass `--test-args consts/const-eval/dont_promote_unstable_const_fn.rs`
[00:49:13] error: 1 errors occurred comparing output.
[00:49:13] status: exit code: 1
[00:49:13] status: exit code: 1
[00:49:13] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-eval/dont_promote_unstable_const_fn.nll/a" "-Zborrowck=mir" "-Ztwo-phase-borrows" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-eval/dont_promote_unstable_const_fn.nll/auxiliary" "-A" "unused"
[00:49:13] ------------------------------------------
[00:49:13] 
[00:49:13] ------------------------------------------
[00:49:13] stderr:
[00:49:13] stderr:
[00:49:13] ------------------------------------------
[00:49:13] {"message":"`foo` is not yet stable as a const fn","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs","byte_start":867,"byte_end":872,"line_start":25,"line_end":25,"column_start":25,"column_end":30,"is_primary":true,"text":[{"text":"const fn bar() -> u32 { foo() } //~ ERROR `foo` is not yet stable as a const fn","highlight_start":25,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"in Nightly builds, add `#![feature(foo)]` to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: `foo` is not yet stable as a const fn\n  --> /checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs:25:25\n   |\nLL | const fn bar() -> u32 { foo() } //~ ERROR `foo` is not yet stable as a const fn\n   |                         ^^^^^\n   |\n   = help: in Nightly builds, add `#![feature(foo)]` to the crate attributes to enable\n\n"}
[00:49:13] {"message":"borrowed value does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n