\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs","byte_start":1113,"byte_end":1165,"line_start":33,"line_end":33,"column_start":26,"column_end":78,"is_primary":true,"text":[{"text":"    let x: &'static _ = &std::time::Duration::from_millis(42).subsec_millis();","highlight_start":26,"highlight_end":78}],"label":"temporary value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs","byte_start":1202,"byte_end":1203,"line_start":35,"line_end":35,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"temporary value only lives until here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the static lifetime...","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0597]: borrowed value does not live long enough\n  --> /checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs:33:26\n   |\nLL |     let x: &'static _ = &std::time::Duration::from_millis(42).subsec_millis();\n   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ temporary value does not live long enough\nLL |     //~^ does not live long enough\nLL | }\n   | - temporary value only lives until here\n   |\n   = note: borrowed value must be valid for the static lifetime...\n\n"}
[00:49:13] {"message":"aborting due to 4 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 4 previous errors\n\n"}
[00:49:13] {"message":"For more information about this error, try `rustc --explain E0597`.","code":null,value only lives until here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the static lifetime...","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0597]: borrowed value does not live long enough\n  --> /checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn_cross_crate.rs:18:28\n   |\nLL |     let _: &'static u32 = &foo(); //~ ERROR does not live long enough\n   |                            ^^^^^ temporary value does not live long enough\nLL |     let _x: &'static u32 = &foo(); //~ ERROR does not live long enough\nLL | }\n   | - temporary value only lives until here\n   |\n   = note: borrowed value must be valid for the static lifetime...\n\n"}
[00:49:13] {"message":"borrowed value does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n