\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/uninhabited/uninhabited-matches-feature-gated.rs","byte_start":559,"byte_end":560,"line_start":15,"line_end":15,"column_start":19,"column_end":20,"is_primary":true,"text":[{"text":"    let _ = match x {   //~ ERROR non-exhaustive","highlight_start":19,"highlight_end":20}],"label":"pattern `Err(_)` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `Err(_)` not covered\n  --> /checkout/src/test/ui/uninhabited/uninhabited-matches-feature-gated.rs:15:19\n   |\nLL |     let _ = match x {   //~ ERROR non-exhaustive\n   |                   ^ pattern `Err(_)` not covered\n\n"}
[00:53:48] {"message":"non-exhaustive patterns: type &Void is non-empty","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n