\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/uninhabited/uninhabited-matches-feature-gated.rs","byte_start":1226,"byte_end":1227,"line_start":37,"line_end":37,"column_start":19,"column_end":20,"is_primary":true,"text":[{"text":"    let _ = match x {   //~ ERROR non-exhaustive","highlight_start":19,"highlight_end":20}],"label":"pattern `Err(_)` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `Err(_)` not covered\n  --> /checkout/src/test/ui/uninhabited/uninhabited-matches-feature-gated.rs:37:19\n   |\nLL |     let _ = match x {   //~ ERROR non-exhaustive\n   |                   ^ pattern `Err(_)` not covered\n\n"}
[00:53:48] {"message":"refutable pattern in local binding: `Err(_)` not covered","code":{"code":"E0005","explanation":"\nPatterns used to bind names must be irrefutable, that is, they must guarantee\nthat a name will be extracted in all cases. Erroneous code example:\n\n