\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/uninhabited/uninhabited-matches-feature-gated.rs","byte_start":693,"byte_end":694,"line_start":20,"line_end":20,"column_start":19,"column_end":20,"is_primary":true,"text":[{"text":"    let _ = match x {}; //~ ERROR non-exhaustive","highlight_start":19,"highlight_end":20}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"Please ensure that all possible cases are being handled; possibly adding wildcards or more match arms.","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/uninhabited/uninhabited-matches-feature-gated.rs","byte_start":693,"byte_end":694,"line_start":20,"line_end":20,"column_start":19,"column_end":20,"is_primary":true,"text":[{"text":"    let _ = match x {}; //~ ERROR non-exhaustive","highlight_start":19,"highlight_end":20}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0004]: non-exhaustive patterns: type &Void is non-empty\n  --> /checkout/src/test/ui/uninhabited/uninhabited-matches-feature-gated.rs:20:19\n   |\nLL |     let _ = match x {}; //~ ERROR non-exhaustive\n   |                   ^\n   |\nhelp: Please ensure that all possible cases are being handled; possibly adding wildcards or more match arms.\n  --> /checkout/src/test/ui/uninhabited/uninhabited-matches-feature-gated.rs:20:19\n   |\nLL |     let _ = match x {}; //~ ERROR non-exhaustive\n   |                   ^\n\n"}
[00:53:48] {"message":"non-exhaustive patterns: `&[_]` not covered","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n