\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/suggestions/suggest-methods.rs","byte_start":739,"byte_end":747,"line_start":31,"line_end":31,"column_start":15,"column_end":23,"is_primary":true,"text":[{"text":"    let _ = s.is_emtpy(); //~ ERROR no method named","highlight_start":15,"highlight_end":23}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"did you mean `is_empty`?","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0599]: no method named `is_emtpy` found for type `std::string::String` in the current scope\n  --> /checkout/src/test/ui/suggestions/suggest-methods.rs:31:15\n   |\nLL |     let _ = s.is_emtpy(); //~ ERROR no method named\n   |               ^^^^^^^^\n   |\n   = help: did you mean `is_empty`?\n\n"}
[00:51:58] {"message":"no method named `count_eos` found for type `u32` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n