\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/const-int-wrapping.rs","byte_start":770,"byte_end":793,"line_start":15,"line_end":15,"column_start":28,"column_end":51,"is_primary":true,"text":[{"text":"    let a: &'static i32 = &(5_i32.wrapping_shl(3)); //~ ERROR does not live long enough","highlight_start":28,"highlight_end":51}],"label":"creates a temporary which is freed while still in use","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/consts/const-int-wrapping.rs","byte_start":919,"byte_end":920,"line_start":17,"line_end":17,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"temporary value is freed at the end of this statement","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the static lifetime...","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0716]: temporary value dropped while borrowed\n  --> /checkout/src/test/ui/consts/const-int-wrapping.rs:15:28\n   |\nLL |     let a: &'static i32 = &(5_i32.wrapping_shl(3)); //~ ERROR does not live long enough\n   |                            ^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use\nLL |     let b: &'static i32 = &(5_i32.wrapping_shr(3)); //~ ERROR does not live long enough\nLL | }\n   | - temporary value is freed at the end of this statement\n   |\n   = note: borrowed value must be valid for the static lifetime...\n\n"}
[00:59:39] {"message":"temporary value dropped while borrowed","code":{"code":"E0716","explanation":"\nThis error indicates that a temporary value is being dropped\nwhile a borrow is still in active use.\n\nErroneous code example:\n\n