\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/suggestions/suggest-methods.rs","byte_start":467,"byte_end":478,"line_start":11,"line_end":11,"column_start":1,"column_end":12,"is_primary":false,"text":[{"text":"struct Foo;","highlight_start":1,"highlight_end":12}],"label":"method `bat` not found for this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/suggestions/suggest-methods.rs","byte_start":657,"byte_end":660,"line_start":28,"line_end":28,"column_start":7,"column_end":10,"is_primary":true,"text":[{"text":"    f.bat(1.0); //~ ERROR no method named","highlight_start":7,"highlight_end":10}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"did you mean `bag`?","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0599]: no method named `bat` found for type `Foo` in the current scope\n  --> /checkout/src/test/ui/suggestions/suggest-methods.rs:28:7\n   |\nLL | struct Foo;\n   | ----------- method `bat` not found for this\n...\nLL |     f.bat(1.0); //~ ERROR no method named\n   |       ^^^\n   |\n   = help: did you mean `bag`?\n\n"}
[00:51:58] {"message":"no method named `is_emtpy` found for type `std::string::String` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n