\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":506,"byte_end":515,"line_start":13,"line_end":13,"column_start":1,"column_end":10,"is_primary":false,"text":[{"text":"struct S;","highlight_start":1,"highlight_end":10}],"label":"associated item `B` not found for this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":2611,"byte_end":2615,"line_start":108,"line_end":108,"column_start":5,"column_end":9,"is_primary":true,"text":[{"text":"    S::B; //~ ERROR no associated item named `B` found for type `S` in the current scope","highlight_start":5,"highlight_end":9}],"label":"associated item not found in `S`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"items from traits can only be used if the trait is in scope","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"did you mean `b`?","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"the following trait is implemented but not in scope, perhaps add a `use` for it:","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":506,"byte_end":506,"line_start":13,"line_end":13,"column_start":1,"column_end":1,"is_primary":true,"text":[{"text":"struct S;","highlight_start":1,"highlight_end":1}],"label":null,"suggested_replacement":"use assoc_const::B;\n\n","suggestion_applicability":"Unspecified","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0599]: no associated item named `B` found for type `S` in the current scope\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:108:5\n   |\nLL | struct S;\n   | --------- associated item `B` not found for this\n...\nLL |     S::B; //~ ERROR no associated item named `B` found for type `S` in the current scope\n   |     ^^^^ associated item not found in `S`\n   |\n   = help: items from traits can only be used if the trait is in scope\n   = help: did you mean `b`?\nhelp: the following trait is implemented but not in scope, perhaps add a `use` for it:\n   |\nLL | use assoc_const::B;\n   |\n\n"}
[00:51:58] {"message":"associated constant `A` is private","code":{"code":"E0624","explanation":"\nA private item was used outside of its scope.\n\nErroneous code example:\n\n