\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs","byte_start":2591,"byte_end":2598,"line_start":82,"line_end":82,"column_start":71,"column_end":78,"is_primary":true,"text":[{"text":"    std::rc::Rc::new(&mut Box::new(&no_method_suggested_traits::Foo)).method3();","highlight_start":71,"highlight_end":78}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&no_method_suggested_traits::Foo>>` in the current scope\n  --> /checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs:82:71\n   |\nLL |     std::rc::Rc::new(&mut Box::new(&no_method_suggested_traits::Foo)).method3();\n   |                                                                       ^^^^^^^\n\n"}
[00:51:56] {"messa,"rendered":null},{"message":"to disambiguate the method call, write `CtxtFn::f9(u, 342)` instead","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"candidate #2 is defined in the trait `OtherTrait`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/span/issue-7575.rs","byte_start":679,"byte_end":704,"line_start":20,"line_end":20,"column_start":5,"column_end":30,"is_primary":true,"text":[{"text":"    fn f9(_: usize) -> usize;","highlight_start":5,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"to disambiguate the method call, write `OtherTrait::f9(u, 342)` instead","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"candidate #3 is defined in the trait `UnusedTrait`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/span/issue-7575.rs","byte_start":1022,"byte_end":1047,"line_start":29,"line_end":29,"column_start":5,"column_end":30,"is_primary":true,"text":[{"text":"    fn f9(_: usize) -> usize;","highlight_start":5,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"to disambiguate the method call, write `UnusedTrait::f9(u, 342)` instead","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"items from traits can only be used if the trait is implemented and in scope","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"the following traits define an item `f9`,pl for the type `Myisize`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/span/issue-7575.rs","byte_start":1312,"byte_end":1337,"line_start":51,"line_end":51,"column_start":5,"column_end":30,"is_primary":true,"text":[{"text":"    fn fff(i: isize) -> isize {","highlight_start":5,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"use associated function syntax instead","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/span/issue-7575.rs","byte_start":1718,"byte_end":1723,"line_start":74,"line_end":74,"column_start":28,"column_end":33,"is_primary":true,"text":[{"text":"    u.f8(42) + u.f9(342) + m.fff(42)","highlight_start":28,"highlight_end":33}],"label":null,"suggested_replacement":"Myisize::fff","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0599]: no method named `fff` found for type `Myisize` in the current scope\n  --> /checkout/src/test/ui/span/issue-7575.rs:74:30\n   |\nLL | struct Myisize(isize);\n   | ---------------------- method `fff` not found for this\n...\nLL |     u.f8(42) + u.f9(342) + m.fff(42)\n   |                            --^^^\n   |                            | |\n   |                            | this is an associated function, not a method\n   |                            help: use associated function syntax instead: `Myisize::fff`\n   |\n   = note: found the following associated functions; to be used as methods, functions must have a `self` parameter\nnote: the candidate is defined i.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0599`.\n"}
[00:51:56] ------------------------------------------
[00:51:56] 
[00:51:56] thread '[ui] ui/span/issue-7575.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3205:9
[00:51:56] 
[00:51:56] 
[00:51:56] ---- [ui] ui/suggestions/suggest-methods.rs stdout ----
[00:51:56] diff of stderr:
[00:51:56] 
[00:51:56] 7 LL |     f.bat(1.0); //~ ERROR no method named
[00:51:56] 9    |
[00:51:56] -    = help: did you mean `bar`?
[00:51:56] +    = help: did you mean `bag`?
[00:51:56] 11 
[00:51:56] 11 
[00:51:56] 12 error[E0599]: no method named `is_emtpy` found for type `std::string::String` in the current scope
[00:51:56] 13   --> $DIR/suggest-methods.rs:31:15
[00:51:56] 
[00:51:56] The actual stderr differed from the expected stderr.
[00:51:56] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/suggestions/suggest-methods/suggest-methods.stderr
[00:51:56] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/suggestions/suggest-methods/suggest-methods.stderr
[00:51:56] To update references, rerun the tests and pass the `--bless` flag
[00:51:56] To only update this specific test, also pass `--test-args suggestions/suggest-methods.rs`
[00:51:56] error: 1 errors occurred comparing output.
[00:51:56] status: exit code: 1
[00:51:56] status: exit code: 1
[00:51:56] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/suggestions/suggest-methods.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/suggestions_emtpy(); //~ ERROR no method named\n   |               ^^^^^^^^\n   |\n   = help: did you mean `is_empty`?\n\n"}
[00:51:56] {"message":"no method named `count_eos` found for type `u32` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n