\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":506,"byte_end":515,"line_start":13,"line_end":13,"column_start":1,"column_end":10,"is_primary":false,"text":[{"text":"struct S;","highlight_start":1,"highlight_end":10}],"label":"method `a` not found for this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":1449,"byte_end":1450,"line_start":77,"line_end":77,"column_start":7,"column_end":8,"is_primary":true,"text":[{"text":"    S.a(); //~ ERROR no method named `a` found for type `S` in the current scope","highlight_start":7,"highlight_end":8}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"items from traits can only be used if the trait is implemented and in scope","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"the following trait defines an item `a`, perhaps you need to implement it:\ncandidate #1: `method::A`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"did you mean `b`?","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0599]: no method named `a` found for type `S` in the current scope\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:77:7\n   |\nLL | struct S;\n   | --------- method `a` not found for this\n...\nLL |     S.a(); //~ ERROR no method named `a` found for type `S` in the current scope\n   |       ^\n   |\n   = help: items from traits can only be used if the trait is implemented and in scope\n   = note: the following trait defines an item `a`, perhaps you need to implement it:\n           candidate #1: `method::A`\n   = help: did you mean `b`?\n\n"}
[00:51:58] {"message":"no method named `b` found for type `S` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n