\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/traits/trait-item-privacy.rs","byte_start":1736,"byte_end":1737,"line_start":82,"line_end":82,"column_start":7,"column_end":8,"is_primary":true,"text":[{"text":"    c.a(); //~ ERROR method `a` is private","highlight_start":7,"highlight_end":8}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0624]: method `a` is private\n  --> /checkout/src/test/ui/traits/trait-item-privacy.rs:82:7\n   |\nLL |     c.a(); //~ ERROR method `a` is private\n   |       ^\n\n"}
[00:51:58] {"message":"no function or associated item named `a` found for type `S` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n