\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs","byte_start":2410,"byte_end":2417,"line_start":80,"line_end":80,"column_start":47,"column_end":54,"is_primary":true,"text":[{"text":"    std::rc::Rc::new(&mut Box::new(&1_usize)).method3(); //~ ERROR no method named","highlight_start":47,"highlight_end":54}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&usize>>` in the current scope\n  --> /checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs:80:47\n   |\nLL |     std::rc::Rc::new(&mut Box::new(&1_usize)).method3(); //~ ERROR no method named\n   |                                               ^^^^^^^\n\n"}
[00:51:58] {"message":"no method named `method3` found for type `no_method_suggested_traits::Foo` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n