\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs","byte_start":2672,"byte_end":2679,"line_start":84,"line_end":84,"column_start":40,"column_end":47,"is_primary":true,"text":[{"text":"    no_method_suggested_traits::Bar::X.method3();  //~ ERROR no method named","highlight_start":40,"highlight_end":47}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `method3` found for type `no_method_suggested_traits::Bar` in the current scope\n  --> /checkout/src/test/ui/impl-trait/no-method-suggested-traits.rs:84:40\n   |\nLL |     no_method_suggested_traits::Bar::X.method3();  //~ ERROR no method named\n   |                                        ^^^^^^^\n\n"}
[00:51:58] {"message":"no method named `method3` found for type `std::rc::Rc<&mut std::boxed::Box<&no_method_suggested_traits::Bar>>` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n