\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/const-int-sign.rs","byte_start":507,"byte_end":528,"line_start":12,"line_end":12,"column_start":29,"column_end":50,"is_primary":true,"text":[{"text":"    let x: &'static bool = &(5_i32.is_negative()); //~ ERROR does not live long enough","highlight_start":29,"highlight_end":50}],"label":"creates a temporary which is freed while still in use","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/consts/const-int-sign.rs","byte_start":653,"byte_end":654,"line_start":14,"line_end":14,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"temporary value is freed at the end of this statement","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the static lifetime...","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0716]: temporary value dropped while borrowed\n  --> /checkout/src/test/ui/consts/const-int-sign.rs:12:29\n   |\nLL |     let x: &'static bool = &(5_i32.is_negative()); //~ ERROR does not live long enough\n   |                             ^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use\nLL |     let y: &'static bool = &(5_i32.is_positive()); //~ ERROR does not live long enough\nLL | }\n   | - temporary value is freed at the end of this statement\n   |\n   = note: borrowed value must be valid for the static lifetime...\n\n"}
[00:59:39] {"message":"temporary value dropped while borrowed","code":{"code":"E0716","explanation":"\nThis error indicates that a temporary value is being dropped\nwhile a borrow is still in active use.\n\nErroneous code example:\n\n