\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/suggestions/suggest-methods.rs","byte_start":915,"byte_end":924,"line_start":35,"line_end":35,"column_start":19,"column_end":28,"is_primary":true,"text":[{"text":"    let _ = 63u32.count_eos(); //~ ERROR no method named","highlight_start":19,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"did you mean `count_zeros`?","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0599]: no method named `count_eos` found for type `u32` in the current scope\n  --> /checkout/src/test/ui/suggestions/suggest-methods.rs:35:19\n   |\nLL |     let _ = 63u32.count_eos(); //~ ERROR no method named\n   |                   ^^^^^^^^^\n   |\n   = help: did you mean `count_zeros`?\n\n"}
[00:51:58] {"message":"no method named `count_o` found for type `u32` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n