\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/empty/empty-struct-braces-pat-2.rs","byte_start":700,"byte_end":706,"line_start":25,"line_end":25,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"        Empty1() => () //~ ERROR expected tuple struct/variant, found struct `Empty1`","highlight_start":9,"highlight_end":15}],"label":"did you mean `XEmpty6`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0532]: expected tuple struct/variant, found struct `Empty1`\n  --> /checkout/src/test/ui/empty/empty-struct-braces-pat-2.rs:25:9\n   |\nLL |         Empty1() => () //~ ERROR expected tuple struct/variant, found struct `Empty1`\n   |         ^^^^^^ did you mean `XEmpty6`?\n\n"}
[01:05:24] {"message":"expected tuple struct/variant, found struct `XEmpty1`","code":{"code":"E0532","explanation":"\nPattern arm did not match expected kind.\n\nErroneous code example:\n\n