\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/empty/empty-struct-braces-expr.rs","byte_start":911,"byte_end":920,"line_start":28,"line_end":28,"column_start":14,"column_end":23,"is_primary":true,"text":[{"text":"    let e3 = E::Empty3(); //~ ERROR expected function, found struct variant `E::Empty3`","highlight_start":14,"highlight_end":23}],"label":"did you mean `E::Empty3 { /* fields */ }`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0423]: expected function, found struct variant `E::Empty3`\n  --> /checkout/src/test/ui/empty/empty-struct-braces-expr.rs:28:14\n   |\nLL |     let e3 = E::Empty3(); //~ ERROR expected function, found struct variant `E::Empty3`\n   |              ^^^^^^^^^ did you mean `E::Empty3 { /* fields */ }`?\n\n"}
[00:52:49] {"message":"expected value, found struct `XEmpty1`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n