\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/empty/empty-struct-braces-expr.rs","byte_start":1150,"byte_end":1160,"line_start":32,"line_end":32,"column_start":15,"column_end":25,"is_primary":true,"text":[{"text":"    let xe3 = XE::Empty3; //~ ERROR no variant named `Empty3` found for type","highlight_start":15,"highlight_end":25}],"label":"variant not found in `empty_struct::XE`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"did you mean `empty_struct::XE::XEmpty3`?","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0599]: no variant named `Empty3` found for type `empty_struct::XE` in the current scope\n  --> /checkout/src/test/ui/empty/empty-struct-braces-expr.rs:32:15\n   |\nLL |     let xe3 = XE::Empty3; //~ ERROR no variant named `Empty3` found for type\n   |               ^^^^^^^^^^ variant not found in `empty_struct::XE`\n   |\n   = note: did you mean `empty_struct::XE::XEmpty3`?\n\n"}
[00:52:49] {"message":"no variant named `Empty3` found for type `empty_struct::XE` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n