\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/empty/empty-struct-braces-expr.rs","byte_start":685,"byte_end":691,"line_start":25,"line_end":25,"column_start":14,"column_end":20,"is_primary":true,"text":[{"text":"    let e1 = Empty1; //~ ERROR expected value, found struct `Empty1`","highlight_start":14,"highlight_end":20}],"label":"did you mean `XEmpty2`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0423]: expected value, found struct `Empty1`\n  --> /checkout/src/test/ui/empty/empty-struct-braces-expr.rs:25:14\n   |\nLL |     let e1 = Empty1; //~ ERROR expected value, found struct `Empty1`\n   |              ^^^^^^ did you mean `XEmpty2`?\n\n"}
[01:05:24] {"message":"expected function, found struct `Empty1`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n