\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/empty/empty-struct-braces-expr.rs","byte_start":1073,"byte_end":1080,"line_start":31,"line_end":31,"column_start":15,"column_end":22,"is_primary":true,"text":[{"text":"    let xe1 = XEmpty1(); //~ ERROR expected function, found struct `XEmpty1`","highlight_start":15,"highlight_end":22}],"label":"did you mean `XEmpty2`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0423]: expected function, found struct `XEmpty1`\n  --> /checkout/src/test/ui/empty/empty-struct-braces-expr.rs:31:15\n   |\nLL |     let xe1 = XEmpty1(); //~ ERROR expected function, found struct `XEmpty1`\n   |               ^^^^^^^ did you mean `XEmpty2`?\n\n"}
[00:52:49] {"message":"no variant named `Empty3` found for type `empty_struct::XE` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n