plain
[00:46:27] ....................................................................................................
[00:46:30] .......................i............................................................................
[00:46:35] ....................................................................................................
[00:46:40] ................................i...........i.......................................................
[00:46:43] ...................................................iiiii............................................
[00:46:46] ......................................F.............................................................
[00:46:48] ....................................................................................................
[00:46:50] ...........F........................................................................................
[00:46:55] ....................................................................................................
[00:46:58] ...........................................i........................................................
[00:47:01] ..........i.........................................................................................
[00:47:04] ....................................................................................................
---
[00:47:46] .........................................................................................i..........
[00:47:49] ....................................................................................................
[00:47:52] ....................................................................................................
[00:47:55] ....................................................................................................
[00:47:57] ......................................................F...........F.................................
[00:48:01] ...............................i.........F..........................................................
[00:48:17] ....................................................................................................
[00:48:24] ....................................................................................................
[00:48:32] ....................................................................................................
[00:48:46] ....................................................................................................
---
[00:52:49] diff of stderr:
[00:52:49] 
[00:52:49] 11   --> $DIR/empty-struct-braces-expr.rs:26:14
[00:52:49] 12    |
[00:52:49] 13 LL |     let e1 = Empty1(); //~ ERROR expected function, found struct `Empty1`
[00:52:49] -    |              |
[00:52:49] -    |              |
[00:52:49] -    |              did you mean `XEmpty2`?
[00:52:49] -    |              did you mean `Empty1 { /* fields */ }`?
[00:52:49] +    |              ^^^^^^ did you mean `XEmpty2`?
[00:52:49] 19 error[E0423]: expected value, found struct variant `E::Empty3`
[00:52:49] 20   --> $DIR/empty-struct-braces-expr.rs:27:14
[00:52:49] 
[00:52:49] 41   --> $DIR/empty-struct-braces-expr.rs:31:15
[00:52:49] 41   --> $DIR/empty-struct-braces-expr.rs:31:15
[00:52:49] 42    |
[00:52:49] 43 LL |     let xe1 = XEmpty1(); //~ ERROR expected function, found struct `XEmpty1`
[00:52:49] -    |               |
[00:52:49] -    |               |
[00:52:49] -    |               did you mean `XEmpty2`?
[00:52:49] -    |               did you mean `XEmpty1 { /* fields */ }`?
[00:52:49] +    |               ^^^^^^^ did you mean `XEmpty2`?
[00:52:49] 48 
[00:52:49] 49 error[E0599]: no variant named `Empty3` found for type `empty_struct::XE` in the current scope
[00:52:49] 50   --> $DIR/empty-struct-braces-expr.rs:32:15
[00:52:49] 
[00:52:49] The actual stderr differed from the expected stderr.
[00:52:49] The actual stderr differed from the expected stderr.
[00:52:49] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/empty/empty-struct-braces-expr/empty-struct-braces-expr.stderr
[00:52:49] To update references, rerun the tests and pass the `--bless` flag
[00:52:49] To only update this specific test, also pass `--test-args empty/empty-struct-braces-expr.rs`
[00:52:49] error: 1 errors occurred comparing output.
[00:52:49] status: exit code: 1
[00:52:49] status: exit code: 1
[00:52:49] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/empty/empty-struct-braces-expr.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/empty/empty-struct-braces-expr/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/empty/empty-struct-braces-expr/auxiliary" "-A" "unused"
[00:52:49] ------------------------------------------
[00:52:49] 
[00:52:49] ------------------------------------------
[00:52:49] stderr:
[00:52:49] stderr:
[00:52:49] ------------------------------------------
[00:52:49] {"message":"expected value, found struct `Empty1`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n