\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/empty/empty-struct-braces-pat-2.rs","byte_start":1027,"byte_end":1034,"line_start":34,"line_end":34,"column_start":9,"column_end":16,"is_primary":true,"text":[{"text":"        XEmpty1(..) => () //~ ERROR expected tuple struct/variant, found struct `XEmpty1`","highlight_start":9,"highlight_end":16}],"label":"did you mean `XEmpty6`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0532]: expected tuple struct/variant, found struct `XEmpty1`\n  --> /checkout/src/test/ui/empty/empty-struct-braces-pat-2.rs:34:9\n   |\nLL |         XEmpty1(..) => () //~ ERROR expected tuple struct/variant, found struct `XEmpty1`\n   |         ^^^^^^^ did you mean `XEmpty6`?\n\n"}
[01:05:24] {"message":"aborting due to 4 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 4 previous errors\n\n"}
[01:05:24] {"message":"For more information about this error, try `rustc --explain E0532`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0532`.\n"}
[01:05:24] ------------------------------------------
[01:05:24] 
[01:05:24] thread '[ui] ui/empty/empty-struct-braces-pat-2.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3205:9
[01:05:24] 
[01:05:24] 
[01:05:24] ---- [ui] ui/error-codes/E0423.rs stdout ----
[01out/src/test/ui/error-codes/E0423.rs","byte_start":768,"byte_end":770,"line_start":25,"line_end":25,"column_start":13,"column_end":15,"is_primary":true,"text":[{"text":"    if T {} == T {} { println!(\"Ok\"); }","highlight_start":13,"highlight_end":15}],"label":"expected expression","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: expected expression, found `==`\n  --> /checkout/src/test/ui/error-codes/E0423.rs:25:13\n   |\nLL |     if T {} == T {} { println!(\"Ok\"); }\n   |             ^^ expected expression\n\n"}
[01:05:24] {"message":"expected type, found `0`","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0423.rs","byte_start":916,"byte_end":917,"line_start":31,"line_end":31,"column_start":39,"column_end":40,"is_primary":true,"text":[{"text":"    for _ in std::ops::Range { start: 0, end: 10 } {}","highlight_start":39,"highlight_end":40}],"label":"expecting a type here because of type ascription","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: expected type, found `0`\n  --> /checkout/src/test/ui/error-codes/E0423.rs:31:39\n   |\nLL |     for _ in std::ops::Range { start: 0, end: 10 } {}\n   |                                       ^ expecting a type here because of type ascription\n\n"}
[01:05:24] {"message":"expected function, found struct `Foo`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n