\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/empty/empty-struct-braces-expr.rs","byte_start":1001,"byte_end":1008,"line_start":30,"line_end":30,"column_start":15,"column_end":22,"is_primary":true,"text":[{"text":"    let xe1 = XEmpty1; //~ ERROR expected value, found struct `XEmpty1`","highlight_start":15,"highlight_end":22}],"label":"did you mean `XEmpty2`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/empty/empty-struct-braces-expr.rs","byte_start":1001,"byte_end":1008,"line_start":30,"line_end":30,"column_start":15,"column_end":22,"is_primary":true,"text":[{"text":"    let xe1 = XEmpty1; //~ ERROR expected value, found struct `XEmpty1`","highlight_start":15,"highlight_end":22}],"label":"did you mean `XEmpty1 { /* fields */ }`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0423]: expected value, found struct `XEmpty1`\n  --> /checkout/src/test/ui/empty/empty-struct-braces-expr.rs:30:15\n   |\nLL |     let xe1 = XEmpty1; //~ ERROR expected value, found struct `XEmpty1`\n   |               ^^^^^^^\n   |               |\n   |               did you mean `XEmpty2`?\n   |               did you mean `XEmpty1 { /* fields */ }`?\n\n"}
[00:52:49] {"message":"expected function, found struct `XEmpty1`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n