\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/polonius-smoke-test.rs","byte_start":199,"byte_end":201,"line_start":7,"line_end":7,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    &x //~ ERROR","highlight_start":5,"highlight_end":7}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/polonius-smoke-test.rs","byte_start":212,"byte_end":213,"line_start":8,"line_end":8,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"`x` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"borrowed value must be valid for the static lifetime...","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0597]: `x` does not live long enough\n  --> /checkout/src/test/ui/nll/polonius-smoke-test.rs:7:5\n   |\nLL |     &x //~ ERROR\n   |     ^^ borrowed value does not live long enough\nLL | }\n   | - `x` dropped here while still borrowed\n   |  37% /
none            5.0M     0  5.0M   0% /run/lock
none            7.4G     0  7.4G   0% /run/shm
none            100M     0  100M   0% /run/user
none            768M     0  768M   0% /var/ramfs
