\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs","byte_start":481,"byte_end":485,"line_start":22,"line_end":22,"column_start":11,"column_end":15,"is_primary":true,"text":[{"text":"        #[repr(C)] //~ ERROR `repr` is ambiguous","highlight_start":11,"highlight_end":15}],"label":"ambiguous name","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`repr` could refer to the name imported here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs","byte_start":155,"byte_end":171,"line_start":7,"line_end":7,"column_start":5,"column_end":21,"is_primary":true,"text":[{"text":"use builtin_attrs::*;","highlight_start":5,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"`repr` could also refer to the name defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs","byte_start":481,"byte_end":485,"line_start":22,"line_end":22,"column_start":11,"column_end":15,"is_primary":true,"text":[{"text":"        #[repr(C)] //~ ERROR `repr` is ambiguous","highlight_start":11,"highlight_end":15}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"consider adding an explicit import of `repr` to disambiguate","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0659]: `repr` is ambiguous\n  --> /checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs:22:11\n   |\nLL |         #[repr(C)] //~ ERROR `repr` is ambiguous\n   |           ^^^^ ambiguous name\n   |\nnote: `repr` could refer to the name imported here\n  --> /checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs:7:5\n   |\nLL | use builtin_attrs::*;\n   |     ^^^^^^^^^^^^^^^^\nnote: `repr` could also refer to the name defined here\n  --> /checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs:22:11\n   |\nLL |         #[repr(C)] //~ ERROR `repr` is ambiguous\n   |           ^^^^\n   = note: consider adding an explicit import of `repr` to disambiguate\n\n"}
[01:03:33] {"message":"`feature` is ambiguous","code":{"code":"E0659","explanation":"\nAn item usage is ambiguous.\n\nErroneous code example:\n\n