\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":798,"byte_end":825,"line_start":22,"line_end":22,"column_start":1,"column_end":28,"is_primary":true,"text":[{"text":"#[generate_mod::check_attr] //~ ERROR cannot find type `FromOutside` in this scope","highlight_start":1,"highlight_end":28}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":798,"byte_end":825,"line_start":22,"line_end":22,"column_start":1,"column_end":28,"is_primary":false,"text":[{"text":"#[generate_mod::check_attr] //~ ERROR cannot find type `FromOutside` in this scope","highlight_start":1,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"#[generate_mod::check_attr]","def_site_span":null}}],"children":[],"rendered":"error[E0412]: cannot find type `Outer` in this scope\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:22:1\n   |\nLL | #[generate_mod::check_attr] //~ ERROR cannot find type `FromOutside` in this scope\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope\n\n"}
[01:03:33] {"message":"cannot find type `FromOutside` in this scope","code":{"code":"proc_macro_derive_resolution_fallback","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":983,"byte_end":1008,"line_start":26,"line_end":26,"column_start":10,"column_end":35,"is_primary":true,"text":[{"text":"#[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope","highlight_start":10,"highlight_end":35}],"label":"names from parent modules are not accessible without an explicit import","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"#[warn(proc_macro_derive_resolution_fallback)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: cannot find type `FromOutside` in this scope\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:26:10\n   |\nLL | #[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope\n   |          ^^^^^^^^^^^^^^^^^^^^^^^^^ names from parent modules are not accessible without an explicit import\n   |\n   = note: #[warn(proc_macro_derive_resolution_fallback)] on by default\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>\n\n"}
[01:03:33] {"message":"cannot find type `Outer` in this scope","code":{"code":"proc_macro_derive_resolution_fallback","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":983,"byte_end":1008,"line_start":26,"line_end":26,"column_start":10,"column_end":35,"is_primary":true,"text":[{"text":"#[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope","highlight_start":10,"highlight_end":35}],"label":"names from parent modules are not accessible without an explicit import","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: cannot find type `Outer` in this scope\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:26:10\n   |\nLL | #[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope\n   |          ^^^^^^^^^^^^^^^^^^^^^^^^^ names from parent modules are not accessible without an explicit import\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>\n\n"}
[01:03:33] {"message":"cannot find type `FromOutside` in this scope","code":{"code":"proc_macro_derive_resolution_fallback","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":1352,"byte_end":1377,"line_start":33,"line_end":33,"column_start":14,"column_end":39,"is_primary":true,"text":[{"text":"    #[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope","highlight_start":14,"highlight_end":39}],"label":"names from parent modules are not accessible without an explicit import","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: cannot find type `FromOutside` in this scope\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:33:14\n   |\nLL |     #[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope\n   |              ^^^^^^^^^^^^^^^^^^^^^^^^^ names from parent modules are not accessible without an explicit import\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>\n\n"}
[01:03:33] {"message":"cannot find type `Outer` in this scope","code":{"code":"proc_macro_derive_resolution_fallback","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":1352,"byte_end":1377,"line_start":33,"line_end":33,"column_start":14,"column_end":39,"is_primary":true,"text":[{"text":"    #[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope","highlight_start":14,"highlight_end":39}],"label":"names from parent modules are not accessible without an explicit import","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: cannot find type `Outer` in this scope\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:33:14\n   |\nLL |     #[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope\n   |              ^^^^^^^^^^^^^^^^^^^^^^^^^ names from parent modules are not accessible without an explicit import\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>\n\n"}
[01:03:33] {"message":"cannot find type `FromOutside` in this scope","code":{"code":"proc_macro_derive_resolution_fallback","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":1718,"byte_end":1747,"line_start":40,"line_end":40,"column_start":10,"column_end":39,"is_primary":true,"text":[{"text":"#[derive(generate_mod::CheckDeriveLint)] // OK, lint is suppressed","highlight_start":10,"highlight_end":39}],"label":"names from parent modules are not accessible without an explicit import","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: cannot find type `FromOutside` in this scope\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:40:10\n   |\nLL | #[derive(generate_mod::CheckDeriveLint)] // OK, lint is suppressed\n   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ names from parent modules are not accessible without an explicit import\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>\n\n"}
[01:03:33] {"message":"cannot find type `Outer` in this scope","code":{"code":"proc_macro_derive_resolution_fallback","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":1718,"byte_end":1747,"line_start":40,"line_end":40,"column_start":10,"column_end":39,"is_primary":true,"text":[{"text":"#[derive(generate_mod::CheckDeriveLint)] // OK, lint is suppressed","highlight_start":10,"highlight_end":39}],"label":"names from parent modules are not accessible without an explicit import","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: cannot find type `Outer` in this scope\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:40:10\n   |\nLL | #[derive(generate_mod::CheckDeriveLint)] // OK, lint is suppressed\n   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ names from parent modules are not accessible without an explicit import\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #50504 <https://github.com/rust-lang/rust/issues/50504>\n\n"}
[01:03:33] {"message":"aborting due to 13 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 13 previous errors\n\n"}
[01:03:33] {"message":"Some errors occurred: E0412, E0428.","code":null,"level":"","spans":[],"children":[],"rendered":"Some errors occurred: E0412, E0428.\n"}
[01:03:33] {"message":"For more information about an error, try `rustc --explain E0412`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about an error, try `rustc --explain E0412`.\n"}
[01:03:33] ------------------------------------------
[01:03:33] 
[01:03:33] thread '[ui] ui-fulldeps/proc-macro/generate-mod.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3277:9
[01:03:33] 
[01:03:33] 
[01:03:33] ---- [ui] ui-fulldeps/proc-macro/nested-item-spans.rs stdout ----
[01:03:33] diff of stderr:
[01:03:33] 
[01:03:33] + error[E0425]: cannot find function `bar` in this scope
[01:03:33] +   --> $DIR/nested-item-spans.rs:32:5
[01:03:33] +    |
[01:03:33] + LL |     bar();
[01:03:33] + 
[01:03:33] 1 error[E0308]: mismatched types
[01:03:33] -   --> $DIR/nested-item-spans.rs:20:22
[01:03:33] 3    |
[01:03:33] 3    |
[01:03:33] - LL |         let x: u32 = "x"; //~ ERROR: mismatched types
[01:03:33] -    |
[01:03:33] 7    = note: expected type `u32`
[01:03:33] 8               found type `&'static str`
[01:03:33] 9 
[01:03:33] 9 
[01:03:33] 
[01:03:33] 10 error[E0308]: mismatched types
[01:03:33] -   --> $DIR/nested-item-spans.rs:29:22
[01:03:33] +   --> $DIR/nested-item-spans.rs:20:22
[01:03:33] 12    |
[01:03:33] 13 LL |         let x: u32 = "x"; //~ ERROR: mismatched types
[01:03:33] 
[01:03:33] 16    = note: expected type `u32`
[01:03:33] 17               found type `&'static str`
[01:03:33] 18 
---
[01:03:33] 22 
[01:03:33] 
[01:03:33] 
[01:03:33] The actual stderr differed from the expected stderr.
[01:03:33] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/proc-macro/nested-item-spans/nested-item-spans.stderr
[01:03:33] To update references, rerun the tests and pass the `--bless` flag
[01:03:33] To only update this specific test, also pass `--test-args proc-macro/nested-item-spans.rs`
[01:03:33] error: 1 errors occurred comparing output.
[01:03:33] status: exit code: 1
[01:03:33] status: exit code: 1
[01:03:33] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui-fulldeps/proc-macro/nested-item-spans.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/proc-macro/nested-item-spans/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/proc-macro/nested-item-spans/auxiliary" "-A" "unused"
[01:03:33] ------------------------------------------
[01:03:33] 
[01:03:33] ------------------------------------------
[01:03:33] stderr:
[01:03:33] stderr:
[01:03:33] ------------------------------------------
[01:03:33] {"message":"cannot find function `bar` in this scope","code":{"code":"E0425","explanation":"\nAn unresolved name was used.\n\nErroneous code examples:\n\n