\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":983,"byte_end":1008,"line_start":26,"line_end":26,"column_start":10,"column_end":35,"is_primary":true,"text":[{"text":"#[derive(generate_mod::CheckDerive)] //~ WARN cannot find type `FromOutside` in this scope","highlight_start":10,"highlight_end":35}],"label":"`Outer` redefined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":983,"byte_end":1008,"line_start":26,"line_en//~ WARN cannot find type `FromOutside` in this scope\n   |          ^^^^^^^^^^^^^^^^^^^^^^^^^ `inner` redefined here\n   |\n   = note: `inner` must be defined only once in the type namespace of this module\n\n"}
[01:10:40] {"message":"the name `Alias` is defined multiple times","code":{"code":"E0428","explanation":"\nA type or module has been defined more than once.\n\nErroneous code example:\n\n