\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":1718,"byte_end":1747,"line_start":40,"line_end":40,"column_start":10,"column_end":39,"is_primary":true,"text":[{"text":"#[derive(generate_mod::CheckDeriveLint)] // OK, lint is suppressed","highlight_start":10,"highlight_end":39}],"label":"`Outer` redefined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":1718,"byte_end":1747,"line_start":40,"line_end":40,"column_start":10,"column_end":39,"is_primary":false,"text":[{"text":"#[derive(generate_mod::CheckDeriveLint)] // OK, lint is suppressed","highlight_start":10,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"#[derive(generate_mod::CheckDeriveLint)]","def_site_span":null}},{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":644,"byte_end":667,"line_start":19,"line_end":19,"column_start":1,"column_end":24,"is_primary":false,"text":[{"text":"generate_mod::check!(); //~ ERROR cannot find type `FromOutside` in this scope","highlight_start":1,"highlight_end":24}],"label":"previous definition of the type `Outer` here","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":644,"byte_end":667,"line_start":19,"line_end":19,"column_start":1,"column_end":24,"is_primary":false,"text":[{"text":"generate_mod::check!(); //~ ERROR cannot find type `FromOutside` in this scope","highlight_start":1,"highlight_end":24}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"generate_mod::check!","def_site_span":null}}],"children":[{"message":"`Outer` must be defined only once in the type namespace of this module","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0428]: the name `Outer` is defined multiple times\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:40:10\n   |\nLL | generate_mod::check!(); //~ ERROR cannot find type `FromOutside` in this scope\n   | ----------------------- previous definition of the type `Outer` here\n...\nLL | #[derive(generate_mod::CheckDeriveLint)] // OK, lint is suppressed\n   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Outer` redefined here\n   |\n   = note: `Outer` must be defined only once in the type namespace of this module\n\n"}
[01:04:51] {"message":"the name `inner` is defined multiple times","code":{"code":"E0428","explanation":"\nA type or module has been defined more than once.\n\nErroneous code example:\n\n