\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":798,"byte_end":825,"line_start":22,"line_end":22,"column_start":1,"column_end":28,"is_primary":true,"text":[{"text":"#[generate_mod::check_attr] //~ ERROR cannot find type `FromOutside` in this scope","highlight_start":1,"highlight_end":28}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs","byte_start":798,"byte_end":825,"line_start":22,"line_end":22,"column_start":1,"column_end":28,"is_primary":false,"text":[{"text":"#[generate_mod::check_attr] //~ ERROR cannot find type `FromOutside` in this scope","highlight_start":1,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"#[generate_mod::check_attr]","def_site_span":null}}],"children":[],"rendered":"error[E0412]: cannot find type `FromOutside` in this scope\n  --> /checkout/src/test/ui-fulldeps/proc-macro/generate-mod.rs:22:1\n   |\nLL | #[generate_mod::check_attr] //~ ERROR cannot find type `FromOutside` in this scope\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope\n\n"}
[01:03:33] {"message":"cannot find type `Outer` in this scope","code":{"code":"E0412","explanation":"\nThe type name used is not in scope.\n\nErroneous code examples:\n\n