\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs","byte_start":34,"byte_end":41,"line_start":3,"line_end":3,"column_start":4,"column_end":11,"is_primary":true,"text":[{"text":"#![feature(decl_macro)] //~ ERROR `feature` is ambiguous","highlight_start":4,"highlight_end":11}],"label":"ambiguous name","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`feature` could refer to the name imported here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs","byte_start":155,"byte_end":171,"line_start":7,"line_end":7,"column_start":5,"column_end":21,"is_primary":true,"text":[{"text":"use builtin_attrs::*;","highlight_start":5,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"`feature` could also refer to the name defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs","byte_start":34,"byte_end":41,"line_start":3,"line_end":3,"column_start":4,"column_end":11,"is_primary":true,"text":[{"text":"#![feature(decl_macro)] //~ ERROR `feature` is ambiguous","highlight_start":4,"highlight_end":11}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"consider adding an explicit import of `feature` to disambiguate","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0659]: `feature` is ambiguous\n  --> /checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs:3:4\n   |\nLL | #![feature(decl_macro)] //~ ERROR `feature` is ambiguous\n   |    ^^^^^^^ ambiguous name\n   |\nnote: `feature` could refer to the name imported here\n  --> /checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs:7:5\n   |\nLL | use builtin_attrs::*;\n   |     ^^^^^^^^^^^^^^^^\nnote: `feature` could also refer to the name defined here\n  --> /checkout/src/test/ui-fulldeps/proc-macro/ambiguous-builtin-attrs.rs:3:4\n   |\nLL | #![feature(decl_macro)] //~ ERROR `feature` is ambiguous\n   |    ^^^^^^^\n   = note: consider adding an explicit import of `feature` to disambiguate\n\n"}
[01:03:33] {"message":"cannot find value `Bench` in this scope","code":{"code":"E0425","explanation":"\nAn unresolved name was used.\n\nErroneous code examples:\n\n