\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/underscore-lifetime/underscore-lifetime-binders.rs","byte_start":767,"byte_end":769,"line_start":24,"line_end":24,"column_start":21,"column_end":23,"is_primary":true,"text":[{"text":"fn meh() -> Box<for<'_> Meh<'_>> //~ ERROR invalid lifetime parameter name: `'_`","highlight_start":21,"highlight_end":23}],"label":"'_ is a reserved lifetime name","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0262]: invalid lifetime parameter name: `'_`\n  --> /checkout/src/test/ui/underscore-lifetime/underscore-lifetime-binders.rs:24:21\n   |\nLL | fn meh() -> Box<for<'_> Meh<'_>> //~ ERROR invalid lifetime parameter name: `'_`\n   |                     ^^ '_ is a reserved lifetime name\n\n"}
[00:51:12] {"message":"missing lifetime specifier","code":{"code":"E0106","explanation":"\nThis error indicates that a lifetime is missing from a type. If it is an error\ninside a function signature, the problem may be with failing to adhere to the\nliffetime elision in implementation headers was part of the lifetime elision\nRFC. It is, however, [currently unimplemented][iss15872].\n\n[book-le]: https://doc.rust-lang.org/nightly/book/first-edition/lifetimes.html#lifetime-elision\n[iss15872]: https://github.com/rust-lang/rust/issues/15872\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/underscore-lifetime/underscore-lifetime-binders.rs","byte_start":775,"byte_end":777,"line_start":24,"line_end":24,"column_start":29,"column_end":31,"is_primary":true,"text":[{"text":"fn meh() -> Box<for<'_> Meh<'_>> //~ ERROR invalid lifetime parameter name: `'_`","highlight_start":29,"highlight_end":31}],"label":"expected lifetime parameter","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this function's return type contains a borrowed value, but there is no value for it to be borrowed from","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"consider giving it a 'static lifetime","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0106]: missing lifetime specifier\n  --> /checkout/src/test/ui/underscore-lifetime/underscore-lifetime-binders.rs:24:29\n   |\nLL | fn meh() -> Box<for<'_> Meh<'_>> //~ ERROR invalid lifetime parameter name: `'_`\n   |                             ^^ expected lifetime parameter\n   |\n   = help: this function's return type contains a borrowed value, but there is no value for it to be borrowed from\n   = help: consider giving it a 'static lifetime\n\n"}
[00:51:12] {"message":"missing lifetime specifier","code":{"code":"E0106","explanation":"\nThis error indicates that a lifetime is missing from a type. If it is an error\ninside a function signature, the problem may be with failing to adhere to the\nlifetime elision rules (see below).\n\nHere are some simple examples of where you'll run into this error:\n\n