\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/underscore-lifetime/underscore-lifetime-binders.rs","byte_start":633,"byte_end":635,"line_start":18,"line_end":18,"column_start":8,"column_end":10,"is_primary":true,"text":[{"text":"fn foo<'_> //~ ERROR invalid lifetime parameter name: `'_`","highlight_start":8,"highlight_end":10}],"label":"'_ is a reserved lifetime name","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0262]: invalid lifetime parameter name: `'_`\n  --> /checkout/src/test/ui/underscore-lifetime/underscore-lifetime-binders.rs:18:8\n   |\nLL | fn foo<'_> //~ ERROR invalid lifetime parameter name: `'_`\n   |        ^^ '_ is a reserved lifetime name\n\n"}
[00:51:12] {"message":"invalid lifetime parameter name: `'_`","code":{"code":"E0262","explanation":"\nDeclaring certain lifetime names in parameters is disallowed. For example,\nbecause the `'static` lifetime is a special built-in lifetime name denoting\nthe lifetime of the entire program, this is an error:\n\n