\n\nLifetime elision in implementation headers was part of the lifetime elision\nRFC. It is, however, [currently unimplemented][iss15872].\n\n[book-le]: https://doc.rust-lang.org/nightly/book/first-edition/lifetimes.html#lifetime-elision\n[iss15872]: https://github.com/rust-lang/rust/issues/15872\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/underscore-lifetime/underscore-lifetime-binders.rs","byte_start":507,"byte_end":509,"line_start":12,"line_end":12,"column_start":17,"column_end":19,"is_primary":true,"text":[{"text":"struct Baz<'a>(&'_ &'a u8); //~ ERROR missing lifetime specifier","highlight_start":17,"highlight_end":19}],"label":"expected lifetime parameter","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0106]: missing lifetime specifier\n  --> /checkout/src/test/ui/underscore-lifetime/underscore-lifetime-binders.rs:12:17\n   |\nLL | struct Baz<'a>(&'_ &'a u8); //~ ERROR missing lifetime specifier\n   |                 ^^ expected lifetime parameter\n\n"}
[00:51:12] {"message":"invalid lifetime parameter name: `'_`","code":{"code":"E0262","explanation":"\nDeclaring certain lifetime names in parameters is disallowed. For example,\nbecause the `'static` lifetime is a special built-in lifetime name denoting\nthe lifetime of the entire program, this is an error:\n\n