\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-17905.rs","byte_start":600,"byte_end":618,"line_start":18,"line_end":18,"column_start":18,"column_end":36,"is_primary":true,"text":[{"text":"    fn say(self: &Pair<&str, isize>) {","highlight_start":18,"highlight_end":36}],"label":"lifetime mismatch","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `Pair<&'_ str, _>`\n   found type `Pair<&str, _>`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"the anonymous lifetime #2 defined on the method body at 18:5...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-17905.rs","byte_start":587,"byte_end":657,"line_start":18,"line_end":20,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    fn say(self: &Pair<&str, isize>) {","highlight_start":5,"highlight_end":39},{"text":"        println!(\"{}\", self);","highlight_start":1,"highlight_end":30},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"...does not necessarily outlive the lifetime '_ as defined on the impl at 15:5","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-17905.rs","byte_start":526,"byte_end":526,"line_start":15,"line_end":15,"column_start":5,"column_end":5,"is_primary":true,"text":[{"text":"    &str, //~ ERROR missing lifetime specifier","highlight_start":5,"highlight_end":5}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched method receiver\n  --> /checkout/src/test/ui/issues/issue-17905.rs:18:18\n   |\nLL |     fn say(self: &Pair<&str, isize>) {\n   |                  ^^^^^^^^^^^^^^^^^^ lifetime mismatch\n   |\n   = note: expected type `Pair<&'_ str, _>`\n              found type `Pair<&str, _>`\nnote: the anonymous lifetime #2 defined on the method body at 18:5...\n  --> /checkout/src/test/ui/issues/issue-17905.rs:18:5\n   |\nLL | /     fn say(self: &Pair<&str, isize>) {\nLL | |         println!(\"{}\", self);\nLL | |     }\n   | |_____^\nnote: ...does not necessarily outlive the lifetime '_ as defined on the impl at 15:5\n  --> /checkout/src/test/ui/issues/issue-17905.rs:15:5\n   |\nLL |     &str, //~ ERROR missing lifetime specifier\n   |     ^\n\n"}
[00:51:12] {"message":"mismatched method receiver","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n