plain
travis_time:start:test_ui
Check compiletest suite=ui mode=ui (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
[00:44:12] 
[00:44:12] running 4570 tests
[00:44:15] ......................F............................................................................. 100/4570
[00:44:21] .................................................................................................... 300/4570
[00:44:24] .................................................................................................... 400/4570
[00:44:27] .................................................................................................... 500/4570
[00:44:30] .......................i............................................................................ 600/4570
[00:44:30] .......................i............................................................................ 600/4570
[00:44:35] .................................................................................................... 700/4570
[00:44:40] .................................i..........i....................................................... 800/4570
[00:44:43] ...................................................iiiii............................................ 900/4570
[00:44:46] .................................................................................................... 1000/4570
[00:44:48] .................................................................................................... 1100/4570
[00:44:50] ...........................................................................................F........ 1200/4570
[00:44:55] .................................................................................................... 1400/4570
[00:44:58] .........................................i.......................................................... 1500/4570
[00:45:01] ........i........................................................................................... 1600/4570
[00:45:05] .................................................................................................... 1700/4570
[00:45:05] .................................................................................................... 1700/4570
[00:45:08] .................................................................................................... 1800/4570
[00:45:11] ..........................................i............................................F............ 1900/4570
[00:45:19] .................................................................................................... 2100/4570
[00:45:22] .................................................................................................... 2200/4570
[00:45:26] .................................................................................................... 2300/4570
[00:45:30] .................................................................................................... 2400/4570
---
[00:46:31] ......................................................i............................................. 4200/4570
[00:46:34] .................................................................................................... 4300/4570
[00:46:37] .................................................................................................... 4400/4570
[00:46:40] .................................................................................................... 4500/4570
able_features)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: the feature `impl_header_lifetime_elision` has been stable since 1.31.0 and no longer requires an attribute to enable\n  --> /checkout/src/test/ui/E0705.rs:13:12\n   |\nLL | #![feature(impl_header_lifetime_elision)]\n   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = note: #[warn(stable_features)] on by default\n\n"}
[00:46:42] ------------------------------------------
[00:46:42] 
[00:46:42] thread '[ui] ui/E0705.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3267:9
[00:46:42] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:46:42] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:46:42] 
[00:46:42] ---- [ui] ui/error-codes/E0637.rs stdout ----
[00:46:42] diff of stderr:
[00:46:42] 
[00:46:42] 10 LL | fn foo<'a: '_>(_: &'a u8) {} //~ ERROR invalid lifetime bound name: `'_`
[00:46:42] 11    |            ^^ `'_` is a reserved lifetime name
[00:46:42] 12 
[00:46:42] - error[E0637]: invalid lifetime bound name: `'_`
[00:46:42] -   --> $DIR/E0637.rs:15:10
[00:46:42] -    |
[00:46:42] - LL | impl<'a: '_> Bar<'a> { //~ ERROR invalid lifetime bound name: `'_`
[00:46:42] -    |          ^^ `'_` is a reserved lifetime name
[00:46:42] - error: aborting due to 3 previous errors
[00:46:42] + error: aborting due to 2 previous errors
[00:46:42] 20 
[00:46:42] 21 For more information about this error, try `rustc --explain E0637`.
[00:46:42] 21 For more information about this error, try `rustc --explain E0637`.
[00:46:42] 22 
[00:46:42] 
[00:46:42] 
[00:46:42] The actual stderr differed from the expected stderr.
[00:46:42] Actual stderr saved to /checkout/obj/build/x86_64-unsrc/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-5.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options " "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "5.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[00:46:42] 
[00:46:42] 
[00:46:42] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[00:46:42] Build completed unsuccessfully in 0:03:29
[00:46:42] Build completed unsuccessfully in 0:03:29
[00:46:42] Makefile:58: recipe for target 'check' failed
[00:46:42] make: *** [check] Error 1

The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:0c390918
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
