\n\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/trait-alias-fail.rs","byte_start":931,"byte_end":937,"line_start":23,"line_end":23,"column_start":6,"column_end":12,"is_primary":true,"text":[{"text":"impl Alias1 for () { //~ERROR expected trait, found trait alias","highlight_start":6,"highlight_end":12}],"label":"not a trait","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0404]: expected trait, found trait alias `Alias1`\n  --> /checkout/src/test/ui/trait-alias-fail.rs:23:6\n   |\nLL | impl Alias1 for () { //~ERROR expected trait, found trait alias\n   |      ^^^^^^ not a trait\n\n"}
[00:49:39] {"message":"trait aliases are experimental (see issue #41517)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n