\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/match/match-byte-array-patterns-2.rs","byte_start":519,"byte_end":522,"line_start":14,"line_end":14,"column_start":11,"column_end":14,"is_primary":true,"text":[{"text":"    match buf { //~ ERROR non-exhaustive","highlight_start":11,"highlight_end":14}],"label":"pattern `&[_, _, _, _]` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `&[_, _, _, _]` not covered\n  --> /checkout/src/test/ui/match/match-byte-array-patterns-2.rs:14:11\n   |\nLL |     match buf { //~ ERROR non-exhaustive\n   |           ^^^ pattern `&[_, _, _, _]` not covered\n\n"}
[00:48:15] {"message":"non-exhaustive patterns: `&[]` and `&[_]` not covered","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n