\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/rust-2018/trait-import-suggestions.rs","byte_start":1113,"byte_end":1123,"line_start":42,"line_end":42,"column_start":7,"column_end":17,"is_primary":true,"text":[{"text":"    x.extern_baz(); //~ ERROR no method named `extern_baz`","highlight_start":7,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `extern_baz` found for type `u32` in the current scope\n  --> /checkout/src/test/ui/rust-2018/trait-import-suggestions.rs:42:7\n   |\nLL |     x.extern_baz(); //~ ERROR no method named `extern_baz`\n   |       ^^^^^^^^^^\n\n"}
[00:47:42] {"message":"no function or associated item named `from_str` found for type `u32` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n