\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/rust-2018/trait-import-suggestions.rs","byte_start":1068,"byte_end":1071,"line_start":41,"line_end":41,"column_start":7,"column_end":10,"is_primary":true,"text":[{"text":"    x.bar(); //~ ERROR no method named `bar`","highlight_start":7,"highlight_end":10}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"items from traits can only be used if the trait is in scope","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"the following trait is implemented `extern_baz`","highlight_start":7,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `extern_baz` found for type `u32` in the current scope\n  --> /checkout/src/test/ui/rust-2018/trait-import-suggestions.rs:42:7\n   |\nLL |     x.extern_baz(); //~ ERROR no method named `extern_baz`\n   |       ^^^^^^^^^^\n\n"}
[00:48:15] {"message":"no function or associated item named `from_str` found for type `u32` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n