\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0017.rs","byte_start":596,"byte_end":602,"line_start":15,"line_end":15,"column_start":39,"column_end":45,"is_primary":true,"text":[{"text":"static STATIC_REF: &'static mut i32 = &mut X; //~ ERROR E0017","highlight_start":39,"highlight_end":45}],"label":"cannot borrow as mutable","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0596]: cannot borrow immutable static item `X` as mutable\n  --> /checkout/src/test/ui/error-codes/E0017.rs:15:39\n   |\nLL | static STATIC_REF: &'static mut i32 = &mut X; //~ ERROR E0017\n   |                                       ^^^^^^ cannot borrow as mutable\n\n"}
[00:51:54] {"message":"references in statics may only refer to immutable values","code":{"code":"E0017","explanation":"\nReferences in statics and constants may only refer to immutable values.\nErroneous code example:\n\n