plain
travis_time:end:06f90f88:start=1542643563869916028,finish=1542643567238871105,duration=3368955077
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#Pull-Requests-and-Security-Restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
Setting environment variables from .travis.yml
$ export IMAGE=x86_64-gnu-llvm-5.0
---
[00:49:21] 
[00:49:21] running 5041 tests
[00:49:24] .................................................................................................... 100/5041
[00:49:26] .................................................................................................... 200/5041
[00:49:29] .............................ii............................................ii...................ii.. 300/5041
[00:49:32] ...............................................................................................iii.. 400/5041
[00:49:34] .....iiiiiiii.iii............................iii...........................................i........ 500/5041
[00:49:41] .................................................................................................... 700/5041
[00:49:47] .......................................................................................i...........i 800/5041
[00:49:50] .................................................................................................... 900/5041
[00:49:53] ......iiiii..................ii.iiii................................................................ 1000/5041
[00:49:53] ......iiiii..................ii.iiii................................................................ 1000/5041
[00:49:55] ........................F........................................................iiiiiiii........... 1100/5041
[00:49:58] ................................................F................................................... 1200/5041
[00:50:03] .................................................................................................... 1400/5041
[00:50:05] ......................................i............................................................. 1500/5041
[00:50:08] .......i.........ii..........................................................i...................... 1600/5041
[00:50:11] .................................................................................................... 1700/5041
---
[00:50:28] .................................................................................................... 2200/5041
[00:50:32] .................................................................................................... 2300/5041
[00:50:35] .................................................................................................... 2400/5041
[00:50:39] .................................................................................................... 2500/5041
[00:50:42] ..........................................................................................iiiiiiiii. 2600/5041
[00:50:49] ........................................................ii.......................................... 2800/5041
[00:50:52] .................................................................................................... 2900/5041
[00:50:56] .................................................................................................... 3000/5041
[00:50:59] ...................................................i................................................ 3100/5041
---
[00:51:44] ..........i......................................................................................... 4700/5041
[00:51:48] .................................................................................................... 4800/5041
[00:51:50] .................................................................................................... 4900/5041
[00:51:53] ................................................................................i................... 5000/5041
0:51:54] 23    |                                      ^^^^^^ statics require immutable values
[00:51:54] 24 
[00:51:54] - error: aborting due to 4 previous errors
[00:51:54] + error: aborting due to 5 previous errors
[00:51:54] 26 
[00:51:54] 26 
[00:51:54] 27 Some errors occurred: E0017, E0596.
[00:51:54] 28 For more information about an error, try `rustc --explain E0017`.
[00:51:54] 
[00:51:54] 
[00:51:54] The actual stderr differed from the expected stderr.
[00:51:54] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0017.nll/E0017.nll.stderr
[00:51:54] To update references, rerun the tests and pass the `--bless` flag
[00:51:54] To only update this specific test, also pass `--test-args error-codes/E0017.rs`
[00:51:54] error: 1 errors occurred comparing output.
[00:51:54] status: exit code: 1
[00:51:54] status: exit code: 1
[00:51:54] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/error-codes/E0017.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0017.nll/a" "-Zborrowck=migrate" "-Ztwo-phase-borrows" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0017.nll/auxiliary" "-A" "unused"
[00:51:54] ------------------------------------------
[00:51:54] 
[00:51:54] ------------------------------------------
[00:51:54] stderr:
[00:51:54] stderr:
[00:51:54] -------------------------------.rs","byte_start":596,"byte_end":602,"line_start":15,"line_end":15,"column_start":39,"column_end":45,"is_primary":true,"text":[{"text":"static STATIC_REF: &'static mut i32 = &mut X; //~ ERROR E0017","highlight_start":39,"highlight_end":45}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: cannot mutate statics in the initializer of another static\n  --> /checkout/src/test/ui/error-codes/E0017.rs:15:39\n   |\nLL | static STATIC_REF: &'static mut i32 = &mut X; //~ ERROR E0017\n   |                                       ^^^^^^\n\n"}
[00:51:54] {"message":"references in statics may only refer to immutable values","code":{"code":"E0017","explanation":"\nReferences in statics and constants may only refer to immutable values.\nErroneous code example:\n\n