\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/stable/book/references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/borrowck-for-loop-head-linkage.rs","byte_start":605,"byte_end":629,"line_start":17,"line_end":17,"column_start":9,"column_end":33,"is_primary":true,"text":[{"text":"        vector.extend(repeat(0));      //~ ERROR cannot borrow","highlight_start":9,"highlight_end":33}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-for-loop-head-linkage.rs","byte_start":550,"byte_end":557,"line_start":15,"line_end":15,"column_start":15,"column_end":22,"is_primary":false,"text":[{"text":"    for &x in &vector {","highlight_start":15,"highlight_end":22}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-for-loop-head-linkage.rs","byte_start":550,"byte_end":557,"line_start":15,"line_end":15,"column_start":15,"column_end":22,"is_primary":false,"text":[{"text":"    for &x in &vector {","highlight_start":15,"highlight_end":22}],"label":"immutable borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `vector` as mutable because it is also borrowed as immutable\n  --> /checkout/src/test/ui/borrowck/borrowck-for-loop-head-linkage.rs:17:9\n   |\nLL |     for &x in &vector {\n   |               -------\n   |               |\n   |               immutable borrow occurs here\n   |               immutable borrow later used here\nLL |         let cap = vector.capacity();\nLL |         vector.extend(repeat(0));      //~ ERROR cannot borrow\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here\n\n"}
[00:56:19] {"message":"cannot borrow `vector` as mutable because it is also borrowed as immutable","code":{"code":"E0502","explanation":"\nThis error indicates that you are trying to borrow a variable as mutable when it\nhas already been borrowed as immutable.\n\nExample of erroneous code:\n\n