plain
travis_time:end:019a5d65:start=1542743550227308912,finish=1542743605389370072,duration=55162061160
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#Pull-Requests-and-Security-Restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
Setting environment variables from .travis.yml
$ export IMAGE=x86_64-gnu-llvm-5.0
---
[00:48:33] 
[00:48:33] running 5043 tests
[00:48:35] .................................................................................................... 100/5043
[00:48:38] .................................................................................................... 200/5043
[00:48:41] .............................ii...........................F................ii...................ii.. 300/5043
[00:48:43] .......................F......................................................................iii... 400/5043
[00:48:46] .....iiiiiiii.iii.........F..................iii...........................................i........ 500/5043
[00:48:53] .................................................................................................... 700/5043
[00:48:59] ..................................................................................i...........i..... 800/5043
[00:49:02] .................................................................................................... 900/5043
[00:49:05] .iiiii..................ii.iiii..................................................................... 1000/5043
---
[00:49:40] .................................................................................................... 2200/5043
[00:49:44] .................................................................................................... 2300/5043
[00:49:47] .................................................................................................... 2400/5043
[00:49:51] .................................................................................................... 2500/5043
[00:49:54] .....................................................................................iiiiiiiii...... 2600/5043
[00:49:57] .................................................................................................... 2700/5043
[00:50:00] .........................F.........................ii............................................... 2800/5043
[00:50:07] .................................................................................................... 3000/5043
[00:50:10] ...............................................i.................................................... 3100/5043
[00:50:13] .................................................................................................... 3200/5043
[00:50:15] ..........ii..i..ii................................................................................. 3300/5043
---
[00:50:27] ..................................................................i................................. 3800/5043
[00:50:29] .................................................................................................... 3900/5043
[00:50:32] .................................................................................................... 4000/5043
[00:50:35] .................................................................................................... 4100/5043
[00:50:37] ...................................................................................................i 4200/5043
[00:50:41] iiii.............F........i......................................................................... 4300/5043
[00:50:45] .................................................................................................... 4400/5043
[00:50:48] .....................................................................................F.F............ 4500/5043
[00:50:54] ..........i......................................................................................... 4700/5043
[00:50:58] .................................................................................................... 4800/5043
[00:51:00] .................................................................................................... 4900/5043
[00:51:00] .................................................................................................... 4900/5043
[00:51:03] .............................................................................F....i................. 5000/5043
[00:51:04] failures:
[00:51:04] 
[00:51:04] ---- [ui (nll)] ui/borrowck/borrowck-for-loop-head-linkage.rs stdout ----
[00:51:04] diff of stderr:
[00:51:04] diff of stderr:
[00:51:04] 
[00:51:04] 5    |               -------
[00:51:04] 6    |               |
[00:51:04] 7    |               immutable borrow occurs here
[00:51:04] -    |               immutable borrow used here, in later iteration of loop
[00:51:04] +    |               immutable borrow later used here
[00:51:04] 9 LL |         let cap = vector.capacity();
[00:51:04] 10 LL |         vector.extend(repeat(0));      //~ ERROR cannot borrow
[00:51:04] 
[00:51:04] 17    |               -------
[00:51:04] 18    |               |
[00:51:04] 19    |               immutable borrow occurs here
[00:51:04] 19    |               immutable borrow occurs here
[00:51:04] -    |               immutable borrow used here, in later iteration of loop
[00:51:04] 21 ...
[00:51:04] 21 ...
[00:51:04] 22 LL |         vector[1] = 5;   //~ ERROR cannot borrow
[00:51:04] 
[00:51:04] 
[00:51:04] The actual stderr differed from the expected stderr.
[00:51:04] The actual stderr differed from the expected stderr.
[00:51:04] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/borrowck/borrowck-for-loop-head-linkage.nll/borrowck-for-loop-head-linkage.nll.stderr
[00:51:04] To update references, rerun the tests and pass the `--bless` flag
[00:51:04] To only update this specific test, also pass `--test-args borrowck/borrowck-for-loop-head-linkage.rs`
[00:51:04] error: 1 errors occurred comparing output.
[00:51:04] status: exit code: 1
[00:51:04] status: exit code: 1
[00:51:04] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/borrowck/borrowck-for-loop-head-linkage.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/borrowck/borrowck-for-loop-head-linkage.nll/a" "-Zborrowck=migrate" "-Ztwo-phase-borrows" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/borrowck/borrowck-for-loop-head-linkage.nll/auxiliary" "-A" "unused"
[00:51:04] ------------------------------------------
[00:51:04] 
[00:51:04] ------------------------------------------
[00:51:04] stderr:
[00:51:04] stderr:
[00:51:04] ------------------------------------------
[00:51:04] {"message":"cannot borrow `vector` as mutable because it is also borrowed as immutable","code":{"code":"E0502","explanation":"\nThis error indicates that you are trying to borrow a variable as mutable when it\nhas already been borrowed as immutable.\n\nExample of erroneous code:\n\n