\n\nYou can find more information about borrowing in the rust-book:\nhttp://doc.rust-lang.org/stable/book/references-and-borrowing.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":592,"byte_end":598,"line_start":14,"line_end":14,"column_start":23,"column_end":29,"is_primary":false,"text":[{"text":"    let mut _y = vec![&mut x];","highlight_start":23,"highlight_end":29}],"label":"borrow of `x` occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":697,"byte_end":698,"line_start":16,"line_end":16,"column_start":21,"column_end":22,"is_primary":true,"text":[{"text":"        let mut z = x; //~ ERROR cannot use `x` because it was mutably borrowed","highlight_start":21,"highlight_end":22}],"label":"use of borrowed `x`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":765,"byte_end":767,"line_start":17,"line_end":17,"column_start":9,"column_end":11,"is_primary":false,"text":[{"text":"        _y.push(&mut z);","highlight_start":9,"highlight_end":11}],"label":"borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0503]: cannot use `x` because it was mutably borrowed\n  --> /checkout/src/test/ui/span/regions-escape-loop-via-vec.rs:16:21\n   |\nLL |     let mut _y = vec![&mut x];\n   |                       ------ borrow of `x` occurs here\nLL |     while x < 10 { //~ ERROR cannot use `x` because it was mutably borrowed\nLL |         let mut z = x; //~ ERROR cannot use `x` because it was mutably borrowed\n   |                     ^ use of borrowed `x`\nLL |         _y.push(&mut z);\n   |         -- borrow later used here\n\n"}
[00:56:19] {"message":"`z` does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n