\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/mut-borrow-outside-loop.rs","byte_start":823,"byte_end":838,"line_start":24,"line_end":24,"column_start":27,"column_end":42,"is_primary":false,"text":[{"text":"        let inner_first = &mut inner_void;","highlight_start":27,"highlight_end":42}],"label":"first mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/mut-borrow-outside-loop.rs","byte_start":867,"byte_end":882,"line_start":25,"line_end":25,"column_start":28,"column_end":43,"is_primary":true,"text":[{"text":"        let inner_second = &mut inner_void; //~ ERROR cannot borrow","highlight_start":28,"highlight_end":43}],"label":"second mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/mut-borrow-outside-loop.rs","byte_start":948,"byte_end":959,"line_start":27,"line_end":27,"column_start":9,"column_end":20,"is_primary":false,"text":[{"text":"        inner_first.use_mut();","highlight_start":9,"highlight_end":20}],"label":"first borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0499]: cannot borrow `inner_void` as mutable more than once at a time\n  --> /checkout/simary":true,"text":[{"text":"        let mut z = x; //~ ERROR cannot use `x` because it was mutably borrowed","highlight_start":21,"highlight_end":22}],"label":"use of borrowed `x`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":765,"byte_end":767,"line_start":17,"line_end":17,"column_start":9,"column_end":11,"is_primary":false,"text":[{"text":"        _y.push(&mut z);","highlight_start":9,"highlight_end":11}],"label":"borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0503]: cannot use `x` because it was mutably borrowed\n  --> /checkout/src/test/ui/span/regions-escape-loop-via-vec.rs:16:21\n   |\nLL |     let mut _y = vec![&mut x];\n   |                       ------ borrow of `x` occurs here\nLL |     while x < 10 { //~ ERROR cannot use `x` because it was mutably borrowed\nLL |         let mut z = x; //~ ERROR cannot use `x` because it was mutably borrowed\n   |                     ^ use of borrowed `x`\nLL |         _y.push(&mut z);\n   |         -- borrow later used here\n\n"}
[00:51:19] {"message":"`z` does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n