\n\nYou can find more information about borrowing in the rust-book:\nhttp://doc.rust-lang.org/stable/book/references-and-borrowing.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":125,"byte_end":131,"line_start":4,"line_end":4,"column_start":23,"column_end":29,"is_primary":false,"text":[{"text":"    let mut _y = vec![&mut x];","highlight_start":23,"highlight_end":29}],"label":"borrow of `x` occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":144,"byte_end":145,"line_start":5,"line_end":5,"column_start":11,"column_end":12,"is_primary":true,"text":[{"text":"    while x < 10 { //~ ERROR cannot use `x` because it was mutably borrowed","highlight_start":11,"highlight_end":12}],"label":"use of borrowed `x`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":298,"byte_end":300,"line_start":7,"line_end":7,"column_start":9,"column_end":11,"is_primary":false,"text":[{"text":"        _y.push(&mut z);","highlight_start":9,"highlight_end":11}],"label":"borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0503]: cannot use `x` because it was mutably borrowed\n  --> /checkout/src/test/ui/span/regions-escape-loop-via-vec.rs:5:11\n   |\nLL |     let mut _y = vec![&mut x];\n   |                       ------ borrow of `x` occurs here\nLL |     while x < 10 { //~ ERROR cannot use `x` because it was mutably borrowed\n   |           ^ use of borrowed `x`\nLL |         let mut z = x; //~ ERROR cannot use `x` because it was mutably borrowed\nLL |         _y.push(&mut z);\n   |         -- borrow later used here\n\n"}
[01:30:33] {"message":"cannot use `x` because it was mutably borrowed","code":{"code":"E0503","explanation":"\nA value was used after it was mutably borrowed.\n\nExample of erroneous code:\n\n