\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":773,"byte_end":779,"line_start":17,"line_end":17,"column_start":17,"column_end":23,"is_primary":true,"text":[{"text":"        _y.push(&mut z);","highlight_start":17,"highlight_end":23}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":875,"byte_end":876,"line_start":20,"line_end":20,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    }","highlight_start":5,"highlight_end":6}],"label":"`z` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":765,"byte_end":767,"line_start":17,"line_end":17,"column_start":9,"column_end":11,"is_primary":false,"text":[{"text":"        _y.push(&mut z);","highlight_start":9,"highlight_end":11}],"label":"borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `z` does not live long enough\n  --> /checkout/src/test/ui/span/regions-escape-loop-via-vec.rs:17:17\n   |\nLL |         _y.push(&mut z);\n   |         --      ^^^^^^ borrowed value does not live long enough\n   |         |\n   |         borrow later used here\n...\nLL |     }\n   |     - `z` dropped here while still borrowed\n\n"}
[00:56:19] {"message":"cannot use `x` because it was mutably borrowed","code":{"code":"E0503","explanatiia-vec.rs","byte_start":592,"byte_end":598,"line_start":14,"line_end":14,"column_start":23,"column_end":29,"is_primary":false,"text":[{"text":"    let mut _y = vec![&mut x];","highlight_start":23,"highlight_end":29}],"label":"borrow of `x` occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":839,"byte_end":845,"line_start":19,"line_end":19,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"        x += 1; //~ ERROR cannot assign","highlight_start":9,"highlight_end":15}],"label":"use of borrowed `x`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/span/regions-escape-loop-via-vec.rs","byte_start":765,"byte_end":767,"line_start":17,"line_end":17,"column_start":9,"column_end":11,"is_primary":false,"text":[{"text":"        _y.push(&mut z);","highlight_start":9,"highlight_end":11}],"label":"borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0503]: cannot use `x` because it was mutably borrowed\n  --> /checkout/src/test/ui/span/regions-escape-loop-via-vec.rs:19:9\n   |\nLL |     let mut _y = vec![&mut x];\n   |                       ------ borrow of `x` occurs here\n...\nLL |         _y.push(&mut z);\n   |         -- borrow later used here\nLL |         //~^ ERROR `z` does not live long enough\nLL |         x += 1; //~ ERROR cannot assign\n   |         ^^^^^^ use of borrowed `x`\n\n"}
[00:56:19] {"message":"aborting due to 4 previous errors"est-args vec/vec-mut-iter-borrow.rs`
[00:56:19] error: 1 errors occurred comparing output.
[00:56:19] status: exit code: 1
[00:56:19] status: exit code: 1
[00:56:19] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/vec/vec-mut-iter-borrow.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/vec/vec-mut-iter-borrow.nll/a" "-Zborrowck=migrate" "-Ztwo-phase-borrows" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/vec/vec-mut-iter-borrow.nll/auxiliary" "-A" "unused"
[00:56:19] ------------------------------------------
[00:56:19] 
[00:56:19] ------------------------------------------
[00:56:19] stderr:
[00:56:19] stderr:
[00:56:19] ------------------------------------------
[00:56:19] {"message":"cannot borrow `xs` as mutable more than once at a time","code":{"code":"E0499","explanation":"\nA variable was borrowed as mutable more than once. Erroneous code example:\n\n