\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/mut-borrow-outside-loop.rs","byte_start":645,"byte_end":654,"line_start":16,"line_end":16,"column_start":17,"column_end":26,"is_primary":false,"text":[{"text":"    let first = &mut void;","highlight_start":17,"highlight_end":26}],"label":"first mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/mut-borrow-outside-loop.rs","byte_start":673,"byte_end":682,"line_start":17,"line_end":17,"column_start":18,"column_end":27,"is_primary":true,"text":[{"text":"    let second = &mut void; //~ ERROR cannot borrow","highlight_start":18,"highlight_end":27}],"label":"second mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/mut-borrow-outside-loop.rs","byte_start":712,"byte_end":717,"line_start":18,"line_end":18,"column_start":5,"column_end":10,"is_primary":false,"text":[{"text":"    first.use_mut();","highlight_start":5,"highlight_end":10}],"label":"first borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0499]: cannot borrow `void` as mutable more than once at a time\n  --> /checkout/src/test/ui/borrowck/mut-borrow-outside-loop.rs:17:18\n   |\nLL |     let first = &mut void;\n   |                 --------- first mutable borrow occurs here\nLL |     let second = &mut void; //~ ERROR cannot borrow\n   |                  ^^^^^^^^^ second mutable borrow occurs here\nLL |     first.use_mut();\n   |     ----- first borrow later used here\n\n"}
[00:56:19] {"message":"cannot borrow `inner_void` as mutable more than once at a time","code":{"code":"E0499","explanation":"\nA variable was borrowed as mutable more than once. Erroneous code example:\n\n