\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/imports/local-modularized-tricky-fail-1.rs","byte_start":988,"byte_end":993,"line_start":45,"line_end":45,"column_start":5,"column_end":10,"is_primary":true,"text":[{"text":"    panic!(); //~ ERROR `panic` is ambiguous","highlight_start":5,"highlight_end":10}],"label":"ambiguous name","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`panic` could refer to a macro from prelude","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/imports/local-modularized-tricky-fail-1.rs","byte_start":0,"byte_end":0,"line_start":1,"line_end":1,"column_start":1,"column_end":1,"is_primary":true,"text":[{"text":"// Copyright 2018 The Rust Project Developers. See the COPYRIGHT","highlight_start":1,"highlight_end":1}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"`panic` could also refer to the macro defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/imports/local-modularized-tricky-fail-1.rs","byte_start":665,"byte_end":708,"line_start":21,"line_end":23,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    macro_rules! panic {","highlight_start":5,"highlight_end":25},{"text":"        () => ()","highlight_start":1,"highlight_end":17},{"text":"    }","h|     panic!(); //~ ERROR `panic` is ambiguous\n   |     ^^^^^ ambiguous name\n   |\nnote: `panic` could refer to a macro from prelude\nnote: `panic` could also refer to the macro defined here\n  --> /checkout/src/test/ui/imports/local-modularized-tricky-fail-1.rs:21:5\n   |\nLL | /     macro_rules! panic {\nLL | |         () => ()\nLL | |     }\n   | |_____^\n...\nLL |       define_panic!();\n   |       ---------------- in this macro invocation\n   = help: use `self::panic` to refer to this macro unambiguously\n\n"}
[00:53:40] {"message":"`panic` is ambiguous (macro-expanded name vs less macro-expanded name from outer scope during import/macro resolution)","code":{"code":"E0659","explanation":"\nAn item usage is ambiguous.\n\nErroneous code example:\n\n