\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/exhaustive_integer_patterns.rs","byte_start":2912,"byte_end":2915,"line_start":126,"line_end":126,"column_start":11,"column_end":14,"is_primary":true,"text":[{"text":"    match 0u8 { //~ ERROR non-exhaustive patterns","highlight_start":11,"highlight_end":14}],"label":"pattern `128u8..=255u8` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `128u8..=255u8` not covered\n  --> /checkout/src/test/ui/exhaustive_integer_patterns.rs:126:11\n   |\nLL |     match 0u8 { //~ ERROR non-exhaustive patterns\n   |           ^^^ pattern `128u8..=255u8` not covered\n\n"}
[00:51:08] {"message":"non-exhaustive patterns: `(0u8, Some(_))` and `(2u8..=255u8, Some(_))` not covered","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n