\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs","byte_start":1125,"byte_end":1129,"line_start":32,"line_end":32,"column_start":11,"column_end":15,"is_primary":true,"text":[{"text":"    match t::a { //~ ERROR non-exhaustive patterns: `b` not covered","highlight_start":11,"highlight_end":15}],"label":"pattern `b` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `b` not covered\n  --> /checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs:32:11\n   |\nLL |     match t::a { //~ ERROR non-exhaustive patterns: `b` not covered\n   |           ^^^^ pattern `b` not covered\n\n"}
[00:51:08] {"message":"non-exhaustive patterns: `[]` not covered","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n