\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs","byte_start":683,"byte_end":687,"line_start":19,"line_end":19,"column_start":11,"column_end":15,"is_primary":true,"text":[{"text":"    match true { //~ ERROR non-exhaustive patterns: `false` not covered","highlight_start":11,"highlight_end":15}],"label":"pattern `false` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `false` not covered\n  --> /checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs:19:11\n   |\nLL |     match true { //~ ERROR non-exhaustive patterns: `false` not covered\n   |           ^^^^ pattern `false` not covered\n\n"}
[00:51:08] {"message":"non-exhaustive patterns: `Some(_)` not covered","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n