\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/precise_pointer_size_matching.rs","byte_start":871,"byte_end":877,"line_start":27,"line_end":27,"column_start":11,"column_end":17,"is_primary":true,"text":[{"text":"    match 0isize { //~ ERROR non-exhaustive patterns","highlight_start":11,"highlight_end":17}],"label":"patterns `-9223372036854775808isize..=-6isize` and `21isize..=9223372036854775807isize` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `-9223372036854775808isize..=-6isize` and `21isize..=9223372036854775807isize` not covered\n  --> /checkout/src/test/ui/precise_pointer_size_matching.rs:27:11\n   |\nLL |     match 0isize { //~ ERROR non-exhaustive patterns\n   |           ^^^^^^ patterns `-9223372036854775808isize..=-6isize` and `21isize..=9223372036854775807isize` not covered\n\n"}
[00:46:39] {"message":"non-exhaustive patterns: `0usize` and `21usize..=18446744073709551615usize` not covered","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n