\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs","byte_start":778,"byte_end":786,"line_start":22,"line_end":22,"column_start":11,"column_end":19,"is_primary":true,"text":[{"text":"    match Some(10) { //~ ERROR non-exhaustive patterns: `Some(_)` not covered","highlight_start":11,"highlight_end":19}],"label":"pattern `Some(_)` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `Some(_)` not covered\n  --> /checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs:22:11\n   |\nLL |     match Some(10) { //~ ERROR non-exhaustive patterns: `Some(_)` not covered\n   |           ^^^^^^^^ pattern `Some(_)` not covered\n\n"}
[00:51:08] {"mhlight_start":11,"highlight_end":20}],"label":"patterns `(_, _, -2147483648i32..=3i32)` and `(_, _, 5i32..=2147483647i32)` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `(_, _, -2147483648i32..=3i32)` and `(_, _, 5i32..=2147483647i32)` not covered\n  --> /checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs:25:11\n   |\nLL |     match (2, 3, 4) { //~ ERROR non-exhaustive patterns: `(_, _, _)` not covered\n   |           ^^^^^^^^^ patterns `(_, _, -2147483648i32..=3i32)` and `(_, _, 5i32..=2147483647i32)` not covered\n\n"}
[00:51:08] {"message":"non-exhaustive patterns: `(a, a)` not covered","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n