\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs","byte_start":604,"byte_end":605,"line_start":18,"line_end":18,"column_start":11,"column_end":12,"is_primary":true,"text":[{"text":"    match x { t::b => { } } //~ ERROR non-exhaustive patterns: `a` not covered","highlight_start":11,"highlight_end":12}],"label":"pattern `a` not covered","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0004]: non-exhaustive patterns: `a` not covered\n  --> /checkout/src/test/ui/non-exhaustive/non-exhaustive-match.rs:18:11\n   |\nLL |     match x { t::b => { } } //~ ERROR non-exhaustive patterns: `a` not covered\n   |           ^ pattern `a` not covered\n\n"}
[00:51:08] {"message":"non-exhaustive patterns: `false` not covered","code":{"code":"E0004","explanation":"\nThis error indicates that the compiler cannot guarantee a matching pattern for\none or more possible inputs to a match expression. Guaranteed matches are\nrequired in order to assign values to match expressions, or alternatively,\ndetermine the flow of execution. Erroneous code example:\n\n