plain
travis_time:end:2066a831:start=1543768649367594094,finish=1543768650374595136,duration=1007001042
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#Pull-Requests-and-Security-Restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
Setting environment variables from .travis.yml
$ export IMAGE=x86_64-gnu-llvm-5.0
---
[00:45:52] .................................................................................................... 2600/5107
[00:45:56] .................................................................................................... 2700/5107
[00:45:59] .................................................................................................... 2800/5107
[00:46:02] .................................................................................................... 2900/5107
[00:46:05] ................................F................................................................... 3000/5107
[00:46:09] ...................................F......F..................................i...................... 3100/5107
[00:46:15] ........................................ii..i..ii................................................... 3300/5107
[00:46:18] .................................................................................................... 3400/5107
[00:46:21] .................................................................................................... 3500/5107
[00:46:24] .....................ii............................................................................. 3600/5107
---
[00:47:10] 
[00:47:10] ---- [ui] ui/lint/issue-47390-unused-variable-in-struct-pattern.rs stdout ----
[00:47:10] diff of stderr:
[00:47:10] 
[00:47:10] 44    = note: consider using `_hours_are_suns` instead
[00:47:10] 45 
[00:47:10] 46 warning: value assigned to `hours_are_suns` is never read
[00:47:10]on_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unused variable: `i_think_continually`\n  --> /checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs:36:9\n   |\nLL |     let i_think_continually = 2;\n   |         ^^^^^^^^^^^^^^^^^^^ help: consider using `_i_think_continually` instead\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs:15:9\n   |\nLL | #![warn(unused)] // UI tests pass `-A unused` (#43896)\n   |         ^^^^^^\n   = note: #[warn(unused_variables)] implied by #[warn(unused)]\n\n"}
[00:47:10] {"message":"unused variable: `mut_unused_var`","code":{"code":"unused_variables","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1061,"byte_end":1075,"line_start":43,"line_end":43,"column_start":13,"column_end":27,"is_primary":true,"text":[{"text":"    let mut mut_unused_var = 1;","highlight_start":13,"highlight_end":27}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider using `_mut_unused_var` instead","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1061,"byte_end":1075,"line_start":43,"line_end":43,"column_start":13,"column_end":27,"is_primary":true,"text":[{"text":"    let mut mut_unused_var = 1;","highlight_start":13,"highlight_end":27}],"label":null,"suggested_replacement":"_mut_unused_var","suggestion_aansion":null}],"children":[{"message":"try ignoring the field","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1847,"byte_end":1851,"line_start":78,"line_end":78,"column_start":24,"column_end":28,"is_primary":true,"text":[{"text":"        (Large::Suit { case },) => {}","highlight_start":24,"highlight_end":28}],"label":null,"suggested_replacement":"case: _","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unused variable: `case`\n  --> /checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs:78:24\n   |\nLL |         (Large::Suit { case },) => {}\n   |                        ^^^^ help: try ignoring the field: `case: _`\n\n"}
[00:47:10] {"message":"unused variable: `case`","code":{"code":"unused_variables","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1936,"byte_end":1940,"line_start":83,"line_end":83,"column_start":24,"column_end":28,"is_primary":true,"text":[{"text":"        [Large::Suit { case }] => {}","highlight_start":24,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try ignoring the field","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1936,"byte_end":1940,"line_start":83,"line_end":83,"column_start":24,"column_end":28,"is_primary":true,"text":[{"text":"     nused-variable-in-struct-pattern.rs:88:29\n   |\nLL |         Tuple(Large::Suit { case }, ()) => {}\n   |                             ^^^^ help: try ignoring the field: `case: _`\n\n"}
[00:47:10] {"message":"variable does not need to be mutable","code":{"code":"unused_mut","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1057,"byte_end":1075,"line_start":43,"line_end":43,"column_start":9,"column_end":27,"is_primary":true,"text":[{"text":"    let mut mut_unused_var = 1;","highlight_start":9,"highlight_end":27}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":542,"byte_end":548,"line_start":15,"line_end":15,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"#![warn(unused)] // UI tests pass `-A unused` (#43896)","highlight_start":9,"highlight_end":15}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"#[warn(unused_mut)] implied by #[warn(unused)]","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"remove this `mut`","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.rs","byte_start":1057,"byte_end":1061,"line_start":43,"line_end":43,"column_start":9,"column_end":13,"is_primary":true,"text":[{"te7:10] +                      (maybe it is overwritten before being read)
[00:47:10] 21    |
[00:47:10] 21    |
[00:47:10] 22 LL | fn f4(mut x: i32) { //~ ERROR: value passed to `x` is never read
[00:47:10] 23    |           ^
[00:47:10] 24 
[00:47:10] 25 error: value assigned to `x` is never read
[00:47:10] 25 error: value assigned to `x` is never read
[00:47:10] +                      (maybe it is overwritten before being read)
[00:47:10] 27    |
[00:47:10] 27    |
[00:47:10] 28 LL |     x = 4; //~ ERROR: value assigned to `x` is never read
[00:47:10] 
[00:47:10] The actual stderr differed from the expected stderr.
[00:47:10] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/liveness/liveness-dead/liveness-dead.stderr
[00:47:10] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/liveness/liveness-dead/liveness-dead.stderr
[00:47:10] To update references, rerun the tests and pass the `--bless` flag
[00:47:10] To only update this specific test, also pass `--test-args liveness/liveness-dead.rs`
[00:47:10] error: 1 errors occurred comparing output.
[00:47:10] status: exit code: 1
[00:47:10] status: exit code: 1
[00:47:10] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/liveness/liveness-dead.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/liveness/liveness-dead/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/liveness/liveness-dead/auxiliary" "-A" "unused"
[00:47:10] stdout:
[00:47:10] ----------rwritten before being read)\n  --> /checkout/src/test/ui/liveness/liveness-dead.rs:30:11\n   |\nLL | fn f4(mut x: i32) { //~ ERROR: value passed to `x` is never read\n   |           ^\n\n"}
[00:47:10] {"message":"value assigned to `x` is never read\n                     (maybe it is overwritten before being read)","code":{"code":"unused_assignments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/liveness/liveness-dead.rs","byte_start":925,"byte_end":926,"line_start":37,"line_end":37,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    x = 4; //~ ERROR: value assigned to `x` is never read","highlight_start":5,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: value assigned to `x` is never read\n                     (maybe it is overwritten before being read)\n  --> /checkout/src/test/ui/liveness/liveness-dead.rs:37:5\n   |\nLL |     x = 4; //~ ERROR: value assigned to `x` is never read\n   |     ^\n\n"}
[00:47:10] {"message":"aborting due to 4 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 4 previous errors\n\n"}
[00:47:10] ------------------------------------------
[00:47:10] 
[00:47:10] thread '[ui] ui/liveness/liveness-dead.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3282:9
[00:47:10] 
---
[00:47:10] 
[00:47:10] ------------------------------------------
[00:47:10] stderr:
[00:47:10] ------------------------------------------
[00:47:10] {"message":"unreachable statement","code":{"code":"unreachable_code","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/liveness/liveness-unused.rs","byte_start":1974,"byte_end":1990,"line_start":102,"line_end":102,"column_start":9,"column_end":25,"is_primary":true,"text":[{"text":"        drop(*x as i32); //~ WARNING unreachable statement","highlight_start":9,"highlight_end":25}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/liveness/liveness-unused.rs","byte_start":475,"byte_end":481,"line_start":11,"line_end":11,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"#![warn(unused)]","highlight_start":9,"highlight_end":15}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"#[warn(unreachable_code)] implied by #[warn(unused)]","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"warning: unreachable statement\n  --> /checkout/src/test/ui/liveness/liveness-unused.rs:102:9\n   |\nLL |         drop(*x as i32); //~ WARNING unreachable statement\n   |         ^^^^^^^^^^^^^^^^\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/liveness/liveness-unused.rs:11:9\n   |\nLLe) {\n   |       ^ help: consider using `_x` instead\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/liveness/liveness-unused.rs:12:9\n   |\nLL | #![deny(unused_variables)]\n   |         ^^^^^^^^^^^^^^^^\n\n"}
[00:47:10] {"message":"unused variable: `x`","code":{"code":"unused_variables","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/liveness/liveness-unused.rs","byte_start":697,"byte_end":698,"line_start":22,"line_end":22,"column_start":8,"column_end":9,"is_primary":true,"text":[{"text":"fn f1b(x: &mut isize) {","highlight_start":8,"highlight_end":9}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider using `_x` instead","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/liveness/liveness-unused.rs","byte_start":697,"byte_end":698,"line_start":22,"line_end":22,"column_start":8,"column_end":9,"is_primary":true,"text":[{"text":"fn f1b(x: &mut isize) {","highlight_start":8,"highlight_end":9}],"label":null,"suggested_replacement":"_x","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unused variable: `x`\n  --> /checkout/src/test/ui/liveness/liveness-unused.rs:22:8\n   |\nLL | fn f1b(x: &mut isize) {\n   |        ^ help: consider using `_x` instead\n\n"}
[00:47:10] {"message":"unused variable: `x`","code":{"code":"unused_variables","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/liveness/liveness-unused.rs","byte_start":820,"byte_end":821,"line_start":30,"line_end":30,"column_start":light_start":9,"highlight_end":10}],"label":null,"suggested_replacement":"_x","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unused variable: `x`\n  --> /checkout/src/test/ui/liveness/liveness-unused.rs:35:9\n   |\nLL |     let x = 3;\n   |         ^ help: consider using `_x` instead\n\n"}
[00:47:10] {"message":"variable `x` is assigned to, but never used","code":{"code":"unused_variables","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/liveness/liveness-unused.rs","byte_start":955,"byte_end":956,"line_start":40,"line_end":40,"column_start":13,"column_end":14,"is_primary":true,"text":[{"text":"    let mut x = 3;","highlight_start":13,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider using `_x` instead","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: variable `x` is assigned to, but never used\n  --> /checkout/src/test/ui/liveness/liveness-unused.rs:40:13\n   |\nLL |     let mut x = 3;\n   |             ^\n   |\n   = note: consider using `_x` instead\n\n"}
[00:47:10] {"message":"value assigned to `x` is never read\n                     (maybe it is overwritten before being read)","code":{"code":"unused_assignments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/liveness/liveness-unused.rs","byte_start":1025,"byte_end":1026,"line_start":42,"line_end":42,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    x += 4;","highlight_start":5,"hig
