\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/issue-55850.rs","byte_start":1027,"byte_end":1028,"line_start":38,"line_end":38,"column_start":16,"column_end":17,"is_primary":true,"text":[{"text":"        yield &s[..] //~ ERROR `s` does not live long enough [E0597]","highlight_start":16,"highlight_end":17}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/issue-55850.rs","byte_start":1085,"byte_end":1086,"line_start":39,"line_end":39,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    })","highlight_start":5,"highlight_end":6}],"label":"`s` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `s` does not live long enough\n  --> /checkout/src/test/ui/nll/issue-55850.rs:38:16\n   |\nLL |         yield &s[..] //~ ERROR `s` does not live long enough [E0597]\n   |                ^ borrowed value does not live long enough\nLL |     })\n   |     - `s` dropped here while still borrowed\n\n"}
[01:07:48] {"message":"borrow may still be in use when generator yields","code":{"code":"E0626","explanation":"\nThis error occurs because a borrow in a generator persists across a\nyield point.\n\n