\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":402,"byte_end":404,"line_start":22,"line_end":22,"column_start":13,"column_end":15,"is_primary":true,"text":[{"text":"    let y = &x; //~ ERROR","highlight_start":13,"highlight_end":15}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":457,"byte_end":458,"line_start":25,"line_end":25,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"`x` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":431,"byte_end":443,"line_start":23,"line_end":23,"column_start":16,"column_end":28,"is_primary":false,"text":[{"text":"    let ref z: &'static u32 = y;","highlight_start":16,"highlight_end":28}],"label":"type annotation requires that `x` is borrowed for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `x` does not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:22:13\n   |\nLL |     let y = &x; //~ ERROR\n   |             ^^ borrowed value does not live long enough\nLL |     let ref z: &'static u32 = y;\n   |                ------------ type annotation requires that `x` is borrowed for `'static`\nLL |     **z\nLL | }\n   | - `x` dropped here while still borrowed\n\n"}
[00:48:48] {"message":"`x` does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n