\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs","byte_start":1045,"byte_end":1050,"line_start":32,"line_end":32,"column_start":28,"column_end":33,"is_primary":true,"text":[{"text":"    let _: &'static u32 = &meh(); //~ ERROR does not live long enough","highlight_start":28,"highlight_end":33}],"label":"creates a temporary which is freed while still in use","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs","byte_start":1208,"byte_end":1209,"line_start":35,"line_end":35,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"temporary value is freed at the end of this statement","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs","byte_start":1029,"byte_end":1041,"line_start":32,"line_end":32,"column_start":12,"column_end":24,"is_primary":false,"text":[{"text":"    let _: &'static u32 = &meh(); //~ ERROR does not live long enough","highlight_start":12,"highlight_end":24}],"label":"type annotation requires that borrow lasts for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0716]: temporary value dropped while borrowed\n  --> /checkout/src/test/ui/consts/const-eval/dont_promote_unstable_const_fn.rs:32:28\n   |\nLL |     let _: &'static u32 = &meh(); //~ ERROR does not live long enough\n   |            ------------    ^^^^^ creates a temporary which is freed while still in use\n   |            |\n   |            type annotation requires that borrow lasts for `'static`\n...\nLL | }\n   | - temporary value is freed at the end of this statement\n\n"}
[01:08:49] {"message":"temporary value dropped while borrowed","code":{"code":"E0716","explanation":"\nThis error indicates that a temporary value is being dropped\nwhile a borrow is still in active use.\n\nErroneous code example:\n\n