\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/proc-macro/dollar-crate.rs","byte_start":643,"byte_end":663,"line_start":23,"line_end":23,"column_start":13,"column_end":33,"is_primary":true,"text":[{"text":"            struct D($crate::S); //~ ERROR the name `D` is defined multiple times","highlight_start":13,"highlight_end":33}],"label":"`D` redefined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/proc-macro/dollar-crate.rs","byte_start":735,"byte_end":744,"line_start":27,"line_end":27,"column_start":5,"column_end":14,"is_primary":false,"text":[{"text":"    local!();","highlight_start":5,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"local!","def_site_span":{"file_name":"/checkout/src/test/ui/proc-macro/dollar-crate.rs","byte_start":407,"byte_end":729,"line_start":13,"line_end":25,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    macro_rules! local {","highlight_start":5,"highlight_end":25},{"text":"        () => {","highlight_start":1,"highlight_end":16},{"text":"            dollar_crate::m! {","highlight_start":1,"highlight_end":31},{"text":"                struct M($crate::S);","highlight_start":1,"highlight_end":37},{"text":"            }","highlight_start":1,"highlight_end":14},{"text":"","highlight_start":1,"highlight_end":1},{"text":"            #[dollar_crate::a]","highlight_start":1,"highlight_end":31},{"text":"            struct A($crate::S);","highlight_start":1,"highlight_end":33},{"text":"","highlight_start":1,"highlight_end":1},{"text":"            #[derive(dollar_crate::d)]","highlight_start":1,"highlight_end":39},{"text":"            struct D($crate::S); //~ ERROR the name `D` is defined multiple times","highlight_start":1,"highlight_end":82},{"text":"        };","highlight_start":1,"highlight_end":11},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}},{"file_name":"/checkout/src/test/ui/proc-macro/dollar-crate.rs","byte_start":643,"byte_end":663,"line_start":23,"line_end":23,"column_start":13,"column_end":33,"is_primary":true,"text":[{"text":"            struct D($crate::S); //~ ERROR the name `D` is defined multiple times","highlight_start":13,"highlight_end":33}],"label":"previous definition of the type `D` here","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/proc-macro/dollar-crate.rs","byte_start":735,"byte_end":744,"line_start":27,"line_end":27,"column_start":5,"column_end":14,"is_primary":false,"text":[{"text":"    local!();","highlight_start":5,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"local!","def_site_span":{"file_name":"/checkout/src/test/ui/proc-macro/dollar-crate.rs","byte_start":407,"byte_end":729,"line_start":13,"line_end":25,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    macro_rules! local {","highlight_start":5,"highlight_end":25},{"text":"        () => {","highlight_start":1,"highlight_end":16},{"text":"            dollar_crate::m! {","highlight_start":1,"highlight_end":31},{"text":"                struct M($crate::S);","highlight_start":1,"highlight_end":37},{"text":"            }","highlight_start":1,"highlight_end":14},{"text":"","highlight_start":1,"highlight_end":1},{"text":"            #[dollar_crate::a]","highlight_start":1,"highlight_end":31},{"text":"            struct A($crate::S);","highlight_start":1,"highlight_end":33},{"text":"","highlight_start":1,"highlight_end":1},{"text":"            #[derive(dollar_crate::d)]","highlight_start":1,"highlight_end":39},{"text":"            struct D($crate::S); //~ ERROR the name `D` is defined multiple times","highlight_start":1,"highlight_end":82},{"text":"        };","highlight_start":1,"highlight_end":11},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[{"message":"`D` must be defined only once in the type namespace of this module","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0428]: the name `D` is defined multiple times\n  --> /checkout/src/test/ui/proc-macro/dollar-crate.rs:23:13\n   |\nLL |             struct D($crate::S); //~ ERROR the name `D` is defined multiple times\n   |             ^^^^^^^^^^^^^^^^^^^^\n   |             |\n   |             `D` redefined here\n   |             previous definition of the type `D` here\n...\nLL |     local!();\n   |     --------- in this macro invocation\n   |\n   = note: `D` must be defined only once in the type namespace of this module\n\n"}
[00:44:25] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:503:22
[00:44:25] {"message":"the name `D` is defined multiple times","code":{"code":"E0428","explanation":"\nA type or module has been defined more than once.\n\nErroneous code example:\n\n