\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-42060.rs","byte_start":707,"byte_end":708,"line_start":19,"line_end":19,"column_start":13,"column_end":14,"is_primary":true,"text":[{"text":"    <typeof(q)>::N //~ ERROR attempt to use a non-constant value in a constant","highlight_start":13,"highlight_end":14}],"label":"non-constant value","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0435]: attempt to use a non-constant value in a constant\n  --> /checkout/src/test/ui/issues/issue-42060.rs:19:13\n   |\nLL |     <typeof(q)>::N //~ ERROR attempt to use a non-constant value in a constant\n   |             ^ non-constant value\n\n"}
[01:00:57] {"message":"`typeof` is a reserved keyword but unimplemented","code":{"code":"E0516","explanation":"\nThe `typeof` keyword is currently reserved but unimplemented.\nErroneous code example:\n\n