\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-42060.rs","byte_start":521,"byte_end":526,"line_start":13,"line_end":13,"column_start":23,"column_end":28,"is_primary":true,"text":[{"text":"    let other: typeof(thing) = thing; //~ ERROR attempt to use a non-constant value in a constant","highlight_start":23,"highlight_end":28}],"label":"non-constant value","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0435]: attempt to use a non-constant value in a constant\n  --> /checkout/src/test/ui/issues/issue-42060.rs:13:23\n   |\nLL |     let other: typeof(thing) = thing; //~ ERROR attempt to use a non-constant value in a constant\n   |                       ^^^^^ non-constant value\n\n"}
[00:49:31] {"message":"attempt to use a non-constant value in a constant","code":{"code":"E0435","explanation":"\nA non-constant value was used in a constant expression.\n\nErroneous code example:\n\n