\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-13853.rs","byte_start":1089,"byte_end":1094,"line_start":47,"line_end":47,"column_start":13,"column_end":18,"is_primary":true,"text":[{"text":"    iterate(graph); //~ ERROR mismatched types","highlight_start":13,"highlight_end":18}],"label":"expected reference, found struct `std::vec::Vec`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `&_`\n   found type `std::vec::Vec<Stuff>`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"consider borrowing here","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-13853.rs","byte_start":1089,"byte_end":1094,"line_start":47,"line_end":47,"column_start":13,"column_end":18,"is_primary":true,"text":[{"text":"    iterate(graph); //~ ERROR mismatched types","highlight_start":13,"highlight_end":18}],"label":null,"suggested_replacement":"&graph","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/issues/issue-13853.rs:47:13\n   |\nLL |     iterate(graph); //~ ERROR mismatched types\n   |             ^^^^^\n   |             |\n   |             expected reference, found struct `std::vec::Vec`\n   |             help: consider borrowing here: `&l,E0308\nlet x: i32 = \"I am not a number!\";\n//     ~~~   ~~~~~~~~~~~~~~~~~~~~\n//      |             |\n//      |    initializing expression;\n//      |    compiler infers type `&str`\n//      |\n//    type `i32` assigned to variable `x`\n