\nfn main() {\n    let s1 = String::from(\"hello\");\n\n    let len = calculate_length(&s1);\n\n    println!(\"The length of '{}' is {}.\", s1, len);\n}\n\nfn calculate_length(s: &String) -> uopy` trait","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0382]: use of moved value: `rhs`\n  --> /checkout/src/test/ui/binop/binop-consume-args.rs:30:10\n   |\nLL |     lhs * rhs;\n   |           --- value moved here\nLL |     drop(lhs);  //~ ERROR use of moved value: `lhs`\nLL |     drop(rhs);  //~ ERROR use of moved value: `rhs`\n   |          ^^^ value used here after move\n   |\n   = note: move occurs because `rhs` has type `B`, which does not implement the `Copy` trait\n\n"}
[00:47:45] {"message":"use of moved value: `lhs`","code":{"code":"E0382","explanation":"\nThis error occurs when an attempt is made to use a variable after its contents\nhave been moved elsewhere. For example:\n\n