\n\nWith this approach, x and y share ownership of the data via the `Rc` (reference\ncount type). `RefCell` essentially performs runtime borrow checking: ensuring\nthat at most one writer or multiple readers can access the data at any one time.\n\nIf you wish to learn more about ownership in Rust, start with the chapter in the\nBook:\n\nhttps://doc.rust-lang.org/book/first-edition/ownership.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/binop/binop-consume-args.rs","byte_start":1015,"byte_end":1018,"line_start":29,"line_end":29,"column_start":10,"column_end":13,"is_primary":true,"text":[{"text":"    drop(lhs);  //~ ERROR use of moved value: `lhs`","highlight_start":10,"highlight_end":13}],"label":"value used here after move","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/binop/binop-consume-args.rs","byte_start":995,"byte_end":998,"line_start":28,"line_end":28,"column_start":5,"column_end":8,"is_primary":false,"text":[{"text":"    lhs * rhs;","highlight_start":5,"highlight_end":8}],"label":"value moved here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"move occurs because `lhs` has type `A`, which does not implement the `Copy` trait","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0382]: use of moved value: `lhs`\n  --> /checkout/src/test/ui/binop/binop-consume-args.rs:29:10\n   |\nLL |     lhs * rhs;\n   |     --- value moved here\nLL |     drop(lhs);  //~ ERROR use of moved value: `lhs`\n   |          ^^^ value used here after move\n   |\n   = note: move occurs because `lhs` has type `A`, which does not implement the `Copy` trait\n\n"}
[01:00:57] {"message":"use of moved value: `rhs`","code":{"code":"E0382","explanation":"\nThis error occurs when an attempt is made to use a variable after its contents\nhave been moved elsewhere. For example:\n\n