\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/associated-types/associated-types-multiple-types-one-trait.rs","byte_start":690,"byte_end":696,"line_start":28,"line_end":28,"column_start":5,"column_end":11,"is_primary":true,"text":[{"text":"    want_x(t); //~ ERROR type mismatch","highlight_start":5,"highlight_end":11}],"label":"expected associated type, found u32","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `<T as Foo>::X`\n   found type `u32`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"required by `want_x`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/associated-types/associated-types-multiple-types-one-trait.rs","byte_start":997,"byte_end":1027,"line_start":52,"line_end":52,"column_start":1,"column_end":31,"is_primary":true,"text":[{"text":"fn want_x<T:Foo<X=u32>>(t: &T) { }","highlight_start":1,"highlight_end":31}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0271]: type mismatch resolving `<T as Foo>::X == u32`\n  --> /checkout/src/test/ui/associated-types/associated-types-multiple-types-one-trait.rs:28:5\n   |\nLL |     want_x(t); //~ ERROR type mismatch\n   |     ^^^^^^ expected associated type, found u32\n   |\n   = note: expected type `<T as Foo>::X`\n              found type `u32`\nnote: required by `want_x`\n  --> /checkout/src/test/ui/associated-types/associated-types-multiple-types-one-trait.rs:52:1\n   |\nLL | fn want_x<T:Foo<X=u32>>(t: &T) { }\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:49:31] {"message":"type mismatch resolving `<T as Foo>::Y == i32`","code":{"code":"E0271","explanation":"\nThis is because of a type mismatch between the associated type of some\ntrait (e.g., `T::Bar`, where `T` implements `trait Quux { type Bar; }`)\nand another type `U` that is required to be equal to `T::Bar`, but is not.\nExamples follow.\n\nHere is a basic example:\n\n