\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-42060.rs","byte_start":700,"byte_end":709,"line_start":19,"line_end":19,"column_start":6,"column_end":15,"is_primary":true,"text":[{"text":"    <typeof(q)>::N //~ ERROR attempt to use a non-constant value in a constant","highlight_start":6,"highlight_end":15}],"label":"reserved keyword","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0516]: `typeof` is a reserved keyword but unimplemented\n  --> /checkout/src/test/ui/issues/issue-42060.rs:19:6\n   |\nLL |     <typeof(q)>::N //~ ERROR attempt to use a non-constant value in a constant\n   |      ^^^^^^^^^ reserved keyword\n\n"}
[01:00:57] {"message":"`typeof` is a reserved keyword but unimplemented","code":{"code":"E0516","explanation":"\nThe `typeof` keyword is currently reserved but unimplemented.\nErroneous code example:\n\n