\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/associated-type/associated-type-projection-from-supertrait.rs","byte_start":1611,"byte_end":1616,"line_start":51,"line_end":51,"column_start":28,"column_end":33,"is_primary":true,"text":[{"text":"fn g() { ModelU.chip_paint(Black); } //~ ERROR mismatched types","highlight_start":28,"highlight_end":33}],"label":"expected struct `Blue`, found struct `Black`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `Blue`\n   found type `Black`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/associated-type/associated-type-projection-from-supertrait.rs:51:28\n   |\nLL | fn g() { ModelU.chip_paint(Black); } //~ ERROR mismatched types\n   |                            ^^^^^ expected struct `Blue`, found struct `Black`\n   |\n   = note: expected type `Blue`\n              found type `Black`\n\n"}
[00:49:31] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n