\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/closure-access-spans.rs","byte_start":718,"byte_end":724,"line_start":22,"line_end":22,"column_start":13,"column_end":19,"is_primary":false,"text":[{"text":"    let r = &mut x;","highlight_start":13,"highlight_end":19}],"label":"first mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-access-spans.rs","byte_start":730,"byte_end":732,"line_start":23,"line_end":23,"column_start":5,"column_end":7,"is_primary":true,"text":[{"text":"    || x = 2; //~ ERROR","highlight_start":5,"highlight_end":7}],"label":"second mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-access-spans.rs","byte_start":73ine_end":35,"column_start":13,"column_end":14,"is_primary":true,"text":[{"text":"    move || x; //~ ERROR","highlight_start":13,"highlight_end":14}],"label":"use of borrowed `x`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/closure-access-spans.rs","byte_start":985,"byte_end":986,"line_start":36,"line_end":36,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    r.use_ref();","highlight_start":5,"highlight_end":6}],"label":"borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0503]: cannot use `x` because it was mutably borrowed\n  --> /checkout/src/test/ui/nll/closure-access-spans.rs:35:13\n   |\nLL |     let r = &mut x;\n   |             ------ borrow of `x` occurs here\nLL |     move || x; //~ ERROR\n   |             ^ use of borrowed `x`\nLL |     r.use_ref();\n   |     - borrow later used here\n\n"}
[00:47:58] {"message":"cannot move out of `x` because it is borrowed","code":{"code":"E0505","explanation":"\nA value was moved out while it was still borrowed.\n\nErroneous code example:\n\n