\n\nWith this approach, x and y share ownership of the data via the `Rc` (reference\ncount type). `RefCell` essentially performs runtime borrow checking: ensuring\nthat at most one writer or multiple readers can access the data at any one time.\n\nIf you wish to learn more about ownership in Rust, start with the chapter in the\nBook:\n\nhttps://doc.rust-lang.org/book/first-edition/ownership.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/borrowck-move-out-from-array.rs","byte_start":840,"byte_end":842,"line_start":26,"line_end":26,"column_start":10,"column_end":12,"is_primary":false,"text":[{"text":"    let [_x, _] = a;","highlight_start":10,"highlight_end":12}],"label":"value moved here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/chec the chapter in the\nBook:\n\nhttps://doc.rust-lang.org/book/first-edition/ownership.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/borrowck-reinit.rs","byte_start":627,"byte_end":628,"line_start":17,"line_end":17,"column_start":10,"column_end":11,"is_primary":false,"text":[{"text":"    drop(x);","highlight_start":10,"highlight_end":11}],"label":"value moved here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-reinit.rs","byte_start":646,"byte_end":647,"line_start":18,"line_end":18,"column_start":16,"column_end":17,"is_primary":true,"text":[{"text":"    let _ = (1,x); //~ ERROR use of moved value: `x` (Ast)","highlight_start":16,"highlight_end":17}],"label":"value used here after partial move","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"move occurs because `x` has type `std::boxed::Box<i32>`, which does not implement the `Copy` trait","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0382]: use of moved value: `x` (Mir)\n  --> /checkout/src/test/ui/borrowck/borrowck-reinit.rs:18:16\n   |\nLL |     drop(x);\n   |          - value moved here\nLL |     let _ = (1,x); //~ ERROR use of moved value: `x` (Ast)\n   |                ^ value used here after partial move\n   |\n   = note: move occurs because `x` has type `std::boxed::Box<i32>`, which does not implement the `Copy` trait\n\n"}
[00:47:58] {"message":"aborting due to 2 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aboove occurs because `line1.origin` has type `Point`, which does not implement the `Copy` trait","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0382]: use of moved value: `line1.origin`\n  --> /checkout/src/test/ui/borrowck/borrowck-uninit-field-access.rs:39:13\n   |\nLL |     let _moved = line1.origin;\n   |                  ------------ value moved here\nLL |     let _ = line1.origin.x + 1; //[ast]~ ERROR use of moved value: `line1.origin.x`\n   |             ^^^^^^^^^^^^^^ value used here after move\n   |\n   = note: move occurs because `line1.origin` has type `Point`, which does not implement the `Copy` trait\n\n"}
[00:47:58] {"message":"use of moved value: `line2`","code":{"code":"E0382","explanation":"\nThis error occurs when an attempt is made to use a variable after its contents\nhave been moved elsewhere. For example:\n\n