plain
[00:51:04] 
[00:51:04] ---- [ui] ui/issues/issue-22644.rs stdout ----
[00:51:04] diff of stderr:
[00:51:04] 
[00:51:04] 89 LL |     println!("{}", a: &mut 4); //~ ERROR expected type, found `4`
[00:51:04] 90    |                            ^ expecting a type here because of type ascription
[00:51:04] - error: aborting due to 9 previous errors
[00:51:04] - error: aborting due to 9 previous errors
[00:51:04] + error[E0658]: type ascription is experimental (see issue #23416)
[00:51:04] +   --> $DIR/issue-22644.rs:10:20
[00:51:04] +    |
[00:51:04] + LL |     println!("{}", a: usize > long_name);
[00:51:04] +    |
[00:51:04] +    = help: add #![feature(type_ascription)] to the crate attributes to enable
[00:51:04] 93 
[00:51:04] 93 
[00:51:04] + error[E0658]: type ascription is experimental (see issue #23416)
[00:51:04] +   --> $DIR/issue-22644.rs:11:22
[00:51:04] +    |
[00:51:04] + LL |     println!("{}{}", a: usize < long_name, long_name);
[00:51:04] +    |
[00:51:04] +    = help: add #![feature(type_ascription)] to the crate attributes to enable
[00:51:04] + 
[00:51:04] + 
[00:51:04] + error[E0658]: type ascription is experimental (see issue #23416)
[00:51:04] +   --> $DIR/issue-22644.rs:13:20
[00:51:04] +    |
[00:51:04] + LL |     println!("{}", a: usize < 4); //~ ERROR `<` is interpreted as a start of generic
[00:51:04] +    |
[00:51:04] +    = help: add #![feature(type_ascription)] to the crate attributes to enable
[00:51:04] + 
[00:51:04] + error: aborting due to 12 previous errors
[00:51:04] + error: aborting due to 12 previous errors
[00:51:04] + 
[00:51:04] + For more information about this error, try `rustc --explain E0658`.
[00:51:04] 94 
[00:51:04] 
[00:51:04] 
[00:51:04] The actual stderr differed from the expected stderr.
[00:51:04] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-22644/issue-22644.stderr
[00:51:04] To update references, rerun the tests and pass the `--bless` flag
[00:51:04] To only update this specific test, also pass `--test-args issues/issue-22644.rs`
[00:51:04] error: 1 errors occurred comparing output.
[00:51:04] status: exit code: 1
[00:51:04] status: exit code: 1
[00:51:04] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/issues/issue-22644.rs" "--target=x86_64-unknown-linux-musl" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-22644/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-musl/native/rust-test-helpers" "-Clinker=/musl-x86_64/bin/musl-gcc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-22644/auxiliary" "-A" "unused"
[00:51:04] ------------------------------------------
[00:51:04] 
[00:51:04] ------------------------------------------
[00:51:04] stderr:
[00:51:04] stderr:
[00:51:04] ------------------------------------------
[00:51:04] {"message":"`<` is interpreted as a start of generic arguments for `usize`, not a comparison","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":141,"byte_end":150,"line_start":6,"line_end":6,"column_start":33,"column_end":42,"is_primary":false,"text":[{"text":"    println!(\"{}\", a as usize < long_name); //~ ERROR `<` is interpreted as a start of generic","highlight_start":33,"highlight_end":42}],"label":"interpreted as generic arguments","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":139,"byte_end":140,"line_start":6,"line_end":6,"column_start":31,"column_end":32,"is_primary":true,"text":[{"text":"    println!(\"{}\", a as usize < long_name); //~ ERROR `<` is interpreted as a start of generic","highlight_start":31,"highlight_end":32}],"label":"not interpreted as comparison","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try comparing the cast value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":128,"byte_end":138,"line_start":6,"line_end":6,"column_start":20,"column_end":30,"is_primary":true,"text":[{"text":"    println!(\"{}\", a as usize < long_name); //~ ERROR `<` is interpreted as a start of generic","highlight_start":20,"highlight_end":30}],"label":null,"suggested_replacement":"(a as usize)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: `<` is interpreted as a start of generic arguments for `usize`, not a comparison\n  --> /checkout/src/test/ui/issues/issue-22644.rs:6:31\n   |\nLL |     println!(\"{}\", a as usize < long_name); //~ ERROR `<` is interpreted as a start of generic\n   |                    ---------- ^ --------- interpreted as generic arguments\n   |                    |          |\n   |                    |          not interpreted as comparison\n   |                    help: try comparing the cast value: `(a as usize)`\n\n"}
[00:51:04] {"message":"`<` is interpreted as a start of generic arguments for `usize`, not a comparison","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":238,"byte_end":258,"line_start":7,"line_end":7,"column_start":35,"column_end":55,"is_primary":false,"text":[{"text":"    println!(\"{}{}\", a as usize < long_name, long_name);","highlight_start":35,"highlight_end":55}],"label":"interpreted as generic arguments","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":236,"byte_end":237,"line_start":7,"line_end":7,"column_start":33,"column_end":34,"is_primary":true,"text":[{"text":"    println!(\"{}{}\", a as usize < long_name, long_name);","highlight_start":33,"highlight_end":34}],"label":"not interpreted as comparison","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try comparing the cast value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":225,"byte_end":235,"line_start":7,"line_end":7,"column_start":22,"column_end":32,"is_primary":true,"text":[{"text":"    println!(\"{}{}\", a as usize < long_name, long_name);","highlight_start":22,"highlight_end":32}],"label":null,"suggested_replacement":"(a as usize)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: `<` is interpreted as a start of generic arguments for `usize`, not a comparison\n  --> /checkout/src/test/ui/issues/issue-22644.rs:7:33\n   |\nLL |     println!(\"{}{}\", a as usize < long_name, long_name);\n   |                      ---------- ^ -------------------- interpreted as generic arguments\n   |                      |          |\n   |                      |          not interpreted as comparison\n   |                      help: try comparing the cast value: `(a as usize)`\n\n"}
[00:51:04] {"message":"`<` is interpreted as a start of generic arguments for `usize`, not a comparison","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":349,"byte_end":350,"line_start":9,"line_end":9,"column_start":33,"column_end":34,"is_primary":false,"text":[{"text":"    println!(\"{}\", a as usize < 4); //~ ERROR `<` is interpreted as a start of generic","highlight_start":33,"highlight_end":34}],"label":"interpreted as generic arguments","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":347,"byte_end":348,"line_start":9,"line_end":9,"column_start":31,"column_end":32,"is_primary":true,"text":[{"text":"    println!(\"{}\", a as usize < 4); //~ ERROR `<` is interpreted as a start of generic","highlight_start":31,"highlight_end":32}],"label":"not interpreted as comparison","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try comparing the cast value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":336,"byte_end":346,"line_start":9,"line_end":9,"column_start":20,"column_end":30,"is_primary":true,"text":[{"text":"    println!(\"{}\", a as usize < 4); //~ ERROR `<` is interpreted as a start of generic","highlight_start":20,"highlight_end":30}],"label":null,"suggested_replacement":"(a as usize)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: `<` is interpreted as a start of generic arguments for `usize`, not a comparison\n  --> /checkout/src/test/ui/issues/issue-22644.rs:9:31\n   |\nLL |     println!(\"{}\", a as usize < 4); //~ ERROR `<` is interpreted as a start of generic\n   |                    ---------- ^ - interpreted as generic arguments\n   |                    |          |\n   |                    |          not interpreted as comparison\n   |                    help: try comparing the cast value: `(a as usize)`\n\n"}
[00:51:04] {"message":"`<` is interpreted as a start of generic arguments for `usize`, not a comparison","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":478,"byte_end":498,"line_start":11,"line_end":11,"column_start":33,"column_end":53,"is_primary":false,"text":[{"text":"    println!(\"{}{}\", a: usize < long_name, long_name);","highlight_start":33,"highlight_end":53}],"label":"interpreted as generic arguments","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":476,"byte_end":477,"line_start":11,"line_end":11,"column_start":31,"column_end":32,"is_primary":true,"text":[{"text":"    println!(\"{}{}\", a: usize < long_name, long_name);","highlight_start":31,"highlight_end":32}],"label":"not interpreted as comparison","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try comparing the cast value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":467,"byte_end":475,"line_start":11,"line_end":11,"column_start":22,"column_end":30,"is_primary":true,"text":[{"text":"    println!(\"{}{}\", a: usize < long_name, long_name);","highlight_start":22,"highlight_end":30}],"label":null,"suggested_replacement":"(a: usize)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: `<` is interpreted as a start of generic arguments for `usize`, not a comparison\n  --> /checkout/src/test/ui/issues/issue-22644.rs:11:31\n   |\nLL |     println!(\"{}{}\", a: usize < long_name, long_name);\n   |                      -------- ^ -------------------- interpreted as generic arguments\n   |                      |        |\n   |                      |        not interpreted as comparison\n   |                      help: try comparing the cast value: `(a: usize)`\n\n"}
[00:51:04] {"message":"`<` is interpreted as a start of generic arguments for `usize`, not a comparison","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":587,"byte_end":588,"line_start":13,"line_end":13,"column_start":31,"column_end":32,"is_primary":false,"text":[{"text":"    println!(\"{}\", a: usize < 4); //~ ERROR `<` is interpreted as a start of generic","highlight_start":31,"highlight_end":32}],"label":"interpreted as generic arguments","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":585,"byte_end":586,"line_start":13,"line_end":13,"column_start":29,"column_end":30,"is_primary":true,"text":[{"text":"    println!(\"{}\", a: usize < 4); //~ ERROR `<` is interpreted as a start of generic","highlight_start":29,"highlight_end":30}],"label":"not interpreted as comparison","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try comparing the cast value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":576,"byte_end":584,"line_start":13,"line_end":13,"column_start":20,"column_end":28,"is_primary":true,"text":[{"text":"    println!(\"{}\", a: usize < 4); //~ ERROR `<` is interpreted as a start of generic","highlight_start":20,"highlight_end":28}],"label":null,"suggested_replacement":"(a: usize)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: `<` is interpreted as a start of generic arguments for `usize`, not a comparison\n  --> /checkout/src/test/ui/issues/issue-22644.rs:13:29\n   |\nLL |     println!(\"{}\", a: usize < 4); //~ ERROR `<` is interpreted as a start of generic\n   |                    -------- ^ - interpreted as generic arguments\n   |                    |        |\n   |                    |        not interpreted as comparison\n   |                    help: try comparing the cast value: `(a: usize)`\n\n"}
[00:51:04] {"message":"`<` is interpreted as a start of generic arguments for `usize`, not a comparison","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":802,"byte_end":803,"line_start":19,"line_end":19,"column_start":20,"column_end":21,"is_primary":false,"text":[{"text":"                   4);","highlight_start":20,"highlight_end":21}],"label":"interpreted as generic arguments","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":730,"byte_end":731,"line_start":18,"line_end":18,"column_start":20,"column_end":21,"is_primary":true,"text":[{"text":"                   < //~ ERROR `<` is interpreted as a start of generic","highlight_start":20,"highlight_end":21}],"label":"not interpreted as comparison","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try comparing the cast value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":662,"byte_end":710,"line_start":15,"line_end":17,"column_start":20,"column_end":25,"is_primary":true,"text":[{"text":"    println!(\"{}\", a","highlight_start":20,"highlight_end":21},{"text":"                   as","highlight_start":1,"highlight_end":22},{"text":"                   usize","highlight_start":1,"highlight_end":25}],"label":null,"suggested_replacement":"(a\n                   as\n                   usize)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: `<` is interpreted as a start of generic arguments for `usize`, not a comparison\n  --> /checkout/src/test/ui/issues/issue-22644.rs:18:20\n   |\nLL |                    < //~ ERROR `<` is interpreted as a start of generic\n   |                    ^ not interpreted as comparison\nLL |                    4);\n   |                    - interpreted as generic arguments\nhelp: try comparing the cast value\n   |\nLL |     println!(\"{}\", (a\nLL |                    as\nLL |                    usize)\n   |\n\n"}
[00:51:04] {"message":"`<` is interpreted as a start of generic arguments for `usize`, not a comparison","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":969,"byte_end":970,"line_start":28,"line_end":28,"column_start":20,"column_end":21,"is_primary":false,"text":[{"text":"                   5);","highlight_start":20,"highlight_end":21}],"label":"interpreted as generic arguments","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":897,"byte_end":898,"line_start":27,"line_end":27,"column_start":20,"column_end":21,"is_primary":true,"text":[{"text":"                   < //~ ERROR `<` is interpreted as a start of generic","highlight_start":20,"highlight_end":21}],"label":"not interpreted as comparison","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try comparing the cast value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":825,"byte_end":877,"line_start":20,"line_end":26,"column_start":20,"column_end":25,"is_primary":true,"text":[{"text":"    println!(\"{}\", a","highlight_start":20,"highlight_end":21},{"text":"","highlight_start":1,"highlight_end":1},{"text":"","highlight_start":1,"highlight_end":1},{"text":"                   as","highlight_start":1,"highlight_end":22},{"text":"","highlight_start":1,"highlight_end":1},{"text":"","highlight_start":1,"highlight_end":1},{"text":"                   usize","highlight_start":1,"highlight_end":25}],"label":null,"suggested_replacement":"(a\n\n\n                   as\n\n\n                   usize)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: `<` is interpreted as a start of generic arguments for `usize`, not a comparison\n  --> /checkout/src/test/ui/issues/issue-22644.rs:27:20\n   |\nLL |                    < //~ ERROR `<` is interpreted as a start of generic\n   |                    ^ not interpreted as comparison\nLL |                    5);\n   |                    - interpreted as generic arguments\nhelp: try comparing the cast value\n   |\nLL |     println!(\"{}\", (a\nLL | \nLL | \nLL |                    as\nLL | \nLL | \n ...\n\n"}
[00:51:04] {"message":"`<` is interpreted as a start of generic arguments for `usize`, not a shift","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":1007,"byte_end":1016,"line_start":30,"line_end":30,"column_start":34,"column_end":43,"is_primary":false,"text":[{"text":"    println!(\"{}\", a as usize << long_name); //~ ERROR `<` is interpreted as a start of generic","highlight_start":34,"highlight_end":43}],"label":"interpreted as generic arguments","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":1004,"byte_end":1006,"line_start":30,"line_end":30,"column_start":31,"column_end":33,"is_primary":true,"text":[{"text":"    println!(\"{}\", a as usize << long_name); //~ ERROR `<` is interpreted as a start of generic","highlight_start":31,"highlight_end":33}],"label":"not interpreted as shift","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try shifting the cast value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":993,"byte_end":1003,"line_start":30,"line_end":30,"column_start":20,"column_end":30,"is_primary":true,"text":[{"text":"    println!(\"{}\", a as usize << long_name); //~ ERROR `<` is interpreted as a start of generic","highlight_start":20,"highlight_end":30}],"label":null,"suggested_replacement":"(a as usize)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: `<` is interpreted as a start of generic arguments for `usize`, not a shift\n  --> /checkout/src/test/ui/issues/issue-22644.rs:30:31\n   |\nLL |     println!(\"{}\", a as usize << long_name); //~ ERROR `<` is interpreted as a start of generic\n   |                    ---------- ^^ --------- interpreted as generic arguments\n   |                    |          |\n   |                    |          not interpreted as shift\n   |                    help: try shifting the cast value: `(a as usize)`\n\n"}
[00:51:04] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:495:22
[00:51:04] {"message":"expected type, found `4`","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-22644.rs","byte_start":1098,"byte_end":1099,"line_start":32,"line_end":32,"column_start":28,"column_end":29,"is_primary":true,"text":[{"text":"    println!(\"{}\", a: &mut 4); //~ ERROR expected type, found `4`","highlight_start":28,"highlight_end":29}],"label":"expecting a type here because of type ascription","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: expected type, found `4`\n  --> /checkout/src/test/ui/issues/issue-22644.rs:32:28\n   |\nLL |     println!(\"{}\", a: &mut 4); //~ ERROR expected type, found `4`\n   |                            ^ expecting a type here because of type ascription\n\n"}
[00:51:04] {"message":"type ascription is experimental (see issue #23416)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n