\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/transmute/main.rs","byte_start":662,"byte_end":671,"line_start":24,"line_end":24,"column_start":17,"column_end":26,"is_primary":true,"text":[{"text":"    let x: u8 = transmute(\"test\"); //~ ERROR cannot transmute between types of different sizes","highlight_start":17,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"source type: `&str` (64 bits)","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"target type: `u8` (8 bits)","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0512]: cannot transmute between types of different sizes, or dependently-sized types\n  --> /checkout/src/test/ui/transmute/main.rs:24:17\n   |\nLL |     let x: u8 = transmute(\"test\"); //~ ERROR cannot transmute between types of different sizes\n   |                 ^^^^^^^^^\n   |\n   = note: source type: `&str` (64 bits)\n   = note: target type: `u8` (8 bits)\n\n"}
[00:52:07] {"message":"cannot transmute between types of different sizes, or dependently-sized types","code":{"code":"E0512","explanation":"\nTransmute with two differently sized types was attempted. Erroneous code\nexample:\n\n