\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/transmute/main.rs","byte_start":434,"byte_end":443,"line_start":16,"line_end":16,"column_start":5,"column_end":14,"is_primary":true,"text":[{"text":"    transmute(x) //~ ERROR cannot transmute between types of different sizes","highlight_start":5,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"source type: `<C as TypeConstructor<'a>>::T` (size can vary because of <C as TypeConstructor>::T)","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"target type: `<C as TypeConstructor<'b>>::T` (size can vary because of <C as TypeConstructor>::T)","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0512]: cannot transmute between types of different sizes, or dependently-sized types\n  --> /checkout/src/test/ui/transmute/main.rs:16:5\n   |\nLL |     transmute(x) //~ ERROR cannot transmute between types of different sizes\n   |     ^^^^^^^^^\n   |\n   = note: source type: `<C as TypeConstructor<'a>>::T` (size can vary because of <C as TypeConstructor>::T)\n   = note: target type: `<C as TypeConstructor<'b>>::T` (size can vary because of <C as TypeConstructor>::T)\n\n"}
[01:02:32] {"message":"cannot transmute between types of different sizes, or dependently-sized types","code":{"code":"E0512","explanation":"\nTransmute with two differently sized types was attempted. Erroneous code\nexample:\n\n