\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/regions/regions-creating-enums.rs","byte_start":925,"byte_end":943,"line_start":33,"line_end":33,"column_start":16,"column_end":34,"is_primary":true,"text":[{"text":"        return &Ast::Num((*f)(x)); //~ ERROR borrowed value does not live long enough","highlight_start":16,"highlight_end":34}],"label":"returns a reference to data owned by the current function","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/regions/regions-creating-enums.rs","byte_start":926,"byte_end":943,"line_start":33,"line_end":33,"column_start":17,"column_end":34,"is_primary":false,"text":[{"text":"        return &Ast::Num((*f)(x)); //~ ERROR borrowed value does not live long enough","highlight_start":17,"highlight_end":34}],"label":"temporary value created here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0515]: cannot return reference to temporary value\n  --> /checkout/src/test/ui/regions/regions-creating-enums.rs:33:16\n   |\nLL |         return &Ast::Num((*f)(x)); //~ ERROR borrowed value does not live long enough\n   |                ^-----------------\n   |                ||\n   |                |temporary value created here\n   |                returns a reference to data owned by the current function\n\n"}
[01:20:28] {"message":"cannot return reference to temporary value","code":{"code":"E0515","explanation":"\nCannot return value that references local variable\n\nLocal variables, function parameters and temporaries are all dropped before the\nend of the function body. So a reference to them cannot be returned.\n\n