\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/stable/book/references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/borrowck-loan-rcvr.rs","byte_start":934,"byte_end":936,"line_start":33,"line_end":33,"column_start":14,"column_end":16,"is_primary":true,"text":[{"text":"    p.blockm(|| { //~ ERROR cannot borrow `p` as mutable","highlight_start":14,"highlight_end":16}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-loan-rcvr.rs","byte_start":925,"byte_end":926,"line_start":33,"line_end":33,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    p.blockm(|| { //~ ERROR cannot borrow `p` as mutable","highlight_start":5,"highlight_end":6}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-loan-rcvr.rs","byte_start":986,"byte_end":987,"line_start":34,"line_end":34,"column_start":9,"column_end":10,"is_primary":false,"text":[{"text":"        p.x = 10;","highlight_start":9,"highlight_end":10}],"label":"second borrow occurs due to use of `p` in closure","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-loan-rcvr.rs","byte_start":927,"byte_end":933,"line_start":33,"line_end":33,"column_start":7,"column_end":13,"is_primary":false,"text":[{"text":"    p.blockm(|| { //~ ERROR cannot borrow `p` as mutable","highlight_start":7,"highlight_end":13}],"label":"immutable borrow later used by call","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `p` as mutable because it is also borrowed as immutable\n  --> /checkout/src/test/ui/borrowck/borrowck-loan-rcvr.rs:33:14\n   |\nLL |     p.blockm(|| { //~ ERROR cannot borrow `p` as mutable\n   |     - ------ ^^ mutable borrow occurs here\n   |     | |\n   |     | immutable borrow later used by call\n   |     immutable borrow occurs here\nLL |         p.x = 10;\n   |         - second borrow occurs due to use of `p` in closure\n\n"}
[01:20:27] {"message":"cannot borrow `p` as immutable because it is also borrowed as mutable","code":{"code":"E0502","explanation":"\nThis error indicates that you are trying to borrow a variable as mutable when it\nhas already been borrowed as immutable.\n\nExample of erroneous code:\n\n