\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":5093,"byte_end":5120,"line_start":81,"line_end":81,"column_start":11,"column_end":38,"is_primary":true,"text":[{"text":"type A9 = extern \"amdgpu-kernel\" fn(); //~ ERROR amdgpu-kernel ABI is experimental and subject to change","highlight_start":11,"highlight_end":38}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_amdgpu_kernel)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: amdgpu-kernel ABI is experimental and subject to change (see issue #51575)\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:81:11\n   |\nLL | type A9 = extern \"amdgpu-kernel\" fn(); //~ ERROR amdgpu-kernel ABI is experimental and subject to change\n   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_amdgpu_kernel)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"intrinsics are subject to change","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n