\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":2364,"byte_end":2398,"line_start":39,"line_end":39,"column_start":5,"column_end":39,"is_primary":true,"text":[{"text":"    extern \"x86-interrupt\" fn dm7() {} //~ ERROR x86-interrupt ABI is experimental","highlight_start":5,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_x86_interrupt)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: x86-interrupt ABI is experimental and subject to change (see issue #40180)\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:39:5\n   |\nLL |     extern \"x86-interrupt\" fn dm7() {} //~ ERROR x86-interrupt ABI is experimental\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_x86_interrupt)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"thiscall is experimental and subject to change","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n