\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":4919,"byte_end":4946,"line_start":79,"line_end":79,"column_start":11,"column_end":38,"is_primary":true,"text":[{"text":"type A7 = extern \"x86-interrupt\" fn(); //~ ERROR x86-interrupt ABI is experimental","highlight_start":11,"highlight_end":38}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_x86_interrupt)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: x86-interrupt ABI is experimental and subject to change (see issue #40180)\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:79:11\n   |\nLL | type A7 = extern \"x86-interrupt\" fn(); //~ ERROR x86-interrupt ABI is experimental\n   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_x86_interrupt)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"thiscall is experimental and subject to change","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n