\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":312,"byte_end":350,"line_start":12,"line_end":12,"column_start":1,"column_end":39,"is_primary":true,"text":[{"text":"extern \"platform-intrinsic\" fn f2() {} //~ ERROR platform intrinsics are experimental","highlight_start":1,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(platform_intrinsics)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: platform intrinsics are experimental and possibly buggy (see issue #27731)\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:12:1\n   |\nLL | extern \"platform-intrinsic\" fn f2() {} //~ ERROR platform intrinsics are experimental\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(platform_intrinsics)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"vectorcall is experimental and subject to change","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n