\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":4735,"byte_end":4765,"line_start":77,"line_end":77,"column_start":11,"column_end":41,"is_primary":true,"text":[{"text":"type A5 = extern \"msp430-interrupt\" fn(); //~ ERROR msp430-interrupt ABI is experimental","highlight_start":11,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_msp430_interrupt)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: msp430-interrupt ABI is experimental and subject to change (see issue #38487)\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:77:11\n   |\nLL | type A5 = extern \"msp430-interrupt\" fn(); //~ ERROR msp430-interrupt ABI is experimental\n   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_msp430_interrupt)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"PTX ABIs are experimental and subject to change (see issue #38788)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n