\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":4560,"byte_end":4584,"line_start":75,"line_end":75,"column_start":11,"column_end":35,"is_primary":true,"text":[{"text":"type A3 = extern \"vectorcall\" fn(); //~ ERROR vectorcall is experimental and subject to change","highlight_start":11,"highlight_end":35}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_vectorcall)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: vectorcall is experimental and subject to change\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:75:11\n   |\nLL | type A3 = extern \"vectorcall\" fn(); //~ ERROR vectorcall is experimental and subject to change\n   |           ^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_vectorcall)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"rust-call ABI is subject to change (see issue #29625)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n