\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":4386,"byte_end":4414,"line_start":73,"line_end":73,"column_start":11,"column_end":39,"is_primary":true,"text":[{"text":"type A1 = extern \"rust-intrinsic\" fn(); //~ ERROR intrinsics are subject to change","highlight_start":11,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(intrinsics)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: intrinsics are subject to change\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:73:11\n   |\nLL | type A1 = extern \"rust-intrinsic\" fn(); //~ ERROR intrinsics are subject to change\n   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(intrinsics)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"platform intrinsics are experimental and possibly buggy (see issue #27731)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n