\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":4824,"byte_end":4849,"line_start":78,"line_end":78,"column_start":11,"column_end":36,"is_primary":true,"text":[{"text":"type A6 = extern \"ptx-kernel\" fn (); //~ ERROR PTX ABIs are experimental and subject to change","highlight_start":11,"highlight_end":36}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_ptx)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: PTX ABIs are experimental and subject to change (see issue #38788)\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:78:11\n   |\nLL | type A6 = extern \"ptx-kernel\" fn (); //~ ERROR PTX ABIs are experimental and subject to change\n   |           ^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_ptx)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"x86-interrupt ABI is experimental and subject to change (see issue #40180)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n