\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":5732,"byte_end":5752,"line_start":91,"line_end":91,"column_start":1,"column_end":21,"is_primary":true,"text":[{"text":"extern \"thiscall\" {} //~ ERROR thiscall is experimental and subject to change","highlight_start":1,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_thiscall)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: thiscall is experimental and subject to change\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:91:1\n   |\nLL | extern \"thiscall\" {} //~ ERROR thiscall is experimental and subject to change\n   | ^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_thiscall)] to the crate attributes to enable\n\n"}
[01:22:06] {"message":"amdgpu-kernel ABI is experimental and subject to change (see issue #51575)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n