\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":234,"byte_end":268,"line_start":11,"line_end":11,"column_start":1,"column_end":35,"is_primary":true,"text":[{"text":"extern \"rust-intrinsic\" fn f1() {} //~ ERROR intrinsics are subject to change","highlight_start":1,"highlight_end":35}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(intrinsics)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: intrinsics are subject to change\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:11:1\n   |\nLL | extern \"rust-intrinsic\" fn f1() {} //~ ERROR intrinsics are subject to change\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(intrinsics)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"platform intrinsics are experimental and possibly buggy (see issue #27731)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n