\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":5002,"byte_end":5024,"line_start":80,"line_end":80,"column_start":11,"column_end":33,"is_primary":true,"text":[{"text":"type A8 = extern \"thiscall\" fn(); //~ ERROR thiscall is experimental and subject to change","highlight_start":11,"highlight_end":33}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_thiscall)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: thiscall is experimental and subject to change\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:80:11\n   |\nLL | type A8 = extern \"thiscall\" fn(); //~ ERROR thiscall is experimental and subject to change\n   |           ^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_thiscall)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"amdgpu-kernel ABI is experimental and subject to change (see issue #51575)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n