\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":488,"byte_end":517,"line_start":14,"line_end":14,"column_start":1,"column_end":30,"is_primary":true,"text":[{"text":"extern \"rust-call\" fn f4() {} //~ ERROR rust-call ABI is subject to change","highlight_start":1,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(unboxed_closures)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: rust-call ABI is subject to change (see issue #29625)\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:14:1\n   |\nLL | extern \"rust-call\" fn f4() {} //~ ERROR rust-call ABI is subject to change\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(unboxed_closures)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"msp430-interrupt ABI is experimental and subject to change (see issue #38487)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n