\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":3306,"byte_end":3334,"line_start":55,"line_end":55,"column_start":5,"column_end":33,"is_primary":true,"text":[{"text":"    extern \"thiscall\" fn m8() {} //~ ERROR thiscall is experimental and subject to change","highlight_start":5,"highlight_end":33}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_thiscall)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: thiscall is experimental and subject to change\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:55:5\n   |\nLL |     extern \"thiscall\" fn m8() {} //~ ERROR thiscall is experimental and subject to change\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_thiscall)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"amdgpu-kernel ABI is experimental and subject to change (see issue #51575)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n