\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-abi.rs","byte_start":814,"byte_end":842,"line_start":18,"line_end":18,"column_start":1,"column_end":29,"is_primary":true,"text":[{"text":"extern \"thiscall\" fn f8() {} //~ ERROR thiscall is experimental and subject to change","highlight_start":1,"highlight_end":29}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(abi_thiscall)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: thiscall is experimental and subject to change\n  --> /checkout/src/test/ui/feature-gates/feature-gate-abi.rs:18:1\n   |\nLL | extern \"thiscall\" fn f8() {} //~ ERROR thiscall is experimental and subject to change\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(abi_thiscall)] to the crate attributes to enable\n\n"}
[01:22:05] {"message":"amdgpu-kernel ABI is experimental and subject to change (see issue #51575)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n