\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/utf8_idents.rs","byte_start":234,"byte_end":236,"line_start":10,"line_end":10,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    δ: usize //~ ERROR non-ascii idents are not fully supported","highlight_start":5,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(non_ascii_idents)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: non-ascii idents are not fully supported. (see issue #55467)\n  --> /checkout/src/test/ui/utf8_idents.rs:10:5\n   |\nLL |     δ: usize //~ ERROR non-ascii idents are nonux-gnu/stage2/lib/rustlib
46352 ./src/llvm-emscripten/test/MC
44784 ./src/test/ui
44224 ./obj/build/x86_64-unknown-linux-gnu/stage1-rustc
42808 ./obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib
