\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-duplicate-bounds.rs","byte_start":251,"byte_end":260,"line_start":18,"line_end":18,"column_start":39,"column_end":48,"is_primary":true,"text":[{"text":"impl Foo for dyn Send + Sync + Send + SyncAlias {}","highlight_start":39,"highlight_end":48}],"label":"non-auto additional trait","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0225]: only auto traits can be used as additional traits in a trait object\n  --> /checkout/src/test/ui/lint/lint-duplicate-bounds.rs:18:39\n   |\nLL | impl Foo for dyn Send + Sync + Send + SyncAlias {}\n   |                                       ^^^^^^^^^ non-auto additional trait\n\n"}
[01:18:42] {"message":"For more information about this error, try `rustc --explain E0225`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0225`.\n"}
[01:18:42] 
[01:18:42] ------------------------------------------
[01:18:42] 
[01:18:42] 
[01:18:42] thread '[ui] ui/lint/lint-duplicate-bounds.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3369:9
[01:18:42] 
[01:18:42] ---- [ui] ui/type/type-alias-bounds.rs stdout ----
[01:18:42] diff of stderr:
[01:18:42] 
[01:18:42] - warning: duplicate auto trait `::marker[0]::Send[0]` found in type parameter bounds
[01:18:42] -    |
[01:18:42] -    |
[01:18:42] - LL | type SVec<T: Send + Send> = Vec<T>;
[01:18:42] -    |              ^^^^   ^^^^ subsequent use of auto trait
[01:18:42] -    |              first use of auto trait
[01:18:42] -    |
[01:18:42] -    |
[01:18:42] -    = note: #[warn(duplicate_auto_traits_in_bounds)] on by default
[01:18:42] - 
[01:18:42] 11 warning: bounds on generic parameters are not enforced in type aliases
[01:18:42] 13    |
[01:18:42] 
[01:18:42] 
[01:18:42] The actual stderr differed from the expected stderr.
[01:18:42] The actual stderr differed from the expected stderr.
[01:18:42] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type/type-alias-bounds/type-alias-bounds.stderr
[01:18:42] To update references, rerun the tests and pass the `--bless` flag
[01:18:42] To only update this specific test, also pass `--test-args type/type-alias-bounds.rs`
[01:18:42] error: 1 errors occurred comparing output.
[01:18:42] status: exit code: 0
[01:18:42] status: exit code: 0
[01:18:42] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/type/type-alias-bounds.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type/type-alias-bounds/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type/type-alias-bounds/auxiliary" "-A" "unused"
[01:18:42] ------------------------------------------
[01:18:42] 
[01:18:42] ------------------------------------------
[01:18:42] stderr:
[01:18:42] stderr:
[01:18:42] ------------------------------------------
[01:18:42] {"message":"bounds on generic parameters are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":145,"byte_end":149,"line_start":8,"line_end":8,"column_start":14,"column_end":18,"is_primary":true,"text":[{"text":"type SVec<T: Send + Send> = Vec<T>;","highlight_start":14,"highlight_end":18}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":152,"byte_end":156,"line_start":8,"line_end":8,"column_start":21,"column_end":25,"is_primary":true,"text":[{"text":"type SVec<T: Send + Send> = Vec<T>;","highlight_start":21,"highlight_end":25}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"#[warn(type_alias_bounds)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"the bound will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: bounds on generic parameters are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:8:14\n   |\nLL | type SVec<T: Send + Send> = Vec<T>;\n   |              ^^^^   ^^^^\n   |\n   = note: #[warn(type_alias_bounds)] on by default\n   = help: the bound will not be checked when the type alias is used, and should be removed\n\n"}
[01:18:42] {"message":"where clauses are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":280,"byte_end":287,"line_start":10,"line_end":10,"column_start":21,"column_end":28,"is_primary":true,"text":[{"text":"type S2Vec<T> where T: Send = Vec<T>;","highlight_start":21,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the clause will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: where clauses are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:10:21\n   |\nLL | type S2Vec<T> where T: Send = Vec<T>;\n   |                     ^^^^^^^\n   |\n   = help: the clause will not be checked when the type alias is used, and should be removed\n\n"}
[01:18:42] {"message":"bounds on generic parameters are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":393,"byte_end":395,"line_start":12,"line_end":12,"column_start":19,"column_end":21,"is_primary":true,"text":[{"text":"type VVec<'b, 'a: 'b + 'b> = (&'b u32, Vec<&'a i32>);","highlight_start":19,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":398,"byte_end":400,"line_start":12,"line_end":12,"column_start":24,"column_end":26,"is_primary":true,"text":[{"text":"type VVec<'b, 'a: 'b + 'b> = (&'b u32, Vec<&'a i32>);","highlight_start":24,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the bound will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: bounds on generic parameters are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:12:19\n   |\nLL | type VVec<'b, 'a: 'b + 'b> = (&'b u32, Vec<&'a i32>);\n   |                   ^^   ^^\n   |\n   = help: the bound will not be checked when the type alias is used, and should be removed\n\n"}
[01:18:42] {"message":"bounds on generic parameters are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":538,"byte_end":540,"line_start":14,"line_end":14,"column_start":18,"column_end":20,"is_primary":true,"text":[{"text":"type WVec<'b, T: 'b + 'b> = (&'b u32, Vec<T>);","highlight_start":18,"highlight_end":20}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":543,"byte_end":545,"line_start":14,"line_end":14,"column_start":23,"column_end":25,"is_primary":true,"text":[{"text":"type WVec<'b, T: 'b + 'b> = (&'b u32, Vec<T>);","highlight_start":23,"highlight_end":25}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the bound will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: bounds on generic parameters are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:14:18\n   |\nLL | type WVec<'b, T: 'b + 'b> = (&'b u32, Vec<T>);\n   |                  ^^   ^^\n   |\n   = help: the bound will not be checked when the type alias is used, and should be removed\n\n"}
[01:18:42] {"message":"where clauses are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":684,"byte_end":689,"line_start":16,"line_end":16,"column_start":25,"column_end":30,"is_primary":true,"text":[{"text":"type W2Vec<'b, T> where T: 'b, T: 'b = (&'b u32, Vec<T>);","highlight_start":25,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":691,"byte_end":696,"line_start":16,"line_end":16,"column_start":32,"column_end":37,"is_primary":true,"text":[{"text":"type W2Vec<'b, T> where T: 'b, T: 'b = (&'b u32, Vec<T>);","highlight_start":32,"highlight_end":37}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the clause will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: where clauses are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:16:25\n   |\nLL | type W2Vec<'b, T> where T: 'b, T: 'b = (&'b u32, Vec<T>);\n   |                         ^^^^^  ^^^^^\n   |\n   = help: the clause will not be checked when the type alias is used, and should be removed\n\n"}
[01:18:42] {"message":"bounds on generic parameters are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":1886,"byte_end":1891,"line_start":47,"line_end":47,"column_start":12,"column_end":17,"is_primary":true,"text":[{"text":"type T1<U: Bound> = U::Assoc; //~ WARN not enforced in type aliases","highlight_start":12,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the bound will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"use fully disambiguated paths (i.e., `<T as Trait>::Assoc`) to refer to associated types in type aliases","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":1895,"byte_end":1903,"line_start":47,"line_end":47,"column_start":21,"column_end":29,"is_primary":true,"text":[{"text":"type T1<U: Bound> = U::Assoc; //~ WARN not enforced in type aliases","highlight_start":21,"highlight_end":29}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"warning: bounds on generic parameters are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:47:12\n   |\nLL | type T1<U: Bound> = U::Assoc; //~ WARN not enforced in type aliases\n   |            ^^^^^\n   |\n   = help: the bound will not be checked when the type alias is used, and should be removed\nhelp: use fully disambiguated paths (i.e., `<T as Trait>::Assoc`) to refer to associated types in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:47:21\n   |\nLL | type T1<U: Bound> = U::Assoc; //~ WARN not enforced in type aliases\n   |                     ^^^^^^^^\n\n"}
[01:18:42] {"message":"where clauses are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":1960,"byte_end":1968,"line_start":48,"line_end":48,"column_start":18,"column_end":26,"is_primary":true,"text":[{"text":"type T2<U> where U: Bound = U::Assoc;  //~ WARN not enforced in type aliases","highlight_start":18,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the clause will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null},{"message":"use fully disambiguated paths (i.e., `<T as Trait>::Assoc`) to refer to associated types in type aliases","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":1971,"byte_end":1979,"line_start":48,"line_end":48,"column_start":29,"column_end":37,"is_primary":true,"text":[{"text":"type T2<U> where U: Bound = U::Assoc;  //~ WARN not enforced in type aliases","highlight_start":29,"highlight_end":37}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"warning: where clauses are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:48:18\n   |\nLL | type T2<U> where U: Bound = U::Assoc;  //~ WARN not enforced in type aliases\n   |                  ^^^^^^^^\n   |\n   = help: the clause will not be checked when the type alias is used, and should be removed\nhelp: use fully disambiguated paths (i.e., `<T as Trait>::Assoc`) to refer to associated types in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:48:29\n   |\nLL | type T2<U> where U: Bound = U::Assoc;  //~ WARN not enforced in type aliases\n   |                             ^^^^^^^^\n\n"}
[01:18:42] {"message":"bounds on generic parameters are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":2200,"byte_end":2205,"line_start":56,"line_end":56,"column_start":12,"column_end":17,"is_primary":true,"text":[{"text":"type T5<U: Bound> = <U as Bound>::Assoc;  //~ WARN not enforced in type aliases","highlight_start":12,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the bound will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: bounds on generic parameters are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:56:12\n   |\nLL | type T5<U: Bound> = <U as Bound>::Assoc;  //~ WARN not enforced in type aliases\n   |            ^^^^^\n   |\n   = help: the bound will not be checked when the type alias is used, and should be removed\n\n"}
[01:18:42] {"message":"bounds on generic parameters are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/type/type-alias-bounds.rs","byte_start":2280,"byte_end":2285,"line_start":57,"line_end":57,"column_start":12,"column_end":17,"is_primary":true,"text":[{"text":"type T6<U: Bound> = ::std::vec::Vec<U>;  //~ WARN not enforced in type aliases","highlight_start":12,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the bound will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: bounds on generic parameters are not enforced in type aliases\n  --> /checkout/src/test/ui/type/type-alias-bounds.rs:57:12\n   |\nLL | type T6<U: Bound> = ::std::vec::Vec<U>;  //~ WARN not enforced in type aliases\n   |            ^^^^^\n   |\n   = help: the bound will not be checked when the type alias is used, and should be removed\n\n"}
[01:18:42] ------------------------------------------
[01:18:42] 
[01:18:42] thread '[ui] ui/type/type-alias-bounds.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3369:9
[01:18:42] 
---
[01:18:42] 
[01:18:42] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:496:22
[01:18:42] 
[01:18:42] 
[01:18:42] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[01:18:42] 
[01:18:42] 
[01:18:42] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[01:18:42] Build completed unsuccessfully in 0:04:31
[01:18:42] Build completed unsuccessfully in 0:04:31
[01:18:42] make: *** [check] Error 1
[01:18:42] Makefile:48: recipe for target 'check' failed
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:03040a12
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Sun Mar 24 22:19:10 UTC 2019
