\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/check-static-values-constraints.rs","byte_start":2958,"byte_end":2965,"line_start":97,"line_end":97,"column_start":9,"column_end":16,"is_primary":true,"text":[{"text":"    box MyOwned, //~ ERROR allocations are not allowed in statics","highlight_start":9,"highlight_end":16}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0019]: static contains unimplemented expression type\n  --> /checkout/src/test/ui/check-static-values-constraints.rs:97:9\n   |\nLL |     box MyOwned, //~ ERROR allocations are not allowed in statics\n   |         ^^^^^^^\n\n"}
[01:31:44] {"message":"allocations are not allowed in statics","code":{"code":"E0010","explanation":"\nThe value of statics and constants must be known at compile time, and they live\nfor the entire lifetime of a program. Creating a boxed value allocates memory on\nthe heap at runtime, and therefore cannot be done at compile time. Erroneous\ncode example:\n\n