\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/promote_const_let.rs","byte_start":68,"byte_end":70,"line_start":4,"line_end":4,"column_start":9,"column_end":11,"is_primary":true,"text":[{"text":"        &y //~ ERROR does not live long enough","highlight_start":9,"highlight_end":11}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/consts/promote_const_let.rs","byte_start":111,"byte_end":112,"line_start":5,"line_end":5,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    };","highlight_start":5,"highlight_end":6}],"label":"`y` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/consts/promote_const_let.rs","byte_start":23,"byte_end":35,"line_start":2,"line_end":2,"column_start":12,"column_end":24,"is_primary":false,"text":[{"text":"    let x: &'static u32 = {","highlight_start":12,"highlight_end":24}],"label":"type annotation requires that `y` is borrowed for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `y` does not live long enough\n  --> /checkout/src/test/ui/consts/promote_const_let.rs:4:9\n   |\nLL |     let x: &'static u32 = {\n   |            ------------ type annotation requires that `y` is borrowed for `'static`\nLL |         let y = 42;\nLL |         &y //~ ERROR does not live long enough\n   |         ^^ borrowed value does not live long enough\nLL |     };\n   |     - `y` dropped here while still borrowed\n\n"}
[01:31:44] {"message":"temporary value dropped while borrowed","code":{"code":"E0716","explanation":"\nThis error indicates that a temporary value is being dropped\nwhile a borrow is still in active use.\n\nErroneous code example:\n\n