\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/check-static-values-constraints.rs","byte_start":2954,"byte_end":2965,"line_start":97,"line_end":97,"column_start":5,"column_end":16,"is_primary":true,"text":[{"text":"    box MyOwned, //~ ERROR allocations are not allowed in statics","highlight_start":5,"highlight_end":16}],"label":"allocation not allowed in statics","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0010]: allocations are not allowed in statics\n  --> /checkout/src/test/ui/check-static-values-constraints.rs:97:5\n   |\nLL |     box MyOwned, //~ ERROR allocations are not allowed in statics\n   |     ^^^^^^^^^^^ allocation not allowed in statics\n\n"}
[01:31:44] {"message":"static contains unimplemented expression type","code":{"code":"E0019","explanation":"\nA function call isn't allowed in the const's initialization expression\nbecause the expression's value must be known at compile-time. Erroneous code\nexample:\n\n