\n\nCertain Rust types must be cast before passing them to a variadic function,\nbecause of arcane ABI rules dictated by the C standard. To fix the error,\ncast the value to the type specified by the error message (which you may need\nto import from `std::os::raw`).\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/variadic/variadic-ffi-3.rs","byte_start":810,"byte_end":814,"line_start":24,"line_end":24,"column_start":19,"column_end":23,"is_primary":true,"text":[{"text":"        foo(1, 2, 3f32); //~ ERROR can't pass `f32` to variadic function","highlight_start":19,"highlight_end":23}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"cast the value to `c_double`","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/variadic/variadic-ffi-3.rs","byte_start":810,"byte_end":814,"line_start":24,"line_end":24,"column_start":19,"column_end":23,"is_primary":true,"text":[{"text":"        foo(1, 2, 3f32); //~ ERROR can't pass `f32` to variadic function","highlight_start":19,"highlight_end":23}],"label":null,"suggested_replacement":"3f32 as c_double","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0617]: can't pass `f32` to variadic function\n  --> /checkout/src/test/ui/variadic/variadic-ffi-3.rs:24:19\n   |\nLL |         foo(1, 2, 3f32); //~ ERROR can't pass `f32` to variadic function\n   |                   ^^^^ help: cast the value to `c_double`: `3f32 as c_double`\n\n"}
[01:02:39] {"message":"can't pass `bool` to variadic function","code":{"code":"E0617","explanation":"\nAttempted to pass an invalid type of variable into a variadic function.\n\nErroneous code example:\n\n