\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/variadic/variadic-ffi-3.rs","byte_start":13,"byte_end":42,"line_start":2,"line_end":2,"column_start":5,"column_end":34,"is_primary":false,"text":[{"text":"    fn foo(f: isize, x: u8, ...);","highlight_start":5,"highlight_end":34}],"label":"defined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/variadic/variadic-ffi-3.rs","byte_start":261,"byte_end":267,"line_start":12,"line_end":12,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"        foo(1); //~ ERROR: this function takes at least 2 parameters but 1 parameter was supplied","highlight_start":9,"highlight_end":15}],"label":"expected at least 2 parameters","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0060]: this function takes at least 2 parameters but 1 parameter was supplied\n  --> /checkout/src/test/ui/variadic/variadic-ffi-3.rs:12:9\n   |\nLL |     fn foo(f: isize, x: u8, ...);\n   |     ----------------------------- defined here\n...\nLL |         foo(1); //~ ERROR: this function takes at least 2 parameters but 1 parameter was supplied\n   |         ^^^^^^ expected at least 2 parameters\n\n"}
[01:02:39] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n