\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/variadic/variadic-ffi-3.rs","byte_start":13,"byte_end":42,"line_start":2,"line_end":2,"column_start":5,"column_end":34,"is_primary":false,"text":[{"text":"    fn foo(f: isize, x: u8, ...);","highlight_start":5,"highlight_end":34}],"label":"defined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/variadic/variadic-ffi-3.rs","byte_start":162,"byte_end":167,"line_start":11,"line_end":11,"column_start":9,"column_end":14,"is_primary":true,"text":[{"text":"        foo(); //~ ERROR: this function takes at least 2 parameters but 0 parameters were supplied","highlight_start":9,"highlight_end":14}],"label":"expected at least 2 parameters","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0060]: this function takes at least 2 parameters but 0 parameters were supplied\n  --> /checkout/src/test/ui/variadic/variadic-ffi-3.rs:11:9\n   |\nLL |     fn foo(f: isize, x: u8, ...);\n   |     ----------------------------- defined here\n...\nLL |         foo(); //~ ERROR: this function takes at least 2 parameters but 0 parameters were supplied\n   |         ^^^^^ expected at least 2 parameters\n\n"}
[01:02:39] {"message":"this function takes at least 2 parameters but 1 parameter was supplied","code":{"code":"E0060","explanation":"\nExternal C functions are allowed to be variadic. However, a variadic function\ntakes a minimum number of arguments. For example, consider C's variadic `printf`\nfunction:\n\n