\n\nCertain Rust types must be cast before passing them to a variadic function,\nbecause of arcane ABI rules dictated by the C standard. To fix the error,\ncast the value to the type specified by the error message (which you may need\nto import from `std::os::raw`).\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/variadic/variadic-ffi-3.rs","byte_start":1028,"byte_end":1031,"line_start":27,"line_end":27,"column_start":19,"column_end":22,"is_primary":true,"text":[{"text":"        foo(1, 2, 1u8); //~ ERROR can't pass `u8` to variadic function","highlight_start":19,"highlight_end":22}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"cast the value to `c_uint`","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/variadic/variadic-ffi-3.rs","byte_start":1028,"byte_end":1031,"line_start":27,"line_end":27,"column_start":19,"column_end":22,"is_primary":true,"text":[{"text":"        foo(1, 2, 1u8); //~ ERROR can't pass `u8` to variadic function","highlight_start":19,"highlight_end":22}],"label":null,"suggested_replacement":"1u8 as c_uint","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0617]: can't pass `u8` to variadic function\n  --> /checkout/src/test/ui/variadic/variadic-ffi-3.rs:27:19\n   |\nLL |         foo(1, 2, 1u8); //~ ERROR can't pass `u8` to variadic function\n   |                   ^^^ help: cast the value to `c_uint`: `1u8 as c_uint`\n\n"}
[01:02:39] {"message":"can't pass `i16` to variadic function","code":{"code":"E0617","explanation":"\nAttempted to pass an invalid type of variable into a variadic function.\n\nErroneous code example:\n\n