\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/methods/method-deref-to-same-trait-object-with-separate-params.rs","byte_start":3905,"byte_end":3906,"line_start":137,"line_end":137,"column_start":24,"column_end":25,"is_primary":true,"text":[{"text":"    let _seetype: () = z; //~ ERROR mismatched types","highlight_start":24,"highlight_end":25}],"label":"expected (), found u8","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `()`\n   found type `u8`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/methods/method-deref-to-same-trait-object-with-separate-params.rs:137:24\n   |\nLL |     let _seetype: () = z; //~ ERROR mismatched types\n   |                        ^ expected (), found u8\n   |\n   = note: expected type `()`\n              found type `u8`\n\n"}
[01:08:54] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n