\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/methods/method-deref-to-same-trait-object-with-separate-params.rs","byte_start":2641,"byte_end":2642,"line_start":102,"line_end":102,"column_start":24,"column_end":25,"is_primary":true,"text":[{"text":"    let _seetype: () = z; //~ ERROR mismatched types","highlight_start":24,"highlight_end":25}],"label":"expected (), found u64","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `()`\n   found type `u64`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/methods/method-deref-to-same-trait-object-with-separate-params.rs:102:24\n   |\nLL |     let _seetype: () = z; //~ ERROR mismatched types\n   |                        ^ expected (), found u64\n   |\n   = note: expected type `()`\n              found type `u64`\n\n"}
[01:08:54] {"message":"multiple applicable items in scope","code":{"code":"E0034","explanation":"\nThe compiler doesn't know what method to call because more than one method\nhas the same prototype. Erroneous code example:\n\n