\n\nRemember this solution is unsafe! You will have to ensure that accesses to the\ncell are synchronized.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/unsafe/ranged_ints3_const.rs","byte_start":272,"byte_end":276,"line_start":12,"line_end":12,"column_start":13,"column_end":17,"is_primary":true,"text":[{"text":"    let y = &x.0; //~ ERROR cannot borrow a constant which may contain interior mutability","highlight_start":13,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0492]: cannot borrow a constant which may contain interior mutability, create a static instead\n  --> /checkout/src/test/ui/unsafe/ranged_ints3_const.rs:12:13\n   |\nLL |     let y = &x.0; //~ ERROR cannot borrow a constant which may contain interior mutability\n   |             ^^^^\n\n"}
[01:00:53] {"message":"cannot borrow a constant which may contain interior mutability, create a static instead","code":{"code":"E0492","explanation":"\nA borrow of a constant containing interior mutability was attempted. Erroneous\ncode example:\n\n