\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-rustc-attrs-1.rs","byte_start":106,"byte_end":123,"line_start":5,"line_end":5,"column_start":1,"column_end":18,"is_primary":true,"text":[{"text":"#[rustc_variance] //~ ERROR the `#[rustc_variance]` attribute is just used for rustc unit tests and will never be stable","highlight_start":1,"highlight_end":18}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(rustc_attrs)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: the `#[rustc_variance]` attribute is just used for rustc unit tests and will never be stable (see issue #29642)\n  --> /checkout/src/test/ui/feature-gates/feature-gate-rustc-attrs-1.rs:5:1\n   |\nLL | #[rustc_variance] //~ ERROR the `#[rustc_variance]` attribute is just used for rustc unit tests and will never be stable\n   | ^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(rustc_attrs)] to the crate attributes to enable\n\n"}
[01:00:53] {"message":"the `#[rustc_error]` attribute is just used for rustc unit tests and will never be stable (see issue #29642)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n