\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":734,"byte_end":755,"line_start":29,"line_end":29,"column_start":5,"column_end":26,"is_primary":true,"text":[{"text":"use issue_52891::b::inner; //~ ERROR `inner` is defined multiple times","highlight_start":5,"highlight_end":26}],"label":"`inner` reimported here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":707,"byte_end":728,"line_start":28,"line_end":28,"column_start":5,"column_end":26,"is_primary":false,"text":[{"text":"use issue_52891::a::inner;","highlight_start":5,"highlight_end":26}],"label":"previous import of the module `inner` here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`inner` must be defined only once in the type namespace of this module","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"you can use `as` to change the binding name of the import","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":734,"byte_end":755,"line_start":29,"line_end":29,"column_start":5,"column_end":26,"is_primary":true,"text":[{"text":"use issue_52891::b::inner; //~ ERROR `inner` is defined multiple times","highlight_start":5,"highlight_end":26}],"label":null,"suggested_replacement":"issue_52891::b::inner as other_inner","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0252]: the name `inner` is defined multiple times\n  --> /checkout/src/test/ui/issues/issue-52891.rs:29:5\n   |\nLL | use issue_52891::a::inner;\n   |     --------------------- previous import of the module `inner` here\nLL | use issue_52891::b::inner; //~ ERROR `inner` is defined multiple times\n   |     ^^^^^^^^^^^^^^^^^^^^^ `inner` reimported here\n   |\n   = note: `inner` must be defined only once in the type namespace of this module\nhelp: you can use `as` to change the binding name of the import\n   |\nLL | use issue_52891::b::inner as other_inner; //~ ERROR `inner` is defined multiple times\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[01:01:49] {"message":"the name `issue_52891` is defined multiple times","code":{"code":"E0254","explanation":"\nAttempt was made to import an item whereas an extern crate with this name has\nalready been imported.\n\nErroneous code example:\n\n