\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":658,"byte_end":659,"line_start":26,"line_end":26,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    a}; //~ ERROR `a` is defined multiple times","highlight_start":5,"highlight_end":6}],"label":"`a` reimported here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":192,"byte_end":206,"line_start":11,"line_end":11,"column_start":5,"column_end":19,"is_primary":false,"text":[{"text":"use issue_52891::a;","highlight_start":5,"highlight_end":19}],"label":"previous import of the module `a` here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`a` must be defined only once in the type namespace of this module","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"remove unnecessary import","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":652,"byte_end":659,"line_start":25,"line_end":26,"column_start":6,"column_end":6,"is_primary":true,"text":[{"text":"    m,","highlight_start":6,"highlight_end":7},{"text":"    a}; //~ ERROR `a` is defined multiple times","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":"","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0252]: the name `a` is defined multiple times\n  --> /checkout/src/test/ui/issues/issue-52891.rs:26:5\n   |\nLL |   use issue_52891::a;\n   |       -------------- previous import of the module `a` here\n...\nLL |       m,\n   |  ______-\nLL | |     a}; //~ ERROR `a` is defined multiple times\n   | |     ^\n   | |     |\n   | |_____`a` reimported here\n   |       help: remove unnecessary import\n   |\n   = note: `a` must be defined only once in the type namespace of this module\n\n"}
[01:01:49] {"message":"the name `inner` is defined multiple times","code":{"code":"E0252","explanation":"\nTwo items of the same name cannot be imported without rebinding one of the\nitems under a new local name.\n\nErroneous code example:\n\n