\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":429,"byte_end":430,"line_start":16,"line_end":16,"column_start":25,"column_end":26,"is_primary":true,"text":[{"text":"use issue_52891::{f, g, a}; //~ ERROR `a` is defined multiple times","highlight_start":25,"highlight_end":26}],"label":"`a` reimported here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":192,"byte_end":206,"line_start":11,"line_end":11,"column_start":5,"column_end":19,"is_primary":false,"text":[{"text":"use issue_52891::a;","highlight_start":5,"highlight_end":19}],"label":"previous import of the module `a` here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`a` must be defined only once in the type namespace of this module","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"remove unnecessary import","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-52891.rs","byte_start":427,"byte_end":430,"line_start":16,"line_end":16,"column_start":23,"column_end":26,"is_primary":true,"text":[{"text":"use issue_52891::{f, g, a}; //~ ERROR `a` is defined multiple times","highlight_start":23,"highlight_end":26}],"label":null,"suggested_replacement":"","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0252]: the name `a` is defined multiple times\n  --> /checkout/src/test/ui/issues/issue-52891.rs:16:25\n   |\nLL | use issue_52891::a;\n   |     -------------- previous import of the module `a` here\n...\nLL | use issue_52891::{f, g, a}; //~ ERROR `a` is defined multiple times\n   |                       --^\n   |                       | |\n   |                       | `a` reimported here\n   |                       help: remove unnecessary import\n   |\n   = note: `a` must be defined only once in the type namespace of this module\n\n"}
[01:01:49] {"message":"the name `a` is defined multiple times","code":{"code":"E0252","explanation":"\nTwo items of the same name cannot be imported without rebinding one of the\nitems under a new local name.\n\nErroneous code example:\n\n