\n\nYou can find more information about borrowing in the rust-book:\nhttp://doc.rust-lang.org/stable/book/references-and-borrowing.html\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/augmented-assignments.rs","byte_start":182,"byte_end":183,"line_start":13,"line_end":13,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    x   //~ error: use of moved value: `x`","highlight_start":5,"highlight_end":6}],"label":"borrow of `x` occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/augmented-assignments.rs","byte_start":268,"byte_end":269,"line_start":16,"line_end":16,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    x;  //~ value moved here","highlight_start":5,"highlight_end":6}],"label":"move out of `x` occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/augmented-assignments.rs","byte_start":182,"byte_end":269,"line_start":13,"line_end":16,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    x   //~ error: use of moved value: `x`","highlight_start":5,"highlight_end":43},{"text":"    //~^ value used here after move","highlight_start":1,"highlight_end":36},{"text":"    +=","highlight_start":1,"highlight_end":7},{"text":"    x;  //~ value moved here","highlight_start":1,"highlight_end":6}],"label":"borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0505]: cannot move out of `x` because it is borrowed\n  --> /checkout/src/test/ui/augmented-assignments.rs:16:5\n   |\nLL |       x   //~ error: use of moved value: `x`\n   |       -\n   |       |\n   |  _____borrow of `x` occurs here\n   | |\nLL | |     //~^ value used here after move\nLL | |     +=\nLL | |     x;  //~ value moved here\n   | |     ^\n   | |     |\n   | |_____move out of `x` occurs here\n   |       borrow later used here\n\n"}
[01:24:59] {"message":"cannot borrow `y` as mutable, as it is not declared as mutable","code":{"code":"E0596","explanation":"\nThis error occurs because you tried to mutably borrow a non-mutable variable.\n\nExample of erroneous code:\n\n