plain
travis_time:end:01a171eb:start=1548906169294153839,finish=1548906171569075314,duration=2274921475
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
Setting environment variables from .travis.yml
$ export IMAGE=x86_64-gnu-llvm-6.0
---
[01:06:23] .................................................................................................... 2700/5355
[01:06:28] .................................................................................................... 2800/5355
[01:06:32] .................................................................................................... 2900/5355
[01:06:35] .................................................................................................... 3000/5355
[01:06:39] .................................................................................F.................. 3100/5355
[01:06:43] ..................................................F................................................. 3200/5355
[01:06:51] .................................................................................ii...i..ii......... 3400/5355
[01:06:55] .................................................................................................... 3500/5355
[01:06:59] .................................................................................................... 3600/5355
[01:07:03] .............................................................................ii..................... 3700/5355
---
[01:08:10] ---- [ui] ui/lint/lint-group-nonstandard-style.rs stdout ----
[01:08:10] diff of stderr:
[01:08:10] 
[01:08:10] 11    |                 ^^^^^^^^^^^^^^^^^
[01:08:10] 12    = note: #[warn(non_camel_case_types)] implied by #[warn(nonstandard_style)]
[01:08:10] 13 
[01:08:10] - error: static variable `bad` should have an upper case name
[01:08:10] -    |
[01:08:10] -    |
[01:08:10] - LL |         static bad: isize = 1; //~ ERROR should have an upper
[01:08:10] -    |                ^^^ help: convert the identifier to upper case: `BAD`
[01:08:10] - note: lint level defined here
[01:08:10] -   --> $DIR/lint-group-nonstandard-style.rs:10:14
[01:08:10] -    |
[01:08:10] -    |
[01:08:10] - LL |     #[forbid(nonstandard_style)]
[01:08:10] -    |              ^^^^^^^^^^^^^^^^^
[01:08:10] -    = note: #[forbid(non_upper_case_globals)] implied by #[forbid(nonstandard_style)]
[01:08:10] - 
[01:08:10] 27 error: function `CamelCase` should have a snake case name
[01:08:10] 29    |
[01:08:10] 
[01:08:10] 
[01:08:10] 62 LL |         #![warn(nonstandard_style)]
[01:08:10] 63    |                 ^^^^^^^^^^^^^^^^^
[01:08:10] 64    = note: #[warn(non_snake_case)] implied by #[warn(nonstandard_style)]
[01:08:10] + 
[01:08:10] + error: static variable `bad` should have an upper case name
[01:08:10] +    |
[01:08:10] +    |
[01:08:10] + LL |         static bad: isize = 1; //~ ERROR should have an upper
[01:08:10] +    |                ^^^ help: convert the identifier to upper case: `BAD`
[01:08:10] + note: lint level defined here
[01:08:10] +   --> $DIR/lint-group-nonstandard-style.rs:10:14
[01:08:10] +    |
[01:08:10] +    |
[01:08:10] + LL |     #[forbid(nonstandard_style)]
[01:08:10] +    |              ^^^^^^^^^^^^^^^^^
[01:08:10] +    = note: #[forbid(non_upper_case_globals)] implied by #[forbid(nonstandard_style)]
[01:08:10] 66 error: aborting due to 3 previous errors
[01:08:10] 67 
[01:08:10] 
[01:08:10] 
[01:08:10] 
[01:08:10] The actual stderr differed from the expected stderr.
[01:08:10] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/lint-group-nonstandard-style/lint-group-nonstandard-style.stderr
[01:08:10] To update references, rerun the tests and pass the `--bless` flag
[01:08:10] To only update this specific test, also pass `--test-args lint/lint-group-nonstandard-style.rs`
[01:08:10] error: 1 errors occurred comparing output.
[01:08:10] status: exit code: 1
[01:08:10] status: exit code: 1
[01:08:10] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/lint-group-nonstandard-style/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/lint-group-nonstandard-style/auxiliary" "-A" "unused"
[01:08:10] ------------------------------------------
[01:08:10] 
[01:08:10] ------------------------------------------
[01:08:10] stderr:
[01:08:10] stderr:
[01:08:10] ------------------------------------------
[01:08:10] {"message":"type `snake_case` should have a camel case name","code":{"code":"non_camel_case_types","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":457,"byte_end":467,"line_start":22,"line_end":22,"column_start":16,"column_end":26,"is_primary":true,"text":[{"text":"        struct snake_case; //~ WARN should have a camel","highlight_start":16,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":365,"byte_end":382,"line_start":18,"line_end":18,"column_start":17,"column_end":34,"is_primary":true,"text":[{"text":"        #![warn(nonstandard_style)]","highlight_start":17,"highlight_end":34}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"#[warn(non_camel_case_types)] implied by #[warn(nonstandard_style)]","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"convert the identifier to camel case","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":457,"byte_end":467,"line_start":22,"line_end":22,"column_start":16,"column_end":26,"is_primary":true,"text":[{"text":"        struct snake_case; //~ WARN should have a camel","highlight_start":16,"highlight_end":26}],"label":null,"suggested_replacement":"SnakeCase","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: type `snake_case` should have a camel case name\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:22:16\n   |\nLL |         struct snake_case; //~ WARN should have a camel\n   |                ^^^^^^^^^^ help: convert the identifier to camel case: `SnakeCase`\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:18:17\n   |\nLL |         #![warn(nonstandard_style)]\n   |                 ^^^^^^^^^^^^^^^^^\n   = note: #[warn(non_camel_case_types)] implied by #[warn(nonstandard_style)]\n\n"}
[01:08:10] {"message":"function `CamelCase` should have a snake case name","code":{"code":"non_snake_case","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":53,"byte_end":62,"line_start":4,"line_end":4,"column_start":4,"column_end":13,"is_primary":true,"text":[{"text":"fn CamelCase() {} //~ ERROR should have a snake","highlight_start":4,"highlight_end":13}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":8,"byte_end":25,"line_start":1,"line_end":1,"column_start":9,"column_end":26,"is_primary":true,"text":[{"text":"#![deny(nonstandard_style)]","highlight_start":9,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"#[deny(non_snake_case)] implied by #[deny(nonstandard_style)]","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"convert the identifier to snake case","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":53,"byte_end":62,"line_start":4,"line_end":4,"column_start":4,"column_end":13,"is_primary":true,"text":[{"text":"fn CamelCase() {} //~ ERROR should have a snake","highlight_start":4,"highlight_end":13}],"label":null,"suggested_replacement":"camel_case","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error: function `CamelCase` should have a snake case name\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:4:4\n   |\nLL | fn CamelCase() {} //~ ERROR should have a snake\n   |    ^^^^^^^^^ help: convert the identifier to snake case: `camel_case`\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:1:9\n   |\nLL | #![deny(nonstandard_style)]\n   |         ^^^^^^^^^^^^^^^^^\n   = note: #[deny(non_snake_case)] implied by #[deny(nonstandard_style)]\n\n"}
[01:08:10] {"message":"function `CamelCase` should have a snake case name","code":{"code":"non_snake_case","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":219,"byte_end":228,"line_start":12,"line_end":12,"column_start":12,"column_end":21,"is_primary":true,"text":[{"text":"        fn CamelCase() {} //~ ERROR should have a snake","highlight_start":12,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":174,"byte_end":191,"line_start":10,"line_end":10,"column_start":14,"column_end":31,"is_primary":true,"text":[{"text":"    #[forbid(nonstandard_style)]","highlight_start":14,"highlight_end":31}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"#[forbid(non_snake_case)] implied by #[forbid(nonstandard_style)]","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"convert the identifier to snake case","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":219,"byte_end":228,"line_start":12,"line_end":12,"column_start":12,"column_end":21,"is_primary":true,"text":[{"text":"        fn CamelCase() {} //~ ERROR should have a snake","highlight_start":12,"highlight_end":21}],"label":null,"suggested_replacement":"camel_case","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error: function `CamelCase` should have a snake case name\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:12:12\n   |\nLL |         fn CamelCase() {} //~ ERROR should have a snake\n   |            ^^^^^^^^^ help: convert the identifier to snake case: `camel_case`\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:10:14\n   |\nLL |     #[forbid(nonstandard_style)]\n   |              ^^^^^^^^^^^^^^^^^\n   = note: #[forbid(non_snake_case)] implied by #[forbid(nonstandard_style)]\n\n"}
[01:08:10] {"message":"function `CamelCase` should have a snake case name","code":{"code":"non_snake_case","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":397,"byte_end":406,"line_start":20,"line_end":20,"column_start":12,"column_end":21,"is_primary":true,"text":[{"text":"        fn CamelCase() {} //~ WARN should have a snake","highlight_start":12,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":365,"byte_end":382,"line_start":18,"line_end":18,"column_start":17,"column_end":34,"is_primary":true,"text":[{"text":"        #![warn(nonstandard_style)]","highlight_start":17,"highlight_end":34}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"#[warn(non_snake_case)] implied by #[warn(nonstandard_style)]","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"convert the identifier to snake case","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":397,"byte_end":406,"line_start":20,"line_end":20,"column_start":12,"column_end":21,"is_primary":true,"text":[{"text":"        fn CamelCase() {} //~ WARN should have a snake","highlight_start":12,"highlight_end":21}],"label":null,"suggested_replacement":"camel_case","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: function `CamelCase` should have a snake case name\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:20:12\n   |\nLL |         fn CamelCase() {} //~ WARN should have a snake\n   |            ^^^^^^^^^ help: convert the identifier to snake case: `camel_case`\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:18:17\n   |\nLL |         #![warn(nonstandard_style)]\n   |                 ^^^^^^^^^^^^^^^^^\n   = note: #[warn(non_snake_case)] implied by #[warn(nonstandard_style)]\n\n"}
[01:08:10] {"message":"static variable `bad` should have an upper case name","code":{"code":"non_upper_case_globals","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":280,"byte_end":283,"line_start":14,"line_end":14,"column_start":16,"column_end":19,"is_primary":true,"text":[{"text":"        static bad: isize = 1; //~ ERROR should have an upper","highlight_start":16,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":174,"byte_end":191,"line_start":10,"line_end":10,"column_start":14,"column_end":31,"is_primary":true,"text":[{"text":"    #[forbid(nonstandard_style)]","highlight_start":14,"highlight_end":31}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"#[forbid(non_upper_case_globals)] implied by #[forbid(nonstandard_style)]","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"convert the identifier to upper case","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-group-nonstandard-style.rs","byte_start":280,"byte_end":283,"line_start":14,"line_end":14,"column_start":16,"column_end":19,"is_primary":true,"text":[{"text":"        static bad: isize = 1; //~ ERROR should have an upper","highlight_start":16,"highlight_end":19}],"label":null,"suggested_replacement":"BAD","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error: static variable `bad` should have an upper case name\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:14:16\n   |\nLL |         static bad: isize = 1; //~ ERROR should have an upper\n   |                ^^^ help: convert the identifier to upper case: `BAD`\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/lint-group-nonstandard-style.rs:10:14\n   |\nLL |     #[forbid(nonstandard_style)]\n   |              ^^^^^^^^^^^^^^^^^\n   = note: #[forbid(non_upper_case_globals)] implied by #[forbid(nonstandard_style)]\n\n"}
[01:08:10] 
[01:08:10] ------------------------------------------
[01:08:10] 
[01:08:10] thread '[ui] ui/lint/lint-group-nonstandard-style.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3291:9
---
[01:08:10] 
[01:08:10] 44    |  |____________|
[01:08:10] 45    |               help: remove this `mut`
[01:08:10] 46 
[01:08:10] - warning: denote infinite loops with `loop { ... }`
[01:08:10] -   --> $DIR/suggestions.rs:46:5
[01:08:10] - LL |     while true {
[01:08:10] - LL |     while true {
[01:08:10] -    |     ^^^^^^^^^^ help: use `loop`
[01:08:10] -    |
[01:08:10] -    = note: #[warn(while_true)] on by default
[01:08:10] - 
[01:08:10] - warning: the `warp_factor:` in this pattern is redundant
[01:08:10] -   --> $DIR/suggestions.rs:61:23
[01:08:10] -    |
[01:08:10] - LL |             Equinox { warp_factor: warp_factor } => {}
[01:08:10] -    |                       |
[01:08:10] -    |                       help: remove this
[01:08:10] -    |
[01:08:10] -    = note: #[warn(non_shorthand_field_patterns)] on by default
[01:08:10] -    = note: #[warn(non_shorthand_field_patterns)] on by default
[01:08:10] - 
[01:08:10] 65 error: const items should never be #[no_mangle]
[01:08:10] 66   --> $DIR/suggestions.rs:6:14
[01:08:10] 
[01:08:10] 114    |     ------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[01:08:10] 115    |     |
[01:08:10] 116    |     help: remove this attribute
[01:08:10] 116    |     help: remove this attribute
[01:08:10] + 
[01:08:10] + warning: denote infinite loops with `loop { ... }`
[01:08:10] +   --> $DIR/suggestions.rs:46:5
[01:08:10] + LL |     while true {
[01:08:10] + LL |     while true {
[01:08:10] +    |     ^^^^^^^^^^ help: use `loop`
[01:08:10] +    |
[01:08:10] +    = note: #[warn(while_true)] on by default
[01:08:10] + 
[01:08:10] + warning: the `warp_factor:` in this pattern is redundant
[01:08:10] +   --> $DIR/suggestions.rs:61:23
[01:08:10] +    |
[01:08:10] + LL |             Equinox { warp_factor: warp_factor } => {}
[01:08:10] +    |                       |
[01:08:10] +    |                       help: remove this
[01:08:10] +    |
[01:08:10] +    = note: #[warn(non_shorthand_field_patterns)] on by default
[01:08:10] +    = note: #[warn(non_shorthand_field_patterns)] on by default
[01:08:10] 117 
[01:08:10] 118 error: aborting due to 3 previous errors
[01:08:10] 119 
[01:08:10] 
[01:08:10] 
[01:08:10] The actual stderr differed from the expected stderr.
[01:08:10] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/suggestions/suggestions.stderr
[01:08:10] To update references, rerun the tests and pass the `--bless` flag
[01:08:10] To only update this specific test, also pass `--test-args lint/suggestions.rs`
[01:08:10] error: 1 errors occurred comparing output.
[01:08:10] status: exit code: 1
[01:08:10] status: exit code: 1
[01:08:10] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/lint/suggestions.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/suggestions/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/suggestions/auxiliary" "-A" "unused"
[01:08:10] ------------------------------------------
[01:08:10] 
[01:08:10] ------------------------------------------
[01:08:10] stderr:
[01:08:10] stderr:
[01:08:10] ------------------------------------------
[01:08:10] {"message":"unnecessary parentheses around assigned value","code":{"code":"unused_parens","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1530,"byte_end":1555,"line_start":49,"line_end":49,"column_start":31,"column_end":56,"is_primary":true,"text":[{"text":"        let mut registry_no = (format!(\"NX-{}\", 74205));","highlight_start":31,"highlight_end":56}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":40,"byte_end":53,"line_start":3,"line_end":3,"column_start":21,"column_end":34,"is_primary":true,"text":[{"text":"#![warn(unused_mut, unused_parens)] // UI tests pass `-A unused`—see Issue #43896","highlight_start":21,"highlight_end":34}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1530,"byte_end":1555,"line_start":49,"line_end":49,"column_start":31,"column_end":56,"is_primary":true,"text":[{"text":"        let mut registry_no = (format!(\"NX-{}\", 74205));","highlight_start":31,"highlight_end":56}],"label":null,"suggested_replacement":"format!(\"NX-{}\", 74205)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unnecessary parentheses around assigned value\n  --> /checkout/src/test/ui/lint/suggestions.rs:49:31\n   |\nLL |         let mut registry_no = (format!(\"NX-{}\", 74205));\n   |                               ^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove these parentheses\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/suggestions.rs:3:21\n   |\nLL | #![warn(unused_mut, unused_parens)] // UI tests pass `-A unused`—see Issue #43896\n   |                     ^^^^^^^^^^^^^\n\n"}
[01:08:10] {"message":"use of deprecated attribute `no_debug`: the `#[no_debug]` attribute was an experimental feature that has been deprecated due to lack of demand. See https://github.com/rust-lang/rust/issues/29721","code":{"code":"deprecated","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1295,"byte_end":1306,"line_start":42,"line_end":42,"column_start":1,"column_end":12,"is_primary":true,"text":[{"text":"#[no_debug] // should suggest removal of deprecated attribute","highlight_start":1,"highlight_end":12}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"#[warn(deprecated)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"remove this attribute","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1295,"byte_end":1306,"line_start":42,"line_end":42,"column_start":1,"column_end":12,"is_primary":true,"text":[{"text":"#[no_debug] // should suggest removal of deprecated attribute","highlight_start":1,"highlight_end":12}],"label":null,"suggested_replacement":"","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: use of deprecated attribute `no_debug`: the `#[no_debug]` attribute was an experimental feature that has been deprecated due to lack of demand. See https://github.com/rust-lang/rust/issues/29721\n  --> /checkout/src/test/ui/lint/suggestions.rs:42:1\n   |\nLL | #[no_debug] // should suggest removal of deprecated attribute\n   | ^^^^^^^^^^^ help: remove this attribute\n   |\n   = note: #[warn(deprecated)] on by default\n\n"}
[01:08:10] {"message":"variable does not need to be mutable","code":{"code":"unused_mut","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1512,"byte_end":1527,"line_start":49,"line_end":49,"column_start":13,"column_end":28,"is_primary":true,"text":[{"text":"        let mut registry_no = (format!(\"NX-{}\", 74205));","highlight_start":13,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":28,"byte_end":38,"line_start":3,"line_end":3,"column_start":9,"column_end":19,"is_primary":true,"text":[{"text":"#![warn(unused_mut, unused_parens)] // UI tests pass `-A unused`—see Issue #43896","highlight_start":9,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"remove this `mut`","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1512,"byte_end":1516,"line_start":49,"line_end":49,"column_start":13,"column_end":17,"is_primary":true,"text":[{"text":"        let mut registry_no = (format!(\"NX-{}\", 74205));","highlight_start":13,"highlight_end":17}],"label":null,"suggested_replacement":"","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: variable does not need to be mutable\n  --> /checkout/src/test/ui/lint/suggestions.rs:49:13\n   |\nLL |         let mut registry_no = (format!(\"NX-{}\", 74205));\n   |             ----^^^^^^^^^^^\n   |             |\n   |             help: remove this `mut`\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/suggestions.rs:3:9\n   |\nLL | #![warn(unused_mut, unused_parens)] // UI tests pass `-A unused`—see Issue #43896\n   |         ^^^^^^^^^^\n\n"}
[01:08:10] {"message":"variable does not need to be mutable","code":{"code":"unused_mut","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1816,"byte_end":1830,"line_start":55,"line_end":56,"column_start":13,"column_end":11,"is_primary":true,"text":[{"text":"        let mut","highlight_start":13,"highlight_end":16},{"text":"\t        b = 1;","highlight_start":1,"highlight_end":11}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove this `mut`","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1816,"byte_end":1829,"line_start":55,"line_end":56,"column_start":13,"column_end":10,"is_primary":true,"text":[{"text":"        let mut","highlight_start":13,"highlight_end":16},{"text":"\t        b = 1;","highlight_start":1,"highlight_end":10}],"label":null,"suggested_replacement":"","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: variable does not need to be mutable\n  --> /checkout/src/test/ui/lint/suggestions.rs:55:13\n   |\nLL |            let mut\n   |   _____________^\n   |  |_____________|\n   | ||\nLL | ||             b = 1;\n   | ||____________-^\n   |  |____________|\n   |               help: remove this `mut`\n\n"}
[01:08:10] {"message":"const items should never be #[no_mangle]","code":{"code":"no_mangle_const_items","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":140,"byte_end":167,"line_start":6,"line_end":6,"column_start":14,"column_end":41,"is_primary":true,"text":[{"text":"#[no_mangle] const DISCOVERY: usize = 1;","highlight_start":14,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"#[deny(no_mangle_const_items)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"try a static value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":140,"byte_end":145,"line_start":6,"line_end":6,"column_start":14,"column_end":19,"is_primary":true,"text":[{"text":"#[no_mangle] const DISCOVERY: usize = 1;","highlight_start":14,"highlight_end":19}],"label":null,"suggested_replacement":"pub static","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: const items should never be #[no_mangle]\n  --> /checkout/src/test/ui/lint/suggestions.rs:6:14\n   |\nLL | #[no_mangle] const DISCOVERY: usize = 1;\n   |              -----^^^^^^^^^^^^^^^^^^^^^^\n   |              |\n   |              help: try a static value: `pub static`\n   |\n   = note: #[deny(no_mangle_const_items)] on by default\n\n"}
[01:08:10] {"message":"functions generic over types must be mangled","code":{"code":"no_mangle_generic_items","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":295,"byte_end":322,"line_start":12,"line_end":12,"column_start":1,"column_end":28,"is_primary":true,"text":[{"text":"pub fn defiant<T>(_t: T) {}","highlight_start":1,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"#[warn(no_mangle_generic_items)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"remove this attribute","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":250,"byte_end":262,"line_start":10,"line_end":10,"column_start":1,"column_end":13,"is_primary":true,"text":[{"text":"#[no_mangle]","highlight_start":1,"highlight_end":13}],"label":null,"suggested_replacement":"","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: functions generic over types must be mangled\n  --> /checkout/src/test/ui/lint/suggestions.rs:12:1\n   |\nLL | #[no_mangle]\n   | ------------ help: remove this attribute\nLL | //~^ HELP remove this attribute\nLL | pub fn defiant<T>(_t: T) {}\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = note: #[warn(no_mangle_generic_items)] on by default\n\n"}
[01:08:10] {"message":"const items should never be #[no_mangle]","code":{"code":"no_mangle_const_items","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":639,"byte_end":672,"line_start":22,"line_end":22,"column_start":18,"column_end":51,"is_primary":true,"text":[{"text":"    #[no_mangle] pub const DAUNTLESS: bool = true;","highlight_start":18,"highlight_end":51}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try a static value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":639,"byte_end":648,"line_start":22,"line_end":22,"column_start":18,"column_end":27,"is_primary":true,"text":[{"text":"    #[no_mangle] pub const DAUNTLESS: bool = true;","highlight_start":18,"highlight_end":27}],"label":null,"suggested_replacement":"pub static","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: const items should never be #[no_mangle]\n  --> /checkout/src/test/ui/lint/suggestions.rs:22:18\n   |\nLL |     #[no_mangle] pub const DAUNTLESS: bool = true;\n   |                  ---------^^^^^^^^^^^^^^^^^^^^^^^^\n   |                  |\n   |                  help: try a static value: `pub static`\n\n"}
[01:08:10] {"message":"functions generic over types must be mangled","code":{"code":"no_mangle_generic_items","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":779,"byte_end":802,"line_start":25,"line_end":25,"column_start":18,"column_end":41,"is_primary":true,"text":[{"text":"    #[no_mangle] pub fn val_jean<T>() {}","highlight_start":18,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove this attribute","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":766,"byte_end":778,"line_start":25,"line_end":25,"column_start":5,"column_end":17,"is_primary":true,"text":[{"text":"    #[no_mangle] pub fn val_jean<T>() {}","highlight_start":5,"highlight_end":17}],"label":null,"suggested_replacement":"","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: functions generic over types must be mangled\n  --> /checkout/src/test/ui/lint/suggestions.rs:25:18\n   |\nLL |     #[no_mangle] pub fn val_jean<T>() {}\n   |     ------------ ^^^^^^^^^^^^^^^^^^^^^^^\n   |     |\n   |     help: remove this attribute\n\n"}
[01:08:10] {"message":"const items should never be #[no_mangle]","code":{"code":"no_mangle_const_items","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":979,"byte_end":1015,"line_start":30,"line_end":30,"column_start":18,"column_end":54,"is_primary":true,"text":[{"text":"    #[no_mangle] pub(crate) const VETAR: bool = true;","highlight_start":18,"highlight_end":54}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try a static value","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":979,"byte_end":995,"line_start":30,"line_end":30,"column_start":18,"column_end":34,"is_primary":true,"text":[{"text":"    #[no_mangle] pub(crate) const VETAR: bool = true;","highlight_start":18,"highlight_end":34}],"label":null,"suggested_replacement":"pub static","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: const items should never be #[no_mangle]\n  --> /checkout/src/test/ui/lint/suggestions.rs:30:18\n   |\nLL |     #[no_mangle] pub(crate) const VETAR: bool = true;\n   |                  ----------------^^^^^^^^^^^^^^^^^^^^\n   |                  |\n   |                  help: try a static value: `pub static`\n\n"}
[01:08:10] {"message":"functions generic over types must be mangled","code":{"code":"no_mangle_generic_items","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1122,"byte_end":1154,"line_start":33,"line_end":33,"column_start":18,"column_end":50,"is_primary":true,"text":[{"text":"    #[no_mangle] pub(crate) fn crossfield<T>() {}","highlight_start":18,"highlight_end":50}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove this attribute","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1109,"byte_end":1121,"line_start":33,"line_end":33,"column_start":5,"column_end":17,"is_primary":true,"text":[{"text":"    #[no_mangle] pub(crate) fn crossfield<T>() {}","highlight_start":5,"highlight_end":17}],"label":null,"suggested_replacement":"","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: functions generic over types must be mangled\n  --> /checkout/src/test/ui/lint/suggestions.rs:33:18\n   |\nLL |     #[no_mangle] pub(crate) fn crossfield<T>() {}\n   |     ------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |     |\n   |     help: remove this attribute\n\n"}
[01:08:10] {"message":"denote infinite loops with `loop { ... }`","code":{"code":"while_true","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1426,"byte_end":1436,"line_start":46,"line_end":46,"column_start":5,"column_end":15,"is_primary":true,"text":[{"text":"    while true {","highlight_start":5,"highlight_end":15}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"#[warn(while_true)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"use `loop`","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":1426,"byte_end":1436,"line_start":46,"line_end":46,"column_start":5,"column_end":15,"is_primary":true,"text":[{"text":"    while true {","highlight_start":5,"highlight_end":15}],"label":null,"suggested_replacement":"loop","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: denote infinite loops with `loop { ... }`\n  --> /checkout/src/test/ui/lint/suggestions.rs:46:5\n   |\nLL |     while true {\n   |     ^^^^^^^^^^ help: use `loop`\n   |\n   = note: #[warn(while_true)] on by default\n\n"}
[01:08:10] {"message":"the `warp_factor:` in this pattern is redundant","code":{"code":"non_shorthand_field_patterns","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":2007,"byte_end":2031,"line_start":61,"line_end":61,"column_start":23,"column_end":47,"is_primary":true,"text":[{"text":"            Equinox { warp_factor: warp_factor } => {}","highlight_start":23,"highlight_end":47}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"#[warn(non_shorthand_field_patterns)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"remove this","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/suggestions.rs","byte_start":2007,"byte_end":2019,"line_start":61,"line_end":61,"column_start":23,"column_end":35,"is_primary":true,"text":[{"text":"            Equinox { warp_factor: warp_factor } => {}","highlight_start":23,"highlight_end":35}],"label":null,"suggested_replacement":"warp_factor","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: the `warp_factor:` in this pattern is redundant\n  --> /checkout/src/test/ui/lint/suggestions.rs:61:23\n   |\nLL |             Equinox { warp_factor: warp_factor } => {}\n   |                       ------------^^^^^^^^^^^^\n   |                       |\n   |                       help: remove this\n   |\n   = note: #[warn(non_shorthand_field_patterns)] on by default\n\n"}
[01:08:10] 
[01:08:10] ------------------------------------------
[01:08:10] 
[01:08:10] thread '[ui] ui/lint/suggestions.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3291:9
---
[01:08:10] 
[01:08:10] 
[01:08:10] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:495:22
[01:08:10] 
[01:08:10] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[01:08:10] 
[01:08:10] 
[01:08:10] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[01:08:10] Build completed unsuccessfully in 0:04:31
[01:08:10] Build completed unsuccessfully in 0:04:31
[01:08:10] make: *** [check] Error 1
[01:08:10] Makefile:48: recipe for target 'check' failed
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:001d549a
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Thu Jan 31 04:51:12 UTC 2019
---
travis_time:end:05bb5b20:start=1548910274514980775,finish=1548910274522732630,duration=7751855
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:21c54d94
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:260188f8
$ dmesg | grep -i kill
