\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":6782,"byte_end":6800,"line_start":216,"line_end":216,"column_start":9,"column_end":27,"is_primary":true,"text":[{"text":"        type Check = Priv; //~ ERROR private type `aliases_pub::Priv` in public interface","highlight_start":9,"highlight_end":27}],"label":"can't leak private type","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":5697,"byte_end":5697,"line_start":179,"line_end":179,"column_start":5,"column_end":5,"is_primary":false,"text":[{"text":"    struct Priv;","highlight_start":5,"highlight_end":5}],"label":"`aliases_pub::Priv` declared as private","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0446]: private type `aliases_pub::Priv` in public interface\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:216:9\n   |\nLL |     struct Priv;\n   |     - `aliases_pub::Priv` declared as private\n...\nLL |         type Check = Priv; //~ ERROR private type `aliases_pub::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^ can't leak private type\n\n"}
[01:00:52] {"message":"private trait `aliases_priv::PrivTr1` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":7251,"byte_end":7283,"line_start":240,"line_end":240,"column_start":5,"column_end":37,"is_primary":true,"text":[{"text":"    pub trait Tr1: PrivUseAliasTr {}","highlight_start":5,"highlight_end":37}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `aliases_priv::PrivTr1` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:240:5\n   |\nLL |     pub trait Tr1: PrivUseAliasTr {}\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `aliases_priv::PrivTr1<aliases_priv::Priv2>` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":7397,"byte_end":7440,"line_start":243,"line_end":243,"column_start":5,"column_end":48,"is_primary":true,"text":[{"text":"    pub trait Tr2: PrivUseAliasTr<PrivAlias> {}","highlight_start":5,"highlight_end":48}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `aliases_priv::PrivTr1<aliases_priv::Priv2>` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:243:5\n   |\nLL |     pub trait Tr2: PrivUseAliasTr<PrivAlias> {}\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `aliases_priv::Priv2` in public interface (error E0446)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":7397,"byte_end":7440,"line_start":243,"line_end":243,"column_start":5,"column_end":48,"is_primary":true,"text":[{"text":"    pub trait Tr2: PrivUseAliasTr<PrivAlias> {}","highlight_start":5,"highlight_end":48}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private type `aliases_priv::Priv2` in public interface (error E0446)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:243:5\n   |\nLL |     pub trait Tr2: PrivUseAliasTr<PrivAlias> {}\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"bounds on generic parameters are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":1773,"byte_end":1779,"line_start":50,"line_end":50,"column_start":23,"column_end":29,"is_primary":true,"text":[{"text":"    pub type Alias<T: PrivTr> = T; //~ ERROR private trait `traits::PrivTr` in public interface","highlight_start":23,"highlight_end":29}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"#[warn(type_alias_bounds)] on by default","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"the bound will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: bounds on generic parameters are not enforced in type aliases\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:50:23\n   |\nLL |     pub type Alias<T: PrivTr> = T; //~ ERROR private trait `traits::PrivTr` in public interface\n   |                       ^^^^^^\n   |\n   = note: #[warn(type_alias_bounds)] on by default\n   = help: the bound will not be checked when the type alias is used, and should be removed\n\n"}
[01:00:52] {"message":"where-clauses are not enforced in type aliases","code":{"code":"type_alias_bounds","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":2857,"byte_end":2866,"line_start":75,"line_end":75,"column_start":29,"column_end":38,"is_primary":true,"text":[{"text":"    pub type Alias<T> where T: PrivTr = T;","highlight_start":29,"highlight_end":38}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"the clause will not be checked when the type alias is used, and should be removed","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"warning: where-clauses are not enforced in type aliases\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:75:29\n   |\nLL |     pub type Alias<T> where T: PrivTr = T;\n   |                             ^^^^^^^^^\n   |\n   = help: the clause will not be checked when the type alias is used, and should be removed\n\n"}
[01:00:52] {"message":"For more information about this error, try `rustc --explain E0446`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0446`.\n"}
[01:00:52] 
[01:00:52] ------------------------------------------
[01:00:52] 
---
[01:00:52] 
[01:00:52] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:502:22
[01:00:52] 
[01:00:52] 
[01:00:52] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[01:00:52] 
[01:00:52] 
[01:00:52] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[01:00:52] Build completed unsuccessfully in 0:04:03
[01:00:52] Build completed unsuccessfully in 0:04:03
[01:00:52] make: *** [check] Error 1
[01:00:52] Makefile:48: recipe for target 'check' failed
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:00651684
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Fri Feb  8 13:18:15 UTC 2019
---
travis_time:end:35776c30:start=1549631896238925418,finish=1549631896243553236,duration=4627818
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:2ea0fdba
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb --batch -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:1f61aae4
travis_time:start:1f61aae4
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:09f03850
$ dmesg | grep -i kill
