\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":1583,"byte_end":1601,"line_start":41,"line_end":41,"column_start":9,"column_end":27,"is_primary":true,"text":[{"text":"        type Alias = Priv; //~ ERROR private type `types::Priv` in public interface","highlight_start":9,"highlight_end":27}],"label":"can't leak private type","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":262,"byte_end":262,"line_start":9,"line_end":9,"column_start":5,"column_end":5,"is_primary":false,"text":[{"text":"    struct Priv;","highlight_start":5,"highlight_end":5}],"label":"`types::Priv` declared as private","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0446]: private type `types::Priv` in public interface\n  --> /checkbeing phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":1958,"byte_end":1982,"line_start":53,"line_end":53,"column_start":5,"column_end":29,"is_primary":true,"text":[{"text":"    pub trait Tr1: PrivTr {} //~ ERROR private trait `traits::PrivTr` in public interface","highlight_start":5,"highlight_end":29}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:53:5\n   |\nLL |     pub trait Tr1: PrivTr {} //~ ERROR private trait `traits::PrivTr` in public interface\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":2076,"byte_end":2103,"line_start":55,"line_end":55,"column_start":5,"column_end":32,"is_primary":true,"text":[{"text":"    pub trait Tr2<T: PrivTr> {} //~ ERROR private trait `traits::PrivTr` in public interface","highlight_start":5,"highlight_end":32}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:55:5\n   |\nLL |     pub trait Tr2<T: PrivTr> {} //~ ERROR private trait `traits::PrivTr` in public interface\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":2201,"byte_end":2480,"line_start":57,"line_end":63,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    pub trait Tr3 {","highlight_start":5,"highlight_end":20},{"text":"        //~^ ERROR private trait `traits::PrivTr` in public interface","highlight_start":1,"highlight_end":70},{"text":"        //~| WARNING hard error","highlight_start":1,"highlight_end":32},{"text":"        type Alias: PrivTr;","highlight_start":1,"highlight_end":28},{"text":"        fn f<T: PrivTr>(arg: T) {} //~ ERROR private trait `traits::PrivTr` in public interface","highlight_start":1,"highlight_end":96},{"text":"        //~^ WARNING hard error","highlight_start":1,"highlight_end":32},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:57:5\n   |\nLL | /     pub trait Tr3 {\nLL | |         //~^ ERROR private trait `traits::PrivTr` in public interface\nLL | |         //~| WARNING hard error\nLL | |         type Alias: PrivTr;\nLL | |         fn f<T: PrivTr>(arg: T) {} //~ ERROR private trait `traits::PrivTr` in public interface\nLL | |         //~^ WARNING hard error\nLL | |     }\n   | |_____^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":2355,"byte_end":2381,"line_start":61,"line_end":61,"column_start":9,"column_end":35,"is_primary":true,"text":[{"text":"        fn f<T: PrivTr>(arg: T) {} //~ ERROR private trait `traits::PrivTr` in public interface","highlight_start":9,"highlight_end":35}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:61:9\n   |\nLL | ^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":2608,"byte_end":2643,"line_start":66,"line_end":66,"column_start":5,"column_end":40,"is_primary":true,"text":[{"text":"    impl<T: PrivTr> PubTr for Pub<T> {} //~ ERROR private trait `traits::PrivTr` in public interface","highlight_start":5,"highlight_end":40}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:66:5\n   |\nLL |     impl<T: PrivTr> PubTr for Pub<T> {} //~ ERROR private trait `traits::PrivTr` in public interface\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits_where::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":2833,"byte_end":2871,"line_start":75,"line_end":75,"column_start":5,"column_end":43,"is_primary":true,"text":[{"text":"    pub type Alias<T> where T: PrivTr = T;","highlight_start":5,"highlight_end":43}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits_where::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:75:5\n   |\nLL |     pub type Alias<T> where T: PrivTr = T;\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits_where::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":3052,"byte_end":3087,"line_start":79,"line_end":79,"column_start":5,"column_end":40,"is_primary":true,"text":[{"text":"    pub trait Tr2<T> where T: PrivTr {}","highlight_start":5,"highlight_end":40}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits_where::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:79:5\n   |\nLL |     pub trait Tr2<T> where T: PrivTr {}\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits_where::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":3224,"byte_end":3258,"line_start":83,"line_end":83,"column_start":9,"column_end":43,"is_primary":true,"text":[{"text":"        fn f<T>(arg: T) where T: PrivTr {}","highlight_start":9,"highlight_end":43}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits_where::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:83:9\n   |\nLL |         fn f<T>(arg: T) where T: PrivTr {}\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits_where::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":3377,"byte_end":3410,"line_start":87,"line_end":87,"column_start":5,"column_end":38,"is_primary":true,"text":[{"text":"    impl<T> Pub<T> where T: PrivTr {}","highlight_start":5,"highlight_end":38}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits_where::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:87:5\n   |\nLL |     impl<T> Pub<T> where T: PrivTr {}\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `traits_where::PrivTr` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":3523,"byte_end":3566,"line_start":90,"line_end":90,"column_start":5,"column_end":48,"is_primary":true,"text":[{"text":"    impl<T> PubTr for Pub<T> where T: PrivTr {}","highlight_start":5,"highlight_end":48}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `traits_where::PrivTr` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:90:5\n   |\nLL |     impl<T> PubTr for Pub<T> where T: PrivTr {}\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private trait `generics::PrivTr<generics::Pub>` in public interface (error E0445)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":3806,"byte_end":3835,"line_start":101,"line_end":101,"column_start":5,"column_end":34,"is_primary":true,"text":[{"text":"    pub trait Tr1: PrivTr<Pub> {}","highlight_start":5,"highlight_end":34}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private trait `generics::PrivTr<generics::Pub>` in public interface (error E0445)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:101:5\n   |\nLL |     pub trait Tr1: PrivTr<Pub> {}\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `generics::Priv` in public interface (error E0446)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":3959,"byte_end":3988,"line_start":104,"line_end":104,"column_start":5,"column_end":34,"is_primary":true,"text":[{"text":"    pub trait Tr2: PubTr<Priv> {} //~ ERROR private type `generics::Priv` in public interface","highlight_start":5,"highlight_end":34}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private type `generics::Priv` in public interface (error E0446)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:104:5\n   |\nLL |     pub trait Tr2: PubTr<Priv> {} //~ ERROR private type `generics::Priv` in public interface\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `generics::Priv` in public interface (error E0446)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":4085,"byte_end":4119,"line_start":106,"line_end":106,"column_start":5,"column_end":39,"is_primary":true,"text":[{"text":"    pub trait Tr3: PubTr<[Priv; 1]> {} //~ ERROR private type `generics::Priv` in public interface","highlight_start":5,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private type `generics::Priv` in public interface (error E0446)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:106:5\n   |\nLL |     pub trait Tr3: PubTr^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `impls::Priv` in public interface","code":{"code":"E0446","explanation":"\nA private type was used in a public type signature. Erroneous code example:\n\n