\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":4777,"byte_end":4795,"line_start":135,"line_end":135,"column_start":9,"column_end":27,"is_primary":true,"text":[{"text":"        type Alias = Priv; //~ ERROR private type `impls::Priv` in public interface","highlight_start":9,"highlight_end":27}],"label":"can't leak private type","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":4362,"byte_end":4362,"line_start":113,"line_end":113,"column_start":5,"column_end":5,"is_primary":false,"text":[{"text":"    struct Priv;","highlight_start":5,"highlight_end":5}],"label":"`impls::Priv` declared as private","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0446]: private type `impls::Priv` in public interface\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:135:9\n   |\nLL |     struct Priv;\n   |     - `impls::Priv` declared as private\n...\nLL |         type Alias = Priv; //~ ERROR private type `impls::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^ can't leak private type\n\n"}
[01:00:52] {"message":"private type `aliases_pub::Priv` in public interface (error E0446)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":6316,"byte_end":6338,"line_start":206,"line_end":206,"column_start":9,"column_end":31,"is_primary":true,"text":[{"text":"        pub fn f(arg: Priv) {} //~ ERROR private type `aliases_pub::Priv` in public interface","highlight_start":9,"highlight_end":31}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private type `aliases_pub::Priv` in public interface (error E0446)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:206:9\n   |\nLL |         pub fn f(arg: Priv) {} //~ ERROR private type `aliases_pub::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `aliases_pub::Priv` in public interface","code":{"code":"E0446","explanation":"\nA private type was used in a public type signature. Erroneous code example:\n\n