\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":849,"byte_end":867,"line_start":26,"line_end":26,"column_start":9,"column_end":27,"is_primary":true,"text":[{"text":"        type Alias = Priv; //~ ERROR private type `types::Priv` in public interface","highlight_start":9,"highlight_end":27}],"label":"can't leak private type","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":262,"byte_end":262,"line_start":9,"line_end":9,"column_start":5,"column_end":5,"is_primary":false,"text":[{"text":"    struct Priv;","highlight_start":5,"highlight_end":5}],"label":"`types::Priv` declared as private","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0446]: private type `types::Priv` in public interface\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:26:9\n   |\nLL |     struct Priv;\n   |     - `types::Priv` declared as private\n...\nLL |         type Alias = Priv; //~ ERROR private type `types::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^ can't leak private type\n\n"}
[01:00:52] {"message":"private type `types::Priv` in public interface (error E0446)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":933,"byte_end":952,"line_start":27,"line_end":27,"column_start":9,"column_end":28,"is_primary":true,"text":[{"text":"        fn f1(arg: Priv) {} //~ ERROR private type `types::Priv` in public interface","highlight_start":9,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":nrn.rs:29:9\n   |\nLL |         fn f2() -> Priv { panic!() } //~ ERROR private type `types::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `types::Priv` in public interface (error E0446)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":1195,"byte_end":1215,"line_start":33,"line_end":33,"column_start":9,"column_end":29,"is_primary":true,"text":[{"text":"        pub static ES: Priv; //~ ERROR private type `types::Priv` in public interface","highlight_start":9,"highlight_end":29}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private type `types::Priv` in public interface (error E0446)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:33:9\n   |\nLL |         pub static ES: Priv; //~ ERROR private type `types::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `types::Priv` in public interface (error E0446)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":1313,"byte_end":1335,"line_start":35,"line_end":35,"column_start":9,"column_end":31,"is_primary":true,"text":[{"text":"        pub fn ef1(arg: Priv); //~ ERROR private type `types::Priv` in public interface","highlight_start":9,"highlight_end":31}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private type `types::Priv` in public interface (error E0446)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:35:9\n   |\nLL |         pub fn ef1(arg: Priv); //~ ERROR private type `types::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `types::Priv` in public interface (error E0446)","code":{"code":"private_in_public","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":1433,"byte_end":1454,"line_start":37,"line_end":37,"column_start":9,"column_end":30,"is_primary":true,"text":[{"text":"        pub fn ef2() -> Priv; //~ ERROR private type `types::Priv` in public interface","highlight_start":9,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: private type `types::Priv` in public interface (error E0446)\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:37:9\n   |\nLL |         pub fn ef2() -> Priv; //~ ERROR private type `types::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^^^^\n   |\n   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!\n   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>\n\n"}
[01:00:52] {"message":"private type `types::Priv` in public interface","code":{"code":"E0446","explanation":"\nA private type was used in a public type signature. Erroneous code example:\n\n