\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":6491,"byte_end":6509,"line_start":210,"line_end":210,"column_start":9,"column_end":27,"is_primary":true,"text":[{"text":"        type Check = Priv; //~ ERROR private type `aliases_pub::Priv` in public interface","highlight_start":9,"highlight_end":27}],"label":"can't leak private type","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/privacy/private-in-public-warn.rs","byte_start":5697,"byte_end":5697,"line_start":179,"line_end":179,"column_start":5,"column_end":5,"is_primary":false,"text":[{"text":"    struct Priv;","highlight_start":5,"highlight_end":5}],"label":"`aliases_pub::Priv` declared as private","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0446]: private type `aliases_pub::Priv` in public interface\n  --> /checkout/src/test/ui/privacy/private-in-public-warn.rs:210:9\n   |\nLL |     struct Priv;\n   |     - `aliases_pub::Priv` declared as private\n...\nLL |         type Check = Priv; //~ ERROR private type `aliases_pub::Priv` in public interface\n   |         ^^^^^^^^^^^^^^^^^^ can't leak private type\n\n"}
[01:00:52] {"message":"private type `aliases_pub::Priv` in public interface","code":{"code":"E0446","explanation":"\nA private type was used in a public type signature. Erroneous code example:\n\n