\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/hygiene/no_implicit_prelude.rs","byte_start":172,"byte_end":175,"line_start":11,"line_end":11,"column_start":9,"column_end":12,"is_primary":true,"text":[{"text":"        Vec::new(); //~ ERROR failed to resolve","highlight_start":9,"highlight_end":12}],"label":"use of undeclared type or module `Vec`","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/hygiene/no_implicit_prelude.rs","byte_start":93,"byte_end":105,"line_start":5,"line_end":5,"column_start":14,"column_end":26,"is_primary":false,"text":[{"text":"    fn f() { ::bar::m!(); }","highlight_start":14,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"::bar::m!","def_site_span":{"file_name":"/checkout/src/test/ui/hygiene/no_implicit_prelude.rs","byte_start":148,"byte_end":276,"line_start":10,"line_end":13,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    pub macro m() {","highlight_start":5,"highlight_end":20},{"text":"        Vec::new(); //~ ERROR failed to resolve","highlight_start":1,"highlight_end":48},{"text":"        ().clone() //~ ERROR no method named `clone` found","highlight_start":1,"highlight_end":59},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[],"rendered":"error[E0433]: failed to resolve: use of undeclared type or module `Vec`\n  --> /checkout/src/test/ui/hygiene/no_implicit_prelude.rs:11:9\n   |\nLL |     fn f() { ::bar::m!(); }\n   |              ------------ in this macro invocation\n...\nLL |         Vec::new(); //~ ERROR failed to resolve\n   |         ^^^ use of undeclared type or module `Vec`\n\n"}
[01:02:26] {"message":"cannot find macro `print!` in this scope","code":null,"level":"error","spans":[{"file_name":"<::std::macros::println macros>","byte_start":10,"byte_end":15,"line_start":1,"line_end":1,"column_start":11,"column_end":16,"is_primary":true,"text":[{"text":"(  ) => ( print ! ( \"\\n\" ) ) ; ( $ ( $ arg : tt ) * ) => (","highlight_start":11,"highlight_end":16}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/hygiene/no_implicit_prelude.rs","byte_start":319,"byte_end":330,"line_start":16,"line_end":16,"column_start":9,"column_end":20,"is_primary":false,"text":[{"text":"        println!(); // OK on 2015 edition (at least for now)","highlight_start":9,"highlight_end":20}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"println!","def_site_span":{"file_name":"<::std::macros::println macros>","byte_start":0,"byte_end":129,"line_start":1,"line_end":2,"column_start":1,"column_end":71,"is_primary":false,"text":[{"text":"(  ) => ( print ! ( \"\\n\" ) ) ; ( $ ( $ arg : tt ) * ) => (","highlight_start":1,"highlight_end":59},{"text":"{ $ crate :: io :: _print ( format_args_nl ! ( $ ( $ arg ) * ) ) ; } )","highlight_start":1,"highlight_end":71}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[{"message":"have you added the `#[macro_use]` on the module/import?","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: cannot find macro `print!` in this scope\n  --> /checkout/src/test/ui/hygiene/no_implicit_prelude.rs:16:9\n   |\nLL |         println!(); // OK on 2015 edition (at least for now)\n   |         ^^^^^^^^^^^\n   |\n   = help: have you added the `#[macro_use]` on the module/import?\n   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)\n\n"}
[01:02:26] {"message":"no method named `clone` found for type `()` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n