\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1839,"byte_end":1841,"line_start":82,"line_end":82,"column_start":40,"column_end":42,"is_primary":true,"text":[{"text":"    let (y, _): (&'static u32, u32) = (&x, 44); //~ ERROR","highlight_start":40,"highlight_end":42}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1858,"byte_end":1859,"line_start":83,"line_end":83,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"`x` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1816,"byte_end":1835,"line_start":82,"line_end":82,"column_start":17,"column_end":36,"is_primary":false,"text":[{"text":"    let (y, _): (&'static u32, u32) = (&x, 44); //~ ERROR","highlight_start":17,"highlight_end":36}],"label":"type annotation requires that `x` is borrowed for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `x` does not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:82:40\n   |\nLL |     let (y, _): (&'static u32, u32) = (&x, 44); //~ ERROR\n   |                 -------------------    ^^ borrowed value does not live long enough\n   |                 |\n   |                 type annotation requires that `x` is borrowed for `'static`\nLL | }\n   | - `x` dropped here while still borrowed\n\n"}
[01:05:24] {"message":"`x` does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n