\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2378,"byte_end":2380,"line_start":100,"line_end":100,"column_start":17,"column_end":19,"is_primary":true,"text":[{"text":"        value1: &x, //~ ERROR","highlight_start":17,"highlight_end":19}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2420,"byte_end":2421,"line_start":103,"line_end":103,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"`x` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2330,"byte_end":2350,"line_start":99,"line_end":99,"column_start":42,"column_end":62,"is_primary":false,"text":[{"text":"    let Double { value1: _, value2: _ }: Double<&'static u32> = Double {","highlight_start":42,"highlight_end":62}],"label":"type annotation requires that `x` is borrowed for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `x` does not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:100:17\n   |\nLL |     let Double { value1: _, value2: _ }: Double<&'static u32> = Double {\n   |                                          -------------------- type annotation requires that `x` is borrowed for `'static`\nLL |         value1: &x, //~ ERROR\n   |                 ^^ borrowed value does not live long enough\n...\nLL | }\n   | - `x` dropped here while still borrowed\n\n"}
[01:05:24] {"message":"lifetime may not live long enough","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2465,"byte_end":2467,"line_start":105,"line_end":105,"column_start":43,"column_end":45,"is_primary":false,"text":[{"text":"fn static_to_a_to_static_through_variable<'a>(x: &'a u32) -> &'static u32 {","highlight_start":43,"highlight_end":45}],"label":"lifetime `'a` defined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2535,"byte_end":2536,"line_start":113,"line_end":113,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    y //~ ERROR","highlight_start":5,"highlight_end":6}],"label":"returning this value requires that `'a` must outlive `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: lifetime may not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:113:5\n   |\nLL | fn static_to_a_to_static_through_variable<'a>(x: &'a u32) -> &'static u32 {\n   |                                           -- lifetime `'a` defined here\n...\nLL |     y //~ ERROR\n   |     ^ returning this value requires that `'a` must outlive `'static`\n\n"}
[01:05:24] {"message":"lifetime may not live long enough","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2589,"byte_end":2591,"line_start":116,"line_end":116,"column_start":40,"column_end":42,"is_primary":false,"text":[{"text":"fn static_to_a_to_static_through_tuple<'a>(x: &'a u32) -> &'static u32 {","highlight_start":40,"highlight_end":42}],"label":"lifetime `'a` defined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2679,"byte_end":2680,"line_start":125,"line_end":125,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    y //~ ERROR","highlight_start":5,"highlight_end":6}],"label":"returning this value requires that `'a` must outlive `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: lifetime may not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:125:5\n   |\nLL | fn static_to_a_to_static_through_tuple<'a>(x: &'a u32) -> &'static u32 {\n   |                                        -- lifetime `'a` defined here\n...\nLL |     y //~ ERROR\n   |     ^ returning this value requires that `'a` must outlive `'static`\n\n"}
[01:05:24] {"message":"lifetime may not live long enough","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2734,"byte_end":2736,"line_start":128,"line_end":128,"column_start":41,"column_end":43,"is_primary":false,"text":[{"text":"fn static_to_a_to_static_through_struct<'a>(_x: &'a u32) -> &'static u32 {","highlight_start":41,"highlight_end":43}],"label":"lifetime `'a` defined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2843,"byte_end":2844,"line_start":130,"line_end":130,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    y //~ ERROR","highlight_start":5,"highlight_end":6}],"label":"returning this value requires that `'a` must outlive `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: lifetime may not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:130:5\n   |\nLL | fn static_to_a_to_static_through_struct<'a>(_x: &'a u32) -> &'static u32 {\n   |                                         -- lifetime `'a` defined here\nLL |     let Single { value: y }: Single<&'a u32> = Single { value: &22 };\nLL |     y //~ ERROR\n   |     ^ returning this value requires that `'a` must outlive `'static`\n\n"}
[01:05:24] {"message":"lifetime may not live long enough","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2885,"byte_end":2887,"line_start":133,"line_end":133,"column_start":28,"column_end":30,"is_primary":false,"text":[{"text":"fn a_to_static_then_static<'a>(x: &'a u32) -> &'static u32 {","highlight_start":28,"highlight_end":30}],"label":"lifetime `'a` defined here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2936,"byte_end":2955,"line_start":134,"line_end":134,"column_start":18,"column_end":37,"is_primary":true,"text":[{"text":"    let (y, _z): (&'static u32, u32) = (x, 44); //~ ERROR","highlight_start":18,"highlight_end":37}],"label":"type annotation requires that `'a` must outlive `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: lifetime may not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:134:18\n   |\nLL | fn a_to_static_then_static<'a>(x: &'a u32) -> &'static u32 {\n   |                            -- lifetime `'a` defined here\nLL |     let (y, _z): (&'static u32, u32) = (x, 44); //~ ERROR\n   |                  ^^^^^^^^^^^^^^^^^^^ type annotation requires that `'a` must outlive `'static`\n\n"}
[01:05:24] {"message":"aborting due to 19 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 19 previous errors\n\n"}
[01:05:24] {"message":"Some errors occurred: E0597, E0716.","code":null,"level":"","spans":[],"children":[],"rendered":"Some errors occurred: E0597, E0716.\n"}
[01:05:24] 
[01:05:24] ------------------------------------------
[01:05:24] 
[01:05:24] thread '[ui] ui/nll/user-annotations/patterns.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3295:9
[01:05:24] thread '[ui] ui/nll/user-annotations/patterns.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3295:9
[01:05:24] 
[01:05:24] ---- [ui] ui/span/issue-15480.rs stdout ----
[01:05:24] diff of stderr:
[01:05:24] 
[01:05:24] 1 error[E0597]: borrowed value does not live long enough
[01:05:24] 3    |
[01:05:24] 3    |
[01:05:24] - LL |         &id(3)
[01:05:24] -    |          ^^^^^ temporary valsting" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/span/issue-15480/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/span/issue-15480/auxiliary" "-A" "unused"
[01:05:24] ------------------------------------------
[01:05:24] 
[01:05:24] ------------------------------------------
[01:05:24] stderr:
[01:05:24] stderr:
[01:05:24] ------------------------------------------
[01:05:24] {"message":"borrowed value does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n