\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":266,"byte_end":268,"line_start":16,"line_end":16,"column_start":9,"column_end":11,"is_primary":true,"text":[{"text":"    y = &x; //~ ERROR","highlight_start":9,"highlight_end":11}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":280,"byte_end":281,"line_start":17,"line_end":17,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"`x` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":228,"byte_end":256,"line_start":15,"line_end":15,"column_start":17,"column_end":45,"is_primary":false,"text":[{"text":"    let (y, z): (&'static u32, &'static u32);","highlight_start":17,"highlight_end":45}],"label":"type annotation requires that `x` is borrowed for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `x` does not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:16:9\n   |\nLL |     let (y, z): (&'static u32, &'static u32);\n   |                 ---------------------------- type annotation requires that `x` is borrowed for `'static`\nLL |     y = &x; //~ ERROR\n   |         ^^ borrowed value does not live long enough\nLL | }\n   | - `x` dropped here while still borrowed\n\n"}
[01:05:24] {"message":"`x` does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n