\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/coercion/coerce-expect-unsized-ascribed.rs","byte_start":664,"byte_end":708,"line_start":15,"line_end":15,"column_start":13,"column_end":57,"is_primary":true,"text":[{"text":"    let _ = box match true { true => 'a', false => 'b' }: Box<Debug>; //~ ERROR mismatched types","highlight_start":13,"highlight_end":57}],"label":"expected trait std::fmt::Debug, found char","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `std::boxed::Box<dyn std::fmt::Debug>`\n   found type `std::boxed::Box<char>`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/coercion/coerce-expect-unsized-ascribed.rs:15:13\n   |\nLL |     let _ = box match true { true => 'a', false => 'b' }: Box<Debug>; //~ ERROR mismatched types\n   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected trait std::fmt::Debug, found char\n   |\n   = note: expected type `std::boxed::Box<dyn std::fmt::Debug>`\n              found type `std::boxed::Box<char>`\n\n"}
[01:05:24] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n