\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/coercion/coerce-expect-unsized-ascribed.rs","byte_start":1340,"byte_end":1363,"line_start":26,"line_end":26,"column_start":13,"column_end":36,"is_primary":true,"text":[{"text":"    let _ = Box::new(|x| (x as u8)): Box<Fn(i32) -> _>; //~ ERROR mismatched types","highlight_start":13,"highlight_end":36}],"label":"expected trait std::ops::Fn, found closure","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected type `std::boxed::Box<dyn std::ops::Fn(i32) -> _>`\n   found type `std::boxed::Box<[closure@/checkout/src/test/ui/coercion/coerce-expect-unsized-ascribed.rs:26:22: 26:35]>`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0308]: mismatched types\n  --> /checkout/src/test/ui/coercion/coerce-expect-unsized-ascribed.rs:26:13\n   |\nLL |     let _ = Box::new(|x| (x as u8)): Box<Fn(i32) -> _>; //~ ERROR mismatched types\n   |             ^^^^^^^^^^^^^^^^^^^^^^^ expected trait std::ops::Fn, found closure\n   |\n   = note: expected type `std::boxed::Box<dyn std::ops::Fn(i32) -> _>`\n              found type `std::boxed::Box<[closure@/checkout/src/test/ui/coercion/coerce-expect-unsized-ascribed.rs:26:22: 26:35]>`\n\n"}
[01:05:24] {"message":"mismatched types","code":{"code":"E0308","explanation":"\nThis error occurs when the compiler was unable to infer the concrete type of a\nvariable. It can occur for several cases, the most common of which is a\nmismatch in the expected type that the compiler inferred for a variable's\ninitializing expression, and the actual type explicitly assigned to the\nvariable.\n\nFor example:\n\n