\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1035,"byte_end":1037,"line_start":53,"line_end":53,"column_start":10,"column_end":12,"is_primary":true,"text":[{"text":"    _y = &x; //~ ERROR","highlight_start":10,"highlight_end":12}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1049,"byte_end":1050,"line_start":54,"line_end":54,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"`x` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1006,"byte_end":1024,"line_start":52,"line_end":52,"column_start":36,"column_end":54,"is_primary":false,"text":[{"text":"    let Single2 { value: mut _y }: Single2<StaticU32>;","highlight_start":36,"highlight_end":54}],"label":"type annotation requires that `x` is borrowed for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `x` does not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:53:10\n   |\nLL |     let Single2 { value: mut _y }: Single2<StaticU32>;\n   |                                    ------------------ type annotation requires that `x` is borrowed for `'static`\nLL |     _y = &x; //~ ERROR\n   |          ^^ borrowed value does not live long enough\nLL | }\n   | - `x` dropped here while still borrowed\n\n"}
[01:05:24] {"message":"`x` does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n