\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1998,"byte_end":2000,"line_start":87,"line_end":87,"column_start":69,"column_end":71,"is_primary":true,"text":[{"text":"    let Single { value: y }: Single<&'static u32> = Single { value: &x }; //~ ERROR","highlight_start":69,"highlight_end":71}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":2014,"byte_end":2015,"line_start":88,"line_end":88,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"`x` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1959,"byte_end":1979,"line_start":87,"line_end":87,"column_start":30,"column_end":50,"is_primary":false,"text":[{"text":"    let Single { value: y }: Single<&'static u32> = Single { value: &x }; //~ ERROR","highlight_start":30,"highlight_end":50}],"label":"type annotation requires that `x` is borrowed for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `x` does not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:87:69\n   |\nLL |     let Single { value: y }: Single<&'static u32> = Single { value: &x }; //~ ERROR\n   |                              --------------------                   ^^ borrowed value does not live long enough\n   |                              |\n   |                              type annotation requires that `x` is borrowed for `'static`\nLL | }\n   | - `x` dropped here while still borrowed\n\n"}
[01:05:24] {"message":"`x` does not live long enough","code":{"code":"E0597","explanation":"\nThis error occurs because a borrow was made inside a variable which has a\ngreater lifetime than the borrowed one.\n\nExample of erroneous code:\n\n