\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1221,"byte_end":1223,"line_start":63,"line_end":63,"column_start":27,"column_end":29,"is_primary":true,"text":[{"text":"    let _: &'static u32 = &x; //~ ERROR","highlight_start":27,"highlight_end":29}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1625,"byte_end":1626,"line_start":73,"line_end":73,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"}","highlight_start":1,"highlight_end":2}],"label":"`x` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/user-annotations/patterns.rs","byte_start":1206,"byte_end":1218,"line_start":63,"line_end":63,"column_start":12,"column_end":24,"is_primary":false,"text":[{"text":"    let _: &'static u32 = &x; //~ ERROR","highlight_start":12,"highlight_end":24}],"label":"type annotation requires that `x` is borrowed for `'static`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0597]: `x` does not live long enough\n  --> /checkout/src/test/ui/nll/user-annotations/patterns.rs:63:27\n   |\nLL |     let _: &'static u32 = &x; //~ ERROR\n   |            ------------   ^^ borrowed value does not live long enough\n   |            |\n   |            type annotation requires that `x` is borrowed for `'static`\n...\nLL | }\n   | - `x` dropped here while still borrowed\n\n"}
[01:05:24] {"message":"temporary value dropped while borrowed","code":{"code":"E0716","explanation":"\nThis error indicates that a temporary value is being dropped\nwhile a borrow is still in active use.\n\nErroneous code example:\n\n