\n\nStatics are shared everywhere, and if they refer to mutable data one might\nviolate memory safety since holding multiple mutable references to shared data\nis not allowed.\n\nIf you really want global mutable state, try using `static mut` or a global\n`UnsafeCell`.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0388.rs","byte_start":67,"byte_end":73,"line_start":4,"line_end":4,"column_start":30,"column_end":36,"is_primary":true,"text":[{"text":"const CR: &'static mut i32 = &mut C; //~ ERROR E0017","highlight_start":30,"highlight_end":36}],"label":"constants require immutable values","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0017]: references in constants may only refer to immutable values\n  --> /checkout/src/test/ui/error-codes/E0388.rs:4:30\n   |\nLL | const CR: &'static mut i32 = &mut C; //~ ERROR E0017\n   |                              ^^^^^^ constants require immutable values\n\n"}
[01:29:35] {"message":"references in statics may only refer to immutable values","code":{"code":"E0017","explanation":"\nReferences in statics and constants may only refer to immutable values.\nErroneous code example:\n\n