\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/min_const_fn/min_const_fn.rs","byte_start":5658,"byte_end":5678,"line_start":134,"line_end":134,"column_start":23,"column_end":43,"is_primary":true,"text":[{"text":"const fn no_rpit() -> impl std::fmt::Debug {} //~ ERROR `impl Trait` in const fn is unstable","highlight_start":23,"highlight_end":43}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(const_fn)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0723]: `impl Trait` in const fn is unstable (see issue #57563)\n  --> /checkout/src/test/ui/consts/min_const_fn/min_const_fn.rs:134:23\n   |\nLL | const fn no_rpit() -> impl std::fmt::Debug {} //~ ERROR `impl Trait` in const fn is unstable\n   |                       ^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: add #![feature(const_fn)] to the crate attributes to enable\n\n"}
[01:20:51] {"message":"trait bounds other than `Sized` on const fn parameters are unstable (see issue #57563)","code":{"code":"E0723","explanation":"\nAn feature unstable in `const` contexts was used.\n\nErroneous code example:\n\n