\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-41974.rs","byte_start":51,"byte_end":80,"line_start":7,"line_end":7,"column_start":1,"column_end":30,"is_primary":true,"text":[{"text":"impl<T> Drop for T where T: A { //~ ERROR E0119","highlight_start":1,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"conflicting implementation in crate `alloc`:\n- impl<T, A> std::ops::Drop for std::boxed::Box<T, A>\n  where A: std::default::Default, A: std::alloc::Alloc, T: ?Sized;","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"downstream crates may implement trait `A` for type `std::boxed::Box<_, _>`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0119]: conflicting implementations of trait `std::ops::Drop` for type `std::boxed::Box<_, _>`:\n  --> /checkout/src/test/ui/issues/issue-41974.rs:7:1\n   |\nLL | impl<T> Drop for T where T: A { //~ ERROR E0119\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = note: conflicting implementation in crate `alloc`:\n           - impl<T, A> std::ops::Drop for std::boxed::Box<T, A>\n             where A: std::default::Default, A: std::alloc::Alloc, T: ?Sized;\n   = note: downstream crates may implement trait `A` for type `std::boxed::Box<_, _>`\n\n"}
[01:01:00] {"message":"the Drop trait may only be implemented on structures","code":{"code":"E0120","explanation":"\nAn attempt was made to implement Drop on a trait, which is not allowed: only\nstructs and enums can implement Drop. An example causing this error:\n\n