\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-46101.rs","byte_start":53,"byte_end":66,"line_start":3,"line_end":3,"column_start":10,"column_end":23,"is_primary":true,"text":[{"text":"#[derive(Foo::Anything)] //~ ERROR failed to resolve: partially resolved path in a derive macro","highlight_start":10,"highlight_end":23}],"label":"partially resolved path in a derive macro","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0433]: failed to resolve: partially resolved path in a derive macro\n  --> /checkout/src/test/ui/issues/issue-46101.rs:3:10\n   |\nLL | #[derive(Foo::Anything)] //~ ERROR failed to resolve: partially resolved path in a derive macro\n   |          ^^^^^^^^^^^^^ partially resolved path in a derive macro\n\n"}
[01:03:56] {"message":"`main` function not found in crate `issue_46101`","code":{"code":"E0601","explanation":"\nNo `main` function was found in a binary crate. To fix this error, add a\n`main` function. For example:\n\n