\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/book/ch04-02-references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/borrowck-report-with-custom-diagnostic.rs","byte_start":554,"byte_end":560,"line_start":21,"line_end":21,"column_start":21,"column_end":27,"is_primary":true,"text":[{"text":"            let z = &mut x; //~ ERROR cannot borrow","highlight_start":21,"highlight_end":27}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-report-with-custom-diagnostic.rs","byte_start":484,"byte_end":486,"line_start":19,"line_end":19,"column_start":21,"column_end":23,"is_primary":false,"text":[{"text":"            let y = &x;","highlight_start":21,"highlight_end":23}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-report-with-custom-diagnostic.rs","byte_start":667,"byte_end":668,"line_start":24,"line_end":24,"column_start":13,"column_end":14,"is_primary":false,"text":[{"text":"            y.use_ref();","highlight_start":13,"highlight_end":14}],"label":"immutable borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `x` as mutable because it is also borrowed as immutable\n  --> /checkout/src/test/ui/borrowck/borrowck-report-with-custom-diagnostic.rs:21:21\n   |\nLL |             let y = &x;\n   |                     -- immutable borrow occurs here\nLL |             //~^ immutable borrow occurs here\nLL |             let z = &mut x; //~ ERROR cannot borrow\n   |                     ^^^^^^ mutable borrow occurs here\n...\nLL |             y.use_ref();\n   |             - immutable borrow later used here\n\n"}
[01:38:31] {"message":"cannot borrow `x` as mutable more than once at a time","code":{"code":"E0499","explanation":"\nA variable was borrowed as mutable more than once. Erroneous code example:\n\n