\n"},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/nll/issue-55850.rs","byte_start":583,"byte_end":584,"line_start":28,"line_end":28,"column_start":16,"column_end":17,"is_primary":true,"text":[{"text":"        yield &s[..] //~ ERROR `s` does not live long enough [E0597]","highlight_start":16,"highlight_end":17}],"label":"borrowed value does not live long enough","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/nll/issue-55850.rs","byte_start":641,"byte_end":642,"line_start":29,"line_end":29,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    })","highlight_start":5,"highlight_end":6}],"label":"`s` dropped here while still borrowed","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"this error has been downgraded to a warning for backwards compatibility with previous releases","code":null,"level":"warning","spans":[],"children":[],"rendered":null},{"message":"this represents potential undefined behavior in your code and this warning will become a hard error in the future","code":null,"level":"warning","spans":[],"children":[],"rendered":null}],"rendered":"warning[E0597]: `s` does not live long enough\n  --> /checkout/src/test/ui/nll/issue-55850.rs:28:16\n   |\nLL |         yield &s[..] //~ ERROR `s` does not live long enough [E0597]\n   |                ^ borrowed value does not live long enough\nLL |     })\n   |     - `s` dropped here while still borrowed\n   |\n   = warning: this error has been downgraded to a warning for backwards compatibility with previous releases\n   = warning: this represents potential undefined behavior in your code and this warning will become a hard error in the future\n\n"}
[01:38:31] {"message":"borrow may still be in use when generator yields","code":{"code":"E0626","explanation":"\nThis error occurs because a borrow in a generator persists across a\nyield point.\n\n