\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/regions/regions-ref-in-fn-arg.rs","byte_start":110,"byte_end":111,"line_start":5,"line_end":5,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    x //~^ ERROR borrowed value does not live long enough","highlight_start":5,"highlight_end":6}],"label":"returns a value referencing data owned by the current function","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/regions/regions-ref-in-fn-arg.rs","byte_start":63,"byte_end":72,"line_start":4,"line_end":4,"column_start":13,"column_end":22,"is_primary":false,"text":[{"text":"fn arg_item(box ref x: Box<isize>) -> &'static isize {","highlight_start":13,"highlight_end":22}],"label":"function parameter borrowed here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0515]: cannot return value referencing function parameter\n  --> /checkout/src/test/ui/regions/regions-ref-in-fn-arg.rs:5:5\n   |\nLL | fn arg_item(box ref x: Box<isize>) -> &'static isize {\n   |             --------- function parameter borrowed here\nLL |     x //~^ ERROR borrowed value does not live long enough\n   |     ^ returns a value referencing data owned by the current function\n\n"}
[01:38:31] {"message":"cannot return value referencing function parameter","code":{"code":"E0515","explanation":"\nCannot return value that references local variable\n\nLocal variables, function parameters and temporaries are all dropped before the\nend of the function body. So a reference to them cannot be returned.\n\n