\n\nFor more information on the rust ownership system, take a look at\nhttps://doc.rust-lang.org/book/ch04-02-references-and-borrowing.html.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/borrowck/borrowck-report-with-custom-diagnostic.rs","byte_start":232,"byte_end":234,"line_start":8,"line_end":8,"column_start":13,"column_end":15,"is_primary":true,"text":[{"text":"    let z = &x; //~ ERROR cannot borrow","highlight_start":13,"highlight_end":15}],"label":"immutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-report-with-custom-diagnostic.rs","byte_start":176,"byte_end":182,"line_start":6,"line_end":6,"column_start":13,"column_end":19,"is_primary":false,"text":[{"text":"    let y = &mut x;","highlight_start":13,"highlight_end":19}],"label":"mutable borrow occurs here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/borrowck/borrowck-report-with-custom-diagnostic.rs","byte_start":319,"byte_end":320,"line_start":11,"line_end":11,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    y.use_mut();","highlight_start":5,"highlight_end":6}],"label":"mutable borrow later used here","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `x` as immutable because it is also borrowed as mutable\n  --> /checkout/src/test/ui/borrowck/borrowck-report-with-custom-diagnostic.rs:8:13\n   |\nLL |     let y = &mut x;\n   |             ------ mutable borrow occurs here\nLL |     //~^ mutable borrow occurs here\nLL |     let z = &x; //~ ERROR cannot borrow\n   |             ^^ immutable borrow occurs here\n...\nLL |     y.use_mut();\n   |     - mutable borrow later used here\n\n"}
[01:38:31] {"message":"cannot borrow `x` as mutable because it is also borrowed as immutable","code":{"code":"E0502","explanation":"\nThis error indicates that you are trying to borrow a variable as mutable when it\nhas already been borrowed as immutable.\n\nExample of erroneous code:\n\n