\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":106,"byte_end":117,"line_start":5,"line_end":5,"column_start":21,"column_end":32,"is_primary":true,"text":[{"text":"            pub use self::$name;","highlight_start":21,"highlight_end":32}],"label":"`issue_56411_aux` reimported here","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":295,"byte_end":320,"line_start":13,"line_end":13,"column_start":1,"column_end":26,"is_primary":false,"text":[{"text":"import!(issue_56411_aux);","highlight_start":1,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"import!","def_site_span":{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":0,"byte_end":293,"line_start":1,"line_end":11,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! import {","highlight_start":1,"highlight_end":22},{"text":"    ( $($name:ident),* ) => {","highlight_start":1,"highlight_end":30},{"text":"        $(","highlight_start":1,"highlight_end":11},{"text":"            mod $name;","highlight_start":1,"highlight_end":23},{"text":"            pub use self::$name;","highlight_start":1,"highlight_end":33},{"text":"            //~^ ERROR the name `issue_56411_aux` is defined multiple times","highlight_start":1,"highlight_end":76},{"text":"            //~| ERROR `issue_56411_aux` is private, and cannot be re-exported","highlight_start":1,"highlight_end":79},{"text":"","highlight_start":1,"highlight_end":1},{"text":"        )*","highlight_start":1,"highlight_end":11},{"text":"    }","highlight_start":1,"highlight_end":6},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}},{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":75,"byte_end":85,"line_start":4,"line_end":4,"column_start":13,"column_end":23,"is_primary":false,"text":[{"text":"            mod $name;","highlight_start":13,"highlight_end":23}],"label":"previous definition of the module `issue_56411_aux` here","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":295,"byte_end":320,"line_start":13,"line_end":13,"column_start":1,"column_end":26,"is_primary":false,"text":[{"text":"import!(issue_56411_aux);","highlight_start":1,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"import!","def_site_span":{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":0,"byte_end":293,"line_start":1,"line_end":11,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! import {","highlight_start":1,"highlight_end":22},{"text":"    ( $($name:ident),* ) => {","highlight_start":1,"highlight_end":30},{"text":"        $(","highlight_start":1,"highlight_end":11},{"text":"            mod $name;","highlight_start":1,"highlight_end":23},{"text":"            pub use self::$name;","highlight_start":1,"highlight_end":33},{"text":"            //~^ ERROR the name `issue_56411_aux` is defined multiple times","highlight_start":1,"highlight_end":76},{"text":"            //~| ERROR `issue_56411_aux` is private, and cannot be re-exported","highlight_start":1,"highlight_end":79},{"text":"","highlight_start":1,"highlight_end":1},{"text":"        )*","highlight_start":1,"highlight_end":11},{"text":"    }","highlight_start":1,"highlight_end":6},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}},{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":106,"byte_end":117,"line_start":5,"line_end":5,"column_start":21,"column_end":32,"is_primary":true,"text":[{"text":"            pub use self::$name;","highlight_start":21,"highlight_end":32}],"label":"you can use `as` to change the binding name of the import","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":295,"byte_end":320,"line_start":13,"line_end":13,"column_start":1,"column_end":26,"is_primary":false,"text":[{"text":"import!(issue_56411_aux);","highlight_start":1,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"import!","def_site_span":{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":0,"byte_end":293,"line_start":1,"line_end":11,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! import {","highlight_start":1,"highlight_end":22},{"text":"    ( $($name:ident),* ) => {","highlight_start":1,"highlight_end":30},{"text":"        $(","highlight_start":1,"highlight_end":11},{"text":"            mod $name;","highlight_start":1,"highlight_end":23},{"text":"            pub use self::$name;","highlight_start":1,"highlight_end":33},{"text":"            //~^ ERROR the name `issue_56411_aux` is defined multiple times","highlight_start":1,"highlight_end":76},{"text":"            //~| ERROR `issue_56411_aux` is private, and cannot be re-exported","highlight_start":1,"highlight_end":79},{"text":"","highlight_start":1,"highlight_end":1},{"text":"        )*","highlight_start":1,"highlight_end":11},{"text":"    }","highlight_start":1,"highlight_end":6},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[{"message":"`issue_56411_aux` must be defined only once in the type namespace of this module","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0255]: the name `issue_56411_aux` is defined multiple times\n  --> /checkout/src/test/ui/issues/issue-56411.rs:5:21\n   |\nLL |             mod $name;\n   |             ---------- previous definition of the module `issue_56411_aux` here\nLL |             pub use self::$name;\n   |                     ^^^^^^^^^^^\n   |                     |\n   |                     `issue_56411_aux` reimported here\n   |                     you can use `as` to change the binding name of the import\n...\nLL | import!(issue_56411_aux);\n   | ------------------------- in this macro invocation\n   |\n   = note: `issue_56411_aux` must be defined only once in the type namespace of this module\n\n"}
[01:18:31] {"message":"`issue_56411_aux` is private, and cannot be re-exported","code":{"code":"E0365","explanation":"\nPrivate modules cannot be publicly re-exported. This error indicates that you\nattempted to `pub use` a module that was not itself public.\n\nErroneous code example:\n\n