\n\nPlease be sure that a file corresponding to the module exists. If you\nwant to use a module named `file_that_doesnt_exist`, you need to have a file\nnamed `file_that_doesnt_exist.rs` or `file_that_doesnt_exist/mod.rs` in the\nsame directory.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":303,"byte_end":318,"line_start":13,"line_end":13,"column_start":9,"column_end":24,"is_primary":true,"text":[{"text":"import!(issue_56411_aux);","highlight_start":9,"highlight_end":24}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"name the file either issue_56411_aux.rs or issue_56411_aux/mod.rs inside the directory \"/checkout/src/test/ui/issues\"","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0583]: file not found for module `issue_56411_aux`\n  --> /checkout/src/test/ui/issues/issue-56411.rs:13:9\n   |\nLL | import!(issue_56411_aux);\n   |         ^^^^^^^^^^^^^^^\n   |\n   = help: name the file either issue_56411_aux.rs or issue_56411_aux/mod.rs inside the directory \"/checkout/src/test/ui/issues\"\n\n"}
[01:17:40] {"message":"For more information about this error, try `rustc --explain E0583`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0583`.\n"}
[01:17:40] 
[01:17:40] ------------------------------------------
[01:17:40] 
---
[01:17:40] 54 error: unused doc comment
[01:17:40] -   --> $DIR/useless_comment.rs:28:5
[01:17:40] +   --> $DIR/useless-comment.rs:28:5
[01:17:40] 56    |
[01:17:40] 57 LL |     #[doc = "foo"]
[01:17:40] 
[01:17:40] 61    |     - rustdoc does not generate documentation for expressions
[01:17:40] 62 
[01:17:40] 63 error: unused doc comment
[01:17:40] 63 error: unused doc comment
[01:17:40] -   --> $DIR/useless_comment.rs:29:5
[01:17:40] +   --> $DIR/useless-comment.rs:29:5
[01:17:40] 65    |
[01:17:40] 66 LL |     #[doc = "bar"]
[01:17:40] 
[01:17:40] 69    |     - rustdoc does not generate documentation for expressions
[01:17:40] 70 
[01:17:40] 71 error: unused doc comment
---
[01:17:40] 81 error: unused doc comment
[01:17:40] -   --> $DIR/useless_comment.rs:35:13
[01:17:40] +   --> $DIR/useless-comment.rs:35:13
[01:17:40] 83    |
[01:17:40] 84 LL |     let x = /** comment */ 47;
[01:17:40] 85    |             ^^^^^^^^^^^^^^ -- rustdoc does not generate documentation for expressions
[01:17:40] 86 
[01:17:40] 87 error: unused doc comment
[01:17:40] -   --> $DIR/useless_comment.rs:37:5
[01:17:40] +   --> $DIR/useless-comment.rs:37:5
[01:17:40] +   --> $DIR/useless-comment.rs:37:5
[01:17:40] 89    |
[01:17:40] 90 LL |       /// dox
[01:17:40] 91    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[01:17:40] 
[01:17:40] 
[01:17:40] The actual stderr differed from the expected stderr.
[01:17:40] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/useless-comment/useless-comment.stderr
[01:17:40] To update references, rerun the tests and pass the `--bless` flag
[01:17:40] To only update this specific test, also pass `--test-args useless-comment.rs`
[01:17:40] error: 1 errors occurred comparing output.
[01:17:40] status: exit code: 1
[01:17:40] status: exit code: 1
[01:17:40] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/useless-comment.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/useless-comment/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/useless-comment/auxiliary" "-A" "unused"
[01:17:40] ------------------------------------------
[01:17:40] 
[01:17:40] ------------------------------------------
[01:17:40] stderr:
[01:17:40] stderr:
[01:17:40] ------------------------------------------
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":138,"byte_end":145,"line_start":10,"line_end":10,"column_start":1,"column_end":8,"is_primary":false,"text":[{"text":"mac!();","highlight_start":1,"highlight_end":8}],"label":"rustdoc does not generate documentation for macro expansions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":101,"byte_end":137,"line_start":9,"line_end":9,"column_start":1,"column_end":37,"is_primary":true,"text":[{"text":"/// foo //~ ERROR unused doc comment","highlight_start":1,"highlight_end":37}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":43,"byte_end":62,"line_start":3,"line_end":3,"column_start":9,"column_end":28,"is_primary":true,"text":[{"text":"#![deny(unused_doc_comments)]","highlight_start":9,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"to document an item produced by a macro, the macro must produce the documentation as part of its expansion","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:9:1\n   |\nLL | /// foo //~ ERROR unused doc comment\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL | mac!();\n   | ------- rustdoc does not generate documentation for macro expansions\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/useless-comment.rs:3:9\n   |\nLL | #![deny(unused_doc_comments)]\n   |         ^^^^^^^^^^^^^^^^^^^\n   = help: to document an item produced by a macro, the macro must produce the documentation as part of its expansion\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":201,"byte_end":212,"line_start":14,"line_end":14,"column_start":5,"column_end":16,"is_primary":false,"text":[{"text":"    let x = 12;","highlight_start":5,"highlight_end":16}],"label":"rustdoc does not generate documentation for statements","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":162,"byte_end":196,"line_start":13,"line_end":13,"column_start":5,"column_end":39,"is_primary":true,"text":[{"text":"    /// a //~ ERROR unused doc comment","highlight_start":5,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:13:5\n   |\nLL |     /// a //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |     let x = 12;\n   |     ----------- rustdoc does not generate documentation for statements\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":303,"byte_end":394,"line_start":19,"line_end":23,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    match x {","highlight_start":5,"highlight_end":14},{"text":"        /// c //~ ERROR unused doc comment","highlight_start":1,"highlight_end":43},{"text":"        1 => {},","highlight_start":1,"highlight_end":17},{"text":"        _ => {}","highlight_start":1,"highlight_end":16},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":218,"byte_end":298,"line_start":16,"line_end":18,"column_start":5,"column_end":23,"is_primary":true,"text":[{"text":"    /// multi-line //~ unused doc comment","highlight_start":5,"highlight_end":42},{"text":"    /// doc comment","highlight_start":1,"highlight_end":20},{"text":"    /// that is unused","highlight_start":1,"highlight_end":23}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:16:5\n   |\nLL | /     /// multi-line //~ unused doc comment\nLL | |     /// doc comment\nLL | |     /// that is unused\n   | |______________________^\nLL | /     match x {\nLL | |         /// c //~ ERROR unused doc comment\nLL | |         1 => {},\nLL | |         _ => {}\nLL | |     }\n   | |_____- rustdoc does not generate documentation for expressions\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":364,"byte_end":371,"line_start":21,"line_end":21,"column_start":9,"column_end":16,"is_primary":false,"text":[{"text":"        1 => {},","highlight_start":9,"highlight_end":16}],"label":"rustdoc does not generate documentation for match arms","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":321,"byte_end":355,"line_start":20,"line_end":20,"column_start":9,"column_end":43,"is_primary":true,"text":[{"text":"        /// c //~ ERROR unused doc comment","highlight_start":9,"highlight_end":43}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:20:9\n   |\nLL |         /// c //~ ERROR unused doc comment\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |         1 => {},\n   |         ------- rustdoc does not generate documentation for match arms\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":441,"byte_end":450,"line_start":26,"line_end":26,"column_start":5,"column_end":14,"is_primary":false,"text":[{"text":"    unsafe {}","highlight_start":5,"highlight_end":14}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":400,"byte_end":436,"line_start":25,"line_end":25,"column_start":5,"column_end":41,"is_primary":true,"text":[{"text":"    /// foo //~ ERROR unused doc comment","highlight_start":5,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:25:5\n   |\nLL |     /// foo //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |     unsafe {}\n   |     --------- rustdoc does not generate documentation for expressions\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":552,"byte_end":553,"line_start":30,"line_end":30,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    3;","highlight_start":5,"highlight_end":6}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":456,"byte_end":470,"line_start":28,"line_end":28,"column_start":5,"column_end":19,"is_primary":true,"text":[{"text":"    #[doc = \"foo\"] //~ ERROR unused doc comment","highlight_start":5,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:28:5\n   |\nLL |     #[doc = \"foo\"] //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^\nLL |     #[doc = \"bar\"] //~ ERROR unused doc comment\nLL |     3;\n   |     - rustdoc does not generate documentation for expressions\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":552,"byte_end":553,"line_start":30,"line_end":30,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    3;","highlight_start":5,"highlight_end":6}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":504,"byte_end":518,"line_start":29,"line_end":29,"column_start":5,"column_end":19,"is_primary":true,"text":[{"text":"    #[doc = \"bar\"] //~ ERROR unused doc comment","highlight_start":5,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:29:5\n   |\nLL |     #[doc = \"bar\"] //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^\nLL |     3;\n   |     - rustdoc does not generate documentation for expressions\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":601,"byte_end":608,"line_start":33,"line_end":33,"column_start":5,"column_end":12,"is_primary":false,"text":[{"text":"    mac!();","highlight_start":5,"highlight_end":12}],"label":"rustdoc does not generate documentation for macro expansions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":560,"byte_end":596,"line_start":32,"line_end":32,"column_start":5,"column_end":41,"is_primary":true,"text":[{"text":"    /// bar //~ ERROR unused doc comment","highlight_start":5,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"to document an item produced by a macro, the macro must produce the documentation as part of its expansion","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:32:5\n   |\nLL |     /// bar //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |     mac!();\n   |     ------- rustdoc does not generate documentation for macro expansions\n   |\n   = help: to document an item produced by a macro, the macro must produce the documentation as part of its expansion\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":637,"byte_end":639,"line_start":35,"line_end":35,"column_start":28,"column_end":30,"is_primary":false,"text":[{"text":"    let x = /** comment */ 47; //~ ERROR unused doc comment","highlight_start":28,"highlight_end":30}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":622,"byte_end":636,"line_start":35,"line_end":35,"column_start":13,"column_end":27,"is_primary":true,"text":[{"text":"    let x = /** comment */ 47; //~ ERROR unused doc comment","highlight_start":13,"highlight_end":27}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:35:13\n   |\nLL |     let x = /** comment */ 47; //~ ERROR unused doc comment\n   |             ^^^^^^^^^^^^^^ -- rustdoc does not generate documentation for expressions\n\n"}
[01:17:40] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":716,"byte_end":724,"line_start":38,"line_end":40,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    {","highlight_start":5,"highlight_end":6},{"text":"","highlight_start":1,"highlight_end":1},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":675,"byte_end":711,"line_start":37,"line_end":37,"column_start":5,"column_end":41,"is_primary":true,"text":[{"text":"    /// dox //~ ERROR unused doc comment","highlight_start":5,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:37:5\n   |\nLL |       /// dox //~ ERROR unused doc comment\n   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL | /     {\nLL | |\nLL | |     }\n   | |_____- rustdoc does not generate documentation for expressions\n\n"}
[01:17:40] 
[01:17:40] ------------------------------------------
[01:17:40] 
[01:17:40] thread '[ui] ui/useless-comment.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3325:9
---
[01:17:40] 
[01:17:40] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:496:22
[01:17:40] 
[01:17:40] 
[01:17:40] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[01:17:40] 
[01:17:40] 
[01:17:40] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[01:17:40] Build completed unsuccessfully in 0:04:40
[01:17:40] Build completed unsuccessfully in 0:04:40
[01:17:40] Makefile:48: recipe for target 'check' failed
[01:17:40] make: *** [check] Error 1
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:0fb0446a
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Tue Mar 12 20:38:46 UTC 2019
---
travis_time:end:01a85ba0:start=1552423127892868562,finish=1552423127898307734,duration=5439172
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:1c737a78
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb --batch -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:03d27720
travis_time:start:03d27720
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:390ae381
$ dmesg | grep -i kill
