\n\nSee the 'Use Declarations' section of the reference for more information\non this topic:\n\nhttps://doc.rust-lang.org/reference.html#use-declarations\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":106,"byte_end":117,"line_start":5,"line_end":5,"column_start":21,"column_end":32,"is_primary":true,"text":[{"text":"            pub use self::$name;","highlight_start":21,"highlight_end":32}],"label":"re-export of private `issue_56411_aux`","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":295,"byte_end":320,"line_start":13,"line_end":13,"column_start":1,"column_end":26,"is_primary":false,"text":[{"text":"import!(issue_56411_aux);","highlight_start":1,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"import!","def_site_span":{"file_name":"/checkout/src/test/ui/issues/issue-56411.rs","byte_start":0,"byte_end":293,"line_start":1,"line_end":11,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! import {","highlight_start":1,"highlight_end":22},{"text":"    ( $($name:ident),* ) => {","highlight_start":1,"highlight_end":30},{"text":"        $(","highlight_start":1,"highlight_end":11},{"text":"            mod $name;","highlight_start":1,"highlight_end":23},{"text":"            pub use self::$name;","highlight_start":1,"highlight_end":33},{"text":"            //~^ ERROR the name `issue_56411_aux` is defined multiple times","highlight_start":1,"highlight_end":76},{"text":"            //~| ERROR `issue_56411_aux` is private, and cannot be re-exported","highlight_start":1,"highlight_end":79},{"text":"","highlight_start":1,"highlight_end":1},{"text":"        )*","highlight_start":1,"highlight_end":11},{"text":"    }","highlight_start":1,"highlight_end":6},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[{"message":"consider declaring type or module `issue_56411_aux` with `pub`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0365]: `issue_56411_aux` is private, and cannot be re-exported\n  --> /checkout/src/test/ui/issues/issue-56411.rs:5:21\n   |\nLL |             pub use self::$name;\n   |                     ^^^^^^^^^^^ re-export of private `issue_56411_aux`\n...\nLL | import!(issue_56411_aux);\n   | ------------------------- in this macro invocation\n   |\n   = note: consider declaring type or module `issue_56411_aux` with `pub`\n\n"}
[01:18:31] {"message":"aborting due to 2 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 2 previous errors\n\n"}
[01:18:31] {"message":"Some errors occurred: E0255, E0365.","code":null,"level":"","spans":[],"children":[],"rendered":"Some errors occurred: E0255, E0365.\n"}
[01:18:31] 
[01:18:31] ------------------------------------------
[01:18:31] 
[01:18:31] thread '[ui] ui/issues/issue-56411.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3325:9
---
[01:18:31] 63 error: unused doc comment
[01:18:31] -   --> $DIR/useles-comment.rs:29:5
[01:18:31] +   --> $DIR/useless-comment.rs:29:5
[01:18:31] 65    |
[01:18:31] 66 LL |     #[doc = "bar"]
[01:18:31] 
[01:18:31] 
[01:18:31] The actual stderr differed from the expected stderr.
[01:18:31] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/useless-comment/useless-comment.stderr
[01:18:31] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/useless-comment/useless-comment.stderr
[01:18:31] To update references, rerun the tests and pass the `--bless` flag
[01:18:31] To only update this specific test, also pass `--test-args useless-comment.rs`
[01:18:31] error: 1 errors occurred comparing output.
[01:18:31] status: exit code: 1
[01:18:31] status: exit code: 1
[01:18:31] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/useless-comment.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/useless-comment/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/useless-comment/auxiliary" "-A" "unused"
[01:18:31] ------------------------------------------
[01:18:31] 
[01:18:31] ------------------------------------------
[01:18:31] stderr:
[01:18:31] stderr:
[01:18:31] ------------------------------------------
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":138,"byte_end":145,"line_start":10,"line_end":10,"column_start":1,"column_end":8,"is_primary":false,"text":[{"text":"mac!();","highlight_start":1,"highlight_end":8}],"label":"rustdoc does not generate documentation for macro expansions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":101,"byte_end":137,"line_start":9,"line_end":9,"column_start":1,"column_end":37,"is_primary":true,"text":[{"text":"/// foo //~ ERROR unused doc comment","highlight_start":1,"highlight_end":37}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":43,"byte_end":62,"line_start":3,"line_end":3,"column_start":9,"column_end":28,"is_primary":true,"text":[{"text":"#![deny(unused_doc_comments)]","highlight_start":9,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"to document an item produced by a macro, the macro must produce the documentation as part of its expansion","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:9:1\n   |\nLL | /// foo //~ ERROR unused doc comment\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL | mac!();\n   | ------- rustdoc does not generate documentation for macro expansions\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/useless-comment.rs:3:9\n   |\nLL | #![deny(unused_doc_comments)]\n   |         ^^^^^^^^^^^^^^^^^^^\n   = help: to document an item produced by a macro, the macro must produce the documentation as part of its expansion\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":201,"byte_end":212,"line_start":14,"line_end":14,"column_start":5,"column_end":16,"is_primary":false,"text":[{"text":"    let x = 12;","highlight_start":5,"highlight_end":16}],"label":"rustdoc does not generate documentation for statements","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":162,"byte_end":196,"line_start":13,"line_end":13,"column_start":5,"column_end":39,"is_primary":true,"text":[{"text":"    /// a //~ ERROR unused doc comment","highlight_start":5,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:13:5\n   |\nLL |     /// a //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |     let x = 12;\n   |     ----------- rustdoc does not generate documentation for statements\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":303,"byte_end":394,"line_start":19,"line_end":23,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    match x {","highlight_start":5,"highlight_end":14},{"text":"        /// c //~ ERROR unused doc comment","highlight_start":1,"highlight_end":43},{"text":"        1 => {},","highlight_start":1,"highlight_end":17},{"text":"        _ => {}","highlight_start":1,"highlight_end":16},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":218,"byte_end":298,"line_start":16,"line_end":18,"column_start":5,"column_end":23,"is_primary":true,"text":[{"text":"    /// multi-line //~ unused doc comment","highlight_start":5,"highlight_end":42},{"text":"    /// doc comment","highlight_start":1,"highlight_end":20},{"text":"    /// that is unused","highlight_start":1,"highlight_end":23}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:16:5\n   |\nLL | /     /// multi-line //~ unused doc comment\nLL | |     /// doc comment\nLL | |     /// that is unused\n   | |______________________^\nLL | /     match x {\nLL | |         /// c //~ ERROR unused doc comment\nLL | |         1 => {},\nLL | |         _ => {}\nLL | |     }\n   | |_____- rustdoc does not generate documentation for expressions\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":364,"byte_end":371,"line_start":21,"line_end":21,"column_start":9,"column_end":16,"is_primary":false,"text":[{"text":"        1 => {},","highlight_start":9,"highlight_end":16}],"label":"rustdoc does not generate documentation for match arms","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":321,"byte_end":355,"line_start":20,"line_end":20,"column_start":9,"column_end":43,"is_primary":true,"text":[{"text":"        /// c //~ ERROR unused doc comment","highlight_start":9,"highlight_end":43}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:20:9\n   |\nLL |         /// c //~ ERROR unused doc comment\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |         1 => {},\n   |         ------- rustdoc does not generate documentation for match arms\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":441,"byte_end":450,"line_start":26,"line_end":26,"column_start":5,"column_end":14,"is_primary":false,"text":[{"text":"    unsafe {}","highlight_start":5,"highlight_end":14}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":400,"byte_end":436,"line_start":25,"line_end":25,"column_start":5,"column_end":41,"is_primary":true,"text":[{"text":"    /// foo //~ ERROR unused doc comment","highlight_start":5,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:25:5\n   |\nLL |     /// foo //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |     unsafe {}\n   |     --------- rustdoc does not generate documentation for expressions\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":552,"byte_end":553,"line_start":30,"line_end":30,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    3;","highlight_start":5,"highlight_end":6}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":456,"byte_end":470,"line_start":28,"line_end":28,"column_start":5,"column_end":19,"is_primary":true,"text":[{"text":"    #[doc = \"foo\"] //~ ERROR unused doc comment","highlight_start":5,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:28:5\n   |\nLL |     #[doc = \"foo\"] //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^\nLL |     #[doc = \"bar\"] //~ ERROR unused doc comment\nLL |     3;\n   |     - rustdoc does not generate documentation for expressions\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":552,"byte_end":553,"line_start":30,"line_end":30,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    3;","highlight_start":5,"highlight_end":6}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":504,"byte_end":518,"line_start":29,"line_end":29,"column_start":5,"column_end":19,"is_primary":true,"text":[{"text":"    #[doc = \"bar\"] //~ ERROR unused doc comment","highlight_start":5,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:29:5\n   |\nLL |     #[doc = \"bar\"] //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^\nLL |     3;\n   |     - rustdoc does not generate documentation for expressions\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":601,"byte_end":608,"line_start":33,"line_end":33,"column_start":5,"column_end":12,"is_primary":false,"text":[{"text":"    mac!();","highlight_start":5,"highlight_end":12}],"label":"rustdoc does not generate documentation for macro expansions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":560,"byte_end":596,"line_start":32,"line_end":32,"column_start":5,"column_end":41,"is_primary":true,"text":[{"text":"    /// bar //~ ERROR unused doc comment","highlight_start":5,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"to document an item produced by a macro, the macro must produce the documentation as part of its expansion","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:32:5\n   |\nLL |     /// bar //~ ERROR unused doc comment\n   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |     mac!();\n   |     ------- rustdoc does not generate documentation for macro expansions\n   |\n   = help: to document an item produced by a macro, the macro must produce the documentation as part of its expansion\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":637,"byte_end":639,"line_start":35,"line_end":35,"column_start":28,"column_end":30,"is_primary":false,"text":[{"text":"    let x = /** comment */ 47; //~ ERROR unused doc comment","highlight_start":28,"highlight_end":30}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":622,"byte_end":636,"line_start":35,"line_end":35,"column_start":13,"column_end":27,"is_primary":true,"text":[{"text":"    let x = /** comment */ 47; //~ ERROR unused doc comment","highlight_start":13,"highlight_end":27}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:35:13\n   |\nLL |     let x = /** comment */ 47; //~ ERROR unused doc comment\n   |             ^^^^^^^^^^^^^^ -- rustdoc does not generate documentation for expressions\n\n"}
[01:18:31] {"message":"unused doc comment","code":{"code":"unused_doc_comments","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":716,"byte_end":724,"line_start":38,"line_end":40,"column_start":5,"column_end":6,"is_primary":false,"text":[{"text":"    {","highlight_start":5,"highlight_end":6},{"text":"","highlight_start":1,"highlight_end":1},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":"rustdoc does not generate documentation for expressions","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/useless-comment.rs","byte_start":675,"byte_end":711,"line_start":37,"line_end":37,"column_start":5,"column_end":41,"is_primary":true,"text":[{"text":"    /// dox //~ ERROR unused doc comment","highlight_start":5,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: unused doc comment\n  --> /checkout/src/test/ui/useless-comment.rs:37:5\n   |\nLL |       /// dox //~ ERROR unused doc comment\n   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nLL | /     {\nLL | |\nLL | |     }\n   | |_____- rustdoc does not generate documentation for expressions\n\n"}
[01:18:31] 
[01:18:31] ------------------------------------------
[01:18:31] 
[01:18:31] thread '[ui] ui/useless-comment.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3325:9
---
[01:18:31] 
[01:18:31] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:496:22
[01:18:31] 
[01:18:31] 
[01:18:31] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[01:18:31] 
[01:18:31] 
[01:18:31] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[01:18:31] Build completed unsuccessfully in 0:04:52
[01:18:31] Build completed unsuccessfully in 0:04:52
[01:18:31] Makefile:48: recipe for target 'check' failed
[01:18:31] make: *** [check] Error 1
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:09c079e6
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Wed Mar 13 01:15:08 UTC 2019
