\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/parser/struct-literal-restrictions-in-lamda.rs","byte_start":115,"byte_end":118,"line_start":12,"line_end":12,"column_start":14,"column_end":17,"is_primary":true,"text":[{"text":"    while || Foo { //~ ERROR expected value, found struct `Foo`","highlight_start":14,"highlight_end":17}],"label":"did you mean `(Foo { /* fields */ })`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0423]: expected value, found struct `Foo`\n  --> /checkout/src/test/ui/parser/struct-literal-restrictions-in-lamda.rs:12:14\n   |\nLL |     while || Foo { //~ ERROR expected value, found struct `Foo`\n   |              ^^^ did you mean `(Foo { /* fields */ })`?\n\n"}
[01:17:14] {"message":"no method named `hi` found for type `()` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n