\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0423.rs","byte_start":54,"byte_end":57,"line_start":4,"line_end":4,"column_start":13,"column_end":16,"is_primary":true,"text":[{"text":"    let f = Foo(); //~ ERROR E0423","highlight_start":13,"highlight_end":16}],"label":"did you mean `Foo { /* fields */ }`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"a function with a similar name exists","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/error-codes/E0423.rs","byte_start":54,"byte_end":57,"line_start":4,"line_end":4,"column_start":13,"column_end":16,"is_primary":true,"text":[{"text":"    let f = Foo(); //~ ERROR E0423","highlight_start":13,"highlight_end":16}],"label":null,"suggested_replacement":"foo","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0423]: expected function, found struct `Foo`\n  --> /checkout/src/test/ui/error-codes/E0423.rs:4:13\n   |\nLL |     let f = Foo(); //~ ERROR E0423\n   |             ^^^\n   |             |\n   |             did you mean `Foo { /* fields */ }`?\n   |             help: a function with a similar name exists: `foo`\n\n"}
[01:17:14] {"message":"expected value, found struct `S`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n