\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/parser/struct-literal-in-while.rs","byte_start":112,"byte_end":115,"line_start":12,"line_end":12,"column_start":11,"column_end":14,"is_primary":true,"text":[{"text":"    while Foo { //~ ERROR expected value, found struct `Foo`","highlight_start":11,"highlight_end":14}],"label":"did you mean `(Foo { /* fields */ })`?","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0423]: expected value, found struct `Foo`\n  --> /checkout/src/test/ui/parser/struct-literal-in-while.rs:12:11\n   |\nLL |     while Foo { //~ ERROR expected value, found struct `Foo`\n   |           ^^^ did you mean `(Foo { /* fields */ })`?\n\n"}
[01:17:14] {"message":"no method named `hi` found for type `()` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n