\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/simd-type.rs","byte_start":70,"byte_end":83,"line_start":5,"line_end":5,"column_start":1,"column_end":14,"is_primary":true,"text":[{"text":"struct empty; //~ ERROR SIMD vector cannot be empty","highlight_start":1,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0075]: SIMD vector cannot be empty\n  --> /checkout/src/test/ui/simd-type.rs:5:1\n   |\nLL | struct empty; //~ ERROR SIMD vector cannot be empty\n   | ^^^^^^^^^^^^^\n\n"}
[01:14:55] {"message":"SIMD vector should be homogeneous","code":{"code":"E0076","explanation":"\nWhen using the `#[simd]` attribute to automatically use SIMD operations in tuple\nstruct, the types in the struct must all be of the same type, or the compiler\nwill trigger this error.\n\nThis will cause an error:\n\n