\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":431,"byte_end":438,"line_start":27,"line_end":27,"column_start":12,"column_end":19,"is_primary":true,"text":[{"text":"    if let Example(_) = y { //~ ERROR expected tuple struct/variant, found enum","highlight_start":12,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try using one of the enum's variants","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":431,"byte_end":438,"line_start":27,"line_end":27,"column_start":12,"column_end":19,"is_primary":true,"text":[{"text":"    if let Example(_) = y { //~ ERROR expected tuple struct/variant, found enum","highlight_start":12,"highlight_end":19}],"label":null,"suggested_replacement":"Example::Ex","suggestion_applicability":"MaybeIncorrect","expansion":null},{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":431,"byte_end":438,"line_start":27,"line_end":27,"column_start":12,"column_end":19,"is_primary":true,"text":[{"text":"    if let Example(_) = y { //~ ERROR expected tuple struct/variant, found enum","highlight_start":12,"highlight_end":19}],"label":null,"suggested_replacement":"Example::NotEx","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0532]: expected tuple struct/variant, found enum `Example`\n  --> /checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs:27:12\n   |\nLL |     if let Example(_) = y { //~ ERROR expected tuple struct/variant, found enum\n   |            ^^^^^^^\nhelp: try using one of the enum's variants\n   |\nLL |     if let Example::Ex(_) = y { //~ ERROR expected tuple struct/variant, found enum\n   |            ^^^^^^^^^^^\nLL |     if let Example::NotEx(_) = y { //~ ERROR expected tuple struct/variant, found enum\n   |            ^^^^^^^^^^^^^^\n\n"}
[01:15:56] {"message":"expected function, found enum `Void`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n