\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":266,"byte_end":272,"line_start":21,"line_end":21,"column_start":12,"column_end":18,"is_primary":true,"text":[{"text":"    if let Option(_) = x { //~ ERROR expected tuple struct/variant, found enum","highlight_start":12,"highlight_end":18}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try using one of the enum's variants","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":266,"byte_end":272,"line_start":21,"line_end":21,"column_start":12,"column_end":18,"is_primary":true,"text":[{"text":"    if let Option(_) = x { //~ ERROR expected tuple struct/variant, found enum","highlight_start":12,"highlight_end":18}],"label":null,"suggested_replacement":"std::option::Option::None","suggestion_applicability":"MaybeIncorrect","expansion":null},{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":266,"byte_end":272,"line_start":21,"line_end":21,"column_start":12,"column_end":18,"is_primary":true,"text":[{"text":"    if let Option(_) = x { //~ ERROR expected tuple struct/variant, found enum","highlight_start":12,"highlight_end":18}],"label":null,"suggested_replacement":"std::option::Option::Some","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0532]: expected tuple struct/variant, found enum `Option`\n  --> /checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs:21:12\n   |\nLL |     if let Option(_) = x { //~ ERROR expected tuple struct/variant, found enum\n   |            ^^^^^^\nhelp: try using one of the enum's variants\n   |\nLL |     if let std::option::Option::None(_) = x { //~ ERROR expected tuple struct/variant, found enum\n   |            ^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |     if let std::option::Option::Some(_) = x { //~ ERROR expected tuple struct/variant, found enum\n   |            ^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[01:15:56] {"message":"expected tuple struct/variant, found enum `Example`","code":{"code":"E0532","explanation":"\nPattern arm did not match expected kind.\n\nErroneous code example:\n\n