\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":203,"byte_end":209,"line_start":19,"line_end":19,"column_start":13,"column_end":19,"is_primary":true,"text":[{"text":"    let x = Option(1); //~ ERROR expected function, found enum","highlight_start":13,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try using one of the enum's variants","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":203,"byte_end":209,"line_start":19,"line_end":19,"column_start":13,"column_end":19,"is_primary":true,"text":[{"text":"    let x = Option(1); //~ ERROR expected function, found enum","highlight_start":13,"highlight_end":19}],"label":null,"suggested_replacement":"std::option::Option::None","suggestion_applicability":"MaybeIncorrect","expansion":null},{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":203,"byte_end":209,"line_start":19,"line_end":19,"column_start":13,"column_end":19,"is_primary":true,"text":[{"text":"    let x = Option(1); //~ ERROR expected function, found enum","highlight_start":13,"highlight_end":19}],"label":null,"suggested_replacement":"std::option::Option::Some","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0423]: expected function, found enum `Option`\n  --> /checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs:19:13\n   |\nLL |     let x = Option(1); //~ ERROR expected function, found enum\n   |             ^^^^^^\nhelp: try using one of the enum's variants\n   |\nLL |     let x = std::option::Option::None(1); //~ ERROR expected function, found enum\n   |             ^^^^^^^^^^^^^^^^^^^^^^^^^\nLL |     let x = std::option::Option::Some(1); //~ ERROR expected function, found enum\n   |             ^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[01:15:56] {"message":"expected tuple struct/variant, found enum `Option`","code":{"code":"E0532","explanation":"\nPattern arm did not match expected kind.\n\nErroneous code example:\n\n