\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs","byte_start":550,"byte_end":554,"line_start":31,"line_end":31,"column_start":13,"column_end":17,"is_primary":true,"text":[{"text":"    let y = Void(); //~ ERROR expected function, found enum","highlight_start":13,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0423]: expected function, found enum `Void`\n  --> /checkout/src/test/ui/did_you_mean/issue-43871-enum-instead-of-variant.rs:31:13\n   |\nLL |     let y = Void(); //~ ERROR expected function, found enum\n   |             ^^^^\n\n"}
[01:15:56] {"message":"expected function, found enum `ManyVariants`","code":{"code":"E0423","explanation":"\nAn identifier was used like a function name or a value was expected and the\nidentifier exists but it belongs to a different namespace.\n\nFor (an erroneous) example, here a `struct` variant name were used as a\nfunction:\n\n