\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":885,"byte_end":894,"line_start":21,"line_end":21,"column_start":24,"column_end":33,"is_primary":true,"text":[{"text":"    ($ty: ty) => ($ty::AssocItem)","highlight_start":24,"highlight_end":33}],"label":"associated item not found in `u8`","suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":1133,"byte_end":1141,"line_start":31,"line_end":31,"column_start":9,"column_end":17,"is_primary":false,"text":[{"text":"        pat!(u8) => {}","highlight_start":9,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"pat!","def_site_span":{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":843,"byte_end":1052,"line_start":20,"line_end":24,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! pat {","highlight_start":1,"highlight_end":19},{"text":"    ($ty: ty) => ($ty::AssocItem)","highlight_start":1,"highlight_end":34},{"text":"    //~^ ERROR missing angle brackets in associated item path","highlight_start":1,"highlight_end":62},{"text":"    //~| ERROR no associated item named `AssocItem` found for type `u8` in the current scope","highlight_start":1,"highlight_end":93},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[],"rendered":"error[E0599]: no associated item named `AssocItem` found for type `u8` in the current scope\n  --> /checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs:21:24\n   |\nLL |     ($ty: ty) => ($ty::AssocItem)\n   |                        ^^^^^^^^^ associated item not found in `u8`\n...\nLL |         pat!(u8) => {}\n   |         -------- in this macro invocation\n\n"}
[01:15:51] {"message":"no associated item named `AssocItem` found for type `u8` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n