plain
travis_time:end:0b1acef7:start=1553259270535831503,finish=1553259365188036606,duration=94652205103
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
$ export GCP_CACHE_BUCKET=rust-lang-ci-cache
Setting environment variables from .travis.yml
---
[01:15:51] diff of stderr:
[01:15:51] 
[01:15:51] 65   --> $DIR/bad-assoc-pat.rs:21:24
[01:15:51] 66    |
[01:15:51] 67 LL |     ($ty: ty) => ($ty::AssocItem)
[01:15:51] -    |                   |
[01:15:51] -    |                   associated item not found in `u8`
[01:15:51] +    |                        ^^^^^^^^^ associated item not found in `u8`
[01:15:51] 71 ...
[01:15:51] 71 ...
[01:15:51] 72 LL |         pat!(u8) => {}
[01:15:51] 
[01:15:51] 76   --> $DIR/bad-assoc-pat.rs:32:16
[01:15:51] 77    |
[01:15:51] 77    |
[01:15:51] 78 LL |         ty!()::AssocItem => {}
[01:15:51] -    |         |
[01:15:51] -    |         associated item not found in `u8`
[01:15:51] +    |                ^^^^^^^^^ associated item not found in `u8`
[01:15:51] 82 
[01:15:51] 82 
[01:15:51] 83 error: aborting due to 12 previous errors
[01:15:51] 84 
[01:15:51] 
[01:15:51] 
[01:15:51] The actual stderr differed from the expected stderr.
[01:15:51] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/did_you_mean/bad-assoc-pat/bad-assoc-pat.stderr
[01:15:51] To update references, rerun the tests and pass the `--bless` flag
[01:15:51] To only update this specific test, also pass `--test-args did_you_mean/bad-assoc-pat.rs`
[01:15:51] error: 1 errors occurred comparing output.
[01:15:51] status: exit code: 1
[01:15:51] status: exit code: 1
[01:15:51] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/did_you_mean/bad-assoc-pat/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/did_you_mean/bad-assoc-pat/auxiliary" "-A" "unused"
[01:15:51] ------------------------------------------
[01:15:51] 
[01:15:51] ------------------------------------------
[01:15:51] stderr:
[01:15:51] stderr:
[01:15:51] ------------------------------------------
[01:15:51] {"message":"missing angle brackets in associated item path","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":36,"byte_end":51,"line_start":3,"line_end":3,"column_start":9,"column_end":24,"is_primary":true,"text":[{"text":"        [u8]::AssocItem => {}","highlight_start":9,"highlight_end":24}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":36,"byte_end":51,"line_start":3,"line_end":3,"column_start":9,"column_end":24,"is_primary":true,"text":[{"text":"        [u8]::AssocItem => {}","highlight_start":9,"highlight_end":24}],"label":null,"suggested_replacement":"<[u8]>::AssocItem","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error: missing angle brackets in associated item path\n  --> /checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs:3:9\n   |\nLL |         [u8]::AssocItem => {}\n   |         ^^^^^^^^^^^^^^^ help: try: `<[u8]>::AssocItem`\n\n"}
[01:15:51] {"message":"missing angle brackets in associated item path","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":231,"byte_end":250,"line_start":6,"line_end":6,"column_start":9,"column_end":28,"is_primary":true,"text":[{"text":"        (u8, u8)::AssocItem => {}","highlight_start":9,"highlight_end":28}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":231,"byte_end":250,"line_start":6,"line_end":6,"column_start":9,"column_end":28,"is_primary":true,"text":[{"text":"        (u8, u8)::AssocItem => {}","highlight_start":9,"highlight_end":28}],"label":null,"suggested_replacement":"<(u8, u8)>::AssocItem","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error: missing angle brackets in associated item path\n  --> /checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs:6:9\n   |\nLL |         (u8, u8)::AssocItem => {}\n   |         ^^^^^^^^^^^^^^^^^^^ help: try: `<(u8, u8)>::AssocItem`\n\n"}
[01:15:51] {"message":"missing angle brackets in associated item path","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":432,"byte_end":444,"line_start":9,"line_end":9,"column_start":9,"column_end":21,"is_primary":true,"text":[{"text":"        _::AssocItem => {}","highlight_start":9,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":432,"byte_end":444,"line_start":9,"line_end":9,"column_start":9,"column_end":21,"is_primary":true,"text":[{"text":"        _::AssocItem => {}","highlight_start":9,"highlight_end":21}],"label":null,"suggested_replacement":"<_>::AssocItem","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error: missing angle brackets in associated item path\n  --> /checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs:9:9\n   |\nLL |         _::AssocItem => {}\n   |         ^^^^^^^^^^^^ help: try: `<_>::AssocItem`\n\n"}
[01:15:51] {"message":"missing angle brackets in associated item path","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":645,"byte_end":661,"line_start":14,"line_end":14,"column_start":10,"column_end":26,"is_primary":true,"text":[{"text":"        &(u8,)::AssocItem => {}","highlight_start":10,"highlight_end":26}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":645,"byte_end":661,"line_start":14,"line_end":14,"column_start":10,"column_end":26,"is_primary":true,"text":[{"text":"        &(u8,)::AssocItem => {}","highlight_start":10,"highlight_end":26}],"label":null,"suggested_replacement":"<(u8,)>::AssocItem","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error: missing angle brackets in associated item path\n  --> /checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs:14:10\n   |\nLL |         &(u8,)::AssocItem => {}\n   |          ^^^^^^^^^^^^^^^^ help: try: `<(u8,)>::AssocItem`\n\n"}
[01:15:51] {"message":"missing angle brackets in associated item path","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":1156,"byte_end":1172,"line_start":32,"line_end":32,"column_start":9,"column_end":25,"is_primary":true,"text":[{"text":"        ty!()::AssocItem => {}","highlight_start":9,"highlight_end":25}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"try","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":1156,"byte_end":1172,"line_start":32,"line_end":32,"column_start":9,"column_end":25,"is_primary":true,"text":[{"text":"        ty!()::AssocItem => {}","highlight_start":9,"highlight_end":25}],"label":null,"suggested_replacement":"<ty!()>::AssocItem","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error: missing angle brackets in associated item path\n  --> /checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs:32:9\n   |\nLL |         ty!()::AssocItem => {}\n   |         ^^^^^^^^^^^^^^^^ help: try: `<ty!()>::AssocItem`\n\n"}
[01:15:51] {"message":"missing angle brackets in associated item path","code":null,"level":"error","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":880,"byte_end":894,"line_start":21,"line_end":21,"column_start":19,"column_end":33,"is_primary":true,"text":[{"text":"    ($ty: ty) => ($ty::AssocItem)","highlight_start":19,"highlight_end":33}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":{"span":{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":1133,"byte_end":1141,"line_start":31,"line_end":31,"column_start":9,"column_end":17,"is_primary":false,"text":[{"text":"        pat!(u8) => {}","highlight_start":9,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"pat!","def_site_span":{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":843,"byte_end":1052,"line_start":20,"line_end":24,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! pat {","highlight_start":1,"highlight_end":19},{"text":"    ($ty: ty) => ($ty::AssocItem)","highlight_start":1,"highlight_end":34},{"text":"    //~^ ERROR missing angle brackets in associated item path","highlight_start":1,"highlight_end":62},{"text":"    //~| ERROR no associated item named `AssocItem` found for type `u8` in the current scope","highlight_start":1,"highlight_end":93},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[{"message":"try","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":880,"byte_end":894,"line_start":21,"line_end":21,"column_start":19,"column_end":33,"is_primary":true,"text":[{"text":"    ($ty: ty) => ($ty::AssocItem)","highlight_start":19,"highlight_end":33}],"label":null,"suggested_replacement":"<$ty>::AssocItem","suggestion_applicability":"MaybeIncorrect","expansion":{"span":{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":1133,"byte_end":1141,"line_start":31,"line_end":31,"column_start":9,"column_end":17,"is_primary":false,"text":[{"text":"        pat!(u8) => {}","highlight_start":9,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null},"macro_decl_name":"pat!","def_site_span":{"file_name":"/checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs","byte_start":843,"byte_end":1052,"line_start":20,"line_end":24,"column_start":1,"column_end":2,"is_primary":false,"text":[{"text":"macro_rules! pat {","highlight_start":1,"highlight_end":19},{"text":"    ($ty: ty) => ($ty::AssocItem)","highlight_start":1,"highlight_end":34},{"text":"    //~^ ERROR missing angle brackets in associated item path","highlight_start":1,"highlight_end":62},{"text":"    //~| ERROR no associated item named `AssocItem` found for type `u8` in the current scope","highlight_start":1,"highlight_end":93},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}}}],"children":[],"rendered":null}],"rendered":"error: missing angle brackets in associated item path\n  --> /checkout/src/test/ui/did_you_mean/bad-assoc-pat.rs:21:19\n   |\nLL |     ($ty: ty) => ($ty::AssocItem)\n   |                   ^^^^^^^^^^^^^^ help: try: `<$ty>::AssocItem`\n...\nLL |         pat!(u8) => {}\n   |         -------- in this macro invocation\n\n"}
[01:15:51] {"message":"no associated item named `AssocItem` found for type `[u8]` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n