\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/inaccessible-test-modules.rs","byte_start":148,"byte_end":159,"line_start":5,"line_end":5,"column_start":5,"column_end":16,"is_primary":true,"text":[{"text":"use __test as x; //~ ERROR unresolved import `__test`","highlight_start":5,"highlight_end":16}],"label":"no `__test` in the root","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"a similar name exists in the module","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/inaccessible-test-modules.rs","byte_start":148,"byte_end":154,"line_start":5,"line_end":5,"column_start":5,"column_end":11,"is_primary":true,"text":[{"text":"use __test as x; //~ ERROR unresolved import `__test`","highlight_start":5,"highlight_end":11}],"label":null,"suggested_replacement":"test","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0432]: unresolved import `__test`\n  --> /checkout/src/test/ui/inaccessible-test-modules.rs:5:5\n   |\nLL | use __test as x; //~ ERROR unresolved import `__test`\n   |     ------^^^^^\n   |     |\n   |     no `__test` in the root\n   |     help: a similar name exists in the module: `test`\n\n"}
[01:14:36] {"message":"unresolved import `__test_reexports`","code":{"code":"E0432","explanation":"\nAn import was unresolved.\n\nErroneous code example:\n\n