\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/test-inner-fn.rs","byte_start":194,"byte_end":319,"line_start":12,"line_end":16,"column_start":5,"column_end":6,"is_primary":true,"text":[{"text":"    fn foo() {","highlight_start":5,"highlight_end":15},{"text":"        #[test] //~ ERROR cannot test inner items [unnameable_test_items]","highlight_start":1,"highlight_end":74},{"text":"        fn bar() {}","highlight_start":1,"highlight_end":20},{"text":"        bar();","highlight_start":1,"highlight_end":15},{"text":"    }","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(rustc_private)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: use of unstable library feature 'rustc_private': this crate is being loaded from the sysroot, an unstable location; did you mean to load this crate from crates.io via `Cargo.toml` instead? (see issue #27812)\n  --> /checkout/src/test/ui/lint/test-inner-fn.rs:12:5\n   |\nLL | /     fn foo() {\nLL | |         #[test] //~ ERROR cannot test inner items [unnameable_test_items]\nLL | |         fn bar() {}\nLL | |         bar();\nLL | |     }\n   | |_____^\n   |\n   = help: add #![feature(rustc_private)] to the crate attributes to enable\n\n"}
[01:14:03] {"message":"use of unstable library feature 'rustc_private': this crate is being loaded from the sysroot, an unstable location; did you mean to load this crate from crates.io via `Cargo.toml` instead? (see issue #27812)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n