\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/rfc-1937-termination-trait/termination-trait-test-wrong-type.rs","byte_start":64,"byte_end":150,"line_start":6,"line_end":8,"column_start":1,"column_end":2,"is_primary":true,"text":[{"text":"fn can_parse_zero_as_f32() -> Result<f32, ParseIntError> { //~ ERROR","highlight_start":1,"highlight_end":69},{"text":"    \"0\".parse()","highlight_start":1,"highlight_end":16},{"text":"}","highlight_start":1,"highlight_end":2}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"add #![feature(rustc_private)] to the crate attributes to enable","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error[E0658]: use of unstable library feature 'rustc_private': this crate is being loaded from the sysroot, an unstable location; did you mean to load this crate from crates.io via `Cargo.toml` instead? (see issue #27812)\n  --> /checkout/src/test/ui/rfc-1937-termination-trait/termination-trait-test-wrong-type.rs:6:1\n   |\nLL | / fn can_parse_zero_as_f32() -> Result<f32, ParseIntError> { //~ ERROR\nLL | |     \"0\".parse()\nLL | | }\n   | |_^\n   |\n   = help: add #![feature(rustc_private)] to the crate attributes to enable\n\n"}
[01:14:03] {"message":"use of unstable library feature 'rustc_private': this crate is being loaded from the sysroot, an unstable location; did you mean to load this crate from crates.io via `Cargo.toml` instead? (see issue #27812)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n