\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/inaccessible-test-modules.rs","byte_start":202,"byte_end":223,"line_start":6,"line_end":6,"column_start":5,"column_end":26,"is_primary":true,"text":[{"text":"use __test_reexports as y; //~ ERROR unresolved import `__test_reexports`","highlight_start":5,"highlight_end":26}],"label":"no `__test_reexports` in the root","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"a similar name exists in the module","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/inaccessible-test-modules.rs","byte_start":202,"byte_end":218,"line_start":6,"line_end":6,"column_start":5,"column_end":21,"is_primary":true,"text":[{"text":"use __test_reexports as y; //~ ERROR unresolved import `__test_reexports`","highlight_start":5,"highlight_end":21}],"label":null,"suggested_replacement":"__test_reexports","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0432]: unresolved import `__test_reexports`\n  --> /checkout/src/test/ui/inaccessible-test-modules.rs:6:5\n   |\nLL | use __test_reexports as y; //~ ERROR unresolved import `__test_reexports`\n   |     ----------------^^^^^\n   |     |\n   |     no `__test_reexports` in the root\n   |     help: a similar name exists in the module: `__test_reexports`\n\n"}
[01:14:03] {"message":"use of unstable library feature 'rustc_private': this crate is being loaded from the sysroot, an unstable location; did you mean to load this crate from crates.io via `Cargo.toml` instead? (see issue #27812)","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n