plain
travis_time:end:042fe55c:start=1554369988204855411,finish=1554369989579721576,duration=1374866165
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
$ export GCP_CACHE_BUCKET=rust-lang-ci-cache
Setting environment variables from .travis.yml
---
[01:09:00] .................................................................................................... 2900/5522
[01:09:04] .................................................................................................... 3000/5522
[01:09:07] .................................................................................................... 3100/5522
[01:09:10] .................................................................................................... 3200/5522
[01:09:14] ......................................F............................................................. 3300/5522
[01:09:21] ................................................................................................ii.. 3500/5522
[01:09:25] .i..ii.............................................................................................. 3600/5522
[01:09:29] .................................................................................................... 3700/5522
[01:09:32] .................................................................................................... 3800/5522
---
[01:10:37] 80 
[01:10:37] 
[01:10:37] 
[01:10:37] The actual stderr differed from the expected stderr.
[01:10:37] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/lint-unnecessary-parens/lint-unnecessary-parens.stderr
[01:10:37] To update references, rerun the tests and pass the `--bless` flag
[01:10:37] To only update this specific test, also pass `--test-args lint/lint-unnecessary-parens.rs`
[01:10:37] error: 1 errors occurred comparing output.
[01:10:37] status: exit code: 1
[01:10:37] status: exit code: 1
[01:10:37] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/lint/lint-unnecessary-parens.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/lint-unnecessary-parens/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/lint-unnecessary-parens/auxiliary" "-A" "unused"
[01:10:37] ------------------------------------------
[01:10:37] 
[01:10:37] ------------------------------------------
[01:10:37] stderr:
[01:10:37] stderr:
[01:10:37] ------------------------------------------
[01:10:37] {"message":"unnecessary parentheses around `return` value","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":179,"byte_end":182,"line_start":10,"line_end":10,"column_start":12,"column_end":15,"is_primary":true,"text":[{"text":"    return (1); //~ ERROR unnecessary parentheses around `return` value","highlight_start":12,"highlight_end":15}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":8,"byte_end":21,"line_start":1,"line_end":1,"column_start":9,"column_end":22,"is_primary":true,"text":[{"text":"#![deny(unused_parens)]","highlight_start":9,"highlight_end":22}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":179,"byte_end":182,"line_start":10,"line_end":10,"column_start":12,"column_end":15,"is_primary":true,"text":[{"text":"    return (1); //~ ERROR unnecessary parentheses around `return` value","highlight_start":12,"highlight_end":15}],"label":null,"suggested_replacement":"1","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around `return` value\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:10:12\n   |\nLL |     return (1); //~ ERROR unnecessary parentheses around `return` value\n   |            ^^^ help: remove these parentheses\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:1:9\n   |\nLL | #![deny(unused_parens)]\n   |         ^^^^^^^^^^^^^\n\n"}
[01:10:37] {"message":"unnecessary parentheses around `return` value","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":276,"byte_end":285,"line_start":13,"line_end":13,"column_start":12,"column_end":21,"is_primary":true,"text":[{"text":"    return (X { y }); //~ ERROR unnecessary parentheses around `return` value","highlight_start":12,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":276,"byte_end":285,"line_start":13,"line_end":13,"column_start":12,"column_end":21,"is_primary":true,"text":[{"text":"    return (X { y }); //~ ERROR unnecessary parentheses around `return` value","highlight_start":12,"highlight_end":21}],"label":null,"suggested_replacement":"X { y }","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around `return` value\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:13:12\n   |\nLL |     return (X { y }); //~ ERROR unnecessary parentheses around `return` value\n   |            ^^^^^^^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around function argument","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":377,"byte_end":383,"line_start":18,"line_end":18,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"    bar((true)); //~ ERROR unnecessary parentheses around function argument","highlight_start":9,"highlight_end":15}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":377,"byte_end":383,"line_start":18,"line_end":18,"column_start":9,"column_end":15,"is_primary":true,"text":[{"text":"    bar((true)); //~ ERROR unnecessary parentheses around function argument","highlight_start":9,"highlight_end":15}],"label":null,"suggested_replacement":"true","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around function argument\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:18:9\n   |\nLL |     bar((true)); //~ ERROR unnecessary parentheses around function argument\n   |         ^^^^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around `if` condition","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":453,"byte_end":459,"line_start":20,"line_end":20,"column_start":8,"column_end":14,"is_primary":true,"text":[{"text":"    if (true) {} //~ ERROR unnecessary parentheses around `if` condition","highlight_start":8,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":453,"byte_end":459,"line_start":20,"line_end":20,"column_start":8,"column_end":14,"is_primary":true,"text":[{"text":"    if (true) {} //~ ERROR unnecessary parentheses around `if` condition","highlight_start":8,"highlight_end":14}],"label":null,"suggested_replacement":"true","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around `if` condition\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:20:8\n   |\nLL |     if (true) {} //~ ERROR unnecessary parentheses around `if` condition\n   |        ^^^^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around `while` condition","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":529,"byte_end":535,"line_start":21,"line_end":21,"column_start":11,"column_end":17,"is_primary":true,"text":[{"text":"    while (true) {} //~ ERROR unnecessary parentheses around `while` condition","highlight_start":11,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":529,"byte_end":535,"line_start":21,"line_end":21,"column_start":11,"column_end":17,"is_primary":true,"text":[{"text":"    while (true) {} //~ ERROR unnecessary parentheses around `while` condition","highlight_start":11,"highlight_end":17}],"label":null,"suggested_replacement":"true","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around `while` condition\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:21:11\n   |\nLL |     while (true) {} //~ ERROR unnecessary parentheses around `while` condition\n   |           ^^^^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around `match` scrutinee","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":608,"byte_end":614,"line_start":22,"line_end":22,"column_start":11,"column_end":17,"is_primary":true,"text":[{"text":"    match (true) { //~ ERROR unnecessary parentheses around `match` scrutinee","highlight_start":11,"highlight_end":17}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":608,"byte_end":614,"line_start":22,"line_end":22,"column_start":11,"column_end":17,"is_primary":true,"text":[{"text":"    match (true) { //~ ERROR unnecessary parentheses around `match` scrutinee","highlight_start":11,"highlight_end":17}],"label":null,"suggested_replacement":"true","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around `match` scrutinee\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:22:11\n   |\nLL |     match (true) { //~ ERROR unnecessary parentheses around `match` scrutinee\n   |           ^^^^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around `if let` scrutinee","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":713,"byte_end":716,"line_start":25,"line_end":25,"column_start":16,"column_end":19,"is_primary":true,"text":[{"text":"    if let 1 = (1) {} //~ ERROR unnecessary parentheses around `if let` scrutinee","highlight_start":16,"highlight_end":19}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":713,"byte_end":716,"line_start":25,"line_end":25,"column_start":16,"column_end":19,"is_primary":true,"text":[{"text":"    if let 1 = (1) {} //~ ERROR unnecessary parentheses around `if let` scrutinee","highlight_start":16,"highlight_end":19}],"label":null,"suggested_replacement":"1","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around `if let` scrutinee\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:25:16\n   |\nLL |     if let 1 = (1) {} //~ ERROR unnecessary parentheses around `if let` scrutinee\n   |                ^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around `while let` scrutinee","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":798,"byte_end":801,"line_start":26,"line_end":26,"column_start":19,"column_end":22,"is_primary":true,"text":[{"text":"    while let 1 = (2) {} //~ ERROR unnecessary parentheses around `while let` scrutinee","highlight_start":19,"highlight_end":22}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":798,"byte_end":801,"line_start":26,"line_end":26,"column_start":19,"column_end":22,"is_primary":true,"text":[{"text":"    while let 1 = (2) {} //~ ERROR unnecessary parentheses around `while let` scrutinee","highlight_start":19,"highlight_end":22}],"label":null,"suggested_replacement":"2","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around `while let` scrutinee\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:26:19\n   |\nLL |     while let 1 = (2) {} //~ ERROR unnecessary parentheses around `while let` scrutinee\n   |                   ^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around method argument","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":1201,"byte_end":1207,"line_start":40,"line_end":40,"column_start":24,"column_end":30,"is_primary":true,"text":[{"text":"    X { y: false }.foo((true)); //~ ERROR unnecessary parentheses around method argument","highlight_start":24,"highlight_end":30}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":1201,"byte_end":1207,"line_start":40,"line_end":40,"column_start":24,"column_end":30,"is_primary":true,"text":[{"text":"    X { y: false }.foo((true)); //~ ERROR unnecessary parentheses around method argument","highlight_start":24,"highlight_end":30}],"label":null,"suggested_replacement":"true","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around method argument\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:40:24\n   |\nLL |     X { y: false }.foo((true)); //~ ERROR unnecessary parentheses around method argument\n   |                        ^^^^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around assigned value","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":1285,"byte_end":1288,"line_start":42,"line_end":42,"column_start":18,"column_end":21,"is_primary":true,"text":[{"text":"    let mut _a = (0); //~ ERROR unnecessary parentheses around assigned value","highlight_start":18,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":1285,"byte_end":1288,"line_start":42,"line_end":42,"column_start":18,"column_end":21,"is_primary":true,"text":[{"text":"    let mut _a = (0); //~ ERROR unnecessary parentheses around assigned value","highlight_start":18,"highlight_end":21}],"label":null,"suggested_replacement":"0","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around assigned value\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:42:18\n   |\nLL |     let mut _a = (0); //~ ERROR unnecessary parentheses around assigned value\n   |                  ^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around assigned value","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":1355,"byte_end":1358,"line_start":43,"line_end":43,"column_start":10,"column_end":13,"is_primary":true,"text":[{"text":"    _a = (0); //~ ERROR unnecessary parentheses around assigned value","highlight_start":10,"highlight_end":13}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":1355,"byte_end":1358,"line_start":43,"line_end":43,"column_start":10,"column_end":13,"is_primary":true,"text":[{"text":"    _a = (0); //~ ERROR unnecessary parentheses around assigned value","highlight_start":10,"highlight_end":13}],"label":null,"suggested_replacement":"0","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around assigned value\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:43:10\n   |\nLL |     _a = (0); //~ ERROR unnecessary parentheses around assigned value\n   |          ^^^ help: remove these parentheses\n\n"}
[01:10:37] {"message":"unnecessary parentheses around assigned value","code":{"code":"unused_parens","explanation":null},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":1426,"byte_end":1429,"line_start":44,"line_end":44,"column_start":11,"column_end":14,"is_primary":true,"text":[{"text":"    _a += (1); //~ ERROR unnecessary parentheses around assigned value","highlight_start":11,"highlight_end":14}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/lint-unnecessary-parens.rs","byte_start":1426,"byte_end":1429,"line_start":44,"line_end":44,"column_start":11,"column_end":14,"is_primary":true,"text":[{"text":"    _a += (1); //~ ERROR unnecessary parentheses around assigned value","highlight_start":11,"highlight_end":14}],"label":null,"suggested_replacement":"1","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: unnecessary parentheses around assigned value\n  --> /checkout/src/test/ui/lint/lint-unnecessary-parens.rs:44:11\n   |\nLL |     _a += (1); //~ ERROR unnecessary parentheses around assigned value\n   |           ^^^ help: remove these parentheses\n\n"}
[01:10:37] 
[01:10:37] ------------------------------------------
[01:10:37] 
[01:10:37] thread '[ui] ui/lint/lint-unnecessary-parens.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3425:9
---
[01:10:37] 
[01:10:37] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:516:22
[01:10:37] 
[01:10:37] 
[01:10:37] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[01:10:37] 
[01:10:37] 
[01:10:37] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[01:10:37] Build completed unsuccessfully in 0:04:17
[01:10:37] Build completed unsuccessfully in 0:04:17
[01:10:37] make: *** [check] Error 1
[01:10:37] Makefile:48: recipe for target 'check' failed
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:11798ebc
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Thu Apr  4 10:37:20 UTC 2019
---
travis_time:end:1dbd7aa9:start=1554374241423718743,finish=1554374241430733287,duration=7014544
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:1315e727
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:08e187e5
$ dmesg | grep -i kill
