\n\nIf the item you are importing is not defined in some super-module of the\ncurrent module, then it must also be declared as public (e.g., `pub fn`).\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-async-await-2015-edition.rs","byte_start":250,"byte_end":255,"line_start":8,"line_end":8,"column_start":13,"column_end":18,"is_primary":true,"text":[{"text":"    let _ = async || { true }; //~ ERROR cannot find value `async` in this scope","highlight_start":13,"highlight_end":18}],"label":"not found in this scope","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0425]: cannot find value `async` in this scope\n  --> /checkout/src/test/ui/feature-gates/feature-gate-async-await-2015-edition.rs:8:13\n   |\nLL |     let _ = async || { true }; //~ ERROR cannot find value `async` in this scope\n   |             ^^^^^ not found in this scope\n\n"}
[01:07:03] {"message":"async fn is unstable","code":{"code":"E0658","explanation":"\nAn unstable feature was used.\n\nErroneous code example:\n\n