\n\nSwitch to the Rust 2018 edition to use `async fn`.\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/feature-gates/feature-gate-async-await-2015-edition.rs","byte_start":17,"byte_end":22,"line_start":3,"line_end":3,"column_start":1,"column_end":6,"is_primary":true,"text":[{"text":"async fn foo() {} //~ ERROR `async fn` is not permitted in the 2015 edition","highlight_start":1,"highlight_end":6}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0670]: `async fn` is not permitted in the 2015 edition\n  --> /checkout/src/test/ui/feature-gates/feature-gate-async-await-2015-edition.rs:3:1\n   |\nLL | async fn foo() {} //~ ERROR `async fn` is not permitted in the 2015 edition\n   | ^^^^^\n\n"}
[01:07:03] {"message":"cannot find struct, variant or union type `async` in this scope","code":{"code":"E0422","explanation":"\nYou are trying to use an identifier that is either undefined or not a struct.\nErroneous code example:\n\n