plain
travis_time:end:00a4b163:start=1554987225468624399,finish=1554987227586275576,duration=2117651177
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
$ export GCP_CACHE_BUCKET=rust-lang-ci-cache
$ export AWS_ACCESS_KEY_ID=AKIA46X5W6CZEJZ6XT55
---
[01:04:19] .................................................................................................... 2800/5530
[01:04:23] .................................................................................................... 2900/5530
[01:04:27] .................................................................................................... 3000/5530
[01:04:30] .................................................................................................... 3100/5530
[01:04:33] .....................................................................F..F........................... 3200/5530
[01:04:41] ..............................i..................................................................... 3400/5530
[01:04:45] .................................................................................................... 3500/5530
[01:04:48] ....ii...i..ii...................................................................................... 3600/5530
[01:04:52] .................................................................................................... 3700/5530
---
[01:06:01] ---- [ui] ui/lint/inclusive-range-pattern-syntax.rs stdout ----
[01:06:01] 
[01:06:01] error: failed to compile fixed code
[01:06:01] status: exit code: 101
[01:06:01] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/lint/inclusive-range-pattern-syntax.fixed" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/inclusive-range-pattern-syntax/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/inclusive-range-pattern-syntax/auxiliary"
[01:06:01] ------------------------------------------
[01:06:01] 
[01:06:01] ------------------------------------------
[01:06:01] stderr:
[01:06:01] stderr:
[01:06:01] ------------------------------------------
[01:06:01] thread 'rustc' panicked at 'src/librustc/hir/map/hir_id_validator.rs:26: 
[01:06:01] ItemLocalIds not assigned densely in ::main[0]. Max ItemLocalId = 33, missing IDs = ["[local_id: 5, node:unknown node (id=32)]"]; seens IDs = ["(HirId { owner: DefIndex(0:3), local_id: 0 } fn main (id=11))", "(HirId { owner: DefIndex(0:3), local_id: 31 } block {\n    let despondency = 2;\n    match despondency { 1 ...2 => { } _ => { } }\n    match &despondency { &(1 ...2) => { } _ => { } }\n} (id=13))", "(HirId { owner: DefIndex(0:3), local_id: 28 } block { } (id=39))", "(HirId { owner: DefIndex(0:3), local_id: 25 } expr 2 (id=35))", "(HirId { owner: DefIndex(0:3), local_id: 22 } expr despondency (id=29))", "(HirId { owner: DefIndex(0:3), local_id: 19 } expr { } (id=26))", "(HirId { owner: DefIndex(0:3), local_id: 16 } block { } (id=22))", "(HirId { owner: DefIndex(0:3), local_id: 13 } expr despondency (id=18))", "(HirId { owner: DefIndex(0:3), local_id: 10 } stmt let despondency = 2; (id=42))", "(HirId { owner: DefIndex(0:3), local_id: 7 } pat _ (id=38))", "(HirId { owner: DefIndex(0:3), local_id: 4 } pat &(1 ...2) (id=31))", "(HirId { owner: DefIndex(0:3), local_id: 1 } local despondency (id=15))", "(HirId { owner: DefIndex(0:3), local_id: 32 } expr {\n    let despondency = 2;\n    match despondency { 1 ...2 => { } _ => { } }\n    match &despondency { &(1 ...2) => { } _ => { } }\n} (id=45))", "(HirId { owner: DefIndex(0:3), local_id: 29 } expr { } (id=40))", "(HirId { owner: DefIndex(0:3), local_id: 26 } block { } (id=36))", "(HirId { owner: DefIndex(0:3), local_id: 23 } expr &despondency (id=30))", "(HirId { owner: DefIndex(0:3), local_id: 20 } expr match despondency { 1 ...2 => { } _ => { } } (id=27))", "(HirId { owner: DefIndex(0:3), local_id: 17 } expr { } (id=23))", "(HirId { owner: DefIndex(0:3), local_id: 14 } expr 1 (id=20))", "(HirId { owner: DefIndex(0:3), local_id: 11 } stmt match despondency { 1 ...2 => { } _ => { } } (id=43))", "(HirId { owner: DefIndex(0:3), local_id: 8 } local despondency (id=14))", "(HirId { owner: DefIndex(0:3), local_id: 2 } pat 1 ...2 (id=19))", "(HirId { owner: DefIndex(0:3), local_id: 33 } unknown node (id=12))", "(HirId { owner: DefIndex(0:3), local_id: 30 } expr match &despondency { &(1 ...2) => { } _ => { } } (id=41))", "(HirId { owner: DefIndex(0:3), local_id: 27 } expr { } (id=37))", "(HirId { owner: DefIndex(0:3), local_id: 24 } expr 1 (id=34))", "(HirId { owner: DefIndex(0:3), local_id: 21 } path segment despondency (id=28))", "(HirId { owner: DefIndex(0:3), local_id: 18 } block { } (id=25))", "(HirId { owner: DefIndex(0:3), local_id: 15 } expr 2 (id=21))", "(HirId { owner: DefIndex(0:3), local_id: 12 } path segment despondency (id=17))", "(HirId { owner: DefIndex(0:3), local_id: 9 } expr 2 (id=16))", "(HirId { owner: DefIndex(0:3), local_id: 6 } pat 1 ...2 (id=33))", "(HirId { owner: DefIndex(0:3), local_id: 3 } pat _ (id=24))"]', src/librustc/util/bug.rs:37:26
[01:06:01] 
[01:06:01] error: internal compiler error: unexpected panic
[01:06:01] 
[01:06:01] note: the compiler unexpectedly panicked. this is a bug.
[01:06:01] note: the compiler unexpectedly panicked. this is a bug.
[01:06:01] 
[01:06:01] note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
[01:06:01] 
[01:06:01] note: rustc 1.35.0-dev running on x86_64-unknown-linux-gnu
[01:06:01] 
[01:06:01] note: compiler flags: -Z threads=1 -Z ui-testing -Z unstable-options -C prefer-dynamic -C rpath
[01:06:01] 
[01:06:01] ------------------------------------------
[01:06:01] 
[01:06:01] thread '[ui] ui/lint/inclusive-range-pattern-syntax.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3425:9
[01:06:01] thread '[ui] ui/lint/inclusive-range-pattern-syntax.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3425:9
[01:06:01] note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
[01:06:01] 
[01:06:01] ---- [ui] ui/lint/issue-54538-unused-parens-lint.rs stdout ----
[01:06:01] 
[01:06:01] error: test compilation failed although it shouldn't!
[01:06:01] status: exit code: 101
[01:06:01] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/issue-54538-unused-parens-lint/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/issue-54538-unused-parens-lint/auxiliary" "-A" "unused"
[01:06:01] ------------------------------------------
[01:06:01] 
[01:06:01] ------------------------------------------
[01:06:01] stderr:
[01:06:01] stderr:
[01:06:01] ------------------------------------------
[01:06:01] {"message":"unnecessary parentheses around pattern","code":{"code":"unused_parens","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":136,"byte_end":139,"line_start":9,"line_end":9,"column_start":9,"column_end":12,"is_primary":true,"text":[{"text":"        (_) => {}         //~ WARNING: unnecessary parentheses around pattern","highlight_start":9,"highlight_end":12}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"lint level defined here","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":85,"byte_end":98,"line_start":5,"line_end":5,"column_start":9,"column_end":22,"is_primary":true,"text":[{"text":"#![warn(unused_parens)]","highlight_start":9,"highlight_end":22}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":136,"byte_end":139,"line_start":9,"line_end":9,"column_start":9,"column_end":12,"is_primary":true,"text":[{"text":"        (_) => {}         //~ WARNING: unnecessary parentheses around pattern","highlight_start":9,"highlight_end":12}],"label":null,"suggested_replacement":"_","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unnecessary parentheses around pattern\n  --> /checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs:9:9\n   |\nLL |         (_) => {}         //~ WARNING: unnecessary parentheses around pattern\n   |         ^^^ help: remove these parentheses\n   |\nnote: lint level defined here\n  --> /checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs:5:9\n   |\nLL | #![warn(unused_parens)]\n   |         ^^^^^^^^^^^^^\n\n"}
[01:06:01] {"message":"unnecessary parentheses around pattern","code":{"code":"unused_parens","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":214,"byte_end":217,"line_start":10,"line_end":10,"column_start":9,"column_end":12,"is_primary":true,"text":[{"text":"        (y) => {}         //~ WARNING: unnecessary parentheses around pattern","highlight_start":9,"highlight_end":12}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":214,"byte_end":217,"line_start":10,"line_end":10,"column_start":9,"column_end":12,"is_primary":true,"text":[{"text":"        (y) => {}         //~ WARNING: unnecessary parentheses around pattern","highlight_start":9,"highlight_end":12}],"label":null,"suggested_replacement":"y","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unnecessary parentheses around pattern\n  --> /checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs:10:9\n   |\nLL |         (y) => {}         //~ WARNING: unnecessary parentheses around pattern\n   |         ^^^ help: remove these parentheses\n\n"}
[01:06:01] {"message":"unnecessary parentheses around pattern","code":{"code":"unused_parens","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":292,"byte_end":299,"line_start":11,"line_end":11,"column_start":9,"column_end":16,"is_primary":true,"text":[{"text":"        (ref r) => {}     //~ WARNING: unnecessary parentheses around pattern","highlight_start":9,"highlight_end":16}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":292,"byte_end":299,"line_start":11,"line_end":11,"column_start":9,"column_end":16,"is_primary":true,"text":[{"text":"        (ref r) => {}     //~ WARNING: unnecessary parentheses around pattern","highlight_start":9,"highlight_end":16}],"label":null,"suggested_replacement":"ref r","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unnecessary parentheses around pattern\n  --> /checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs:11:9\n   |\nLL |         (ref r) => {}     //~ WARNING: unnecessary parentheses around pattern\n   |         ^^^^^^^ help: remove these parentheses\n\n"}
[01:06:01] {"message":"unnecessary parentheses around pattern","code":{"code":"unused_parens","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":370,"byte_end":381,"line_start":12,"line_end":12,"column_start":9,"column_end":20,"is_primary":true,"text":[{"text":"        (e @ 1..=2) => {} //~ WARNING: unnecessary parentheses around outer pattern","highlight_start":9,"highlight_end":20}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":370,"byte_end":381,"line_start":12,"line_end":12,"column_start":9,"column_end":20,"is_primary":true,"text":[{"text":"        (e @ 1..=2) => {} //~ WARNING: unnecessary parentheses around outer pattern","highlight_start":9,"highlight_end":20}],"label":null,"suggested_replacement":"e @ 1..=2","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unnecessary parentheses around pattern\n  --> /checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs:12:9\n   |\nLL |         (e @ 1..=2) => {} //~ WARNING: unnecessary parentheses around outer pattern\n   |         ^^^^^^^^^^^ help: remove these parentheses\n\n"}
[01:06:01] {"message":"unnecessary parentheses around pattern","code":{"code":"unused_parens","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":622,"byte_end":636,"line_start":18,"line_end":18,"column_start":9,"column_end":23,"is_primary":true,"text":[{"text":"        (e @ &(1...2)) => {} //~ WARNING: unnecessary parentheses around outer pattern","highlight_start":9,"highlight_end":23}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":622,"byte_end":636,"line_start":18,"line_end":18,"column_start":9,"column_end":23,"is_primary":true,"text":[{"text":"        (e @ &(1...2)) => {} //~ WARNING: unnecessary parentheses around outer pattern","highlight_start":9,"highlight_end":23}],"label":null,"suggested_replacement":"e @ &(1...2)","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unnecessary parentheses around pattern\n  --> /checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs:18:9\n   |\nLL |         (e @ &(1...2)) => {} //~ WARNING: unnecessary parentheses around outer pattern\n   |         ^^^^^^^^^^^^^^ help: remove these parentheses\n\n"}
[01:06:01] {"message":"unnecessary parentheses around pattern","code":{"code":"unused_parens","explanation":null},"level":"warning","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":710,"byte_end":713,"line_start":19,"line_end":19,"column_start":10,"column_end":13,"is_primary":true,"text":[{"text":"        &(_) => {}           //~ WARNING: unnecessary parentheses around pattern","highlight_start":10,"highlight_end":13}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"remove these parentheses","code":null,"level":"help","spans":[{"file_name":"/checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs","byte_start":710,"byte_end":713,"line_start":19,"line_end":19,"column_start":10,"column_end":13,"is_primary":true,"text":[{"text":"        &(_) => {}           //~ WARNING: unnecessary parentheses around pattern","highlight_start":10,"highlight_end":13}],"label":null,"suggested_replacement":"_","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"warning: unnecessary parentheses around pattern\n  --> /checkout/src/test/ui/lint/issue-54538-unused-parens-lint.rs:19:10\n   |\nLL |         &(_) => {}           //~ WARNING: unnecessary parentheses around pattern\n   |          ^^^ help: remove these parentheses\n\n"}
[01:06:01] thread 'rustc' panicked at 'src/librustc/hir/map/hir_id_validator.rs:26: 
[01:06:01] ItemLocalIds not assigned densely in ::main[0]. Max ItemLocalId = 92, missing IDs = ["[local_id: 1, node:unknown node (id=17)]", "[local_id: 3, node:unknown node (id=21)]", "[local_id: 5, node:unknown node (id=25)]", "[local_id: 7, node:unknown node (id=29)]", "[local_id: 10, node:unknown node (id=36)]", "[local_id: 13, node:unknown node (id=43)]", "[local_id: 15, node:unknown node (id=52)]", "[local_id: 18, node:unknown node (id=55)]", "[local_id: 21, node:unknown node (id=62)]", "[local_id: 25, node:unknown node (id=68)]", "[local_id: 28, node:unknown node (id=75)]", "[local_id: 32, node:unknown node (id=86)]", "[local_id: 36, node:unknown node (id=92)]"]; seens IDs = ["(HirId { owner: DefIndex(0:3), local_id: 0 } fn main (id=13))", "(HirId { owner: DefIndex(0:3), local_id: 61 } stmt match &1 { e@&(1 ...2) => { } &_ => { } e@&(1 ...2) => { } &(1 ...2) => { } } (id=104))", "(HirId { owner: DefIndex(0:3), local_id: 55 } expr { } (id=41))", "(HirId { owner: DefIndex(0:3), local_id: 49 } expr 2 (id=33))", "(HirId { owner: DefIndex(0:3), local_id: 43 } expr { } (id=20))", "(HirId { owner: DefIndex(0:3), local_id: 37 } pat 3 ...4 (id=93))", "(HirId { owner: DefIndex(0:3), local_id: 31 } pat &(1 ...2) (id=85))", "(HirId { owner: DefIndex(0:3), local_id: 92 } unknown node (id=14))", "(HirId { owner: DefIndex(0:3), local_id: 86 } expr { } (id=97))", "(HirId { owner: DefIndex(0:3), local_id: 19 } pat 1 ...2 (id=56))", "(HirId { owner: DefIndex(0:3), local_id: 80 } expr &1 (id=83))", "(HirId { owner: DefIndex(0:3), local_id: 74 } expr 1 (id=77))", "(HirId { owner: DefIndex(0:3), local_id: 68 } block { } (id=64))", "(HirId { owner: DefIndex(0:3), local_id: 62 } expr 1 (id=50))", "(HirId { owner: DefIndex(0:3), local_id: 56 } expr 3 (id=45))", "(HirId { owner: DefIndex(0:3), local_id: 50 } block { } (id=34))", "(HirId { owner: DefIndex(0:3), local_id: 44 } block { } (id=23))", "(HirId { owner: DefIndex(0:3), local_id: 38 } pat &_ (id=98))", "(HirId { owner: DefIndex(0:3), local_id: 26 } pat 1 ...2 (id=69))", "(HirId { owner: DefIndex(0:3), local_id: 87 } block { } (id=100))", "(HirId { owner: DefIndex(0:3), local_id: 20 } pat &_ (id=61))", "(HirId { owner: DefIndex(0:3), local_id: 81 } expr 1 (id=88))", "(HirId { owner: DefIndex(0:3), local_id: 14 } pat 3 ...4 (id=44))", "(HirId { owner: DefIndex(0:3), local_id: 75 } expr 2 (id=78))", "(HirId { owner: DefIndex(0:3), local_id: 8 } local e@1 ...2 (id=30))", "(HirId { owner: DefIndex(0:3), local_id: 69 } expr { } (id=65))", "(HirId { owner: DefIndex(0:3), local_id: 2 } pat _ (id=18))", "(HirId { owner: DefIndex(0:3), local_id: 63 } expr &1 (id=51))", "(HirId { owner: DefIndex(0:3), local_id: 57 } expr 4 (id=46))", "(HirId { owner: DefIndex(0:3), local_id: 51 } expr { } (id=35))", "(HirId { owner: DefIndex(0:3), local_id: 45 } expr { } (id=24))", "(HirId { owner: DefIndex(0:3), local_id: 39 } pat _ (id=99))", "(HirId { owner: DefIndex(0:3), local_id: 33 } pat 1 ...2 (id=87))", "(HirId { owner: DefIndex(0:3), local_id: 27 } pat &(1 ...2) (id=74))", "(HirId { owner: DefIndex(0:3), local_id: 88 } expr { } (id=101))", "(HirId { owner: DefIndex(0:3), local_id: 82 } expr 2 (id=89))", "(HirId { owner: DefIndex(0:3), local_id: 76 } block { } (id=79))", "(HirId { owner: DefIndex(0:3), local_id: 9 } pat 1 ...2 (id=31))", "(HirId { owner: DefIndex(0:3), local_id: 70 } expr 1 (id=70))", "(HirId { owner: DefIndex(0:3), local_id: 64 } expr 1 (id=57))", "(HirId { owner: DefIndex(0:3), local_id: 58 } block { } (id=47))", "(HirId { owner: DefIndex(0:3), local_id: 52 } expr 1 (id=38))", "(HirId { owner: DefIndex(0:3), local_id: 46 } block { } (id=27))", "(HirId { owner: DefIndex(0:3), local_id: 40 } stmt match 1 {\n    _ => { }\n    y => { }\n    ref r => { }\n    e@1 ...2 => { }\n    1 ...2 => { }\n    e@3 ...4 => { }\n} (id=103))", "(HirId { owner: DefIndex(0:3), local_id: 34 } local e@&(3 ...4) (id=90))", "(HirId { owner: DefIndex(0:3), local_id: 89 } expr match &1 { e@&(1 ...2) | e@&(3 ...4) => { } &_ => { } } (id=102))", "(HirId { owner: DefIndex(0:3), local_id: 22 } pat _ (id=63))", "(HirId { owner: DefIndex(0:3), local_id: 83 } expr 3 (id=94))", "(HirId { owner: DefIndex(0:3), local_id: 16 } local e@&(1 ...2) (id=53))", "(HirId { owner: DefIndex(0:3), local_id: 77 } expr { } (id=80))", "(HirId { owner: DefIndex(0:3), local_id: 71 } expr 2 (id=71))", "(HirId { owner: DefIndex(0:3), local_id: 4 } local y (id=22))", "(HirId { owner: DefIndex(0:3), local_id: 65 } expr 2 (id=58))", "(HirId { owner: DefIndex(0:3), local_id: 59 } expr { } (id=48))", "(HirId { owner: DefIndex(0:3), local_id: 53 } expr 2 (id=39))", "(HirId { owner: DefIndex(0:3), local_id: 47 } expr { } (id=28))", "(HirId { owner: DefIndex(0:3), local_id: 41 } expr 1 (id=16))", "(HirId { owner: DefIndex(0:3), local_id: 35 } pat &(3 ...4) (id=91))", "(HirId { owner: DefIndex(0:3), local_id: 29 } pat 1 ...2 (id=76))", "(HirId { owner: DefIndex(0:3), local_id: 90 } block {\n    match 1 {\n        _ => { }\n        y => { }\n        ref r => { }\n        e@1 ...2 => { }\n        1 ...2 => { }\n        e@3 ...4 => { }\n    }\n    match &1 {\n        e@&(1 ...2) => { }\n        &_ => { }\n        e@&(1 ...2) => { }\n        &(1 ...2) => { }\n    }\n    match &1 { e@&(1 ...2) | e@&(3 ...4) => { } &_ => { } }\n} (id=15))", "(HirId { owner: DefIndex(0:3), local_id: 23 } local e@&(1 ...2) (id=66))", "(HirId { owner: DefIndex(0:3), local_id: 84 } expr 4 (id=95))", "(HirId { owner: DefIndex(0:3), local_id: 17 } pat &(1 ...2) (id=54))", "(HirId { owner: DefIndex(0:3), local_id: 78 } expr match &1 { e@&(1 ...2) => { } &_ => { } e@&(1 ...2) => { } &(1 ...2) => { } } (id=81))", "(HirId { owner: DefIndex(0:3), local_id: 11 } pat 1 ...2 (id=37))", "(HirId { owner: DefIndex(0:3), local_id: 72 } block { } (id=72))", "(HirId { owner: DefIndex(0:3), local_id: 66 } block { } (id=59))", "(HirId { owner: DefIndex(0:3), local_id: 60 } expr match 1 {\n    _ => { }\n    y => { }\n    ref r => { }\n    e@1 ...2 => { }\n    1 ...2 => { }\n    e@3 ...4 => { }\n} (id=49))", "(HirId { owner: DefIndex(0:3), local_id: 54 } block { } (id=40))", "(HirId { owner: DefIndex(0:3), local_id: 48 } expr 1 (id=32))", "(HirId { owner: DefIndex(0:3), local_id: 42 } block { } (id=19))", "(HirId { owner: DefIndex(0:3), local_id: 30 } local e@&(1 ...2) (id=84))", "(HirId { owner: DefIndex(0:3), local_id: 91 } expr {\n    match 1 {\n        _ => { }\n        y => { }\n        ref r => { }\n        e@1 ...2 => { }\n        1 ...2 => { }\n        e@3 ...4 => { }\n    }\n    match &1 {\n        e@&(1 ...2) => { }\n        &_ => { }\n        e@&(1 ...2) => { }\n        &(1 ...2) => { }\n    }\n    match &1 { e@&(1 ...2) | e@&(3 ...4) => { } &_ => { } }\n} (id=106))", "(HirId { owner: DefIndex(0:3), local_id: 24 } pat &(1 ...2) (id=67))", "(HirId { owner: DefIndex(0:3), local_id: 85 } block { } (id=96))", "(HirId { owner: DefIndex(0:3), local_id: 79 } expr 1 (id=82))", "(HirId { owner: DefIndex(0:3), local_id: 12 } local e@3 ...4 (id=42))", "(HirId { owner: DefIndex(0:3), local_id: 73 } expr { } (id=73))", "(HirId { owner: DefIndex(0:3), local_id: 6 } local ref r (id=26))", "(HirId { owner: DefIndex(0:3), local_id: 67 } expr { } (id=60))"]', src/librustc/util/bug.rs:37:26
[01:06:01] 
[01:06:01] error: internal compiler error: unexpected panic
[01:06:01] 
[01:06:01] note: the compiler unexpectedly panicked. this is a bug.
[01:06:01] note: the compiler unexpectedly panicked. this is a bug.
[01:06:01] 
[01:06:01] note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
[01:06:01] 
[01:06:01] note: rustc 1.35.0-dev running on x86_64-unknown-linux-gnu
[01:06:01] 
[01:06:01] note: compiler flags: -Z threads=1 -Z ui-testing -Z unstable-options -C prefer-dynamic -C rpath
[01:06:01] 
[01:06:01] ------------------------------------------
[01:06:01] 
[01:06:01] thread '[ui] ui/lint/issue-54538-unused-parens-lint.rs' panicked at 'explicit panic', src/tools/compiletest/src/runtest.rs:3425:9
---
[01:06:01] 
[01:06:01] thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:516:22
[01:06:01] 
[01:06:01] 
[01:06:01] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[01:06:01] 
[01:06:01] 
[01:06:01] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[01:06:01] Build completed unsuccessfully in 0:04:19
[01:06:01] Build completed unsuccessfully in 0:04:19
[01:06:01] Makefile:48: recipe for target 'check' failed
[01:06:01] make: *** [check] Error 1
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:24026e70
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Thu Apr 11 13:59:59 UTC 2019
---
travis_time:end:06c5d242:start=1554991200814950097,finish=1554991200819428758,duration=4478661
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:07fb5ce2
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb --batch -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:0cbfcd34
travis_time:start:0cbfcd34
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:07c07e5c
$ dmesg | grep -i kill
