\n\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/const-generics/invalid-const-arg-for-type-param.rs","byte_start":84,"byte_end":86,"line_start":6,"line_end":6,"column_start":34,"column_end":36,"is_primary":true,"text":[{"text":"    let _: u32 = 5i32.try_into::<32>().unwrap(); //~ ERROR wrong number of const arguments","highlight_start":34,"highlight_end":36}],"label":"unexpected const argument","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0107]: wrong number of const arguments: expected 0, found 1\n  --> /checkout/src/test/ui/const-generics/invalid-const-arg-for-type-param.rs:6:34\n   |\nLL |     let _: u32 = 5i32.try_into::<32>().unwrap(); //~ ERROR wrong number of const arguments\n   |                                  ^^ unexpected const argument\n\n"}
[00:59:57] {"message":"no method named `f` found for type `S` in the current scope","code":{"code":"E0599","explanation":"\nThis error occurs when a method is used on a type which doesn't implement it:\n\nErroneous code example:\n\n