\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/const-generics/invalid-const-arg-for-type-param.rs","byte_start":28,"byte_end":37,"line_start":3,"line_end":3,"column_start":1,"column_end":10,"is_primary":false,"text":[{"text":"struct S;","highlight_start":1,"highlight_end":10}],"label":"method `f` not found for this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"/checkout/src/test/ui/const-generics/invalid-const-arg-for-type-param.rs","byte_start":148,"byte_end":149,"line_start":7,"line_end":7,"column_start":7,"column_end":8,"is_primary":true,"text":[{"text":"    S.f::<0>(); //~ ERROR no method named `f`","highlight_start":7,"highlight_end":8}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error[E0599]: no method named `f` found for type `S` in the current scope\n  --> /checkout/src/test/ui/const-generics/invalid-const-arg-for-type-param.rs:7:7\n   |\nLL | struct S;\n   | --------- method `f` not found for this\n...\nLL |     S.f::<0>(); //~ ERROR no method named `f`\n   |       ^\n\n"}
[01:08:16] {"message":"wrong number of const arguments: expected 0, found 1","code":{"code":"E0107","explanation":"\nThis error means that an incorrect number of generic arguments were provided:\n\n