plain
2019-10-18T21:59:50.1495189Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-10-18T21:59:50.1692384Z ##[command]git config gc.auto 0
2019-10-18T21:59:50.2080878Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-10-18T21:59:50.2126862Z ##[command]git config --get-all http.proxy
2019-10-18T21:59:50.2262121Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/65539/merge:refs/remotes/pull/65539/merge
---
2019-10-18T22:59:17.3651565Z .................................................................................................... 1600/9196
2019-10-18T22:59:22.4907872Z .................................................................................................... 1700/9196
2019-10-18T22:59:35.2704222Z .............................i...............i...................................................... 1800/9196
2019-10-18T22:59:42.7532032Z .................................................................................................... 1900/9196
2019-10-18T22:59:57.0780768Z ...................iiiii............................................................................ 2000/9196
2019-10-18T23:00:07.6775939Z .................................................................................................... 2200/9196
2019-10-18T23:00:10.2507729Z .................................................................................................... 2300/9196
2019-10-18T23:00:15.6469421Z .................................................................................................... 2400/9196
2019-10-18T23:00:37.6673870Z .................................................................................................... 2500/9196
---
2019-10-18T23:03:28.6006436Z ......................i...............i............................................................. 4800/9196
2019-10-18T23:03:39.7771893Z .................................................................................................... 4900/9196
2019-10-18T23:03:45.8946600Z .................................................................................................... 5000/9196
2019-10-18T23:03:56.1472837Z .................................................................................................... 5100/9196
2019-10-18T23:04:02.3933708Z ......................ii.ii......................................................................... 5200/9196
2019-10-18T23:04:12.6468204Z .................................................................................................... 5400/9196
2019-10-18T23:04:22.2332474Z ........................................................................................i........... 5500/9196
2019-10-18T23:04:30.0696953Z .................................................................................................... 5600/9196
2019-10-18T23:04:34.9225953Z .................................................................................................... 5700/9196
2019-10-18T23:04:34.9225953Z .................................................................................................... 5700/9196
2019-10-18T23:04:45.1388721Z .....................................................................................ii...i..ii..... 5800/9196
2019-10-18T23:05:10.5713715Z .................................................................................................... 6000/9196
2019-10-18T23:05:19.0609038Z .................................................................................................... 6100/9196
2019-10-18T23:05:23.5423793Z .................................................................................................... 6200/9196
2019-10-18T23:05:23.5423793Z .................................................................................................... 6200/9196
2019-10-18T23:05:36.8853132Z .......i..ii........................................................................................ 6300/9196
2019-10-18T23:05:55.8146586Z ...................................................................i................................ 6500/9196
2019-10-18T23:05:57.9151591Z .................................................................................................... 6600/9196
2019-10-18T23:06:00.3084034Z .........................................i.......................................................... 6700/9196
2019-10-18T23:06:04.1083308Z .................................................................................................... 6800/9196
---
2019-10-18T23:10:08.2137386Z 
2019-10-18T23:10:08.2138070Z ---- [ui] ui/imports/reexports.rs stdout ----
2019-10-18T23:10:08.2138310Z diff of stderr:
2019-10-18T23:10:08.2138431Z 
2019-10-18T23:10:08.2138777Z 1 error[E0364]: `foo` is private, and cannot be re-exported
2019-10-18T23:10:08.2139128Z -   --> $DIR/reexports.rs:8:17
2019-10-18T23:10:08.2139450Z +   --> $DIR/reexports.rs:10:17
2019-10-18T23:10:08.2139756Z 4 LL |         pub use super::foo;
2019-10-18T23:10:08.2139878Z 5    |                 ^^^^^^^^^^
2019-10-18T23:10:08.2139988Z 
2019-10-18T23:10:08.2140121Z 6    |
2019-10-18T23:10:08.2140121Z 6    |
2019-10-18T23:10:08.2140417Z 7 note: consider marking `foo` as `pub` in the imported module
2019-10-18T23:10:08.2140797Z -   --> $DIR/reexports.rs:8:17
2019-10-18T23:10:08.2141145Z +   --> $DIR/reexports.rs:10:17
2019-10-18T23:10:08.2141430Z 10 LL |         pub use super::foo;
2019-10-18T23:10:08.2141760Z 11    |                 ^^^^^^^^^^
2019-10-18T23:10:08.2141872Z 
2019-10-18T23:10:08.2141996Z 12 
2019-10-18T23:10:08.2141996Z 12 
2019-10-18T23:10:08.2142135Z 13 error[E0603]: module `foo` is private
2019-10-18T23:10:08.2142442Z -   --> $DIR/reexports.rs:30:15
2019-10-18T23:10:08.2142770Z +   --> $DIR/reexports.rs:34:15
2019-10-18T23:10:08.2142944Z 15    |
2019-10-18T23:10:08.2143631Z 16 LL |     use b::a::foo::S;
2019-10-18T23:10:08.2143942Z 
2019-10-18T23:10:08.2144085Z 18 
2019-10-18T23:10:08.2144085Z 18 
2019-10-18T23:10:08.2144234Z 19 error[E0603]: module `foo` is private
2019-10-18T23:10:08.2144637Z -   --> $DIR/reexports.rs:31:15
2019-10-18T23:10:08.2145031Z +   --> $DIR/reexports.rs:35:15
2019-10-18T23:10:08.2145220Z 21    |
2019-10-18T23:10:08.2145406Z 22 LL |     use b::b::foo::S as T;
2019-10-18T23:10:08.2145998Z 
2019-10-18T23:10:08.2146159Z 24 
2019-10-18T23:10:08.2146159Z 24 
2019-10-18T23:10:08.2146737Z 25 warning: A non-empty glob must import something with the glob's visibility
2019-10-18T23:10:08.2147143Z -   --> $DIR/reexports.rs:9:17
2019-10-18T23:10:08.2147475Z +   --> $DIR/reexports.rs:11:17
2019-10-18T23:10:08.2147785Z 28 LL |         pub use super::*;
2019-10-18T23:10:08.2147907Z 29    |                 ^^^^^^^^
2019-10-18T23:10:08.2148015Z 
2019-10-18T23:10:08.2148121Z 
2019-10-18T23:10:08.2148121Z 
2019-10-18T23:10:08.2148262Z The actual stderr differed from the expected stderr.
2019-10-18T23:10:08.2148808Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/imports/reexports/reexports.stderr
2019-10-18T23:10:08.2149170Z To update references, rerun the tests and pass the `--bless` flag
2019-10-18T23:10:08.2149544Z To only update this specific test, also pass `--test-args imports/reexports.rs`
2019-10-18T23:10:08.2149838Z error: 1 errors occurred comparing output.
2019-10-18T23:10:08.2149977Z status: exit code: 1
2019-10-18T23:10:08.2149977Z status: exit code: 1
2019-10-18T23:10:08.2150689Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/imports/reexports.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/imports/reexports" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/imports/reexports/auxiliary" "-A" "unused"
2019-10-18T23:10:08.2151363Z ------------------------------------------
2019-10-18T23:10:08.2151520Z 
2019-10-18T23:10:08.2151848Z ------------------------------------------
2019-10-18T23:10:08.2152016Z stderr:
2019-10-18T23:10:08.2152016Z stderr:
2019-10-18T23:10:08.2152327Z ------------------------------------------
2019-10-18T23:10:08.2152683Z error[E0364]: `foo` is private, and cannot be re-exported
2019-10-18T23:10:08.2153810Z    |
2019-10-18T23:10:08.2154283Z LL |         pub use super::foo; //~ ERROR cannot be re-exported
2019-10-18T23:10:08.2154480Z    |                 ^^^^^^^^^^
2019-10-18T23:10:08.2154633Z    |
2019-10-18T23:10:08.2154633Z    |
2019-10-18T23:10:08.2154806Z note: consider marking `foo` as `pub` in the imported module
2019-10-18T23:10:08.2155403Z    |
2019-10-18T23:10:08.2155793Z LL |         pub use super::foo; //~ ERROR cannot be re-exported
2019-10-18T23:10:08.2155982Z    |                 ^^^^^^^^^^
2019-10-18T23:10:08.2156117Z 
2019-10-18T23:10:08.2156117Z 
2019-10-18T23:10:08.2156293Z error[E0603]: module `foo` is private
2019-10-18T23:10:08.2157267Z    |
2019-10-18T23:10:08.2157267Z    |
2019-10-18T23:10:08.2157395Z LL |     use b::a::foo::S; //~ ERROR `foo`
2019-10-18T23:10:08.2157643Z 
2019-10-18T23:10:08.2157643Z 
2019-10-18T23:10:08.2157763Z error[E0603]: module `foo` is private
2019-10-18T23:10:08.2158487Z    |
2019-10-18T23:10:08.2158487Z    |
2019-10-18T23:10:08.2158621Z LL |     use b::b::foo::S as T; //~ ERROR `foo`
2019-10-18T23:10:08.2159043Z 
2019-10-18T23:10:08.2159043Z 
2019-10-18T23:10:08.2159391Z warning: A non-empty glob must import something with the glob's visibility
2019-10-18T23:10:08.2160232Z    |
2019-10-18T23:10:08.2160232Z    |
2019-10-18T23:10:08.2160613Z LL |         pub use super::*; //~ WARNING must import something with the glob's visibility [unused_imports]
2019-10-18T23:10:08.2160922Z    |
2019-10-18T23:10:08.2161054Z note: lint level defined here
2019-10-18T23:10:08.2161369Z   --> /checkout/src/test/ui/imports/reexports.rs:1:9
2019-10-18T23:10:08.2161543Z    |
---
2019-10-18T23:10:08.2171120Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:537:22
2019-10-18T23:10:08.2171429Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
2019-10-18T23:10:08.2187317Z 
2019-10-18T23:10:08.2188050Z 
2019-10-18T23:10:08.2191976Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2019-10-18T23:10:08.2192726Z 
2019-10-18T23:10:08.2192754Z 
2019-10-18T23:10:08.2202560Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2019-10-18T23:10:08.2202638Z Build completed unsuccessfully in 1:03:39
2019-10-18T23:10:08.2202638Z Build completed unsuccessfully in 1:03:39
2019-10-18T23:10:08.2254076Z == clock drift check ==
2019-10-18T23:10:08.2268670Z   local time: Fri Oct 18 23:10:08 UTC 2019
2019-10-18T23:10:10.2802493Z   network time: == end clock drift check ==
2019-10-18T23:10:11.4344840Z 
2019-10-18T23:10:11.4469984Z ##[error]Bash exited with code '1'.
2019-10-18T23:10:11.4506563Z ##[section]Starting: Checkout
2019-10-18T23:10:11.4508301Z ==============================================================================
2019-10-18T23:10:11.4508357Z Task         : Get sources
2019-10-18T23:10:11.4508422Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
