plain
2019-10-24T06:11:23.5889279Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-10-24T06:11:23.6160285Z ##[command]git config gc.auto 0
2019-10-24T06:11:23.6210363Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-10-24T06:11:23.6275078Z ##[command]git config --get-all http.proxy
2019-10-24T06:11:23.6438022Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/65752/merge:refs/remotes/pull/65752/merge
---
2019-10-24T07:15:15.0257159Z .................................................................................................... 1600/9240
2019-10-24T07:15:20.7758595Z .................................................................................................... 1700/9240
2019-10-24T07:15:33.4716545Z ...................................................i...............i................................ 1800/9240
2019-10-24T07:15:41.8953346Z .................................................................................................... 1900/9240
2019-10-24T07:15:56.5541829Z .........................................iiiii...................................................... 2000/9240
2019-10-24T07:16:07.5437542Z .................................................................................................... 2200/9240
2019-10-24T07:16:10.1703579Z .................................................................................................... 2300/9240
2019-10-24T07:16:14.6276373Z .................................................................................................... 2400/9240
2019-10-24T07:16:38.0900500Z .................................................................................................... 2500/9240
---
2019-10-24T07:19:35.6496424Z .............................................i...............i...................................... 4800/9240
2019-10-24T07:19:45.1492922Z .................................................................................................... 4900/9240
2019-10-24T07:19:53.6816682Z .................................................................................................... 5000/9240
2019-10-24T07:20:00.9113769Z .................................................................................................... 5100/9240
2019-10-24T07:20:11.1834193Z .............................................ii.ii.................................................. 5200/9240
2019-10-24T07:20:21.5830844Z .................................................................................................... 5400/9240
2019-10-24T07:20:31.5424095Z .................................................................................................... 5500/9240
2019-10-24T07:20:39.2869793Z ............i....................................................................................... 5600/9240
2019-10-24T07:20:45.1218053Z .................................................................................................... 5700/9240
2019-10-24T07:20:45.1218053Z .................................................................................................... 5700/9240
2019-10-24T07:20:57.4285093Z .................................................................................................... 5800/9240
2019-10-24T07:21:09.7288481Z .........ii...i..ii...........i..................................................................... 5900/9240
2019-10-24T07:21:32.6469032Z .................................................................................................... 6100/9240
2019-10-24T07:21:38.6546028Z .................................................................................................... 6200/9240
2019-10-24T07:21:38.6546028Z .................................................................................................... 6200/9240
2019-10-24T07:21:52.9656280Z ...............................i..ii................................................................ 6300/9240
2019-10-24T07:22:13.9993029Z .................................................................................................i.. 6500/9240
2019-10-24T07:22:16.2916938Z .................................................................................................... 6600/9240
2019-10-24T07:22:18.5925948Z ........................................................................i........................... 6700/9240
2019-10-24T07:22:21.6519451Z .................................................................................................... 6800/9240
---
2019-10-24T07:26:58.3218937Z  finished in 5.748
2019-10-24T07:26:58.3407383Z Check compiletest suite=codegen mode=codegen (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-24T07:26:58.5109579Z 
2019-10-24T07:26:58.5109838Z running 153 tests
2019-10-24T07:27:01.7238286Z i....iii......iii..iiii....i............................i..i..................i....i...........ii.i. 100/153
2019-10-24T07:27:03.7035244Z i..iiii..............i.........iii.i.........ii......
2019-10-24T07:27:03.7035935Z 
2019-10-24T07:27:03.7038904Z  finished in 5.363
2019-10-24T07:27:03.7233888Z Check compiletest suite=codegen-units mode=codegen-units (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-24T07:27:03.8839215Z 
---
2019-10-24T07:27:05.9739647Z  finished in 2.250
2019-10-24T07:27:05.9942866Z Check compiletest suite=assembly mode=assembly (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-24T07:27:06.1565397Z 
2019-10-24T07:27:06.1566532Z running 9 tests
2019-10-24T07:27:06.1568660Z iiiiiiiii
2019-10-24T07:27:06.1569113Z 
2019-10-24T07:27:06.1569190Z  finished in 0.162
2019-10-24T07:27:06.1751782Z Check compiletest suite=incremental mode=incremental (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-24T07:27:06.3582804Z 
---
2019-10-24T07:27:24.9719874Z  finished in 18.796
2019-10-24T07:27:24.9932972Z Check compiletest suite=debuginfo mode=debuginfo-gdb+lldb (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-24T07:27:25.1715384Z 
2019-10-24T07:27:25.1715770Z running 123 tests
2019-10-24T07:27:50.5723561Z .iiiii...i.....i..i...i..i.i.i..i.ii..i.i.....i..i....ii..........iiii..........i...ii...i.......ii. 100/123
2019-10-24T07:27:55.4121928Z i.i.i......iii.i.....ii
2019-10-24T07:27:55.4124153Z 
2019-10-24T07:27:55.4129464Z  finished in 30.420
2019-10-24T07:27:55.4141963Z Uplifting stage1 rustc (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-24T07:27:55.4143337Z Copying stage2 rustc from stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu / x86_64-unknown-linux-gnu)
---
2019-10-24T07:41:03.8797900Z 
2019-10-24T07:41:03.8798869Z    Doc-tests core
2019-10-24T07:41:09.0170219Z 
2019-10-24T07:41:09.0171874Z running 2406 tests
2019-10-24T07:41:20.2766776Z ......iiiii......................................................................................... 100/2406
2019-10-24T07:41:31.3259592Z ................................................................................ii.................. 200/2406
2019-10-24T07:41:57.3827421Z ..i................................................................................................. 400/2406
2019-10-24T07:41:57.3827421Z ..i................................................................................................. 400/2406
2019-10-24T07:42:08.2960325Z .................................................i..i.................iiii.......................... 500/2406
2019-10-24T07:42:28.7959412Z .................................................................................................... 700/2406
2019-10-24T07:42:39.4064436Z .................................................................................................... 800/2406
2019-10-24T07:42:49.9499150Z .................................................................................................... 900/2406
2019-10-24T07:43:01.2148691Z .................................................................................................... 1000/2406
---
2019-10-24T07:47:19.6434199Z 
2019-10-24T07:47:19.6434478Z running 997 tests
2019-10-24T07:47:42.0529804Z i................................................................................................... 100/997
2019-10-24T07:47:54.2506361Z .................................................................................................... 200/997
2019-10-24T07:48:02.9232769Z ...................iii......i......i...i......i..................................................... 300/997
2019-10-24T07:48:08.7424786Z .................................................................................................... 400/997
2019-10-24T07:48:16.7633722Z ........................................i..i.................................ii..................... 500/997
2019-10-24T07:48:31.9157702Z .................................................................................................... 700/997
2019-10-24T07:48:31.9157702Z .................................................................................................... 700/997
2019-10-24T07:48:39.8995850Z .......................iiii......................................................................... 800/997
2019-10-24T07:48:56.1384346Z .................................................................................................... 900/997
2019-10-24T07:49:04.2158048Z .............................................iiii................................................
2019-10-24T07:49:04.2160967Z 
2019-10-24T07:49:04.2220663Z  finished in 203.079
2019-10-24T07:49:04.2238836Z Testing term stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-24T07:49:04.4515964Z    Compiling term v0.0.0 (/checkout/src/libterm)
---
2019-10-24T08:05:35.3502243Z  finished in 41.906
2019-10-24T08:05:35.3858241Z Check compiletest suite=run-make-fulldeps mode=run-make (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-24T08:05:35.5993933Z 
2019-10-24T08:05:35.5995008Z running 201 tests
2019-10-24T08:06:12.4164856Z ....................i...ii.................................................................i........ 100/201
2019-10-24T08:07:03.6356294Z ...............................iiii.......i...........iiii.iii....................................i. 200/201
2019-10-24T08:07:03.8715450Z test result: ok. 184 passed; 0 failed; 17 ignored; 0 measured; 0 filtered out
2019-10-24T08:07:03.8715492Z 
2019-10-24T08:07:03.8720101Z  finished in 88.486
2019-10-24T08:07:03.8730565Z doc tests for: /checkout/src/doc/rustdoc/src/command-line-arguments.md
---
2019-10-24T08:07:34.2427413Z ---- [ui] rustdoc-ui/intra-links-ambiguity.rs stdout ----
2019-10-24T08:07:34.2427466Z diff of stderr:
2019-10-24T08:07:34.2427499Z 
2019-10-24T08:07:34.2427557Z 23    |
2019-10-24T08:07:34.2427603Z 24 LL | /// [ambiguous] is ambiguous.
2019-10-24T08:07:34.2427647Z 25    |      ^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2428035Z 26 help: to link to the struct, prefix with the item type
2019-10-24T08:07:34.2428096Z 27    |
2019-10-24T08:07:34.2428096Z 27    |
2019-10-24T08:07:34.2428144Z 28 LL | /// [struct@ambiguous] is ambiguous.
2019-10-24T08:07:34.2428239Z 37    |
2019-10-24T08:07:34.2428239Z 37    |
2019-10-24T08:07:34.2428601Z 38 LL | /// [`multi_conflict`] is a three-way conflict.
2019-10-24T08:07:34.2428656Z 39    |      ^^^^^^^^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2428791Z 40 help: to link to the struct, prefix with the item type
2019-10-24T08:07:34.2428838Z 41    |
2019-10-24T08:07:34.2428838Z 41    |
2019-10-24T08:07:34.2429138Z 42 LL | /// [`struct@multi_conflict`] is a three-way conflict.
2019-10-24T08:07:34.2429224Z 55    |
2019-10-24T08:07:34.2429224Z 55    |
2019-10-24T08:07:34.2429269Z 56 LL | /// Ambiguous [type_and_value].
2019-10-24T08:07:34.2429336Z 57    |                ^^^^^^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2429429Z 58 help: to link to the module, prefix with the item type
2019-10-24T08:07:34.2429491Z 59    |
2019-10-24T08:07:34.2429491Z 59    |
2019-10-24T08:07:34.2429536Z 60 LL | /// Ambiguous [module@type_and_value].
2019-10-24T08:07:34.2429607Z 69    |
2019-10-24T08:07:34.2429607Z 69    |
2019-10-24T08:07:34.2429911Z 70 LL | /// Ambiguous non-implied shortcut link [`foo::bar`].
2019-10-24T08:07:34.2429970Z 71    |                                          ^^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2430017Z +    |
2019-10-24T08:07:34.2430090Z 72 help: to link to the enum, prefix with the item type
2019-10-24T08:07:34.2430143Z 73    |
2019-10-24T08:07:34.2430427Z 74 LL | /// Ambiguous non-implied shortcut link [`enum@foo::bar`].
2019-10-24T08:07:34.2430507Z 
2019-10-24T08:07:34.2430553Z The actual stderr differed from the expected stderr.
2019-10-24T08:07:34.2430907Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/intra-links-ambiguity/intra-links-ambiguity.stderr
2019-10-24T08:07:34.2430907Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/intra-links-ambiguity/intra-links-ambiguity.stderr
2019-10-24T08:07:34.2431220Z To update references, rerun the tests and pass the `--bless` flag
2019-10-24T08:07:34.2431520Z To only update this specific test, also pass `--test-args intra-links-ambiguity.rs`
2019-10-24T08:07:34.2431621Z error: 1 errors occurred comparing output.
2019-10-24T08:07:34.2431668Z status: exit code: 1
2019-10-24T08:07:34.2431668Z status: exit code: 1
2019-10-24T08:07:34.2434844Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "/checkout/src/test/rustdoc-ui/intra-links-ambiguity.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/intra-links-ambiguity" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/intra-links-ambiguity/auxiliary"
2019-10-24T08:07:34.2435906Z ------------------------------------------
2019-10-24T08:07:34.2435958Z 
2019-10-24T08:07:34.2436505Z ------------------------------------------
2019-10-24T08:07:34.2436568Z stderr:
2019-10-24T08:07:34.2436568Z stderr:
2019-10-24T08:07:34.2436874Z ------------------------------------------
2019-10-24T08:07:34.2436955Z error: `ambiguous` is both a struct and a function
2019-10-24T08:07:34.2437578Z    |
2019-10-24T08:07:34.2437578Z    |
2019-10-24T08:07:34.2437655Z LL | /// [`ambiguous`] is ambiguous. //~ERROR `ambiguous`
2019-10-24T08:07:34.2438300Z    |      ^^^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2438500Z note: lint level defined here
2019-10-24T08:07:34.2439162Z   --> /checkout/src/test/rustdoc-ui/intra-links-ambiguity.rs:1:9
2019-10-24T08:07:34.2439229Z    |
2019-10-24T08:07:34.2439298Z LL | #![deny(intra_doc_link_resolution_failure)]
2019-10-24T08:07:34.2439298Z LL | #![deny(intra_doc_link_resolution_failure)]
2019-10-24T08:07:34.2439346Z    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2019-10-24T08:07:34.2439393Z help: to link to the struct, prefix with the item type
2019-10-24T08:07:34.2439437Z    |
2019-10-24T08:07:34.2439500Z LL | /// [`struct@ambiguous`] is ambiguous. //~ERROR `ambiguous`
2019-10-24T08:07:34.2439591Z help: to link to the function, add parentheses
2019-10-24T08:07:34.2439798Z    |
2019-10-24T08:07:34.2439798Z    |
2019-10-24T08:07:34.2439875Z LL | /// [`ambiguous()`] is ambiguous. //~ERROR `ambiguous`
2019-10-24T08:07:34.2439950Z 
2019-10-24T08:07:34.2439950Z 
2019-10-24T08:07:34.2440018Z error: `ambiguous` is both a struct and a function
2019-10-24T08:07:34.2440433Z    |
2019-10-24T08:07:34.2440433Z    |
2019-10-24T08:07:34.2440716Z LL | /// [ambiguous] is ambiguous. //~ERROR ambiguous
2019-10-24T08:07:34.2440770Z    |      ^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2440872Z help: to link to the struct, prefix with the item type
2019-10-24T08:07:34.2440915Z    |
2019-10-24T08:07:34.2440915Z    |
2019-10-24T08:07:34.2440961Z LL | /// [struct@ambiguous] is ambiguous. //~ERROR ambiguous
2019-10-24T08:07:34.2441072Z help: to link to the function, add parentheses
2019-10-24T08:07:34.2441114Z    |
2019-10-24T08:07:34.2441114Z    |
2019-10-24T08:07:34.2441158Z LL | /// [ambiguous()] is ambiguous. //~ERROR ambiguous
2019-10-24T08:07:34.2441249Z 
2019-10-24T08:07:34.2441249Z 
2019-10-24T08:07:34.2441446Z error: `multi_conflict` is a struct, a function, and a macro
2019-10-24T08:07:34.2441885Z    |
2019-10-24T08:07:34.2441885Z    |
2019-10-24T08:07:34.2442411Z LL | /// [`multi_conflict`] is a three-way conflict. //~ERROR `multi_conflict`
2019-10-24T08:07:34.2442483Z    |      ^^^^^^^^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2442599Z help: to link to the struct, prefix with the item type
2019-10-24T08:07:34.2442643Z    |
2019-10-24T08:07:34.2442643Z    |
2019-10-24T08:07:34.2443222Z LL | /// [`struct@multi_conflict`] is a three-way conflict. //~ERROR `multi_conflict`
2019-10-24T08:07:34.2443347Z help: to link to the function, add parentheses
2019-10-24T08:07:34.2443417Z    |
2019-10-24T08:07:34.2443417Z    |
2019-10-24T08:07:34.2443952Z LL | /// [`multi_conflict()`] is a three-way conflict. //~ERROR `multi_conflict`
2019-10-24T08:07:34.2444097Z help: to link to the macro, add an exclamation mark
2019-10-24T08:07:34.2444143Z    |
2019-10-24T08:07:34.2444143Z    |
2019-10-24T08:07:34.2444667Z LL | /// [`multi_conflict!`] is a three-way conflict. //~ERROR `multi_conflict`
2019-10-24T08:07:34.2444962Z 
2019-10-24T08:07:34.2445008Z error: `type_and_value` is both a module and a constant
2019-10-24T08:07:34.2445655Z   --> /checkout/src/test/rustdoc-ui/intra-links-ambiguity.rs:33:16
2019-10-24T08:07:34.2445753Z    |
2019-10-24T08:07:34.2445753Z    |
2019-10-24T08:07:34.2445801Z LL | /// Ambiguous [type_and_value]. //~ERROR type_and_value
2019-10-24T08:07:34.2445851Z    |                ^^^^^^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2445963Z help: to link to the module, prefix with the item type
2019-10-24T08:07:34.2446007Z    |
2019-10-24T08:07:34.2446007Z    |
2019-10-24T08:07:34.2446055Z LL | /// Ambiguous [module@type_and_value]. //~ERROR type_and_value
2019-10-24T08:07:34.2446361Z help: to link to the constant, prefix with the item type
2019-10-24T08:07:34.2446406Z    |
2019-10-24T08:07:34.2446406Z    |
2019-10-24T08:07:34.2446612Z LL | /// Ambiguous [const@type_and_value]. //~ERROR type_and_value
2019-10-24T08:07:34.2446711Z 
2019-10-24T08:07:34.2446776Z error: `foo::bar` is both an enum and a function
2019-10-24T08:07:34.2447434Z   --> /checkout/src/test/rustdoc-ui/intra-links-ambiguity.rs:35:42
2019-10-24T08:07:34.2447500Z    |
2019-10-24T08:07:34.2447500Z    |
2019-10-24T08:07:34.2448293Z LL | /// Ambiguous non-implied shortcut link [`foo::bar`]. //~ERROR `foo::bar`
2019-10-24T08:07:34.2448401Z    |                                          ^^^^^^^^^^ ambiguous link
2019-10-24T08:07:34.2448446Z    |
2019-10-24T08:07:34.2448490Z help: to link to the enum, prefix with the item type
2019-10-24T08:07:34.2448554Z    |
2019-10-24T08:07:34.2449174Z LL | /// Ambiguous non-implied shortcut link [`enum@foo::bar`]. //~ERROR `foo::bar`
2019-10-24T08:07:34.2449324Z help: to link to the function, add parentheses
2019-10-24T08:07:34.2449367Z    |
2019-10-24T08:07:34.2449367Z    |
2019-10-24T08:07:34.2449884Z LL | /// Ambiguous non-implied shortcut link [`foo::bar()`]. //~ERROR `foo::bar`
2019-10-24T08:07:34.2450023Z 
2019-10-24T08:07:34.2450068Z error: aborting due to 5 previous errors
2019-10-24T08:07:34.2450100Z 
2019-10-24T08:07:34.2450126Z 
2019-10-24T08:07:34.2450126Z 
2019-10-24T08:07:34.2450731Z ------------------------------------------
2019-10-24T08:07:34.2450777Z 
2019-10-24T08:07:34.2450804Z 
2019-10-24T08:07:34.2451100Z ---- [ui] rustdoc-ui/invalid-syntax.rs stdout ----
2019-10-24T08:07:34.2451179Z diff of stderr:
2019-10-24T08:07:34.2451207Z 
2019-10-24T08:07:34.2451786Z 24 LL | | /// \__________pkt->size___________/          \_result->size_/ \__pkt->size__/
2019-10-24T08:07:34.2451916Z 26    | |_______^
2019-10-24T08:07:34.2451958Z +    |
2019-10-24T08:07:34.2452005Z 27 help: mark blocks that do not contain Rust code as text
2019-10-24T08:07:34.2452273Z 28    |
---
2019-10-24T08:07:34.2465750Z 269 3 |    |     ^^^^^^ did you mean 'baz::foobar`?
2019-10-24T08:07:34.2465784Z 
2019-10-24T08:07:34.2465809Z 
2019-10-24T08:07:34.2465854Z The actual stderr differed from the expected stderr.
2019-10-24T08:07:34.2466473Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/invalid-syntax/invalid-syntax.stderr
2019-10-24T08:07:34.2466767Z To update references, rerun the tests and pass the `--bless` flag
2019-10-24T08:07:34.2467320Z To only update this specific test, also pass `--test-args invalid-syntax.rs`
2019-10-24T08:07:34.2467432Z error: 1 errors occurred comparing output.
2019-10-24T08:07:34.2467476Z status: exit code: 0
2019-10-24T08:07:34.2467476Z status: exit code: 0
2019-10-24T08:07:34.2468969Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "/checkout/src/test/rustdoc-ui/invalid-syntax.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/invalid-syntax" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/invalid-syntax/auxiliary"
2019-10-24T08:07:34.2469824Z ------------------------------------------
2019-10-24T08:07:34.2469862Z 
2019-10-24T08:07:34.2470335Z ------------------------------------------
2019-10-24T08:07:34.2470396Z stderr:
2019-10-24T08:07:34.2470396Z stderr:
2019-10-24T08:07:34.2470704Z ------------------------------------------
2019-10-24T08:07:34.2470754Z error: unknown start of token: \
2019-10-24T08:07:34.2471239Z  --> <doctest>:1:1
2019-10-24T08:07:34.2471322Z   |
2019-10-24T08:07:34.2471597Z 1 | \__________pkt->size___________/          \_result->size_/ \__pkt->size__/
2019-10-24T08:07:34.2471838Z 
2019-10-24T08:07:34.2471955Z error: unknown start of token: \
2019-10-24T08:07:34.2472242Z  --> <doctest>:1:43
2019-10-24T08:07:34.2472672Z   |
2019-10-24T08:07:34.2472672Z   |
2019-10-24T08:07:34.2473137Z 1 | \__________pkt->size___________/          \_result->size_/ \__pkt->size__/
2019-10-24T08:07:34.2473223Z 
2019-10-24T08:07:34.2473263Z error: unknown start of token: \
2019-10-24T08:07:34.2473790Z  --> <doctest>:1:60
2019-10-24T08:07:34.2473849Z   |
2019-10-24T08:07:34.2473849Z   |
2019-10-24T08:07:34.2474114Z 1 | \__________pkt->size___________/          \_result->size_/ \__pkt->size__/
2019-10-24T08:07:34.2474438Z 
2019-10-24T08:07:34.2474481Z warning: could not parse code block as Rust code
2019-10-24T08:07:34.2474831Z   --> /checkout/src/test/rustdoc-ui/invalid-syntax.rs:3:5
2019-10-24T08:07:34.2474882Z    |
2019-10-24T08:07:34.2474882Z    |
2019-10-24T08:07:34.2474922Z LL |   /// 