plain
2019-11-03T20:57:17.2102585Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-11-03T20:57:17.2281336Z ##[command]git config gc.auto 0
2019-11-03T20:57:17.2407942Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-11-03T20:57:17.2413897Z ##[command]git config --get-all http.proxy
2019-11-03T20:57:17.2566586Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/66066/merge:refs/remotes/pull/66066/merge
---
2019-11-03T21:57:22.8353308Z .................................................................................................... 1600/9268
2019-11-03T21:57:28.4984085Z .................................................................................................... 1700/9268
2019-11-03T21:57:41.0993399Z ..............................................................i...............i..................... 1800/9268
2019-11-03T21:57:48.6513989Z .................................................................................................... 1900/9268
2019-11-03T21:58:04.1032567Z ....................................................iiiii........................................... 2000/9268
2019-11-03T21:58:15.1606339Z .................................................................................................... 2200/9268
2019-11-03T21:58:17.7295492Z .................................................................................................... 2300/9268
2019-11-03T21:58:21.2073180Z .................................................................................................... 2400/9268
2019-11-03T21:58:44.6012460Z .................................................................................................... 2500/9268
2019-11-03T21:58:44.6012460Z .................................................................................................... 2500/9268
2019-11-03T21:58:47.3372040Z .................................................................................................... 2600/9268
2019-11-03T21:58:55.2350097Z .................................................................................................... 2700/9268
2019-11-03T21:59:04.2726842Z ....................i............................................................................... 2800/9268
2019-11-03T21:59:13.3384627Z .................................................................................................... 2900/9268
2019-11-03T21:59:17.9447611Z ...................i................................................................................ 3000/9268
2019-11-03T21:59:26.6939006Z .................................................................................................... 3100/9268
2019-11-03T21:59:32.3472231Z .................................................................................................... 3200/9268
2019-11-03T21:59:41.2768715Z .ii................................................................................................. 3300/9268
2019-11-03T21:59:59.1494572Z ..............................................................................................i..... 3500/9268
2019-11-03T22:00:06.0209366Z .........................................i.......................................................... 3600/9268
2019-11-03T22:00:13.0140388Z .................................................................................................... 3700/9268
2019-11-03T22:00:19.7677324Z .................................................................................................... 3800/9268
---
2019-11-03T22:01:38.6194560Z ....................................................i...............i............................... 4800/9268
2019-11-03T22:01:47.6087400Z .................................................................................................... 4900/9268
2019-11-03T22:01:56.2986914Z .................................................................................................... 5000/9268
2019-11-03T22:02:02.3085722Z .................................................................................................... 5100/9268
2019-11-03T22:02:12.8007871Z .....................................................ii.ii...........i.............................. 5200/9268
2019-11-03T22:02:22.5538022Z .................................................................................................... 5400/9268
2019-11-03T22:02:32.7217044Z .................................................................................................... 5500/9268
2019-11-03T22:02:40.0561557Z ..........................i......................................................................... 5600/9268
2019-11-03T22:02:46.5816774Z .................................................................................................... 5700/9268
2019-11-03T22:02:46.5816774Z .................................................................................................... 5700/9268
2019-11-03T22:02:58.4964067Z .................................................................................................... 5800/9268
2019-11-03T22:03:10.2428811Z ...........ii...i..ii...........i................................................................... 5900/9268
2019-11-03T22:03:30.7408374Z .................................................................................................... 6100/9268
2019-11-03T22:03:38.8428131Z .................................................................................................... 6200/9268
2019-11-03T22:03:38.8428131Z .................................................................................................... 6200/9268
2019-11-03T22:03:52.7867762Z ..............................i..ii................................................................. 6300/9268
2019-11-03T22:04:13.2797058Z .................................................................................................i.. 6500/9268
2019-11-03T22:04:15.4855638Z .................................................................................................... 6600/9268
2019-11-03T22:04:17.7458431Z ........................................................................i........................... 6700/9268
2019-11-03T22:04:20.6448639Z .................................................................................................... 6800/9268
---
2019-11-03T22:06:00.3809820Z .................................................................................................... 7400/9268
2019-11-03T22:06:05.8422668Z .................................................................................................... 7500/9268
2019-11-03T22:06:14.8924339Z .................................................................................................... 7600/9268
2019-11-03T22:06:25.9694397Z .................................................................................................... 7700/9268
2019-11-03T22:06:35.5637563Z ...........................................................................................ii......i 7800/9268
2019-11-03T22:06:46.5649386Z .................................................................................................... 8000/9268
2019-11-03T22:07:02.9017679Z .................................................................................................... 8100/9268
2019-11-03T22:07:11.4696017Z .................................................................................................... 8200/9268
2019-11-03T22:07:21.0312042Z .................................................................................................... 8300/9268
---
2019-11-03T22:09:10.5580637Z 17 
2019-11-03T22:09:10.5580815Z + error[E0384]: cannot assign twice to immutable variable `_`
2019-11-03T22:09:10.5581205Z +   --> $DIR/issue-52475.rs:10:18
2019-11-03T22:09:10.5581436Z +    |
2019-11-03T22:09:10.5581626Z + LL |             x = &0; // Materialize a new AllocId
2019-11-03T22:09:10.5581965Z +    |                  ^ cannot assign twice to immutable variable
2019-11-03T22:09:10.5582315Z 18 warning: Constant evaluating a complex constant, this might take some time
2019-11-03T22:09:10.5582686Z 19   --> $DIR/issue-52475.rs:2:18
2019-11-03T22:09:10.5582936Z 20    |
2019-11-03T22:09:10.5583088Z 
2019-11-03T22:09:10.5583088Z 
2019-11-03T22:09:10.5583686Z 34 LL |             n = (n + 1) % 5;
2019-11-03T22:09:10.5583960Z 35    |                 ^^^^^^^^^^^ duplicate interpreter state observed here, const evaluation will never terminate
2019-11-03T22:09:10.5584852Z - error: aborting due to 3 previous errors
2019-11-03T22:09:10.5585210Z + error: aborting due to 4 previous errors
2019-11-03T22:09:10.5585421Z 38 
2019-11-03T22:09:10.5585916Z - Some errors have detailed explanations: E0019, E0080.
2019-11-03T22:09:10.5585916Z - Some errors have detailed explanations: E0019, E0080.
2019-11-03T22:09:10.5586233Z + Some errors have detailed explanations: E0019, E0080, E0384.
2019-11-03T22:09:10.5586724Z 40 For more information about an error, try `rustc --explain E0019`.
2019-11-03T22:09:10.5587145Z 41 
2019-11-03T22:09:10.5587345Z 
2019-11-03T22:09:10.5587501Z 
2019-11-03T22:09:10.5587680Z The actual stderr differed from the expected stderr.
2019-11-03T22:09:10.5588541Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-eval/issue-52475/issue-52475.stderr
2019-11-03T22:09:10.5589011Z To update references, rerun the tests and pass the `--bless` flag
2019-11-03T22:09:10.5589501Z To only update this specific test, also pass `--test-args consts/const-eval/issue-52475.rs`
2019-11-03T22:09:10.5590628Z error: 1 errors occurred comparing output.
2019-11-03T22:09:10.5590796Z status: exit code: 1
2019-11-03T22:09:10.5590796Z status: exit code: 1
2019-11-03T22:09:10.5591633Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/consts/const-eval/issue-52475.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-eval/issue-52475" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-eval/issue-52475/auxiliary" "-A" "unused"
2019-11-03T22:09:10.5592362Z ------------------------------------------
2019-11-03T22:09:10.5592750Z 
2019-11-03T22:09:10.5593647Z ------------------------------------------
2019-11-03T22:09:10.5593985Z stderr:
---
2019-11-03T22:09:10.5596208Z 
2019-11-03T22:09:10.5596434Z error[E0019]: constant contains unimplemented expression type
2019-11-03T22:09:10.5597062Z   --> /checkout/src/test/ui/consts/const-eval/issue-52475.rs:6:9
2019-11-03T22:09:10.5597317Z    |
2019-11-03T22:09:10.5597525Z LL | /         while n < 5 {
2019-11-03T22:09:10.5597711Z LL | |         //~^ ERROR constant contains unimplemented expression type
2019-11-03T22:09:10.5597920Z LL | |         //~| ERROR constant contains unimplemented expression type
2019-11-03T22:09:10.5598131Z LL | |             n = (n + 1) % 5; //~ ERROR evaluation of constant value failed
2019-11-03T22:09:10.5598331Z LL | |             x = &0; // Materialize a new AllocId
2019-11-03T22:09:10.5599038Z    | |_________^
2019-11-03T22:09:10.5599177Z 
2019-11-03T22:09:10.5599355Z error[E0384]: cannot assign twice to immutable variable `_`
2019-11-03T22:09:10.5599739Z   --> /checkout/src/test/ui/consts/const-eval/issue-52475.rs:10:18
2019-11-03T22:09:10.5599739Z   --> /checkout/src/test/ui/consts/const-eval/issue-52475.rs:10:18
2019-11-03T22:09:10.5599961Z    |
2019-11-03T22:09:10.5600148Z LL |             x = &0; // Materialize a new AllocId
2019-11-03T22:09:10.5600310Z    |                  ^ cannot assign twice to immutable variable
2019-11-03T22:09:10.5600637Z warning: Constant evaluating a complex constant, this might take some time
2019-11-03T22:09:10.5601016Z   --> /checkout/src/test/ui/consts/const-eval/issue-52475.rs:2:18
2019-11-03T22:09:10.5601240Z    |
2019-11-03T22:09:10.5601240Z    |
2019-11-03T22:09:10.5601443Z LL |       let _ = [(); {
2019-11-03T22:09:10.5601602Z    |  __________________^
2019-11-03T22:09:10.5601763Z LL | |         //~^ WARNING Constant evaluating a complex constant, this might take some time
2019-11-03T22:09:10.5602047Z LL | |         let mut x = &0;
2019-11-03T22:09:10.5602263Z LL | |         let mut n = 0;
2019-11-03T22:09:10.5602599Z LL | |         0
2019-11-03T22:09:10.5603345Z LL | |     }];
2019-11-03T22:09:10.5603809Z    | |_____^
2019-11-03T22:09:10.5603980Z 
2019-11-03T22:09:10.5603980Z 
2019-11-03T22:09:10.5604123Z error[E0080]: evaluation of constant value failed
2019-11-03T22:09:10.5604655Z   --> /checkout/src/test/ui/consts/const-eval/issue-52475.rs:9:17
2019-11-03T22:09:10.5604883Z    |
2019-11-03T22:09:10.5605029Z LL |             n = (n + 1) % 5; //~ ERROR evaluation of constant value failed
2019-11-03T22:09:10.5605202Z    |                 ^^^^^^^^^^^ duplicate interpreter state observed here, const evaluation will never terminate
2019-11-03T22:09:10.5605490Z error: aborting due to 4 previous errors
2019-11-03T22:09:10.5605628Z 
2019-11-03T22:09:10.5605770Z Some errors have detailed explanations: E0019, E0080, E0384.
2019-11-03T22:09:10.5606193Z For more information about an error, try `rustc --explain E0019`.
---
2019-11-03T22:09:10.5614017Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:537:22
2019-11-03T22:09:10.5614296Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
2019-11-03T22:09:10.5629411Z 
2019-11-03T22:09:10.5630164Z 
2019-11-03T22:09:10.5633693Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2019-11-03T22:09:10.5634308Z 
2019-11-03T22:09:10.5634437Z 
2019-11-03T22:09:10.5642229Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2019-11-03T22:09:10.5642472Z Build completed unsuccessfully in 1:05:09
2019-11-03T22:09:10.5642472Z Build completed unsuccessfully in 1:05:09
2019-11-03T22:09:10.5694516Z == clock drift check ==
2019-11-03T22:09:10.5714422Z   local time: Sun Nov  3 22:09:10 UTC 2019
2019-11-03T22:09:10.8487258Z   network time: Sun, 03 Nov 2019 22:09:10 GMT
2019-11-03T22:09:10.8491606Z == end clock drift check ==
2019-11-03T22:09:13.2868725Z 
2019-11-03T22:09:13.2946674Z ##[error]Bash exited with code '1'.
2019-11-03T22:09:13.2979569Z ##[section]Starting: Checkout
2019-11-03T22:09:13.2981572Z ==============================================================================
2019-11-03T22:09:13.2981642Z Task         : Get sources
2019-11-03T22:09:13.2981686Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
