plain
2019-11-11T22:39:00.0070706Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-11-11T22:39:00.0309920Z ##[command]git config gc.auto 0
2019-11-11T22:39:00.0392459Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-11-11T22:39:00.0445474Z ##[command]git config --get-all http.proxy
2019-11-11T22:39:00.0617347Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/66314/merge:refs/remotes/pull/66314/merge
---
2019-11-11T23:37:37.4702588Z Check compiletest suite=ui mode=ui (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-11-11T23:37:37.9825116Z 
2019-11-11T23:37:37.9825340Z running 9228 tests
2019-11-11T23:37:54.0007365Z .................................................................................................... 100/9228
2019-11-11T23:38:03.2172776Z .......................................FF..F..F..................................................... 200/9228
2019-11-11T23:38:22.4289793Z .................................................................................................... 400/9228
2019-11-11T23:38:28.9277585Z .................................................................................................... 500/9228
2019-11-11T23:38:42.2155451Z ........i...........F............................................................................... 600/9228
2019-11-11T23:38:49.6573355Z .................................................................................................... 700/9228
---
2019-11-11T23:39:32.2465905Z .................................................................................................... 1400/9228
2019-11-11T23:39:38.8171612Z .................................................................................................... 1500/9228
2019-11-11T23:39:45.0018480Z .................................................................................................... 1600/9228
2019-11-11T23:39:54.9263328Z .................................................................................................... 1700/9228
2019-11-11T23:40:03.5626995Z ..i................................................................................................. 1800/9228
2019-11-11T23:40:10.4846294Z ......................................................................................iiiii......... 1900/9228
2019-11-11T23:40:32.0133291Z .................................................................................................... 2100/9228
2019-11-11T23:40:34.4238881Z .................................................................................................... 2200/9228
2019-11-11T23:40:36.9725087Z ..................................F.....................................................F.FFFFF..... 2300/9228
2019-11-11T23:40:46.7722995Z .................................................................................................... 2400/9228
2019-11-11T23:40:46.7722995Z .................................................................................................... 2400/9228
2019-11-11T23:41:03.9600318Z .................................................................................................... 2500/9228
2019-11-11T23:41:07.9641364Z .................................................................................................... 2600/9228
2019-11-11T23:41:19.1735183Z ....................................................i............................................... 2700/9228
2019-11-11T23:41:26.4815483Z .................................................................................................... 2800/9228
2019-11-11T23:41:33.3665423Z ...................................................i................................................ 2900/9228
2019-11-11T23:41:41.6742046Z .................................................................................................... 3000/9228
2019-11-11T23:41:48.0553908Z .................................................................................................... 3100/9228
2019-11-11T23:41:55.4681020Z .................................ii.......................F.F....................................... 3200/9228
2019-11-11T23:42:12.1532888Z .................................................................................................... 3400/9228
2019-11-11T23:42:19.7548158Z .........................i..............................................i........................... 3500/9228
2019-11-11T23:42:25.1427742Z .................................................................................................... 3600/9228
2019-11-11T23:42:30.8363460Z .................................................................................................... 3700/9228
---
2019-11-11T23:43:44.3711301Z ..................................................................................i...............i. 4700/9228
2019-11-11T23:43:51.6037094Z .................................................................................................... 4800/9228
2019-11-11T23:44:01.0059945Z .......................F............................................................................ 4900/9228
2019-11-11T23:44:06.3883663Z .................................................................................................... 5000/9228
2019-11-11T23:44:18.0484123Z .....................................................................................ii.ii.......... 5100/9228
2019-11-11T23:44:21.9869893Z .i.................................................................................................. 5200/9228
2019-11-11T23:44:36.7416011Z .................................................................................................... 5400/9228
2019-11-11T23:44:44.1379388Z ...................................................................i................................ 5500/9228
2019-11-11T23:44:51.8428916Z ...............................................F.................................................... 5600/9228
2019-11-11T23:44:59.8905405Z .................................................................................................... 5700/9228
2019-11-11T23:44:59.8905405Z .................................................................................................... 5700/9228
2019-11-11T23:45:09.3016387Z ....................................................ii...i..ii...........i.......................... 5800/9228
2019-11-11T23:45:32.1765535Z .................................................................................................... 6000/9228
2019-11-11T23:45:40.7883233Z .................................................................................................... 6100/9228
2019-11-11T23:45:40.7883233Z .................................................................................................... 6100/9228
2019-11-11T23:45:45.9117677Z ......................................................................Fi..ii........................ 6200/9228
2019-11-11T23:46:15.5579455Z .................................................................................................... 6400/9228
2019-11-11T23:46:17.7897880Z .......................................i............................................................ 6500/9228
2019-11-11T23:46:20.1088795Z .................................................................................................... 6600/9228
2019-11-11T23:46:22.5830286Z .......................i............................................................................ 6700/9228
---
2019-11-11T23:48:09.3523393Z .................................................................................................... 7400/9228
2019-11-11T23:48:16.5685829Z .................................................................................................... 7500/9228
2019-11-11T23:48:27.1457366Z .................................................................................................... 7600/9228
2019-11-11T23:48:34.0801680Z .................................................................................................... 7700/9228
2019-11-11T23:48:44.4909366Z ...................................FF.FFFFF.F..ii......i............................................ 7800/9228
2019-11-11T23:49:00.1857145Z .................................................................................................... 8000/9228
2019-11-11T23:49:13.5516675Z .................................................................................................... 8100/9228
2019-11-11T23:49:20.3534238Z .................................................................................................... 8200/9228
2019-11-11T23:49:37.3369820Z .................................................................................................... 8300/9228
---
2019-11-11T23:51:14.2905753Z 9 
2019-11-11T23:51:14.2905897Z 
2019-11-11T23:51:14.2906058Z 
2019-11-11T23:51:14.2906226Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:14.2906725Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-bad-clobber/asm-bad-clobber.stderr
2019-11-11T23:51:14.2907230Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:14.2907717Z To only update this specific test, also pass `--test-args asm/asm-bad-clobber.rs`
2019-11-11T23:51:14.2908111Z error: 1 errors occurred comparing output.
2019-11-11T23:51:14.2908270Z status: exit code: 1
2019-11-11T23:51:14.2908270Z status: exit code: 1
2019-11-11T23:51:14.2909328Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/asm/asm-bad-clobber.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-bad-clobber" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-bad-clobber/auxiliary" "-A" "unused"
2019-11-11T23:51:14.2910431Z ------------------------------------------
2019-11-11T23:51:14.2910672Z 
2019-11-11T23:51:14.2911458Z ------------------------------------------
2019-11-11T23:51:14.2912055Z stderr:
2019-11-11T23:51:14.2912055Z stderr:
2019-11-11T23:51:14.2912817Z ------------------------------------------
2019-11-11T23:51:14.2913040Z error[E0664]: clobber should not be surrounded by braces
2019-11-11T23:51:14.2913870Z    |
2019-11-11T23:51:14.2913870Z    |
2019-11-11T23:51:14.2914105Z LL |         asm!("xor %eax, %eax" : : : "{eax}");
2019-11-11T23:51:14.2914416Z 
2019-11-11T23:51:14.2914596Z error: aborting due to previous error
2019-11-11T23:51:14.2914735Z 
2019-11-11T23:51:14.2915255Z For more information about this error, try `rustc --explain E0664`.
---
2019-11-11T23:51:15.1256130Z 15 
2019-11-11T23:51:15.1256184Z 
2019-11-11T23:51:15.1256215Z 
2019-11-11T23:51:15.1256265Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1256645Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-in-bad-modifier/asm-in-bad-modifier.stderr
2019-11-11T23:51:15.1256968Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1257281Z To only update this specific test, also pass `--test-args asm/asm-in-bad-modifier.rs`
2019-11-11T23:51:15.1257389Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1257439Z status: exit code: 1
2019-11-11T23:51:15.1257439Z status: exit code: 1
2019-11-11T23:51:15.1258350Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/asm/asm-in-bad-modifier.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-in-bad-modifier" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-in-bad-modifier/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1258814Z ------------------------------------------
2019-11-11T23:51:15.1258871Z 
2019-11-11T23:51:15.1259118Z ------------------------------------------
2019-11-11T23:51:15.1259169Z stderr:
2019-11-11T23:51:15.1259169Z stderr:
2019-11-11T23:51:15.1259421Z ------------------------------------------
2019-11-11T23:51:15.1259679Z error[E0662]: input operand constraint contains '='
2019-11-11T23:51:15.1260030Z    |
2019-11-11T23:51:15.1260030Z    |
2019-11-11T23:51:15.1260346Z LL |         asm!("mov $1, $0" : "=r"(x) : "=r"(5)); //~ ERROR operand constraint contains '='
2019-11-11T23:51:15.1260472Z 
2019-11-11T23:51:15.1260472Z 
2019-11-11T23:51:15.1260735Z error[E0663]: input operand constraint contains '+'
2019-11-11T23:51:15.1261219Z    |
2019-11-11T23:51:15.1261219Z    |
2019-11-11T23:51:15.1261542Z LL |         asm!("mov $1, $0" : "=r"(y) : "+r"(5)); //~ ERROR operand constraint contains '+'
2019-11-11T23:51:15.1261635Z 
2019-11-11T23:51:15.1261698Z error: aborting due to 2 previous errors
2019-11-11T23:51:15.1261730Z 
2019-11-11T23:51:15.1261781Z Some errors have detailed explanations: E0662, E0663.
---
2019-11-11T23:51:15.1421510Z 9 
2019-11-11T23:51:15.1421540Z 
2019-11-11T23:51:15.1421568Z 
2019-11-11T23:51:15.1421618Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1423466Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-out-no-modifier/asm-out-no-modifier.stderr
2019-11-11T23:51:15.1424039Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1424590Z To only update this specific test, also pass `--test-args asm/asm-out-no-modifier.rs`
2019-11-11T23:51:15.1425342Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1425573Z status: exit code: 1
2019-11-11T23:51:15.1425573Z status: exit code: 1
2019-11-11T23:51:15.1426566Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/asm/asm-out-no-modifier.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-out-no-modifier" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/asm-out-no-modifier/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1427166Z ------------------------------------------
2019-11-11T23:51:15.1427358Z 
2019-11-11T23:51:15.1427761Z ------------------------------------------
2019-11-11T23:51:15.1427974Z stderr:
2019-11-11T23:51:15.1427974Z stderr:
2019-11-11T23:51:15.1428315Z ------------------------------------------
2019-11-11T23:51:15.1428811Z error[E0661]: output operand constraint lacks '=' or '+'
2019-11-11T23:51:15.1429532Z    |
2019-11-11T23:51:15.1429532Z    |
2019-11-11T23:51:15.1429930Z LL |         asm!("mov $1, $0" : "r"(x) : "r"(5)); //~ ERROR output operand constraint lacks '='
2019-11-11T23:51:15.1430252Z 
2019-11-11T23:51:15.1430305Z error: aborting due to previous error
2019-11-11T23:51:15.1430337Z 
2019-11-11T23:51:15.1430665Z For more information about this error, try `rustc --explain E0661`.
---
2019-11-11T23:51:15.1432682Z 
2019-11-11T23:51:15.1432711Z 
2019-11-11T23:51:15.1432761Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1433181Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/issue-51431/issue-51431.stderr
2019-11-11T23:51:15.1433748Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1434259Z To only update this specific test, also pass `--test-args asm/issue-51431.rs`
2019-11-11T23:51:15.1434608Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1434659Z status: exit code: 1
2019-11-11T23:51:15.1434659Z status: exit code: 1
2019-11-11T23:51:15.1435869Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/asm/issue-51431.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/issue-51431" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/asm/issue-51431/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1436706Z ------------------------------------------
2019-11-11T23:51:15.1436892Z 
2019-11-11T23:51:15.1437654Z ------------------------------------------
2019-11-11T23:51:15.1437718Z stderr:
2019-11-11T23:51:15.1437718Z stderr:
2019-11-11T23:51:15.1438201Z ------------------------------------------
2019-11-11T23:51:15.1438441Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1438793Z   --> /checkout/src/test/ui/asm/issue-51431.rs:7:32
2019-11-11T23:51:15.1438852Z    |
2019-11-11T23:51:15.1439085Z LL |         asm! {"mov $0,$1"::"0"("bx"),"1"(0x00)}
2019-11-11T23:51:15.1439223Z 
2019-11-11T23:51:15.1439315Z error: aborting due to previous error
2019-11-11T23:51:15.1439441Z 
2019-11-11T23:51:15.1439808Z For more information about this error, try `rustc --explain E0669`.
---
2019-11-11T23:51:15.1442094Z 21 
2019-11-11T23:51:15.1442159Z 
2019-11-11T23:51:15.1442189Z 
2019-11-11T23:51:15.1442240Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1448529Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/bad/bad-intrinsic-monomorphization/bad-intrinsic-monomorphization.stderr
2019-11-11T23:51:15.1453893Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1454573Z To only update this specific test, also pass `--test-args bad/bad-intrinsic-monomorphization.rs`
2019-11-11T23:51:15.1454876Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1454964Z status: exit code: 1
2019-11-11T23:51:15.1454964Z status: exit code: 1
2019-11-11T23:51:15.1464514Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/bad/bad-intrinsic-monomorphization.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/bad/bad-intrinsic-monomorphization" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/bad/bad-intrinsic-monomorphization/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1465341Z ------------------------------------------
2019-11-11T23:51:15.1465411Z 
2019-11-11T23:51:15.1465666Z ------------------------------------------
2019-11-11T23:51:15.1465719Z stderr:
2019-11-11T23:51:15.1465719Z stderr:
2019-11-11T23:51:15.1466010Z ------------------------------------------
2019-11-11T23:51:15.1466089Z error[E0511]: invalid monomorphization of `cttz` intrinsic: expected basic integer type, found `Foo`
2019-11-11T23:51:15.1466449Z    |
2019-11-11T23:51:15.1466499Z LL |     intrinsics::cttz(v)
2019-11-11T23:51:15.1466548Z    |     ^^^^^^^^^^^^^^^^^^^
2019-11-11T23:51:15.1466580Z 
2019-11-11T23:51:15.1466580Z 
2019-11-11T23:51:15.1466880Z error[E0511]: invalid monomorphization of `fadd_fast` intrinsic: expected basic float type, found `Foo`
2019-11-11T23:51:15.1467845Z    |
2019-11-11T23:51:15.1467845Z    |
2019-11-11T23:51:15.1467954Z LL |     intrinsics::fadd_fast(a, b)
2019-11-11T23:51:15.1468181Z 
2019-11-11T23:51:15.1468181Z 
2019-11-11T23:51:15.1468609Z error[E0511]: invalid monomorphization of `simd_add` intrinsic: expected SIMD input type, found non-SIMD `Foo`
2019-11-11T23:51:15.1469358Z    |
2019-11-11T23:51:15.1469459Z LL |     simd_add(a, b)
2019-11-11T23:51:15.1469539Z    |     ^^^^^^^^^^^^^^
2019-11-11T23:51:15.1469572Z 
---
2019-11-11T23:51:15.1471799Z 9 
2019-11-11T23:51:15.1471854Z 
2019-11-11T23:51:15.1471912Z 
2019-11-11T23:51:15.1471991Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1472768Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0511/E0511.stderr
2019-11-11T23:51:15.1473356Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1473860Z To only update this specific test, also pass `--test-args error-codes/E0511.rs`
2019-11-11T23:51:15.1474161Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1474251Z status: exit code: 1
2019-11-11T23:51:15.1474251Z status: exit code: 1
2019-11-11T23:51:15.1475183Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/error-codes/E0511.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0511" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0511/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1475823Z ------------------------------------------
2019-11-11T23:51:15.1475863Z 
2019-11-11T23:51:15.1476486Z ------------------------------------------
2019-11-11T23:51:15.1476707Z stderr:
2019-11-11T23:51:15.1476707Z stderr:
2019-11-11T23:51:15.1477377Z ------------------------------------------
2019-11-11T23:51:15.1477964Z error[E0511]: invalid monomorphization of `simd_add` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1478711Z    |
2019-11-11T23:51:15.1478711Z    |
2019-11-11T23:51:15.1478824Z LL |     unsafe { simd_add(0, 1); } //~ ERROR E0511
2019-11-11T23:51:15.1478958Z 
2019-11-11T23:51:15.1479018Z error: aborting due to previous error
2019-11-11T23:51:15.1479052Z 
2019-11-11T23:51:15.1479385Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1481080Z 15 
2019-11-11T23:51:15.1481110Z 
2019-11-11T23:51:15.1481152Z 
2019-11-11T23:51:15.1481309Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1481726Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0660/E0660.stderr
2019-11-11T23:51:15.1482294Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1483203Z To only update this specific test, also pass `--test-args error-codes/E0660.rs`
2019-11-11T23:51:15.1483521Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1483593Z status: exit code: 1
2019-11-11T23:51:15.1483593Z status: exit code: 1
2019-11-11T23:51:15.1484525Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/error-codes/E0660.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0660" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0660/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1485159Z ------------------------------------------
2019-11-11T23:51:15.1485200Z 
2019-11-11T23:51:15.1485449Z ------------------------------------------
2019-11-11T23:51:15.1485500Z stderr:
2019-11-11T23:51:15.1485500Z stderr:
2019-11-11T23:51:15.1485755Z ------------------------------------------
2019-11-11T23:51:15.1485810Z error[E0660]: malformed inline assembly
2019-11-11T23:51:15.1488311Z   --> /checkout/src/test/ui/error-codes/E0660.rs:5:5
2019-11-11T23:51:15.1488595Z    |
2019-11-11T23:51:15.1488678Z LL |     asm!("nop" "nop");
2019-11-11T23:51:15.1488823Z 
2019-11-11T23:51:15.1488894Z error[E0660]: malformed inline assembly
2019-11-11T23:51:15.1489267Z   --> /checkout/src/test/ui/error-codes/E0660.rs:7:5
2019-11-11T23:51:15.1489341Z    |
2019-11-11T23:51:15.1489341Z    |
2019-11-11T23:51:15.1489567Z LL |     asm!("nop" "nop" : "=r"(a));
2019-11-11T23:51:15.1489718Z 
2019-11-11T23:51:15.1489899Z error: aborting due to 2 previous errors
2019-11-11T23:51:15.1489963Z 
2019-11-11T23:51:15.1490374Z For more information about this error, try `rustc --explain E0660`.
---
2019-11-11T23:51:15.1492869Z 9 
2019-11-11T23:51:15.1492899Z 
2019-11-11T23:51:15.1492929Z 
2019-11-11T23:51:15.1492996Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1493401Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0662/E0662.stderr
2019-11-11T23:51:15.1494058Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1494584Z To only update this specific test, also pass `--test-args error-codes/E0662.rs`
2019-11-11T23:51:15.1494876Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1494927Z status: exit code: 1
2019-11-11T23:51:15.1494927Z status: exit code: 1
2019-11-11T23:51:15.1495834Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/error-codes/E0662.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0662" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0662/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1496887Z ------------------------------------------
2019-11-11T23:51:15.1497109Z 
2019-11-11T23:51:15.1497472Z ------------------------------------------
2019-11-11T23:51:15.1497526Z stderr:
2019-11-11T23:51:15.1497526Z stderr:
2019-11-11T23:51:15.1497784Z ------------------------------------------
2019-11-11T23:51:15.1498043Z error[E0662]: input operand constraint contains '='
2019-11-11T23:51:15.1498371Z    |
2019-11-11T23:51:15.1498371Z    |
2019-11-11T23:51:15.1498421Z LL |          : "=test"("a") //~ ERROR E0662
2019-11-11T23:51:15.1498502Z 
2019-11-11T23:51:15.1498561Z error: aborting due to previous error
2019-11-11T23:51:15.1498781Z 
2019-11-11T23:51:15.1499132Z For more information about this error, try `rustc --explain E0662`.
---
2019-11-11T23:51:15.1501923Z 16 
2019-11-11T23:51:15.1502198Z 
2019-11-11T23:51:15.1502237Z 
2019-11-11T23:51:15.1502289Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1502718Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0661/E0661.stderr
2019-11-11T23:51:15.1502999Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1503587Z To only update this specific test, also pass `--test-args error-codes/E0661.rs`
2019-11-11T23:51:15.1503895Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1503945Z status: exit code: 1
2019-11-11T23:51:15.1503945Z status: exit code: 1
2019-11-11T23:51:15.1504887Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/error-codes/E0661.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0661" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0661/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1505267Z ------------------------------------------
2019-11-11T23:51:15.1505306Z 
2019-11-11T23:51:15.1505546Z ------------------------------------------
2019-11-11T23:51:15.1505616Z stderr:
2019-11-11T23:51:15.1505616Z stderr:
2019-11-11T23:51:15.1505856Z ------------------------------------------
2019-11-11T23:51:15.1506116Z error[E0661]: output operand constraint lacks '=' or '+'
2019-11-11T23:51:15.1506448Z    |
2019-11-11T23:51:15.1506448Z    |
2019-11-11T23:51:15.1506494Z LL |     asm!("nop" : "r"(a));
2019-11-11T23:51:15.1506586Z 
2019-11-11T23:51:15.1506632Z error[E0282]: type annotations needed
2019-11-11T23:51:15.1506937Z   --> /checkout/src/test/ui/error-codes/E0661.rs:6:9
2019-11-11T23:51:15.1507001Z    |
---
2019-11-11T23:51:15.1511093Z 9 
2019-11-11T23:51:15.1511153Z 
2019-11-11T23:51:15.1511200Z 
2019-11-11T23:51:15.1511283Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1511674Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0663/E0663.stderr
2019-11-11T23:51:15.1512707Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1513270Z To only update this specific test, also pass `--test-args error-codes/E0663.rs`
2019-11-11T23:51:15.1513569Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1513640Z status: exit code: 1
2019-11-11T23:51:15.1513640Z status: exit code: 1
2019-11-11T23:51:15.1514558Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/error-codes/E0663.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0663" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0663/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1515169Z ------------------------------------------
2019-11-11T23:51:15.1515209Z 
2019-11-11T23:51:15.1515467Z ------------------------------------------
2019-11-11T23:51:15.1515826Z stderr:
2019-11-11T23:51:15.1515826Z stderr:
2019-11-11T23:51:15.1516154Z ------------------------------------------
2019-11-11T23:51:15.1516414Z error[E0663]: input operand constraint contains '+'
2019-11-11T23:51:15.1517261Z    |
2019-11-11T23:51:15.1517261Z    |
2019-11-11T23:51:15.1517890Z LL |          : "+test"("a") //~ ERROR E0663
2019-11-11T23:51:15.1519735Z 
2019-11-11T23:51:15.1520496Z error: aborting due to previous error
2019-11-11T23:51:15.1521527Z 
2019-11-11T23:51:15.1522878Z For more information about this error, try `rustc --explain E0663`.
---
2019-11-11T23:51:15.1528135Z 9 
2019-11-11T23:51:15.1528196Z 
2019-11-11T23:51:15.1528227Z 
2019-11-11T23:51:15.1528324Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1528734Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0664/E0664.stderr
2019-11-11T23:51:15.1529229Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1529736Z To only update this specific test, also pass `--test-args error-codes/E0664.rs`
2019-11-11T23:51:15.1530028Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1530248Z status: exit code: 1
2019-11-11T23:51:15.1530248Z status: exit code: 1
2019-11-11T23:51:15.1531484Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/error-codes/E0664.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0664" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0664/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1531908Z ------------------------------------------
2019-11-11T23:51:15.1531961Z 
2019-11-11T23:51:15.1532209Z ------------------------------------------
2019-11-11T23:51:15.1532259Z stderr:
2019-11-11T23:51:15.1532259Z stderr:
2019-11-11T23:51:15.1532951Z ------------------------------------------
2019-11-11T23:51:15.1533020Z error[E0664]: clobber should not be surrounded by braces
2019-11-11T23:51:15.1533378Z    |
2019-11-11T23:51:15.1533378Z    |
2019-11-11T23:51:15.1533640Z LL |          : "{eax}" //~ ERROR E0664
2019-11-11T23:51:15.1533811Z 
2019-11-11T23:51:15.1533873Z error: aborting due to previous error
2019-11-11T23:51:15.1533905Z 
2019-11-11T23:51:15.1534392Z For more information about this error, try `rustc --explain E0664`.
---
2019-11-11T23:51:15.1536850Z 9 
2019-11-11T23:51:15.1536909Z 
2019-11-11T23:51:15.1536959Z 
2019-11-11T23:51:15.1537044Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1537539Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0665/E0665.stderr
2019-11-11T23:51:15.1537829Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1538146Z To only update this specific test, also pass `--test-args error-codes/E0665.rs`
2019-11-11T23:51:15.1538232Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1538296Z status: exit code: 1
2019-11-11T23:51:15.1538296Z status: exit code: 1
2019-11-11T23:51:15.1539108Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/error-codes/E0665.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0665" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/error-codes/E0665/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1539486Z ------------------------------------------
2019-11-11T23:51:15.1539523Z 
2019-11-11T23:51:15.1539776Z ------------------------------------------
2019-11-11T23:51:15.1539826Z stderr:
2019-11-11T23:51:15.1539826Z stderr:
2019-11-11T23:51:15.1540310Z ------------------------------------------
2019-11-11T23:51:15.1540451Z error[E0665]: `Default` cannot be derived for enums, only structs
2019-11-11T23:51:15.1540798Z    |
2019-11-11T23:51:15.1540859Z LL | #[derive(Default)] //~ ERROR E0665
2019-11-11T23:51:15.1540909Z    |          ^^^^^^^
2019-11-11T23:51:15.1540940Z 
---
2019-11-11T23:51:15.1544796Z 21 
2019-11-11T23:51:15.1544856Z 
2019-11-11T23:51:15.1544914Z 
2019-11-11T23:51:15.1544968Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1545386Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/inline-asm-bad-constraint/inline-asm-bad-constraint.stderr
2019-11-11T23:51:15.1545970Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1546601Z To only update this specific test, also pass `--test-args inline-asm-bad-constraint.rs`
2019-11-11T23:51:15.1547031Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1547083Z status: exit code: 1
2019-11-11T23:51:15.1547083Z status: exit code: 1
2019-11-11T23:51:15.1548141Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/inline-asm-bad-constraint.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/inline-asm-bad-constraint" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/inline-asm-bad-constraint/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1548739Z ------------------------------------------
2019-11-11T23:51:15.1548792Z 
2019-11-11T23:51:15.1549258Z ------------------------------------------
2019-11-11T23:51:15.1549472Z stderr:
2019-11-11T23:51:15.1549472Z stderr:
2019-11-11T23:51:15.1549789Z ------------------------------------------
2019-11-11T23:51:15.1550015Z error[E0668]: malformed inline assembly
2019-11-11T23:51:15.1550372Z   --> /checkout/src/test/ui/inline-asm-bad-constraint.rs:21:9
2019-11-11T23:51:15.1550431Z    |
2019-11-11T23:51:15.1550655Z LL |         asm!("" :"={rax"(rax)) //~ ERROR E0668
2019-11-11T23:51:15.1550874Z 
2019-11-11T23:51:15.1550981Z error[E0668]: malformed inline assembly
2019-11-11T23:51:15.1551306Z   --> /checkout/src/test/ui/inline-asm-bad-constraint.rs:29:9
2019-11-11T23:51:15.1551361Z    |
2019-11-11T23:51:15.1551361Z    |
2019-11-11T23:51:15.1551430Z LL |         asm!("callq $0" : : "0"(foo)) //~ ERROR E0668
2019-11-11T23:51:15.1551516Z 
2019-11-11T23:51:15.1551578Z error[E0668]: malformed inline assembly
2019-11-11T23:51:15.1551866Z   --> /checkout/src/test/ui/inline-asm-bad-constraint.rs:36:9
2019-11-11T23:51:15.1551918Z    |
2019-11-11T23:51:15.1551918Z    |
2019-11-11T23:51:15.1551981Z LL |         asm!("addb $1, $0" : "={rax}"((0i32, rax))); //~ ERROR E0668
2019-11-11T23:51:15.1552743Z 
2019-11-11T23:51:15.1552796Z error: aborting due to 3 previous errors
2019-11-11T23:51:15.1552846Z 
2019-11-11T23:51:15.1553360Z For more information about this error, try `rustc --explain E0668`.
---
2019-11-11T23:51:15.1555537Z 45 
2019-11-11T23:51:15.1555614Z 
2019-11-11T23:51:15.1555646Z 
2019-11-11T23:51:15.1555921Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1556510Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/inline-asm-bad-operand/inline-asm-bad-operand.stderr
2019-11-11T23:51:15.1557274Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1557831Z To only update this specific test, also pass `--test-args inline-asm-bad-operand.rs`
2019-11-11T23:51:15.1558129Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1558202Z status: exit code: 1
2019-11-11T23:51:15.1558202Z status: exit code: 1
2019-11-11T23:51:15.1559131Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/inline-asm-bad-operand.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/inline-asm-bad-operand" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/inline-asm-bad-operand/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1559736Z ------------------------------------------
2019-11-11T23:51:15.1559918Z 
2019-11-11T23:51:15.1560262Z ------------------------------------------
2019-11-11T23:51:15.1560458Z stderr:
2019-11-11T23:51:15.1560458Z stderr:
2019-11-11T23:51:15.1560779Z ------------------------------------------
2019-11-11T23:51:15.1560851Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1561319Z   --> /checkout/src/test/ui/inline-asm-bad-operand.rs:21:24
2019-11-11T23:51:15.1561529Z    |
2019-11-11T23:51:15.1561626Z LL |         asm!("" :: "r"("")); //~ ERROR E0669
2019-11-11T23:51:15.1561832Z 
2019-11-11T23:51:15.1561941Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1562264Z   --> /checkout/src/test/ui/inline-asm-bad-operand.rs:26:32
2019-11-11T23:51:15.1562961Z    |
2019-11-11T23:51:15.1562961Z    |
2019-11-11T23:51:15.1563090Z LL |         asm!("ret" : : "{rdi}"(target)); //~ ERROR E0669
2019-11-11T23:51:15.1563220Z 
2019-11-11T23:51:15.1563286Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1563687Z   --> /checkout/src/test/ui/inline-asm-bad-operand.rs:33:29
2019-11-11T23:51:15.1563901Z    |
2019-11-11T23:51:15.1563901Z    |
2019-11-11T23:51:15.1564002Z LL |     unsafe { asm!("" :: "i"(hello)) }; //~ ERROR E0669
2019-11-11T23:51:15.1564110Z 
2019-11-11T23:51:15.1564212Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1564556Z   --> /checkout/src/test/ui/inline-asm-bad-operand.rs:41:38
2019-11-11T23:51:15.1564620Z    |
2019-11-11T23:51:15.1564620Z    |
2019-11-11T23:51:15.1564685Z LL |         asm!("movups $1, %xmm0"::"m"(arr)); //~ ERROR E0669
2019-11-11T23:51:15.1564965Z 
2019-11-11T23:51:15.1565062Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1565407Z   --> /checkout/src/test/ui/inline-asm-bad-operand.rs:48:32
2019-11-11T23:51:15.1565605Z    |
2019-11-11T23:51:15.1565605Z    |
2019-11-11T23:51:15.1565728Z LL |         asm!("mov sp, $0"::"r"(addr)); //~ ERROR E0669
2019-11-11T23:51:15.1565814Z 
2019-11-11T23:51:15.1565874Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1566194Z   --> /checkout/src/test/ui/inline-asm-bad-operand.rs:55:32
2019-11-11T23:51:15.1566247Z    |
2019-11-11T23:51:15.1566247Z    |
2019-11-11T23:51:15.1566297Z LL |         asm!("mov sp, $0"::"r"(addr), //~ ERROR E0669
2019-11-11T23:51:15.1566553Z 
2019-11-11T23:51:15.1566604Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1566924Z   --> /checkout/src/test/ui/inline-asm-bad-operand.rs:56:32
2019-11-11T23:51:15.1566977Z    |
2019-11-11T23:51:15.1566977Z    |
2019-11-11T23:51:15.1567117Z LL | ...                   "r"("hello e0669")); //~ ERROR E0669
2019-11-11T23:51:15.1567226Z 
2019-11-11T23:51:15.1567273Z error: aborting due to 7 previous errors
2019-11-11T23:51:15.1567305Z 
2019-11-11T23:51:15.1567625Z For more information about this error, try `rustc --explain E0669`.
---
2019-11-11T23:51:15.1569414Z 
2019-11-11T23:51:15.1569442Z 
2019-11-11T23:51:15.1569513Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1569850Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-37433/issue-37433.stderr
2019-11-11T23:51:15.1570132Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1570440Z To only update this specific test, also pass `--test-args issues/issue-37433.rs`
2019-11-11T23:51:15.1570528Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1570577Z status: exit code: 1
2019-11-11T23:51:15.1570577Z status: exit code: 1
2019-11-11T23:51:15.1571712Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/issues/issue-37433.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-37433" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-37433/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1572699Z ------------------------------------------
2019-11-11T23:51:15.1572748Z 
2019-11-11T23:51:15.1573274Z ------------------------------------------
2019-11-11T23:51:15.1573487Z stderr:
2019-11-11T23:51:15.1573487Z stderr:
2019-11-11T23:51:15.1573808Z ------------------------------------------
2019-11-11T23:51:15.1574024Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1574380Z   --> /checkout/src/test/ui/issues/issue-37433.rs:7:24
2019-11-11T23:51:15.1574582Z    |
2019-11-11T23:51:15.1574676Z LL |         asm!("" :: "r"(""));
2019-11-11T23:51:15.1574809Z 
2019-11-11T23:51:15.1574865Z error: aborting due to previous error
2019-11-11T23:51:15.1574898Z 
2019-11-11T23:51:15.1575224Z For more information about this error, try `rustc --explain E0669`.
---
2019-11-11T23:51:15.1581057Z 
2019-11-11T23:51:15.1581085Z 
2019-11-11T23:51:15.1581135Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1581826Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-53787-inline-assembler-macro/issue-53787-inline-assembler-macro.stderr
2019-11-11T23:51:15.1582882Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1583573Z To only update this specific test, also pass `--test-args issues/issue-53787-inline-assembler-macro.rs`
2019-11-11T23:51:15.1583883Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1583971Z status: exit code: 1
2019-11-11T23:51:15.1583971Z status: exit code: 1
2019-11-11T23:51:15.1585035Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/issues/issue-53787-inline-assembler-macro.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-53787-inline-assembler-macro" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-53787-inline-assembler-macro/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1585431Z ------------------------------------------
2019-11-11T23:51:15.1585470Z 
2019-11-11T23:51:15.1585966Z ------------------------------------------
2019-11-11T23:51:15.1586169Z stderr:
2019-11-11T23:51:15.1586169Z stderr:
2019-11-11T23:51:15.1586501Z ------------------------------------------
2019-11-11T23:51:15.1586780Z error[E0669]: invalid value for constraint in inline assembly
2019-11-11T23:51:15.1587154Z   --> /checkout/src/test/ui/issues/issue-53787-inline-assembler-macro.rs:23:16
2019-11-11T23:51:15.1587233Z    |
2019-11-11T23:51:15.1587440Z LL |     fake_jump!("FirstFunc"); //~ ERROR invalid value for constraint in inline assembly
2019-11-11T23:51:15.1587608Z 
2019-11-11T23:51:15.1587657Z error: aborting due to previous error
2019-11-11T23:51:15.1587690Z 
2019-11-11T23:51:15.1588030Z For more information about this error, try `rustc --explain E0669`.
---
2019-11-11T23:51:15.1589763Z 
2019-11-11T23:51:15.1589794Z 
2019-11-11T23:51:15.1589856Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1590384Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/macros/macros-nonfatal-errors/macros-nonfatal-errors.stderr
2019-11-11T23:51:15.1590863Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1591458Z To only update this specific test, also pass `--test-args macros/macros-nonfatal-errors.rs`
2019-11-11T23:51:15.1591753Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1591846Z status: exit code: 1
2019-11-11T23:51:15.1591846Z status: exit code: 1
2019-11-11T23:51:15.1593184Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/macros/macros-nonfatal-errors.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/macros/macros-nonfatal-errors" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/macros/macros-nonfatal-errors/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1593857Z ------------------------------------------
2019-11-11T23:51:15.1594064Z 
2019-11-11T23:51:15.1594351Z ------------------------------------------
2019-11-11T23:51:15.1594603Z stderr:
2019-11-11T23:51:15.1594603Z stderr:
2019-11-11T23:51:15.1594916Z ------------------------------------------
2019-11-11T23:51:15.1595241Z error[E0665]: `Default` cannot be derived for enums, only structs
2019-11-11T23:51:15.1595992Z    |
2019-11-11T23:51:15.1596073Z LL | #[derive(Default)] //~ ERROR
2019-11-11T23:51:15.1596123Z    |          ^^^^^^^
2019-11-11T23:51:15.1596166Z 
2019-11-11T23:51:15.1596166Z 
2019-11-11T23:51:15.1596410Z error: inline assembly must be a string literal
2019-11-11T23:51:15.1596792Z   --> /checkout/src/test/ui/macros/macros-nonfatal-errors.rs:13:10
2019-11-11T23:51:15.1596861Z    |
2019-11-11T23:51:15.1596908Z LL |     asm!(invalid); //~ ERROR
2019-11-11T23:51:15.1597178Z 
2019-11-11T23:51:15.1597178Z 
2019-11-11T23:51:15.1597244Z error: concat_idents! requires ident args.
2019-11-11T23:51:15.1597741Z    |
2019-11-11T23:51:15.1597741Z    |
2019-11-11T23:51:15.1597966Z LL |     concat_idents!("not", "idents"); //~ ERROR
2019-11-11T23:51:15.1636262Z 
2019-11-11T23:51:15.1636947Z error: argument must be a string literal
2019-11-11T23:51:15.1638331Z   --> /checkout/src/test/ui/macros/macros-nonfatal-errors.rs:17:17
2019-11-11T23:51:15.1638458Z    |
2019-11-11T23:51:15.1638458Z    |
2019-11-11T23:51:15.1639101Z LL |     option_env!(invalid); //~ ERROR
2019-11-11T23:51:15.1640290Z 
2019-11-11T23:51:15.1640939Z error: expected string literal
2019-11-11T23:51:15.1641861Z   --> /checkout/src/test/ui/macros/macros-nonfatal-errors.rs:18:10
2019-11-11T23:51:15.1642174Z    |
2019-11-11T23:51:15.1642174Z    |
2019-11-11T23:51:15.1643307Z LL |     env!(invalid); //~ ERROR
2019-11-11T23:51:15.1644654Z 
2019-11-11T23:51:15.1645209Z error: expected string literal
2019-11-11T23:51:15.1645930Z   --> /checkout/src/test/ui/macros/macros-nonfatal-errors.rs:19:10
2019-11-11T23:51:15.1646347Z    |
2019-11-11T23:51:15.1646347Z    |
2019-11-11T23:51:15.1646562Z LL |     env!(foo, abr, baz); //~ ERROR
2019-11-11T23:51:15.1646724Z 
2019-11-11T23:51:15.1646724Z 
2019-11-11T23:51:15.1646780Z error: environment variable `RUST_HOPEFULLY_THIS_DOESNT_EXIST` not defined
2019-11-11T23:51:15.1647261Z    |
2019-11-11T23:51:15.1647261Z    |
2019-11-11T23:51:15.1647328Z LL |     env!("RUST_HOPEFULLY_THIS_DOESNT_EXIST"); //~ ERROR
2019-11-11T23:51:15.1647418Z 
2019-11-11T23:51:15.1647486Z error: format argument must be a string literal
2019-11-11T23:51:15.1647771Z   --> /checkout/src/test/ui/macros/macros-nonfatal-errors.rs:22:13
2019-11-11T23:51:15.1647825Z    |
2019-11-11T23:51:15.1647825Z    |
2019-11-11T23:51:15.1647892Z LL |     format!(invalid); //~ ERROR
2019-11-11T23:51:15.1647998Z    |
2019-11-11T23:51:15.1648050Z help: you might be missing a string literal to format with
2019-11-11T23:51:15.1648115Z    |
2019-11-11T23:51:15.1648171Z LL |     format!("{}", invalid); //~ ERROR
2019-11-11T23:51:15.1648171Z LL |     format!("{}", invalid); //~ ERROR
2019-11-11T23:51:15.1648221Z    |             ^^^^^
2019-11-11T23:51:15.1648267Z 
2019-11-11T23:51:15.1648316Z error: argument must be a string literal
2019-11-11T23:51:15.1648947Z   --> /checkout/src/test/ui/macros/macros-nonfatal-errors.rs:24:14
2019-11-11T23:51:15.1649001Z    |
2019-11-11T23:51:15.1649049Z LL |     include!(invalid); //~ ERROR
2019-11-11T23:51:15.1649142Z 
2019-11-11T23:51:15.1649188Z error: argument must be a string literal
2019-11-11T23:51:15.1649462Z   --> /checkout/src/test/ui/macros/macros-nonfatal-errors.rs:26:18
2019-11-11T23:51:15.1649526Z    |
2019-11-11T23:51:15.1649526Z    |
2019-11-11T23:51:15.1649574Z LL |     include_str!(invalid); //~ ERROR
2019-11-11T23:51:15.1649623Z    |                  ^^^^^^^
2019-11-11T23:51:15.1649844Z 
2019-11-11T23:51:15.1650269Z error: couldn't read /checkout/src/test/ui/macros/i'd be quite surprised if a file with this name existed: No such file or directory (os error 2)
2019-11-11T23:51:15.1650766Z    |
2019-11-11T23:51:15.1650766Z    |
2019-11-11T23:51:15.1651101Z LL |     include_str!("i'd be quite surprised if a file with this name existed"); //~ ERROR
2019-11-11T23:51:15.1651204Z 
2019-11-11T23:51:15.1651269Z error: argument must be a string literal
2019-11-11T23:51:15.1651542Z   --> /checkout/src/test/ui/macros/macros-nonfatal-errors.rs:28:20
2019-11-11T23:51:15.1651594Z    |
2019-11-11T23:51:15.1651594Z    |
2019-11-11T23:51:15.1651661Z LL |     include_bytes!(invalid); //~ ERROR
2019-11-11T23:51:15.1651711Z    |                    ^^^^^^^
2019-11-11T23:51:15.1651743Z 
2019-11-11T23:51:15.1652098Z error: couldn't read /checkout/src/test/ui/macros/i'd be quite surprised if a file with this name existed: No such file or directory (os error 2)
2019-11-11T23:51:15.1652784Z    |
2019-11-11T23:51:15.1652784Z    |
2019-11-11T23:51:15.1653132Z LL |     include_bytes!("i'd be quite surprised if a file with this name existed"); //~ ERROR
2019-11-11T23:51:15.1653254Z 
2019-11-11T23:51:15.1653254Z 
2019-11-11T23:51:15.1653303Z error: trace_macros! accepts only `true` or `false`
2019-11-11T23:51:15.1653648Z    |
2019-11-11T23:51:15.1653648Z    |
2019-11-11T23:51:15.1653695Z LL |     trace_macros!(invalid); //~ ERROR
2019-11-11T23:51:15.1653792Z 
2019-11-11T23:51:15.1653841Z error: aborting due to 14 previous errors
2019-11-11T23:51:15.1653872Z 
2019-11-11T23:51:15.1654158Z For more information about this error, try `rustc --explain E0665`.
---
2019-11-11T23:51:15.1655355Z 99 
2019-11-11T23:51:15.1655384Z 
2019-11-11T23:51:15.1655413Z 
2019-11-11T23:51:15.1655462Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1655810Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/non-integer-atomic/non-integer-atomic.stderr
2019-11-11T23:51:15.1656084Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1656381Z To only update this specific test, also pass `--test-args non-integer-atomic.rs`
2019-11-11T23:51:15.1656530Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1656588Z status: exit code: 1
2019-11-11T23:51:15.1656588Z status: exit code: 1
2019-11-11T23:51:15.1657436Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/non-integer-atomic.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/non-integer-atomic" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/non-integer-atomic/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1657794Z ------------------------------------------
2019-11-11T23:51:15.1657834Z 
2019-11-11T23:51:15.1658264Z ------------------------------------------
2019-11-11T23:51:15.1658317Z stderr:
2019-11-11T23:51:15.1658317Z stderr:
2019-11-11T23:51:15.1658567Z ------------------------------------------
2019-11-11T23:51:15.1658718Z error[E0511]: invalid monomorphization of `atomic_load` intrinsic: expected basic integer type, found `bool`
2019-11-11T23:51:15.1659095Z    |
2019-11-11T23:51:15.1659147Z LL |     intrinsics::atomic_load(p);
2019-11-11T23:51:15.1659201Z    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
2019-11-11T23:51:15.1659237Z 
2019-11-11T23:51:15.1659237Z 
2019-11-11T23:51:15.1659309Z error[E0511]: invalid monomorphization of `atomic_store` intrinsic: expected basic integer type, found `bool`
2019-11-11T23:51:15.1659680Z    |
2019-11-11T23:51:15.1659680Z    |
2019-11-11T23:51:15.1659746Z LL |     intrinsics::atomic_store(p, v);
2019-11-11T23:51:15.1659833Z 
2019-11-11T23:51:15.1659833Z 
2019-11-11T23:51:15.1659920Z error[E0511]: invalid monomorphization of `atomic_xchg` intrinsic: expected basic integer type, found `bool`
2019-11-11T23:51:15.1660277Z    |
2019-11-11T23:51:15.1660277Z    |
2019-11-11T23:51:15.1660336Z LL |     intrinsics::atomic_xchg(p, v);
2019-11-11T23:51:15.1660438Z 
2019-11-11T23:51:15.1660438Z 
2019-11-11T23:51:15.1660496Z error[E0511]: invalid monomorphization of `atomic_cxchg` intrinsic: expected basic integer type, found `bool`
2019-11-11T23:51:15.1660853Z    |
2019-11-11T23:51:15.1660853Z    |
2019-11-11T23:51:15.1660904Z LL |     intrinsics::atomic_cxchg(p, v, v);
2019-11-11T23:51:15.1661006Z 
2019-11-11T23:51:15.1661006Z 
2019-11-11T23:51:15.1661065Z error[E0511]: invalid monomorphization of `atomic_load` intrinsic: expected basic integer type, found `Foo`
2019-11-11T23:51:15.1661434Z    |
2019-11-11T23:51:15.1661485Z LL |     intrinsics::atomic_load(p);
2019-11-11T23:51:15.1661537Z    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
2019-11-11T23:51:15.1661571Z 
2019-11-11T23:51:15.1661571Z 
2019-11-11T23:51:15.1661649Z error[E0511]: invalid monomorphization of `atomic_store` intrinsic: expected basic integer type, found `Foo`
2019-11-11T23:51:15.1661996Z    |
2019-11-11T23:51:15.1661996Z    |
2019-11-11T23:51:15.1662059Z LL |     intrinsics::atomic_store(p, v);
2019-11-11T23:51:15.1662146Z 
2019-11-11T23:51:15.1662146Z 
2019-11-11T23:51:15.1662214Z error[E0511]: invalid monomorphization of `atomic_xchg` intrinsic: expected basic integer type, found `Foo`
2019-11-11T23:51:15.1662872Z    |
2019-11-11T23:51:15.1662872Z    |
2019-11-11T23:51:15.1662921Z LL |     intrinsics::atomic_xchg(p, v);
2019-11-11T23:51:15.1663033Z 
2019-11-11T23:51:15.1663033Z 
2019-11-11T23:51:15.1663090Z error[E0511]: invalid monomorphization of `atomic_cxchg` intrinsic: expected basic integer type, found `Foo`
2019-11-11T23:51:15.1663444Z    |
2019-11-11T23:51:15.1663444Z    |
2019-11-11T23:51:15.1663493Z LL |     intrinsics::atomic_cxchg(p, v, v);
2019-11-11T23:51:15.1663588Z 
2019-11-11T23:51:15.1663588Z 
2019-11-11T23:51:15.1663648Z error[E0511]: invalid monomorphization of `atomic_load` intrinsic: expected basic integer type, found `&dyn std::ops::Fn()`
2019-11-11T23:51:15.1663979Z    |
2019-11-11T23:51:15.1664026Z LL |     intrinsics::atomic_load(p);
2019-11-11T23:51:15.1664077Z    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
2019-11-11T23:51:15.1664108Z 
2019-11-11T23:51:15.1664108Z 
2019-11-11T23:51:15.1664180Z error[E0511]: invalid monomorphization of `atomic_store` intrinsic: expected basic integer type, found `&dyn std::ops::Fn()`
2019-11-11T23:51:15.1664680Z    |
2019-11-11T23:51:15.1664680Z    |
2019-11-11T23:51:15.1664741Z LL |     intrinsics::atomic_store(p, v);
2019-11-11T23:51:15.1664919Z 
2019-11-11T23:51:15.1664919Z 
2019-11-11T23:51:15.1664991Z error[E0511]: invalid monomorphization of `atomic_xchg` intrinsic: expected basic integer type, found `&dyn std::ops::Fn()`
2019-11-11T23:51:15.1665344Z    |
2019-11-11T23:51:15.1665344Z    |
2019-11-11T23:51:15.1665404Z LL |     intrinsics::atomic_xchg(p, v);
2019-11-11T23:51:15.1665490Z 
2019-11-11T23:51:15.1665490Z 
2019-11-11T23:51:15.1665550Z error[E0511]: invalid monomorphization of `atomic_cxchg` intrinsic: expected basic integer type, found `&dyn std::ops::Fn()`
2019-11-11T23:51:15.1665918Z    |
2019-11-11T23:51:15.1665918Z    |
2019-11-11T23:51:15.1665969Z LL |     intrinsics::atomic_cxchg(p, v, v);
2019-11-11T23:51:15.1666069Z 
2019-11-11T23:51:15.1666069Z 
2019-11-11T23:51:15.1666136Z error[E0511]: invalid monomorphization of `atomic_load` intrinsic: expected basic integer type, found `[u8; 100]`
2019-11-11T23:51:15.1666490Z    |
2019-11-11T23:51:15.1666540Z LL |     intrinsics::atomic_load(p);
2019-11-11T23:51:15.1666591Z    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
2019-11-11T23:51:15.1666636Z 
2019-11-11T23:51:15.1666636Z 
2019-11-11T23:51:15.1666695Z error[E0511]: invalid monomorphization of `atomic_store` intrinsic: expected basic integer type, found `[u8; 100]`
2019-11-11T23:51:15.1667054Z    |
2019-11-11T23:51:15.1667054Z    |
2019-11-11T23:51:15.1667105Z LL |     intrinsics::atomic_store(p, v);
2019-11-11T23:51:15.1667202Z 
2019-11-11T23:51:15.1667202Z 
2019-11-11T23:51:15.1667275Z error[E0511]: invalid monomorphization of `atomic_xchg` intrinsic: expected basic integer type, found `[u8; 100]`
2019-11-11T23:51:15.1667636Z    |
2019-11-11T23:51:15.1667636Z    |
2019-11-11T23:51:15.1667698Z LL |     intrinsics::atomic_xchg(p, v);
2019-11-11T23:51:15.1667784Z 
2019-11-11T23:51:15.1667784Z 
2019-11-11T23:51:15.1667843Z error[E0511]: invalid monomorphization of `atomic_cxchg` intrinsic: expected basic integer type, found `[u8; 100]`
2019-11-11T23:51:15.1668200Z    |
2019-11-11T23:51:15.1668200Z    |
2019-11-11T23:51:15.1668252Z LL |     intrinsics::atomic_cxchg(p, v, v);
2019-11-11T23:51:15.1668348Z 
2019-11-11T23:51:15.1668398Z error: aborting due to 16 previous errors
2019-11-11T23:51:15.1668432Z 
2019-11-11T23:51:15.1668740Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1670100Z 
2019-11-11T23:51:15.1670130Z 
2019-11-11T23:51:15.1670181Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1670628Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-arithmetic-saturating/simd-intrinsic-generic-arithmetic-saturating.stderr
2019-11-11T23:51:15.1671067Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1671536Z To only update this specific test, also pass `--test-args simd-intrinsic/simd-intrinsic-generic-arithmetic-saturating.rs`
2019-11-11T23:51:15.1671642Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1671695Z status: exit code: 1
2019-11-11T23:51:15.1671695Z status: exit code: 1
2019-11-11T23:51:15.1673104Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/simd-intrinsic/simd-intrinsic-generic-arithmetic-saturating.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-arithmetic-saturating" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-arithmetic-saturating/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1673509Z ------------------------------------------
2019-11-11T23:51:15.1673547Z 
2019-11-11T23:51:15.1673800Z ------------------------------------------
2019-11-11T23:51:15.1673868Z stderr:
2019-11-11T23:51:15.1673868Z stderr:
2019-11-11T23:51:15.1674103Z ------------------------------------------
2019-11-11T23:51:15.1674174Z error[E0511]: invalid monomorphization of `simd_saturating_add` intrinsic: expected element type `f32` of vector type `f32x4` to be a signed or unsigned integer type
2019-11-11T23:51:15.1674565Z    |
2019-11-11T23:51:15.1674565Z    |
2019-11-11T23:51:15.1674613Z LL |         simd_saturating_add(z, z);
2019-11-11T23:51:15.1674711Z 
2019-11-11T23:51:15.1674711Z 
2019-11-11T23:51:15.1674775Z error[E0511]: invalid monomorphization of `simd_saturating_sub` intrinsic: expected element type `f32` of vector type `f32x4` to be a signed or unsigned integer type
2019-11-11T23:51:15.1675180Z    |
2019-11-11T23:51:15.1675180Z    |
2019-11-11T23:51:15.1675228Z LL |         simd_saturating_sub(z, z);
2019-11-11T23:51:15.1675324Z 
2019-11-11T23:51:15.1675370Z error: aborting due to 2 previous errors
2019-11-11T23:51:15.1675401Z 
2019-11-11T23:51:15.1675694Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1676944Z 
2019-11-11T23:51:15.1677204Z 
2019-11-11T23:51:15.1677259Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1677711Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-arithmetic/simd-intrinsic-generic-arithmetic.stderr
2019-11-11T23:51:15.1677992Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1678310Z To only update this specific test, also pass `--test-args simd-intrinsic/simd-intrinsic-generic-arithmetic.rs`
2019-11-11T23:51:15.1678415Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1678464Z status: exit code: 1
2019-11-11T23:51:15.1678464Z status: exit code: 1
2019-11-11T23:51:15.1679552Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/simd-intrinsic/simd-intrinsic-generic-arithmetic.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-arithmetic" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-arithmetic/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1680049Z ------------------------------------------
2019-11-11T23:51:15.1680087Z 
2019-11-11T23:51:15.1680328Z ------------------------------------------
2019-11-11T23:51:15.1680394Z stderr:
2019-11-11T23:51:15.1680394Z stderr:
2019-11-11T23:51:15.1680630Z ------------------------------------------
2019-11-11T23:51:15.1680952Z error[E0511]: invalid monomorphization of `simd_add` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1681334Z    |
2019-11-11T23:51:15.1681380Z LL |         simd_add(0, 0);
2019-11-11T23:51:15.1681452Z    |         ^^^^^^^^^^^^^^
2019-11-11T23:51:15.1681485Z 
2019-11-11T23:51:15.1681485Z 
2019-11-11T23:51:15.1681808Z error[E0511]: invalid monomorphization of `simd_sub` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1682173Z    |
2019-11-11T23:51:15.1682218Z LL |         simd_sub(0, 0);
2019-11-11T23:51:15.1682267Z    |         ^^^^^^^^^^^^^^
2019-11-11T23:51:15.1682298Z 
2019-11-11T23:51:15.1682298Z 
2019-11-11T23:51:15.1682946Z error[E0511]: invalid monomorphization of `simd_mul` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1683320Z    |
2019-11-11T23:51:15.1683320Z    |
2019-11-11T23:51:15.1683383Z LL |         simd_mul(0, 0);
2019-11-11T23:51:15.1683464Z 
2019-11-11T23:51:15.1683464Z 
2019-11-11T23:51:15.1683806Z error[E0511]: invalid monomorphization of `simd_div` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1684163Z    |
2019-11-11T23:51:15.1684163Z    |
2019-11-11T23:51:15.1684209Z LL |         simd_div(0, 0);
2019-11-11T23:51:15.1684301Z 
2019-11-11T23:51:15.1684301Z 
2019-11-11T23:51:15.1684618Z error[E0511]: invalid monomorphization of `simd_shl` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1684980Z    |
2019-11-11T23:51:15.1685026Z LL |         simd_shl(0, 0);
2019-11-11T23:51:15.1685088Z    |         ^^^^^^^^^^^^^^
2019-11-11T23:51:15.1685130Z 
2019-11-11T23:51:15.1685130Z 
2019-11-11T23:51:15.1685446Z error[E0511]: invalid monomorphization of `simd_shr` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1685818Z    |
2019-11-11T23:51:15.1685863Z LL |         simd_shr(0, 0);
2019-11-11T23:51:15.1685911Z    |         ^^^^^^^^^^^^^^
2019-11-11T23:51:15.1685942Z 
2019-11-11T23:51:15.1685942Z 
2019-11-11T23:51:15.1686273Z error[E0511]: invalid monomorphization of `simd_and` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1686623Z    |
2019-11-11T23:51:15.1686623Z    |
2019-11-11T23:51:15.1686682Z LL |         simd_and(0, 0);
2019-11-11T23:51:15.1686761Z 
2019-11-11T23:51:15.1686761Z 
2019-11-11T23:51:15.1687087Z error[E0511]: invalid monomorphization of `simd_or` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1687603Z    |
2019-11-11T23:51:15.1687663Z LL |         simd_or(0, 0);
2019-11-11T23:51:15.1687796Z    |         ^^^^^^^^^^^^^
2019-11-11T23:51:15.1687833Z 
2019-11-11T23:51:15.1687833Z 
2019-11-11T23:51:15.1688179Z error[E0511]: invalid monomorphization of `simd_xor` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1688548Z    |
2019-11-11T23:51:15.1688548Z    |
2019-11-11T23:51:15.1688593Z LL |         simd_xor(0, 0);
2019-11-11T23:51:15.1688687Z 
2019-11-11T23:51:15.1688687Z 
2019-11-11T23:51:15.1688745Z error[E0511]: invalid monomorphization of `simd_shl` intrinsic: unsupported operation on `f32x4` with element `f32`
2019-11-11T23:51:15.1689109Z    |
2019-11-11T23:51:15.1689109Z    |
2019-11-11T23:51:15.1689167Z LL |         simd_shl(z, z);
2019-11-11T23:51:15.1689253Z 
2019-11-11T23:51:15.1689253Z 
2019-11-11T23:51:15.1689320Z error[E0511]: invalid monomorphization of `simd_shr` intrinsic: unsupported operation on `f32x4` with element `f32`
2019-11-11T23:51:15.1689690Z    |
2019-11-11T23:51:15.1689690Z    |
2019-11-11T23:51:15.1689737Z LL |         simd_shr(z, z);
2019-11-11T23:51:15.1689815Z 
2019-11-11T23:51:15.1689815Z 
2019-11-11T23:51:15.1689886Z error[E0511]: invalid monomorphization of `simd_and` intrinsic: unsupported operation on `f32x4` with element `f32`
2019-11-11T23:51:15.1690239Z    |
2019-11-11T23:51:15.1690239Z    |
2019-11-11T23:51:15.1690293Z LL |         simd_and(z, z);
2019-11-11T23:51:15.1690372Z 
2019-11-11T23:51:15.1690372Z 
2019-11-11T23:51:15.1690438Z error[E0511]: invalid monomorphization of `simd_or` intrinsic: unsupported operation on `f32x4` with element `f32`
2019-11-11T23:51:15.1690808Z    |
2019-11-11T23:51:15.1690808Z    |
2019-11-11T23:51:15.1690853Z LL |         simd_or(z, z);
2019-11-11T23:51:15.1690941Z 
2019-11-11T23:51:15.1690941Z 
2019-11-11T23:51:15.1690998Z error[E0511]: invalid monomorphization of `simd_xor` intrinsic: unsupported operation on `f32x4` with element `f32`
2019-11-11T23:51:15.1691363Z    |
2019-11-11T23:51:15.1691363Z    |
2019-11-11T23:51:15.1691408Z LL |         simd_xor(z, z);
2019-11-11T23:51:15.1691502Z 
2019-11-11T23:51:15.1691549Z error: aborting due to 14 previous errors
2019-11-11T23:51:15.1691581Z 
2019-11-11T23:51:15.1691852Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1693577Z 33 
2019-11-11T23:51:15.1693619Z 
2019-11-11T23:51:15.1693648Z 
2019-11-11T23:51:15.1693697Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1694078Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-bitmask/simd-intrinsic-generic-bitmask.stderr
2019-11-11T23:51:15.1694377Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1694860Z To only update this specific test, also pass `--test-args simd-intrinsic/simd-intrinsic-generic-bitmask.rs`
2019-11-11T23:51:15.1695065Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1695123Z status: exit code: 1
2019-11-11T23:51:15.1695123Z status: exit code: 1
2019-11-11T23:51:15.1696080Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/simd-intrinsic/simd-intrinsic-generic-bitmask.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-bitmask" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-bitmask/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1696462Z ------------------------------------------
2019-11-11T23:51:15.1696501Z 
2019-11-11T23:51:15.1696743Z ------------------------------------------
2019-11-11T23:51:15.1696792Z stderr:
2019-11-11T23:51:15.1696792Z stderr:
2019-11-11T23:51:15.1697051Z ------------------------------------------
2019-11-11T23:51:15.1697113Z error[E0511]: invalid monomorphization of `simd_bitmask` intrinsic: bitmask `u16`, expected `u8`
2019-11-11T23:51:15.1697481Z    |
2019-11-11T23:51:15.1697481Z    |
2019-11-11T23:51:15.1697529Z LL |         let _: u16 = simd_bitmask(m2);
2019-11-11T23:51:15.1697627Z 
2019-11-11T23:51:15.1697627Z 
2019-11-11T23:51:15.1697682Z error[E0511]: invalid monomorphization of `simd_bitmask` intrinsic: bitmask `u16`, expected `u8`
2019-11-11T23:51:15.1698047Z    |
2019-11-11T23:51:15.1698047Z    |
2019-11-11T23:51:15.1698107Z LL |         let _: u16 = simd_bitmask(m8);
2019-11-11T23:51:15.1698188Z 
2019-11-11T23:51:15.1698188Z 
2019-11-11T23:51:15.1698260Z error[E0511]: invalid monomorphization of `simd_bitmask` intrinsic: bitmask `u32`, expected `u16`
2019-11-11T23:51:15.1698614Z    |
2019-11-11T23:51:15.1698614Z    |
2019-11-11T23:51:15.1698673Z LL |         let _: u32 = simd_bitmask(m16);
2019-11-11T23:51:15.1698757Z 
2019-11-11T23:51:15.1698757Z 
2019-11-11T23:51:15.1698811Z error[E0511]: invalid monomorphization of `simd_bitmask` intrinsic: bitmask `u64`, expected `u32`
2019-11-11T23:51:15.1699173Z    |
2019-11-11T23:51:15.1699173Z    |
2019-11-11T23:51:15.1699221Z LL |         let _: u64 = simd_bitmask(m32);
2019-11-11T23:51:15.1699329Z 
2019-11-11T23:51:15.1699329Z 
2019-11-11T23:51:15.1699383Z error[E0511]: invalid monomorphization of `simd_bitmask` intrinsic: bitmask `u128`, expected `u64`
2019-11-11T23:51:15.1699758Z    |
2019-11-11T23:51:15.1699758Z    |
2019-11-11T23:51:15.1699807Z LL |         let _: u128 = simd_bitmask(m64);
2019-11-11T23:51:15.1699899Z 
2019-11-11T23:51:15.1699947Z error: aborting due to 5 previous errors
2019-11-11T23:51:15.1699978Z 
2019-11-11T23:51:15.1700253Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1701678Z 27 
2019-11-11T23:51:15.1701716Z 
2019-11-11T23:51:15.1701744Z 
2019-11-11T23:51:15.1701796Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1702217Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-cast/simd-intrinsic-generic-cast.stderr
2019-11-11T23:51:15.1702778Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1703108Z To only update this specific test, also pass `--test-args simd-intrinsic/simd-intrinsic-generic-cast.rs`
2019-11-11T23:51:15.1703218Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1703266Z status: exit code: 1
2019-11-11T23:51:15.1703266Z status: exit code: 1
2019-11-11T23:51:15.1704192Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/simd-intrinsic/simd-intrinsic-generic-cast.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-cast" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-cast/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1704569Z ------------------------------------------
2019-11-11T23:51:15.1704606Z 
2019-11-11T23:51:15.1704845Z ------------------------------------------
2019-11-11T23:51:15.1704894Z stderr:
2019-11-11T23:51:15.1704894Z stderr:
2019-11-11T23:51:15.1705149Z ------------------------------------------
2019-11-11T23:51:15.1705471Z error[E0511]: invalid monomorphization of `simd_cast` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1705851Z    |
2019-11-11T23:51:15.1705851Z    |
2019-11-11T23:51:15.1705907Z LL |         simd_cast::<i32, i32>(0);
2019-11-11T23:51:15.1706004Z 
2019-11-11T23:51:15.1706004Z 
2019-11-11T23:51:15.1706329Z error[E0511]: invalid monomorphization of `simd_cast` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1706686Z    |
2019-11-11T23:51:15.1706686Z    |
2019-11-11T23:51:15.1706734Z LL |         simd_cast::<i32, i32x4>(0);
2019-11-11T23:51:15.1706816Z 
2019-11-11T23:51:15.1706816Z 
2019-11-11T23:51:15.1707145Z error[E0511]: invalid monomorphization of `simd_cast` intrinsic: expected SIMD return type, found non-SIMD `i32`
2019-11-11T23:51:15.1707496Z    |
2019-11-11T23:51:15.1707496Z    |
2019-11-11T23:51:15.1707559Z LL |         simd_cast::<i32x4, i32>(x);
2019-11-11T23:51:15.1707640Z 
2019-11-11T23:51:15.1707640Z 
2019-11-11T23:51:15.1707711Z error[E0511]: invalid monomorphization of `simd_cast` intrinsic: expected return type with length 4 (same as input type `i32x4`), found `i32x8` with length 8
2019-11-11T23:51:15.1708086Z    |
2019-11-11T23:51:15.1708086Z    |
2019-11-11T23:51:15.1708145Z LL |         simd_cast::<_, i32x8>(x);
2019-11-11T23:51:15.1708227Z 
2019-11-11T23:51:15.1708273Z error: aborting due to 4 previous errors
2019-11-11T23:51:15.1708315Z 
2019-11-11T23:51:15.1708586Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1710116Z 111 
2019-11-11T23:51:15.1710145Z 
2019-11-11T23:51:15.1710173Z 
2019-11-11T23:51:15.1710222Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1710626Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-comparison/simd-intrinsic-generic-comparison.stderr
2019-11-11T23:51:15.1710906Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1711225Z To only update this specific test, also pass `--test-args simd-intrinsic/simd-intrinsic-generic-comparison.rs`
2019-11-11T23:51:15.1711343Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1711390Z status: exit code: 1
2019-11-11T23:51:15.1711390Z status: exit code: 1
2019-11-11T23:51:15.1712333Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/simd-intrinsic/simd-intrinsic-generic-comparison.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-comparison" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-comparison/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1713274Z ------------------------------------------
2019-11-11T23:51:15.1713320Z 
2019-11-11T23:51:15.1713590Z ------------------------------------------
2019-11-11T23:51:15.1713639Z stderr:
2019-11-11T23:51:15.1713639Z stderr:
2019-11-11T23:51:15.1713896Z ------------------------------------------
2019-11-11T23:51:15.1714227Z error[E0511]: invalid monomorphization of `simd_eq` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1714605Z    |
2019-11-11T23:51:15.1714605Z    |
2019-11-11T23:51:15.1714653Z LL |         simd_eq::<i32, i32>(0, 0);
2019-11-11T23:51:15.1714752Z 
2019-11-11T23:51:15.1714752Z 
2019-11-11T23:51:15.1715071Z error[E0511]: invalid monomorphization of `simd_ne` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1715511Z    |
2019-11-11T23:51:15.1715511Z    |
2019-11-11T23:51:15.1715558Z LL |         simd_ne::<i32, i32>(0, 0);
2019-11-11T23:51:15.1715648Z 
2019-11-11T23:51:15.1715648Z 
2019-11-11T23:51:15.1715996Z error[E0511]: invalid monomorphization of `simd_lt` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1716353Z    |
2019-11-11T23:51:15.1716353Z    |
2019-11-11T23:51:15.1716416Z LL |         simd_lt::<i32, i32>(0, 0);
2019-11-11T23:51:15.1716496Z 
2019-11-11T23:51:15.1716496Z 
2019-11-11T23:51:15.1716815Z error[E0511]: invalid monomorphization of `simd_le` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1717466Z    |
2019-11-11T23:51:15.1717466Z    |
2019-11-11T23:51:15.1717513Z LL |         simd_le::<i32, i32>(0, 0);
2019-11-11T23:51:15.1717655Z 
2019-11-11T23:51:15.1717655Z 
2019-11-11T23:51:15.1717995Z error[E0511]: invalid monomorphization of `simd_gt` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1718561Z    |
2019-11-11T23:51:15.1718561Z    |
2019-11-11T23:51:15.1718697Z LL |         simd_gt::<i32, i32>(0, 0);
2019-11-11T23:51:15.1718805Z 
2019-11-11T23:51:15.1718805Z 
2019-11-11T23:51:15.1719159Z error[E0511]: invalid monomorphization of `simd_ge` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1719526Z    |
2019-11-11T23:51:15.1719526Z    |
2019-11-11T23:51:15.1719572Z LL |         simd_ge::<i32, i32>(0, 0);
2019-11-11T23:51:15.1719653Z 
2019-11-11T23:51:15.1719653Z 
2019-11-11T23:51:15.1719991Z error[E0511]: invalid monomorphization of `simd_eq` intrinsic: expected SIMD return type, found non-SIMD `i32`
2019-11-11T23:51:15.1720353Z    |
2019-11-11T23:51:15.1720353Z    |
2019-11-11T23:51:15.1720416Z LL |         simd_eq::<_, i32>(x, x);
2019-11-11T23:51:15.1720505Z 
2019-11-11T23:51:15.1720505Z 
2019-11-11T23:51:15.1720844Z error[E0511]: invalid monomorphization of `simd_ne` intrinsic: expected SIMD return type, found non-SIMD `i32`
2019-11-11T23:51:15.1721200Z    |
2019-11-11T23:51:15.1721200Z    |
2019-11-11T23:51:15.1721264Z LL |         simd_ne::<_, i32>(x, x);
2019-11-11T23:51:15.1721346Z 
2019-11-11T23:51:15.1721346Z 
2019-11-11T23:51:15.1721664Z error[E0511]: invalid monomorphization of `simd_lt` intrinsic: expected SIMD return type, found non-SIMD `i32`
2019-11-11T23:51:15.1722028Z    |
2019-11-11T23:51:15.1722028Z    |
2019-11-11T23:51:15.1722085Z LL |         simd_lt::<_, i32>(x, x);
2019-11-11T23:51:15.1722183Z 
2019-11-11T23:51:15.1722183Z 
2019-11-11T23:51:15.1722816Z error[E0511]: invalid monomorphization of `simd_le` intrinsic: expected SIMD return type, found non-SIMD `i32`
2019-11-11T23:51:15.1723218Z    |
2019-11-11T23:51:15.1723218Z    |
2019-11-11T23:51:15.1723266Z LL |         simd_le::<_, i32>(x, x);
2019-11-11T23:51:15.1723364Z 
2019-11-11T23:51:15.1723364Z 
2019-11-11T23:51:15.1723687Z error[E0511]: invalid monomorphization of `simd_gt` intrinsic: expected SIMD return type, found non-SIMD `i32`
2019-11-11T23:51:15.1724053Z    |
2019-11-11T23:51:15.1724053Z    |
2019-11-11T23:51:15.1724101Z LL |         simd_gt::<_, i32>(x, x);
2019-11-11T23:51:15.1724191Z 
2019-11-11T23:51:15.1724191Z 
2019-11-11T23:51:15.1724529Z error[E0511]: invalid monomorphization of `simd_ge` intrinsic: expected SIMD return type, found non-SIMD `i32`
2019-11-11T23:51:15.1724888Z    |
2019-11-11T23:51:15.1724888Z    |
2019-11-11T23:51:15.1724951Z LL |         simd_ge::<_, i32>(x, x);
2019-11-11T23:51:15.1725032Z 
2019-11-11T23:51:15.1725032Z 
2019-11-11T23:51:15.1725095Z error[E0511]: invalid monomorphization of `simd_eq` intrinsic: expected return type with length 4 (same as input type `i32x4`), found `i16x8` with length 8
2019-11-11T23:51:15.1725486Z    |
2019-11-11T23:51:15.1725486Z    |
2019-11-11T23:51:15.1725550Z LL |         simd_eq::<_, i16x8>(x, x);
2019-11-11T23:51:15.1725633Z 
2019-11-11T23:51:15.1725633Z 
2019-11-11T23:51:15.1725695Z error[E0511]: invalid monomorphization of `simd_ne` intrinsic: expected return type with length 4 (same as input type `i32x4`), found `i16x8` with length 8
2019-11-11T23:51:15.1726341Z    |
2019-11-11T23:51:15.1726341Z    |
2019-11-11T23:51:15.1726387Z LL |         simd_ne::<_, i16x8>(x, x);
2019-11-11T23:51:15.1726487Z 
2019-11-11T23:51:15.1726487Z 
2019-11-11T23:51:15.1726550Z error[E0511]: invalid monomorphization of `simd_lt` intrinsic: expected return type with length 4 (same as input type `i32x4`), found `i16x8` with length 8
2019-11-11T23:51:15.1726955Z    |
2019-11-11T23:51:15.1726955Z    |
2019-11-11T23:51:15.1727002Z LL |         simd_lt::<_, i16x8>(x, x);
2019-11-11T23:51:15.1727100Z 
2019-11-11T23:51:15.1727100Z 
2019-11-11T23:51:15.1727162Z error[E0511]: invalid monomorphization of `simd_le` intrinsic: expected return type with length 4 (same as input type `i32x4`), found `i16x8` with length 8
2019-11-11T23:51:15.1727558Z    |
2019-11-11T23:51:15.1727558Z    |
2019-11-11T23:51:15.1727606Z LL |         simd_le::<_, i16x8>(x, x);
2019-11-11T23:51:15.1727686Z 
2019-11-11T23:51:15.1727686Z 
2019-11-11T23:51:15.1727765Z error[E0511]: invalid monomorphization of `simd_gt` intrinsic: expected return type with length 4 (same as input type `i32x4`), found `i16x8` with length 8
2019-11-11T23:51:15.1728124Z    |
2019-11-11T23:51:15.1728124Z    |
2019-11-11T23:51:15.1728187Z LL |         simd_gt::<_, i16x8>(x, x);
2019-11-11T23:51:15.1728268Z 
2019-11-11T23:51:15.1728268Z 
2019-11-11T23:51:15.1728346Z error[E0511]: invalid monomorphization of `simd_ge` intrinsic: expected return type with length 4 (same as input type `i32x4`), found `i16x8` with length 8
2019-11-11T23:51:15.1728727Z    |
2019-11-11T23:51:15.1728727Z    |
2019-11-11T23:51:15.1728790Z LL |         simd_ge::<_, i16x8>(x, x);
2019-11-11T23:51:15.1728870Z 
2019-11-11T23:51:15.1728917Z error: aborting due to 18 previous errors
2019-11-11T23:51:15.1728964Z 
2019-11-11T23:51:15.1729239Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1730497Z 
2019-11-11T23:51:15.1730535Z 
2019-11-11T23:51:15.1730601Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1730985Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-reduction/simd-intrinsic-generic-reduction.stderr
2019-11-11T23:51:15.1731264Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1731599Z To only update this specific test, also pass `--test-args simd-intrinsic/simd-intrinsic-generic-reduction.rs`
2019-11-11T23:51:15.1731690Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1731755Z status: exit code: 1
2019-11-11T23:51:15.1731755Z status: exit code: 1
2019-11-11T23:51:15.1733085Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/simd-intrinsic/simd-intrinsic-generic-reduction.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-reduction" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-reduction/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1733617Z ------------------------------------------
2019-11-11T23:51:15.1733656Z 
2019-11-11T23:51:15.1733900Z ------------------------------------------
2019-11-11T23:51:15.1733970Z stderr:
2019-11-11T23:51:15.1733970Z stderr:
2019-11-11T23:51:15.1734211Z ------------------------------------------
2019-11-11T23:51:15.1734280Z error[E0511]: invalid monomorphization of `simd_reduce_add_ordered` intrinsic: expected return type `f32` (element of input `f32x4`), found `i32`
2019-11-11T23:51:15.1734673Z    |
2019-11-11T23:51:15.1734673Z    |
2019-11-11T23:51:15.1734730Z LL |         simd_reduce_add_ordered(z, 0);
2019-11-11T23:51:15.1734831Z 
2019-11-11T23:51:15.1734831Z 
2019-11-11T23:51:15.1734892Z error[E0511]: invalid monomorphization of `simd_reduce_mul_ordered` intrinsic: expected return type `f32` (element of input `f32x4`), found `i32`
2019-11-11T23:51:15.1735275Z    |
2019-11-11T23:51:15.1735275Z    |
2019-11-11T23:51:15.1735323Z LL |         simd_reduce_mul_ordered(z, 1);
2019-11-11T23:51:15.1735421Z 
2019-11-11T23:51:15.1735421Z 
2019-11-11T23:51:15.1735481Z error[E0511]: invalid monomorphization of `simd_reduce_and` intrinsic: expected return type `u32` (element of input `u32x4`), found `f32`
2019-11-11T23:51:15.1735868Z    |
2019-11-11T23:51:15.1735868Z    |
2019-11-11T23:51:15.1735917Z LL |         let _: f32 = simd_reduce_and(x);
2019-11-11T23:51:15.1736029Z 
2019-11-11T23:51:15.1736029Z 
2019-11-11T23:51:15.1736090Z error[E0511]: invalid monomorphization of `simd_reduce_or` intrinsic: expected return type `u32` (element of input `u32x4`), found `f32`
2019-11-11T23:51:15.1736461Z    |
2019-11-11T23:51:15.1736461Z    |
2019-11-11T23:51:15.1736510Z LL |         let _: f32 = simd_reduce_or(x);
2019-11-11T23:51:15.1736593Z 
2019-11-11T23:51:15.1736593Z 
2019-11-11T23:51:15.1736670Z error[E0511]: invalid monomorphization of `simd_reduce_xor` intrinsic: expected return type `u32` (element of input `u32x4`), found `f32`
2019-11-11T23:51:15.1737032Z    |
2019-11-11T23:51:15.1737032Z    |
2019-11-11T23:51:15.1737097Z LL |         let _: f32 = simd_reduce_xor(x);
2019-11-11T23:51:15.1737188Z 
2019-11-11T23:51:15.1737188Z 
2019-11-11T23:51:15.1737263Z error[E0511]: invalid monomorphization of `simd_reduce_and` intrinsic: unsupported simd_reduce_and from `f32x4` with element `f32` to `f32`
2019-11-11T23:51:15.1737623Z    |
2019-11-11T23:51:15.1737623Z    |
2019-11-11T23:51:15.1737690Z LL |         let _: f32 = simd_reduce_and(z);
2019-11-11T23:51:15.1737773Z 
2019-11-11T23:51:15.1737773Z 
2019-11-11T23:51:15.1737832Z error[E0511]: invalid monomorphization of `simd_reduce_or` intrinsic: unsupported simd_reduce_or from `f32x4` with element `f32` to `f32`
2019-11-11T23:51:15.1738303Z    |
2019-11-11T23:51:15.1738303Z    |
2019-11-11T23:51:15.1738351Z LL |         let _: f32 = simd_reduce_or(z);
2019-11-11T23:51:15.1738516Z 
2019-11-11T23:51:15.1738516Z 
2019-11-11T23:51:15.1738581Z error[E0511]: invalid monomorphization of `simd_reduce_xor` intrinsic: unsupported simd_reduce_xor from `f32x4` with element `f32` to `f32`
2019-11-11T23:51:15.1738991Z    |
2019-11-11T23:51:15.1738991Z    |
2019-11-11T23:51:15.1739039Z LL |         let _: f32 = simd_reduce_xor(z);
2019-11-11T23:51:15.1739139Z 
2019-11-11T23:51:15.1739139Z 
2019-11-11T23:51:15.1739199Z error[E0511]: invalid monomorphization of `simd_reduce_all` intrinsic: unsupported simd_reduce_all from `f32x4` with element `f32` to `bool`
2019-11-11T23:51:15.1739585Z    |
2019-11-11T23:51:15.1739585Z    |
2019-11-11T23:51:15.1739634Z LL |         let _: bool = simd_reduce_all(z);
2019-11-11T23:51:15.1739718Z 
2019-11-11T23:51:15.1739718Z 
2019-11-11T23:51:15.1739943Z error[E0511]: invalid monomorphization of `simd_reduce_any` intrinsic: unsupported simd_reduce_any from `f32x4` with element `f32` to `bool`
2019-11-11T23:51:15.1740327Z    |
2019-11-11T23:51:15.1740327Z    |
2019-11-11T23:51:15.1740394Z LL |         let _: bool = simd_reduce_any(z);
2019-11-11T23:51:15.1740480Z 
2019-11-11T23:51:15.1740545Z error: aborting due to 10 previous errors
2019-11-11T23:51:15.1740578Z 
2019-11-11T23:51:15.1740853Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1742113Z 
2019-11-11T23:51:15.1742501Z 
2019-11-11T23:51:15.1742557Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1743004Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-elements/simd-intrinsic-generic-elements.stderr
2019-11-11T23:51:15.1743305Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1743633Z To only update this specific test, also pass `--test-args simd-intrinsic/simd-intrinsic-generic-elements.rs`
2019-11-11T23:51:15.1743753Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1743802Z status: exit code: 1
2019-11-11T23:51:15.1743802Z status: exit code: 1
2019-11-11T23:51:15.1744721Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/simd-intrinsic/simd-intrinsic-generic-elements.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-elements" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-elements/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1745091Z ------------------------------------------
2019-11-11T23:51:15.1745144Z 
2019-11-11T23:51:15.1745556Z ------------------------------------------
2019-11-11T23:51:15.1745607Z stderr:
2019-11-11T23:51:15.1745607Z stderr:
2019-11-11T23:51:15.1745844Z ------------------------------------------
2019-11-11T23:51:15.1746277Z error[E0511]: invalid monomorphization of `simd_insert` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1746691Z    |
2019-11-11T23:51:15.1746691Z    |
2019-11-11T23:51:15.1746740Z LL |         simd_insert(0, 0, 0);
2019-11-11T23:51:15.1746822Z 
2019-11-11T23:51:15.1746822Z 
2019-11-11T23:51:15.1746899Z error[E0511]: invalid monomorphization of `simd_insert` intrinsic: expected inserted type `i32` (element of input `i32x4`), found `f64`
2019-11-11T23:51:15.1747256Z    |
2019-11-11T23:51:15.1747256Z    |
2019-11-11T23:51:15.1747321Z LL |         simd_insert(x, 0, 1.0);
2019-11-11T23:51:15.1747413Z 
2019-11-11T23:51:15.1747413Z 
2019-11-11T23:51:15.1747471Z error[E0511]: invalid monomorphization of `simd_extract` intrinsic: expected return type `i32` (element of input `i32x4`), found `f32`
2019-11-11T23:51:15.1747857Z    |
2019-11-11T23:51:15.1747857Z    |
2019-11-11T23:51:15.1747903Z LL |         simd_extract::<_, f32>(x, 0);
2019-11-11T23:51:15.1748003Z 
2019-11-11T23:51:15.1748003Z 
2019-11-11T23:51:15.1748329Z error[E0511]: invalid monomorphization of `simd_shuffle2` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1748696Z    |
2019-11-11T23:51:15.1748696Z    |
2019-11-11T23:51:15.1748746Z LL |         simd_shuffle2::<i32, i32>(0, 0, [0; 2]);
2019-11-11T23:51:15.1748859Z 
2019-11-11T23:51:15.1748859Z 
2019-11-11T23:51:15.1749188Z error[E0511]: invalid monomorphization of `simd_shuffle3` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1749565Z    |
2019-11-11T23:51:15.1749565Z    |
2019-11-11T23:51:15.1749613Z LL |         simd_shuffle3::<i32, i32>(0, 0, [0; 3]);
2019-11-11T23:51:15.1749697Z 
2019-11-11T23:51:15.1749697Z 
2019-11-11T23:51:15.1750042Z error[E0511]: invalid monomorphization of `simd_shuffle4` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1750390Z    |
2019-11-11T23:51:15.1750390Z    |
2019-11-11T23:51:15.1750455Z LL |         simd_shuffle4::<i32, i32>(0, 0, [0; 4]);
2019-11-11T23:51:15.1750540Z 
2019-11-11T23:51:15.1750540Z 
2019-11-11T23:51:15.1750884Z error[E0511]: invalid monomorphization of `simd_shuffle8` intrinsic: expected SIMD input type, found non-SIMD `i32`
2019-11-11T23:51:15.1751247Z    |
2019-11-11T23:51:15.1751247Z    |
2019-11-11T23:51:15.1751322Z LL |         simd_shuffle8::<i32, i32>(0, 0, [0; 8]);
2019-11-11T23:51:15.1751408Z 
2019-11-11T23:51:15.1751408Z 
2019-11-11T23:51:15.1751473Z error[E0511]: invalid monomorphization of `simd_shuffle2` intrinsic: expected return element type `i32` (element of input `i32x4`), found `f32x2` with element type `f32`
2019-11-11T23:51:15.1751859Z    |
2019-11-11T23:51:15.1751859Z    |
2019-11-11T23:51:15.1751925Z LL |         simd_shuffle2::<_, f32x2>(x, x, [0; 2]);
2019-11-11T23:51:15.1752012Z 
2019-11-11T23:51:15.1752012Z 
2019-11-11T23:51:15.1752076Z error[E0511]: invalid monomorphization of `simd_shuffle3` intrinsic: expected return element type `i32` (element of input `i32x4`), found `f32x3` with element type `f32`
2019-11-11T23:51:15.1753531Z    |
2019-11-11T23:51:15.1753531Z    |
2019-11-11T23:51:15.1753582Z LL |         simd_shuffle3::<_, f32x3>(x, x, [0; 3]);
2019-11-11T23:51:15.1753691Z 
2019-11-11T23:51:15.1753691Z 
2019-11-11T23:51:15.1753756Z error[E0511]: invalid monomorphization of `simd_shuffle4` intrinsic: expected return element type `i32` (element of input `i32x4`), found `f32x4` with element type `f32`
2019-11-11T23:51:15.1754173Z    |
2019-11-11T23:51:15.1754173Z    |
2019-11-11T23:51:15.1754221Z LL |         simd_shuffle4::<_, f32x4>(x, x, [0; 4]);
2019-11-11T23:51:15.1754335Z 
2019-11-11T23:51:15.1754335Z 
2019-11-11T23:51:15.1754397Z error[E0511]: invalid monomorphization of `simd_shuffle8` intrinsic: expected return element type `i32` (element of input `i32x4`), found `f32x8` with element type `f32`
2019-11-11T23:51:15.1754785Z    |
2019-11-11T23:51:15.1754785Z    |
2019-11-11T23:51:15.1754833Z LL |         simd_shuffle8::<_, f32x8>(x, x, [0; 8]);
2019-11-11T23:51:15.1754934Z 
2019-11-11T23:51:15.1754934Z 
2019-11-11T23:51:15.1754994Z error[E0511]: invalid monomorphization of `simd_shuffle2` intrinsic: expected return type of length 2, found `i32x8` with length 8
2019-11-11T23:51:15.1755364Z    |
2019-11-11T23:51:15.1755364Z    |
2019-11-11T23:51:15.1755413Z LL |         simd_shuffle2::<_, i32x8>(x, x, [0; 2]);
2019-11-11T23:51:15.1755508Z 
2019-11-11T23:51:15.1755508Z 
2019-11-11T23:51:15.1755582Z error[E0511]: invalid monomorphization of `simd_shuffle3` intrinsic: expected return type of length 3, found `i32x4` with length 4
2019-11-11T23:51:15.1755946Z    |
2019-11-11T23:51:15.1755946Z    |
2019-11-11T23:51:15.1756012Z LL |         simd_shuffle3::<_, i32x4>(x, x, [0; 3]);
2019-11-11T23:51:15.1756097Z 
2019-11-11T23:51:15.1756097Z 
2019-11-11T23:51:15.1756155Z error[E0511]: invalid monomorphization of `simd_shuffle4` intrinsic: expected return type of length 4, found `i32x3` with length 3
2019-11-11T23:51:15.1756531Z    |
2019-11-11T23:51:15.1756531Z    |
2019-11-11T23:51:15.1756580Z LL |         simd_shuffle4::<_, i32x3>(x, x, [0; 4]);
2019-11-11T23:51:15.1756695Z 
2019-11-11T23:51:15.1756695Z 
2019-11-11T23:51:15.1756754Z error[E0511]: invalid monomorphization of `simd_shuffle8` intrinsic: expected return type of length 8, found `i32x2` with length 2
2019-11-11T23:51:15.1757414Z    |
2019-11-11T23:51:15.1757414Z    |
2019-11-11T23:51:15.1757463Z LL |         simd_shuffle8::<_, i32x2>(x, x, [0; 8]);
2019-11-11T23:51:15.1757565Z 
2019-11-11T23:51:15.1757612Z error: aborting due to 15 previous errors
2019-11-11T23:51:15.1757644Z 
2019-11-11T23:51:15.1757972Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1759454Z 51 
2019-11-11T23:51:15.1759483Z 
2019-11-11T23:51:15.1759512Z 
2019-11-11T23:51:15.1759563Z The actual stderr differed from the expected stderr.
2019-11-11T23:51:15.1759989Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-select/simd-intrinsic-generic-select.stderr
2019-11-11T23:51:15.1760270Z To update references, rerun the tests and pass the `--bless` flag
2019-11-11T23:51:15.1760584Z To only update this specific test, also pass `--test-args simd-intrinsic/simd-intrinsic-generic-select.rs`
2019-11-11T23:51:15.1760689Z error: 1 errors occurred comparing output.
2019-11-11T23:51:15.1760737Z status: exit code: 1
2019-11-11T23:51:15.1760737Z status: exit code: 1
2019-11-11T23:51:15.1761678Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/simd-intrinsic/simd-intrinsic-generic-select.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-select" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/simd-intrinsic/simd-intrinsic-generic-select/auxiliary" "-A" "unused"
2019-11-11T23:51:15.1762046Z ------------------------------------------
2019-11-11T23:51:15.1762084Z 
2019-11-11T23:51:15.1762327Z ------------------------------------------
2019-11-11T23:51:15.1762670Z stderr:
2019-11-11T23:51:15.1762670Z stderr:
2019-11-11T23:51:15.1763007Z ------------------------------------------
2019-11-11T23:51:15.1763074Z error[E0511]: invalid monomorphization of `simd_select` intrinsic: mismatched lengths: mask length `8` != other vector length `4`
2019-11-11T23:51:15.1763470Z    |
2019-11-11T23:51:15.1763470Z    |
2019-11-11T23:51:15.1763527Z LL |         simd_select(m8, x, x);
2019-11-11T23:51:15.1763622Z 
2019-11-11T23:51:15.1763622Z 
2019-11-11T23:51:15.1763679Z error[E0511]: invalid monomorphization of `simd_select` intrinsic: mask element type is `u32`, expected `i_`
2019-11-11T23:51:15.1764051Z    |
2019-11-11T23:51:15.1764051Z    |
2019-11-11T23:51:15.1764097Z LL |         simd_select(x, x, x);
2019-11-11T23:51:15.1764178Z 
2019-11-11T23:51:15.1764178Z 
2019-11-11T23:51:15.1764251Z error[E0511]: invalid monomorphization of `simd_select` intrinsic: mask element type is `f32`, expected `i_`
2019-11-11T23:51:15.1764609Z    |
2019-11-11T23:51:15.1764609Z    |
2019-11-11T23:51:15.1764673Z LL |         simd_select(z, z, z);
2019-11-11T23:51:15.1764762Z 
2019-11-11T23:51:15.1764762Z 
2019-11-11T23:51:15.1765092Z error[E0511]: invalid monomorphization of `simd_select` intrinsic: expected SIMD argument type, found non-SIMD `u32`
2019-11-11T23:51:15.1765464Z    |
2019-11-11T23:51:15.1765464Z    |
2019-11-11T23:51:15.1765512Z LL |         simd_select(m4, 0u32, 1u32);
2019-11-11T23:51:15.1765609Z 
2019-11-11T23:51:15.1765609Z 
2019-11-11T23:51:15.1765670Z error[E0511]: invalid monomorphization of `simd_select_bitmask` intrinsic: mismatched lengths: mask length `8` != other vector length `4`
2019-11-11T23:51:15.1766040Z    |
2019-11-11T23:51:15.1766040Z    |
2019-11-11T23:51:15.1766217Z LL |         simd_select_bitmask(0u8, x, x);
2019-11-11T23:51:15.1766320Z 
2019-11-11T23:51:15.1766320Z 
2019-11-11T23:51:15.1766777Z error[E0511]: invalid monomorphization of `simd_select_bitmask` intrinsic: expected SIMD argument type, found non-SIMD `u32`
2019-11-11T23:51:15.1767183Z    |
2019-11-11T23:51:15.1767183Z    |
2019-11-11T23:51:15.1767232Z LL |         simd_select_bitmask(0u8, 1u32, 2u32);
2019-11-11T23:51:15.1767337Z 
2019-11-11T23:51:15.1767337Z 
2019-11-11T23:51:15.1767392Z error[E0511]: invalid monomorphization of `simd_select_bitmask` intrinsic: `f32` is not an integral type
2019-11-11T23:51:15.1767762Z    |
2019-11-11T23:51:15.1767762Z    |
2019-11-11T23:51:15.1767812Z LL |         simd_select_bitmask(0.0f32, x, x);
2019-11-11T23:51:15.1767908Z 
2019-11-11T23:51:15.1767908Z 
2019-11-11T23:51:15.1767980Z error[E0511]: invalid monomorphization of `simd_select_bitmask` intrinsic: `&str` is not an integral type
2019-11-11T23:51:15.1768341Z    |
2019-11-11T23:51:15.1768341Z    |
2019-11-11T23:51:15.1768405Z LL |         simd_select_bitmask("x", x, x);
2019-11-11T23:51:15.1768489Z 
2019-11-11T23:51:15.1768535Z error: aborting due to 8 previous errors
2019-11-11T23:51:15.1768584Z 
2019-11-11T23:51:15.1768858Z For more information about this error, try `rustc --explain E0511`.
---
2019-11-11T23:51:15.1777206Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:537:22
2019-11-11T23:51:15.1777380Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
2019-11-11T23:51:15.1777425Z 
2019-11-11T23:51:15.1777454Z 
2019-11-11T23:51:15.1779208Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2019-11-11T23:51:15.1779462Z 
2019-11-11T23:51:15.1779495Z 
2019-11-11T23:51:15.1779563Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2019-11-11T23:51:15.1779620Z Build completed unsuccessfully in 1:05:53
2019-11-11T23:51:15.1779620Z Build completed unsuccessfully in 1:05:53
2019-11-11T23:51:15.1779671Z == clock drift check ==
2019-11-11T23:51:15.1779737Z   local time: Mon Nov 11 23:51:14 UTC 2019
2019-11-11T23:51:15.1779797Z   network time: Mon, 11 Nov 2019 23:51:14 GMT
2019-11-11T23:51:15.1779848Z == end clock drift check ==
2019-11-11T23:51:15.4727909Z 
2019-11-11T23:51:15.4836708Z ##[error]Bash exited with code '1'.
2019-11-11T23:51:15.4878561Z ##[section]Starting: Checkout
2019-11-11T23:51:15.4880726Z ==============================================================================
2019-11-11T23:51:15.4880778Z Task         : Get sources
2019-11-11T23:51:15.4880845Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
