plain
2019-11-23T14:44:27.8437353Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-11-23T14:44:27.8453281Z ##[command]git config gc.auto 0
2019-11-23T14:44:27.8499331Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-11-23T14:44:27.8540359Z ##[command]git config --get-all http.proxy
2019-11-23T14:44:27.8663581Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/66670/merge:refs/remotes/pull/66670/merge
---
2019-11-23T15:33:31.2561630Z .................................................................................................... 1600/9281
2019-11-23T15:33:35.4769065Z .................................................................................................... 1700/9281
2019-11-23T15:33:46.6108886Z ...........................i........................................................................ 1800/9281
2019-11-23T15:33:52.2225717Z .................................................................................................... 1900/9281
2019-11-23T15:34:04.2520439Z ............iiiii................................................................................... 2000/9281
2019-11-23T15:34:12.2595071Z .................................................................................................... 2200/9281
2019-11-23T15:34:14.7292704Z .................................................................................................... 2300/9281
2019-11-23T15:34:19.2892354Z .................................................................................................... 2400/9281
2019-11-23T15:34:36.5565960Z .................................................................................................... 2500/9281
---
2019-11-23T15:36:58.2344464Z ............i...............i....................................................................... 4800/9281
2019-11-23T15:37:06.6957700Z .................................................................................................... 4900/9281
2019-11-23T15:37:11.5742193Z .................................................................................................... 5000/9281
2019-11-23T15:37:19.7502276Z .................................................................................................... 5100/9281
2019-11-23T15:37:25.2078253Z .................ii.ii...........i.................................................................. 5200/9281
2019-11-23T15:37:33.5455009Z .................................................................................................... 5400/9281
2019-11-23T15:37:43.2590696Z ...................................................................................................i 5500/9281
2019-11-23T15:37:50.5966007Z .................................................................................................... 5600/9281
2019-11-23T15:37:55.9819282Z .................................................................................................... 5700/9281
2019-11-23T15:37:55.9819282Z .................................................................................................... 5700/9281
2019-11-23T15:38:05.3697115Z .....................................................................................ii...i..ii..... 5800/9281
2019-11-23T15:38:25.1355922Z .................................................................................................... 6000/9281
2019-11-23T15:38:31.8522220Z .................................................................................................... 6100/9281
2019-11-23T15:38:35.6726851Z .................................................................................................... 6200/9281
2019-11-23T15:38:35.6726851Z .................................................................................................... 6200/9281
2019-11-23T15:38:46.8930340Z ........i..ii....................................................................................... 6300/9281
2019-11-23T15:39:02.3584974Z ............................................................................i....................... 6500/9281
2019-11-23T15:39:04.4406845Z .................................................................................................... 6600/9281
2019-11-23T15:39:06.3849978Z ...................................................................i................................ 6700/9281
2019-11-23T15:39:08.8141806Z .................................................................................................... 6800/9281
---
2019-11-23T15:43:05.8447433Z ---- [ui] ui/codemap_tests/unicode_2.rs stdout ----
2019-11-23T15:43:05.8447480Z diff of stderr:
2019-11-23T15:43:05.8447527Z 
2019-11-23T15:43:05.8447558Z 14    |
2019-11-23T15:43:05.8447593Z 15    = help: valid widths are 8, 16, 32, 64 and 128
2019-11-23T15:43:05.8447831Z - error[E0425]: cannot find value `a̐é` in this scope
2019-11-23T15:43:05.8448008Z + error[E0425]: cannot find value `a̐é` in this scope
2019-11-23T15:43:05.8448184Z 18   --> $DIR/unicode_2.rs:6:13
2019-11-23T15:43:05.8448235Z 19    |
2019-11-23T15:43:05.8448235Z 19    |
2019-11-23T15:43:05.8448382Z 20 LL |     let _ = a̐é;
2019-11-23T15:43:05.8448404Z 
2019-11-23T15:43:05.8448422Z 
2019-11-23T15:43:05.8448471Z The actual stderr differed from the expected stderr.
2019-11-23T15:43:05.8448700Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/codemap_tests/unicode_2/unicode_2.stderr
2019-11-23T15:43:05.8448887Z To update references, rerun the tests and pass the `--bless` flag
2019-11-23T15:43:05.8449099Z To only update this specific test, also pass `--test-args codemap_tests/unicode_2.rs`
2019-11-23T15:43:05.8449157Z error: 1 errors occurred comparing output.
2019-11-23T15:43:05.8449191Z status: exit code: 1
2019-11-23T15:43:05.8449191Z status: exit code: 1
2019-11-23T15:43:05.8449899Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/codemap_tests/unicode_2.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/codemap_tests/unicode_2" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/codemap_tests/unicode_2/auxiliary" "-A" "unused"
2019-11-23T15:43:05.8450207Z ------------------------------------------
2019-11-23T15:43:05.8450233Z 
2019-11-23T15:43:05.8450392Z ------------------------------------------
2019-11-23T15:43:05.8450441Z stderr:
2019-11-23T15:43:05.8450441Z stderr:
2019-11-23T15:43:05.8450782Z ------------------------------------------
2019-11-23T15:43:05.8451221Z error: invalid width `7` for integer literal
2019-11-23T15:43:05.8451586Z    |
2019-11-23T15:43:05.8451586Z    |
2019-11-23T15:43:05.8451829Z LL |     let _ = ("a̐éö̲", 0u7); //~ ERROR invalid width
2019-11-23T15:43:05.8451955Z    |
2019-11-23T15:43:05.8451955Z    |
2019-11-23T15:43:05.8452001Z    = help: valid widths are 8, 16, 32, 64 and 128
2019-11-23T15:43:05.8452034Z 
2019-11-23T15:43:05.8452098Z error: invalid width `42` for integer literal
2019-11-23T15:43:05.8452425Z    |
2019-11-23T15:43:05.8452425Z    |
2019-11-23T15:43:05.8452713Z LL |     let _ = ("아あ", 1i42); //~ ERROR invalid width
2019-11-23T15:43:05.8452809Z    |
2019-11-23T15:43:05.8452809Z    |
2019-11-23T15:43:05.8452876Z    = help: valid widths are 8, 16, 32, 64 and 128
2019-11-23T15:43:05.8453165Z error[E0425]: cannot find value `a̐é` in this scope
2019-11-23T15:43:05.8453424Z   --> /checkout/src/test/ui/codemap_tests/unicode_2.rs:6:13
2019-11-23T15:43:05.8453489Z    |
2019-11-23T15:43:05.8453725Z LL |     let _ = a̐é; //~ ERROR cannot find
---
2019-11-23T15:43:05.8479568Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:537:22
2019-11-23T15:43:05.8479634Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
2019-11-23T15:43:05.8491102Z 
2019-11-23T15:43:05.8491212Z 
2019-11-23T15:43:05.8494428Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2019-11-23T15:43:05.8494915Z 
2019-11-23T15:43:05.8495070Z 
2019-11-23T15:43:05.8504576Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2019-11-23T15:43:05.8504636Z Build completed unsuccessfully in 0:53:08
2019-11-23T15:43:05.8504636Z Build completed unsuccessfully in 0:53:08
2019-11-23T15:43:05.8548820Z == clock drift check ==
2019-11-23T15:43:05.8560947Z   local time: Sat Nov 23 15:43:05 UTC 2019
2019-11-23T15:43:06.3766966Z   network time: Sat, 23 Nov 2019 15:43:06 GMT
2019-11-23T15:43:06.3770366Z == end clock drift check ==
2019-11-23T15:43:07.2796943Z 
2019-11-23T15:43:07.2877973Z ##[error]Bash exited with code '1'.
2019-11-23T15:43:07.2907889Z ##[section]Starting: Checkout
2019-11-23T15:43:07.2909365Z ==============================================================================
2019-11-23T15:43:07.2909406Z Task         : Get sources
2019-11-23T15:43:07.2909441Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
