plain
2020-01-03T11:36:48.9697295Z ##[command]git remote add origin https://github.com/rust-lang/rust
2020-01-03T11:36:48.9880084Z ##[command]git config gc.auto 0
2020-01-03T11:36:48.9969105Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2020-01-03T11:36:49.0015491Z ##[command]git config --get-all http.proxy
2020-01-03T11:36:49.0167554Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/67766/merge:refs/remotes/pull/67766/merge
---
2020-01-03T12:39:00.1612663Z .................................................................................................... 1500/9468
2020-01-03T12:39:06.0834224Z .................................................................................................... 1600/9468
2020-01-03T12:39:11.6370855Z .................................................................................................... 1700/9468
2020-01-03T12:39:21.2416036Z .................................................................................................... 1800/9468
2020-01-03T12:39:29.4864955Z i................................................................................................... 1900/9468
2020-01-03T12:39:36.1904284Z ......................................................................................iiiii......... 2000/9468
2020-01-03T12:39:58.4283253Z .................................................................................................... 2200/9468
2020-01-03T12:40:00.7870099Z .................................................................................................... 2300/9468
2020-01-03T12:40:03.2245498Z .................................................................................................... 2400/9468
2020-01-03T12:40:09.3559917Z .................................................................................................... 2500/9468
---
2020-01-03T12:43:10.3201646Z .................i...............i.................................................................. 4900/9468
2020-01-03T12:43:20.2520525Z .................................................................................................... 5000/9468
2020-01-03T12:43:26.1236221Z ..............................................................i..................................... 5100/9468
2020-01-03T12:43:34.3840553Z .................................................................................................... 5200/9468
2020-01-03T12:43:42.1493013Z .............................ii.ii...........i...................................................... 5300/9468
2020-01-03T12:43:46.0019539Z ..................................................................i..............................F.. 5400/9468
2020-01-03T12:44:01.7087119Z .................................................................................................... 5600/9468
2020-01-03T12:44:08.8830996Z .............i...................................................................................... 5700/9468
2020-01-03T12:44:15.0983233Z .................................................................................................... 5800/9468
2020-01-03T12:44:26.0719951Z .................................................................................................... 5900/9468
2020-01-03T12:44:26.0719951Z .................................................................................................... 5900/9468
2020-01-03T12:44:38.1002065Z .ii...i..ii...........i............................................................................. 6000/9468
2020-01-03T12:44:56.0482681Z .................................................................................................... 6200/9468
2020-01-03T12:45:03.3942032Z .................................................................................................... 6300/9468
2020-01-03T12:45:03.3942032Z .................................................................................................... 6300/9468
2020-01-03T12:45:20.1570652Z ............................i..ii................................................................... 6400/9468
2020-01-03T12:45:40.4635537Z .................................................................................................... 6600/9468
2020-01-03T12:45:42.6176718Z ...i................................................................................................ 6700/9468
2020-01-03T12:45:45.0119486Z .................................................................................................... 6800/9468
2020-01-03T12:45:47.5726022Z ...i................................................................................................ 6900/9468
---
2020-01-03T12:47:27.9326452Z .................................................................................................... 7500/9468
2020-01-03T12:47:32.8809613Z .................................................................................................... 7600/9468
2020-01-03T12:47:38.4121801Z .................................................................................................... 7700/9468
2020-01-03T12:47:48.6764735Z .................................................................................................... 7800/9468
2020-01-03T12:47:56.3910596Z .....................................iiii........................................................... 7900/9468
2020-01-03T12:48:11.5061862Z .................................................................................................... 8100/9468
2020-01-03T12:48:20.2792423Z .................................................................................................... 8200/9468
2020-01-03T12:48:34.6277383Z .................................................................................................... 8300/9468
2020-01-03T12:48:42.7251780Z .................................................................................................... 8400/9468
---
2020-01-03T12:50:40.0994895Z diff of stderr:
2020-01-03T12:50:40.0995161Z 
2020-01-03T12:50:40.0995833Z 27   --> $DIR/issue-67691-unused-field-in-or-pattern.rs:25:20
2020-01-03T12:50:40.0996189Z 28    |
2020-01-03T12:50:40.0996442Z 29 LL |         A { i, ref j } | B { i, ref j } => {
2020-01-03T12:50:40.1003024Z +    |                    ^                ^
2020-01-03T12:50:40.1003089Z 31    |
2020-01-03T12:50:40.1003138Z 32 help: try ignoring the field
2020-01-03T12:50:40.1003201Z 33    |
2020-01-03T12:50:40.1003201Z 33    |
2020-01-03T12:50:40.1003232Z 
2020-01-03T12:50:40.1003692Z 49   --> $DIR/issue-67691-unused-field-in-or-pattern.rs:47:25
2020-01-03T12:50:40.1003747Z 50    |
2020-01-03T12:50:40.1003798Z 51 LL |         Some(A { i, ref j } | B { i, ref j }) => {
2020-01-03T12:50:40.1004198Z +    |                         ^                ^
2020-01-03T12:50:40.1004244Z 53    |
2020-01-03T12:50:40.1004309Z 54 help: try ignoring the field
2020-01-03T12:50:40.1004355Z 55    |
2020-01-03T12:50:40.1004355Z 55    |
2020-01-03T12:50:40.1004383Z 
2020-01-03T12:50:40.1004411Z 
2020-01-03T12:50:40.1004460Z The actual stderr differed from the expected stderr.
2020-01-03T12:50:40.1004892Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/issue-67691-unused-field-in-or-pattern/issue-67691-unused-field-in-or-pattern.stderr
2020-01-03T12:50:40.1004960Z diff of fixed:
2020-01-03T12:50:40.1004996Z 
2020-01-03T12:50:40.1005060Z 12     use MyEnum::*;
2020-01-03T12:50:40.1005160Z 14     match x {
2020-01-03T12:50:40.1005160Z 14     match x {
2020-01-03T12:50:40.1005471Z -         A { i, j: _ } | B { i, j: _ } => { //~ ERROR unused variable
2020-01-03T12:50:40.1005534Z +         A { i, j } | B { i, j } => { //~ ERROR unused variable
2020-01-03T12:50:40.1005651Z 17         }
2020-01-03T12:50:40.1005697Z 18     }
2020-01-03T12:50:40.1005727Z 
2020-01-03T12:50:40.1005727Z 
2020-01-03T12:50:40.1005771Z 22     use MyEnum::*;
2020-01-03T12:50:40.1005876Z 24     match x {
2020-01-03T12:50:40.1005876Z 24     match x {
2020-01-03T12:50:40.1006158Z -         A { i, j: _ } | B { i, j: _ } => { //~ ERROR unused variable
2020-01-03T12:50:40.1006219Z +         A { i, ref j } | B { i, ref j } => { //~ ERROR unused variable
2020-01-03T12:50:40.1006344Z 27         }
2020-01-03T12:50:40.1006389Z 28     }
2020-01-03T12:50:40.1006434Z 
2020-01-03T12:50:40.1006434Z 
2020-01-03T12:50:40.1006478Z 32     use MyEnum::*;
2020-01-03T12:50:40.1006573Z 34     match x {
2020-01-03T12:50:40.1006573Z 34     match x {
2020-01-03T12:50:40.1006881Z -         Some(A { i, j: _ } | B { i, j: _ }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1006943Z +         Some(A { i, j } | B { i, j }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1039358Z 37         }
2020-01-03T12:50:40.1039423Z 38 
2020-01-03T12:50:40.1039456Z 
2020-01-03T12:50:40.1039456Z 
2020-01-03T12:50:40.1039503Z 44     use MyEnum::*;
2020-01-03T12:50:40.1040351Z 46     match x {
2020-01-03T12:50:40.1040351Z 46     match x {
2020-01-03T12:50:40.1041606Z -         Some(A { i, j: _ } | B { i, j: _ }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1041709Z +         Some(A { i, ref j } | B { i, ref j }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1041830Z 49         }
2020-01-03T12:50:40.1041888Z 50 
2020-01-03T12:50:40.1041919Z 
2020-01-03T12:50:40.1041947Z 
2020-01-03T12:50:40.1041947Z 
2020-01-03T12:50:40.1041997Z The actual fixed differed from the expected fixed.
2020-01-03T12:50:40.1042700Z Actual fixed saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/issue-67691-unused-field-in-or-pattern/issue-67691-unused-field-in-or-pattern.fixed
2020-01-03T12:50:40.1043006Z To update references, rerun the tests and pass the `--bless` flag
2020-01-03T12:50:40.1043340Z To only update this specific test, also pass `--test-args lint/issue-67691-unused-field-in-or-pattern.rs`
2020-01-03T12:50:40.1043444Z error: 2 errors occurred comparing output.
2020-01-03T12:50:40.1043494Z status: exit code: 1
2020-01-03T12:50:40.1043494Z status: exit code: 1
2020-01-03T12:50:40.1044632Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/lint/issue-67691-unused-field-in-or-pattern.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/issue-67691-unused-field-in-or-pattern" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lint/issue-67691-unused-field-in-or-pattern/auxiliary" "-A" "unused"
2020-01-03T12:50:40.1045125Z ------------------------------------------
2020-01-03T12:50:40.1045165Z 
2020-01-03T12:50:40.1046116Z ------------------------------------------
2020-01-03T12:50:40.1046186Z stderr:
---
2020-01-03T12:50:40.1047207Z 
2020-01-03T12:50:40.1047263Z error: unused variable: `j`
2020-01-03T12:50:40.1047571Z   --> /checkout/src/test/ui/lint/issue-67691-unused-field-in-or-pattern.rs:15:16
2020-01-03T12:50:40.1047625Z    |
2020-01-03T12:50:40.1047677Z LL |         A { i, j } | B { i, j } => { //~ ERROR unused variable
2020-01-03T12:50:40.1047790Z    |
2020-01-03T12:50:40.1047834Z note: lint level defined here
2020-01-03T12:50:40.1048141Z   --> /checkout/src/test/ui/lint/issue-67691-unused-field-in-or-pattern.rs:4:9
2020-01-03T12:50:40.1048194Z    |
2020-01-03T12:50:40.1048194Z    |
2020-01-03T12:50:40.1048249Z LL | #![deny(unused)]
2020-01-03T12:50:40.1048308Z    |         ^^^^^^
2020-01-03T12:50:40.1048362Z    = note: `#[deny(unused_variables)]` implied by `#[deny(unused)]`
2020-01-03T12:50:40.1048414Z help: try ignoring the field
2020-01-03T12:50:40.1048459Z    |
2020-01-03T12:50:40.1048526Z LL |         A { i, j: _ } | B { i, j: _ } => { //~ ERROR unused variable
2020-01-03T12:50:40.1048611Z 
2020-01-03T12:50:40.1048662Z error: unused variable: `j`
2020-01-03T12:50:40.1048968Z   --> /checkout/src/test/ui/lint/issue-67691-unused-field-in-or-pattern.rs:25:20
2020-01-03T12:50:40.1049022Z    |
2020-01-03T12:50:40.1049022Z    |
2020-01-03T12:50:40.1049080Z LL |         A { i, ref j } | B { i, ref j } => { //~ ERROR unused variable
2020-01-03T12:50:40.1049178Z    |
2020-01-03T12:50:40.1049223Z help: try ignoring the field
2020-01-03T12:50:40.1049280Z    |
2020-01-03T12:50:40.1049280Z    |
2020-01-03T12:50:40.1049339Z LL |         A { i, j: _ } | B { i, j: _ } => { //~ ERROR unused variable
2020-01-03T12:50:40.1049436Z 
2020-01-03T12:50:40.1049481Z error: unused variable: `j`
2020-01-03T12:50:40.1049785Z   --> /checkout/src/test/ui/lint/issue-67691-unused-field-in-or-pattern.rs:35:21
2020-01-03T12:50:40.1050049Z    |
2020-01-03T12:50:40.1050049Z    |
2020-01-03T12:50:40.1050116Z LL |         Some(A { i, j } | B { i, j }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1050212Z    |
2020-01-03T12:50:40.1050266Z help: try ignoring the field
2020-01-03T12:50:40.1050311Z    |
2020-01-03T12:50:40.1050311Z    |
2020-01-03T12:50:40.1050363Z LL |         Some(A { i, j: _ } | B { i, j: _ }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1050457Z 
2020-01-03T12:50:40.1050874Z error: unused variable: `j`
2020-01-03T12:50:40.1051312Z   --> /checkout/src/test/ui/lint/issue-67691-unused-field-in-or-pattern.rs:47:25
2020-01-03T12:50:40.1051508Z    |
2020-01-03T12:50:40.1051508Z    |
2020-01-03T12:50:40.1051573Z LL |         Some(A { i, ref j } | B { i, ref j }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1051688Z    |
2020-01-03T12:50:40.1051741Z help: try ignoring the field
2020-01-03T12:50:40.1051787Z    |
2020-01-03T12:50:40.1051787Z    |
2020-01-03T12:50:40.1051845Z LL |         Some(A { i, j: _ } | B { i, j: _ }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1051933Z 
2020-01-03T12:50:40.1052238Z error: or-patterns are not fully implemented yet
2020-01-03T12:50:40.1052554Z   --> /checkout/src/test/ui/lint/issue-67691-unused-field-in-or-pattern.rs:35:14
2020-01-03T12:50:40.1052608Z    |
2020-01-03T12:50:40.1052608Z    |
2020-01-03T12:50:40.1052660Z LL |         Some(A { i, j } | B { i, j }) => { //~ ERROR unused variable
2020-01-03T12:50:40.1052752Z 
2020-01-03T12:50:40.1052808Z error: aborting due to 5 previous errors
2020-01-03T12:50:40.1052841Z 
2020-01-03T12:50:40.1052868Z 
---
2020-01-03T12:50:40.1054283Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:385:22
2020-01-03T12:50:40.1054347Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
2020-01-03T12:50:40.1090419Z 
2020-01-03T12:50:40.1090872Z 
2020-01-03T12:50:40.1093369Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-7/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "7.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2020-01-03T12:50:40.1093866Z 
2020-01-03T12:50:40.1093898Z 
2020-01-03T12:50:40.1093946Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2020-01-03T12:50:40.1094017Z Build completed unsuccessfully in 1:07:12
2020-01-03T12:50:40.1094017Z Build completed unsuccessfully in 1:07:12
2020-01-03T12:50:40.1094063Z == clock drift check ==
2020-01-03T12:50:40.1106313Z   local time: Fri Jan  3 12:50:40 UTC 2020
2020-01-03T12:50:40.4054463Z   network time: Fri, 03 Jan 2020 12:50:40 GMT
2020-01-03T12:50:40.4055119Z == end clock drift check ==
2020-01-03T12:50:41.5332344Z 
2020-01-03T12:50:41.5452390Z ##[error]Bash exited with code '1'.
2020-01-03T12:50:41.5517426Z ##[section]Starting: Checkout
2020-01-03T12:50:41.5519370Z ==============================================================================
2020-01-03T12:50:41.5519429Z Task         : Get sources
2020-01-03T12:50:41.5519498Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
