plain
2020-02-09T15:15:45.7241417Z ========================== Starting Command Output ===========================
2020-02-09T15:15:45.7243501Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/eb9742a1-a364-4ea2-90d8-de92a9afb089.sh
2020-02-09T15:15:45.7243537Z 
2020-02-09T15:15:45.7246216Z ##[section]Finishing: Disable git automatic line ending conversion
2020-02-09T15:15:45.7251750Z ##[section]Starting: Checkout rust-lang/rust@refs/pull/68963/merge to s
2020-02-09T15:15:45.7253283Z Task         : Get sources
2020-02-09T15:15:45.7253321Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2020-02-09T15:15:45.7253351Z Version      : 1.0.0
2020-02-09T15:15:45.7253380Z Author       : Microsoft
---
2020-02-09T15:15:46.5612326Z ##[command]git remote add origin https://github.com/rust-lang/rust
2020-02-09T15:15:46.5679733Z ##[command]git config gc.auto 0
2020-02-09T15:15:46.5745925Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2020-02-09T15:15:46.5798484Z ##[command]git config --get-all http.proxy
2020-02-09T15:15:46.5915495Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/68963/merge:refs/remotes/pull/68963/merge
---
2020-02-09T16:11:47.6144637Z .................................................................................................... 1700/9622
2020-02-09T16:11:52.6742814Z .................................................................................................... 1800/9622
2020-02-09T16:12:05.5717657Z ............................i....................................................................... 1900/9622
2020-02-09T16:12:13.0847959Z .................................................................................................... 2000/9622
2020-02-09T16:12:27.7302753Z ..................iiiii............................................................................. 2100/9622
2020-02-09T16:12:37.3595288Z .................................................................................................... 2300/9622
2020-02-09T16:12:39.6770522Z .................................................................................................... 2400/9622
2020-02-09T16:12:44.3853892Z .................................................................................................... 2500/9622
2020-02-09T16:13:04.5381663Z .................................................................................................... 2600/9622
---
2020-02-09T16:15:36.2170545Z .....................................................................i...............i.............. 4900/9622
2020-02-09T16:15:43.8273686Z .................................................................................................... 5000/9622
2020-02-09T16:15:51.4785741Z .................................................................................................... 5100/9622
2020-02-09T16:15:55.9103006Z ...........i........................................................................................ 5200/9622
2020-02-09T16:16:06.5065510Z .....................................................................................ii.ii........i. 5300/9622
2020-02-09T16:16:10.1077683Z ..i................................................................................................. 5400/9622
2020-02-09T16:16:22.3096022Z .................................................................................................... 5600/9622
2020-02-09T16:16:30.5724540Z .........................................................................i.......................... 5700/9622
2020-02-09T16:16:37.7246597Z .................................................................................................... 5800/9622
2020-02-09T16:16:44.3166893Z .................................................................................................... 5900/9622
2020-02-09T16:16:44.3166893Z .................................................................................................... 5900/9622
2020-02-09T16:16:54.0684248Z .................................................................ii...i..ii...........i............. 6000/9622
2020-02-09T16:17:15.2294056Z .................................................................................................... 6200/9622
2020-02-09T16:17:23.1773590Z .................................................................................................... 6300/9622
2020-02-09T16:17:23.1773590Z .................................................................................................... 6300/9622
2020-02-09T16:17:30.9054943Z .............................................................................................i..ii.. 6400/9622
2020-02-09T16:17:53.4413728Z .................................................................................................... 6600/9622
2020-02-09T16:18:03.2376109Z ................................................................................i................... 6700/9622
2020-02-09T16:18:05.3523876Z .................................................................................................... 6800/9622
2020-02-09T16:18:07.4823709Z .......................................................................................i............ 6900/9622
---
2020-02-09T16:19:41.7845639Z .................................................................................................... 7600/9622
2020-02-09T16:19:45.7925442Z .................................................................................................... 7700/9622
2020-02-09T16:19:51.4623822Z .................................................................................................... 7800/9622
2020-02-09T16:20:00.2096081Z .................................................................................................... 7900/9622
2020-02-09T16:20:09.7990709Z ...............................................................iiiiiii.i............................ 8000/9622
2020-02-09T16:20:25.3106103Z ...i......i.......................................................................................F. 8200/9622
2020-02-09T16:20:31.1593530Z .................................................................................................... 8300/9622
2020-02-09T16:20:45.3125831Z .................................................................................................... 8400/9622
2020-02-09T16:20:54.5716867Z .................................................................................................... 8500/9622
---
2020-02-09T16:22:58.6360314Z 
2020-02-09T16:22:58.6360356Z 1 error[E0425]: cannot find value `fina` in this scope
2020-02-09T16:22:58.6360583Z 2   --> $DIR/issue-68962.rs:7:5
2020-02-09T16:22:58.6360626Z 3    |
2020-02-09T16:22:58.6360851Z - LL |     fina; //~ ERROR cannot find value
2020-02-09T16:22:58.6360897Z + LL |     fina;
2020-02-09T16:22:58.6360943Z 5    |     ^^^^ help: a local variable with a similar name exists: `r#final`
2020-02-09T16:22:58.6361232Z + 
2020-02-09T16:22:58.6361278Z 7 error[E0425]: cannot find function `f` in this scope
2020-02-09T16:22:58.6361507Z 8   --> $DIR/issue-68962.rs:10:5
2020-02-09T16:22:58.6361553Z 9    |
2020-02-09T16:22:58.6361553Z 9    |
2020-02-09T16:22:58.6361581Z 
2020-02-09T16:22:58.6361623Z 10 LL | fn r#fn() {}
2020-02-09T16:22:58.6361875Z 11    | ------------ similarly named function `r#fn` defined here
2020-02-09T16:22:58.6362208Z - LL |     f(); //~ ERROR cannot find function
2020-02-09T16:22:58.6362274Z + LL |     f();
2020-02-09T16:22:58.6362323Z 14    |     ^ help: a function with a similar name exists: `r#fn`
2020-02-09T16:22:58.6362500Z -  
---
2020-02-09T16:22:58.6362937Z 
2020-02-09T16:22:58.6362972Z 
2020-02-09T16:22:58.6363017Z The actual stderr differed from the expected stderr.
2020-02-09T16:22:58.6363303Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/span/issue-68962/issue-68962.stderr
2020-02-09T16:22:58.6363562Z To update references, rerun the tests and pass the `--bless` flag
2020-02-09T16:22:58.6363811Z To only update this specific test, also pass `--test-args span/issue-68962.rs`
2020-02-09T16:22:58.6363917Z error: 1 errors occurred comparing output.
2020-02-09T16:22:58.6363960Z status: exit code: 1
2020-02-09T16:22:58.6363960Z status: exit code: 1
2020-02-09T16:22:58.6364679Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/span/issue-68962.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/span/issue-68962" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/span/issue-68962/auxiliary" "-A" "unused"
2020-02-09T16:22:58.6365013Z ------------------------------------------
2020-02-09T16:22:58.6365065Z 
2020-02-09T16:22:58.6365276Z ------------------------------------------
2020-02-09T16:22:58.6365322Z stderr:
2020-02-09T16:22:58.6365322Z stderr:
2020-02-09T16:22:58.6365732Z ------------------------------------------
2020-02-09T16:22:58.6365797Z error[E0425]: cannot find value `fina` in this scope
2020-02-09T16:22:58.6366059Z   --> /checkout/src/test/ui/span/issue-68962.rs:7:5
2020-02-09T16:22:58.6366106Z    |
2020-02-09T16:22:58.6366169Z LL |     fina; //~ ERROR cannot find value
2020-02-09T16:22:58.6366223Z    |     ^^^^ help: a local variable with a similar name exists: `r#final`
2020-02-09T16:22:58.6366321Z error[E0425]: cannot find function `f` in this scope
2020-02-09T16:22:58.6366701Z   --> /checkout/src/test/ui/span/issue-68962.rs:10:5
2020-02-09T16:22:58.6366748Z    |
2020-02-09T16:22:58.6366748Z    |
2020-02-09T16:22:58.6366807Z LL | fn r#fn() {}
2020-02-09T16:22:58.6367041Z    | ------------ similarly named function `r#fn` defined here
2020-02-09T16:22:58.6367130Z LL |     f(); //~ ERROR cannot find function
2020-02-09T16:22:58.6367195Z    |     ^ help: a function with a similar name exists: `r#fn`
2020-02-09T16:22:58.6367226Z 
2020-02-09T16:22:58.6367276Z error: aborting due to 2 previous errors
---
2020-02-09T16:22:58.6391553Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:348:22
2020-02-09T16:22:58.6391686Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2020-02-09T16:22:58.6400439Z 
2020-02-09T16:22:58.6400888Z 
2020-02-09T16:22:58.6413040Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-7/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "7.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2020-02-09T16:22:58.6414346Z 
2020-02-09T16:22:58.6414395Z 
2020-02-09T16:22:58.6423063Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2020-02-09T16:22:58.6423139Z Build completed unsuccessfully in 1:01:27
2020-02-09T16:22:58.6423139Z Build completed unsuccessfully in 1:01:27
2020-02-09T16:22:58.6484050Z == clock drift check ==
2020-02-09T16:22:58.6507862Z   local time: Sun Feb  9 16:22:58 UTC 2020
2020-02-09T16:22:58.9472233Z   network time: Sun, 09 Feb 2020 16:22:58 GMT
2020-02-09T16:22:58.9477535Z == end clock drift check ==
2020-02-09T16:22:59.3606857Z 
2020-02-09T16:22:59.3701799Z ##[error]Bash exited with code '1'.
2020-02-09T16:22:59.3712931Z ##[section]Finishing: Run build
2020-02-09T16:22:59.3737469Z ##[section]Starting: Checkout rust-lang/rust@refs/pull/68963/merge to s
2020-02-09T16:22:59.3739164Z Task         : Get sources
2020-02-09T16:22:59.3739221Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2020-02-09T16:22:59.3739264Z Version      : 1.0.0
2020-02-09T16:22:59.3739299Z Author       : Microsoft
2020-02-09T16:22:59.3739299Z Author       : Microsoft
2020-02-09T16:22:59.3739339Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
2020-02-09T16:22:59.3739402Z ==============================================================================
2020-02-09T16:22:59.8045762Z Cleaning any cached credential from repository: rust-lang/rust (GitHub)
2020-02-09T16:22:59.8092982Z ##[section]Finishing: Checkout rust-lang/rust@refs/pull/68963/merge to s
2020-02-09T16:22:59.8202052Z Cleaning up task key
2020-02-09T16:22:59.8202699Z Start cleaning up orphan processes.
2020-02-09T16:22:59.8320357Z Terminate orphan process: pid (3662) (python)
2020-02-09T16:22:59.8583184Z ##[section]Finishing: Finalize Job
